
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://shandley.github.io/awesome-virome/docs/latest/best-practices/tool-combinations/">
      
      
        <link rel="prev" href="../quality-control/">
      
      
        <link rel="next" href="../statistical-analysis/">
      
      
        
      
      
      <link rel="icon" href="../../assets/awesome-virome-hex.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.0">
    
    
      
        <title>Tool Combinations - Awesome-Virome</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.618322db.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      
  
  
    
      
      
    
  
  
  <style>:root{--md-admonition-icon--tool:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22m21.71%2020.29-1.42%201.42a1%201%200%200%201-1.41%200L7%209.85A3.8%203.8%200%200%201%206%2010a4%204%200%200%201-3.78-5.3l2.54%202.54.53-.53%201.42-1.42.53-.53L4.7%202.22A4%204%200%200%201%2010%206a3.8%203.8%200%200%201-.15%201l11.86%2011.88a1%201%200%200%201%200%201.41M2.29%2018.88a1%201%200%200%200%200%201.41l1.42%201.42a1%201%200%200%200%201.41%200l5.47-5.46-2.83-2.83M20%202l-4%202v2l-2.17%202.17%202%202L18%208h2l2-4Z%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#tool-combination-best-practices" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../home.html" title="Awesome-Virome" class="md-header__button md-logo" aria-label="Awesome-Virome" data-md-component="logo">
      
  <img src="../../assets/awesome-virome-hex.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Awesome-Virome
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Tool Combinations
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/shandley/awesome-virome" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    shandley/awesome-virome
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../intro/" class="md-tabs__link">
          
  
  
    
  
  Getting Started

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../fundamentals/" class="md-tabs__link">
          
  
  
    
  
  Fundamentals

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tutorials/" class="md-tabs__link">
          
  
  
    
  
  Tutorials

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../quality-control/" class="md-tabs__link">
          
  
  
    
  
  Best Practices

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tools/overview/" class="md-tabs__link">
          
  
  
    
  
  Core Analysis Tools

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../troubleshooting/" class="md-tabs__link">
          
  
  
    
  
  Troubleshooting

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../api/overview/" class="md-tabs__link">
          
  
  
    
  
  API Reference

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../contributing/guidelines/" class="md-tabs__link">
          
  
  
    
  
  Contributing

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../home.html" title="Awesome-Virome" class="md-nav__button md-logo" aria-label="Awesome-Virome" data-md-component="logo">
      
  <img src="../../assets/awesome-virome-hex.png" alt="logo">

    </a>
    Awesome-Virome
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/shandley/awesome-virome" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    shandley/awesome-virome
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../intro/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../intro/workflows/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Typical Workflows
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../intro/versioning/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Versioning
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../fundamentals/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Fundamentals
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_3" id="__nav_3_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Fundamentals
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../fundamentals/sample-preparation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Sample Preparation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../fundamentals/sequencing-strategies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Sequencing Strategies
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../tutorials/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Tutorials
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_4" id="__nav_4_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Tutorials
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/basic-metagenome-virome/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Basic Metagenome Virome
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/rna-virus-discovery/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RNA Virus Discovery
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/prophage-identification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Prophage Identification
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/comparative-virome/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Comparative Virome Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/host-prediction-workflows/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction Workflows
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Best Practices
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Best Practices
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Control
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Tool Combinations
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Tool Combinations
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#why-combine-tools" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why Combine Tools?
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#viral-identification-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Viral Identification Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Viral Identification Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#recommended-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Recommended Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Recommended Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gold-standard-high-accuracy-moderate-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Gold Standard (High Accuracy, Moderate Coverage)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#balanced-good-accuracy-good-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Balanced (Good Accuracy, Good Coverage)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#high-sensitivity-maximum-coverage-lower-accuracy" class="md-nav__link">
    <span class="md-ellipsis">
      
        High Sensitivity (Maximum Coverage, Lower Accuracy)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tool-specific-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Tool-Specific Considerations
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-filtering-after-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Filtering After Prediction
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#host-prediction-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Host Prediction Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Host Prediction Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tiered-approach-recommended" class="md-nav__link">
    <span class="md-ellipsis">
      
        Tiered Approach (Recommended)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#when-to-use-each-host-prediction-tool" class="md-nav__link">
    <span class="md-ellipsis">
      
        When to Use Each Host Prediction Tool
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Assembly Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#single-assembly-vs-co-assembly-vs-hybrid" class="md-nav__link">
    <span class="md-ellipsis">
      
        Single Assembly vs Co-Assembly vs Hybrid
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Annotation Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Annotation Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#phage-annotation-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      
        Phage Annotation Pipeline
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-classification-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxonomic Classification Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxonomic Classification Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#multi-method-taxonomy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Multi-Method Taxonomy
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#abundance-estimation-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Abundance Estimation Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Abundance Estimation Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#read-mapping-strategy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Mapping Strategy
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#statistical-analysis-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Statistical Analysis Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Statistical Analysis Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#diversity-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diversity Analysis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#differential-abundance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Differential Abundance
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workflow-integration-examples" class="md-nav__link">
    <span class="md-ellipsis">
      
        Workflow Integration Examples
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Integration Examples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#complete-virome-workflow-recommended-stack" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete Virome Workflow (Recommended Stack)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tool-compatibility-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      
        Tool Compatibility Matrix
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-pitfalls-in-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Common Pitfalls in Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Pitfalls in Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#anti-patterns-avoid-these" class="md-nav__link">
    <span class="md-ellipsis">
      
        ❌ Anti-Patterns (Avoid These)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#best-practices" class="md-nav__link">
    <span class="md-ellipsis">
      
        ✅ Best Practices
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#computational-resource-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Computational Resource Considerations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      
        Further Reading
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../statistical-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Statistical Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../validation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../reproducibility/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Core Analysis Tools
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Core Analysis Tools
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/virus-identification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Virus Identification
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/host-prediction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/selection-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Selection Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../troubleshooting/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Troubleshooting
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    API Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            
  
    API Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/endpoints/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Endpoints
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            
  
    Contributing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/guidelines/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guidelines
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/validation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/updating-docs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Updating Docs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#why-combine-tools" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why Combine Tools?
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#viral-identification-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Viral Identification Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Viral Identification Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#recommended-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Recommended Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Recommended Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gold-standard-high-accuracy-moderate-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Gold Standard (High Accuracy, Moderate Coverage)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#balanced-good-accuracy-good-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Balanced (Good Accuracy, Good Coverage)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#high-sensitivity-maximum-coverage-lower-accuracy" class="md-nav__link">
    <span class="md-ellipsis">
      
        High Sensitivity (Maximum Coverage, Lower Accuracy)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tool-specific-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Tool-Specific Considerations
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-filtering-after-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Filtering After Prediction
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#host-prediction-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Host Prediction Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Host Prediction Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tiered-approach-recommended" class="md-nav__link">
    <span class="md-ellipsis">
      
        Tiered Approach (Recommended)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#when-to-use-each-host-prediction-tool" class="md-nav__link">
    <span class="md-ellipsis">
      
        When to Use Each Host Prediction Tool
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assembly Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Assembly Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#single-assembly-vs-co-assembly-vs-hybrid" class="md-nav__link">
    <span class="md-ellipsis">
      
        Single Assembly vs Co-Assembly vs Hybrid
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Annotation Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Annotation Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#phage-annotation-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      
        Phage Annotation Pipeline
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-classification-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Taxonomic Classification Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxonomic Classification Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#multi-method-taxonomy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Multi-Method Taxonomy
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#abundance-estimation-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Abundance Estimation Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Abundance Estimation Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#read-mapping-strategy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Mapping Strategy
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#statistical-analysis-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Statistical Analysis Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Statistical Analysis Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#diversity-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diversity Analysis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#differential-abundance" class="md-nav__link">
    <span class="md-ellipsis">
      
        Differential Abundance
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workflow-integration-examples" class="md-nav__link">
    <span class="md-ellipsis">
      
        Workflow Integration Examples
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow Integration Examples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#complete-virome-workflow-recommended-stack" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete Virome Workflow (Recommended Stack)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tool-compatibility-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      
        Tool Compatibility Matrix
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-pitfalls-in-tool-combinations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Common Pitfalls in Tool Combinations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Pitfalls in Tool Combinations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#anti-patterns-avoid-these" class="md-nav__link">
    <span class="md-ellipsis">
      
        ❌ Anti-Patterns (Avoid These)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#best-practices" class="md-nav__link">
    <span class="md-ellipsis">
      
        ✅ Best Practices
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#computational-resource-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Computational Resource Considerations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      
        Further Reading
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="tool-combination-best-practices">Tool Combination Best Practices</h1>
<blockquote>
<p><strong>Last Updated:</strong> November 29, 2025</p>
</blockquote>
<p>Combining multiple tools is essential for accurate virome analysis. This guide provides evidence-based recommendations for which tool combinations work best for different scenarios.</p>
<h2 id="why-combine-tools">Why Combine Tools?</h2>
<p><strong>Single tools have limitations:</strong>
- <strong>False positives:</strong> 10-30% for viral identification tools
- <strong>False negatives:</strong> Miss 20-40% of viruses
- <strong>Method bias:</strong> Different algorithms favor different virus types</p>
<p><strong>Tool combinations provide:</strong>
- Higher accuracy (consensus predictions)
- Better coverage (union of predictions)
- Confidence assessment (number of tools agreeing)</p>
<h2 id="viral-identification-tool-combinations">Viral Identification Tool Combinations</h2>
<h3 id="recommended-combinations">Recommended Combinations</h3>
<h4 id="gold-standard-high-accuracy-moderate-coverage"><strong>Gold Standard (High Accuracy, Moderate Coverage)</strong></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># VirSorter2 + VIBRANT + geNomad (consensus ≥2)</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="c1"># Expected: 60-70% of viruses, &lt;5% false positives</span>
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="c1"># Run all three tools</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>virsorter<span class="w"> </span>run<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--min-score<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>-w<span class="w"> </span>vs2/<span class="w"> </span>all
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>VIBRANT_run.py<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>-folder<span class="w"> </span>vibrant/
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>genomad<span class="w"> </span>end-to-end<span class="w"> </span>contigs.fa<span class="w"> </span>genomad/<span class="w"> </span>genomad_db/
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a><span class="c1"># Take consensus (≥2 tools agree)</span>
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a><span class="c1"># Implementation in Tutorial 1</span>
</span></code></pre></div>
<p><strong>Best for:</strong>
- Publication-quality datasets
- Novel virus discovery
- Low tolerance for false positives</p>
<p><strong>Performance:</strong>
- Sensitivity: 60-70%
- Specificity: &gt;95%
- Runtime: 4-8 hours (1000 contigs, 16 cores)</p>
<hr />
<h4 id="balanced-good-accuracy-good-coverage"><strong>Balanced (Good Accuracy, Good Coverage)</strong></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># VirSorter2 + geNomad (either tool)</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="c1"># Expected: 75-85% of viruses, ~10% false positives</span>
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>virsorter<span class="w"> </span>run<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--min-score<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>-w<span class="w"> </span>vs2/<span class="w"> </span>all
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>genomad<span class="w"> </span>end-to-end<span class="w"> </span>contigs.fa<span class="w"> </span>genomad/<span class="w"> </span>genomad_db/
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a><span class="c1"># Union: any tool predicts viral</span>
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>cat<span class="w"> </span>vs2/final-viral-combined.fa<span class="w"> </span>genomad/viral.fna<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a><span class="w">    </span>cd-hit-est<span class="w"> </span>-c<span class="w"> </span><span class="m">0</span>.95<span class="w"> </span>-i<span class="w"> </span>-<span class="w"> </span>-o<span class="w"> </span>viral_union.fa
</span></code></pre></div>
<p><strong>Best for:</strong>
- Diversity studies
- Exploratory analysis
- Moderate sample size (10-50 samples)</p>
<p><strong>Performance:</strong>
- Sensitivity: 75-85%
- Specificity: ~90%
- Runtime: 2-4 hours (1000 contigs)</p>
<hr />
<h4 id="high-sensitivity-maximum-coverage-lower-accuracy"><strong>High Sensitivity (Maximum Coverage, Lower Accuracy)</strong></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1"># VirSorter2 (low threshold) + VIBRANT + geNomad + DeepVirFinder</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="c1"># Expected: &gt;90% of viruses, 15-25% false positives</span>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>virsorter<span class="w"> </span>run<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--min-score<span class="w"> </span><span class="m">0</span>.3<span class="w"> </span>-w<span class="w"> </span>vs2/<span class="w"> </span>all
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>VIBRANT_run.py<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>-folder<span class="w"> </span>vibrant/
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>genomad<span class="w"> </span>end-to-end<span class="w"> </span>contigs.fa<span class="w"> </span>genomad/<span class="w"> </span>genomad_db/<span class="w"> </span>--min-score<span class="w"> </span><span class="m">0</span>.7
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>python<span class="w"> </span>dvf.py<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>-o<span class="w"> </span>dvf/<span class="w"> </span>-l<span class="w"> </span><span class="m">1000</span>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="c1"># Union of all predictions</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a><span class="c1"># MUST use CheckV to filter false positives</span>
</span></code></pre></div>
<p><strong>Best for:</strong>
- Highly novel environments
- Pilot studies
- When missing viruses is worse than false positives</p>
<p><strong>Performance:</strong>
- Sensitivity: &gt;90%
- Specificity: ~75-85%
- <strong>Requires:</strong> CheckV filtering to remove false positives</p>
<hr />
<h3 id="tool-specific-considerations">Tool-Specific Considerations</h3>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Strengths</th>
<th>Weaknesses</th>
<th>Best Parameters</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>VirSorter2</strong></td>
<td>Broad virus types, well-tested</td>
<td>Slower, some false positives</td>
<td><code>--min-score 0.5-0.7</code></td>
</tr>
<tr>
<td><strong>VIBRANT</strong></td>
<td>Good for phages, built-in annotation</td>
<td>Misses some virus types</td>
<td>Default settings</td>
</tr>
<tr>
<td><strong>geNomad</strong></td>
<td>Fast, recent database, plasmid detection</td>
<td>Newer (less validated)</td>
<td><code>--min-score 0.8</code></td>
</tr>
<tr>
<td><strong>DeepVirFinder</strong></td>
<td>Works on short contigs</td>
<td>Needs GPU for speed, higher FP</td>
<td><code>--len 1000</code> minimum</td>
</tr>
</tbody>
</table>
<h3 id="quality-filtering-after-prediction">Quality Filtering After Prediction</h3>
<p><strong>Always run CheckV:</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># After any viral identification</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>checkv<span class="w"> </span>end_to_end<span class="w"> </span>viral_contigs.fa<span class="w"> </span>checkv_out/<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="c1"># Recommended filters:</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="c1"># Conservative: Complete + High-quality only</span>
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>awk<span class="w"> </span>-F<span class="s1">&#39;\t&#39;</span><span class="w"> </span><span class="s1">&#39;$8 == &quot;Complete&quot; || $8 == &quot;High-quality&quot; {print $1}&#39;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="w">    </span>checkv_out/quality_summary.tsv<span class="w"> </span>&gt;<span class="w"> </span>hq_ids.txt
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="c1"># Balanced: Complete + High + Medium</span>
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a>awk<span class="w"> </span>-F<span class="s1">&#39;\t&#39;</span><span class="w"> </span><span class="s1">&#39;$8 != &quot;Low-quality&quot; &amp;&amp; $8 != &quot;Not-determined&quot; {print $1}&#39;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a><span class="w">    </span>checkv_out/quality_summary.tsv<span class="w"> </span>&gt;<span class="w"> </span>good_ids.txt
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a>
</span><span id="__span-3-13"><a id="__codelineno-3-13" name="__codelineno-3-13" href="#__codelineno-3-13"></a><span class="c1"># Also filter by contamination</span>
</span><span id="__span-3-14"><a id="__codelineno-3-14" name="__codelineno-3-14" href="#__codelineno-3-14"></a>awk<span class="w"> </span>-F<span class="s1">&#39;\t&#39;</span><span class="w"> </span><span class="s1">&#39;$8 != &quot;Low-quality&quot; &amp;&amp; $10 &lt; 5 {print $1}&#39;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-15"><a id="__codelineno-3-15" name="__codelineno-3-15" href="#__codelineno-3-15"></a><span class="w">    </span>checkv_out/quality_summary.tsv<span class="w"> </span>&gt;<span class="w"> </span>clean_ids.txt
</span></code></pre></div></p>
<h2 id="host-prediction-tool-combinations">Host Prediction Tool Combinations</h2>
<h3 id="tiered-approach-recommended">Tiered Approach (Recommended)</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Tier 1: CRISPR spacers (highest confidence, ~5-10% of phages)</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="c1"># Use PILER-CR or minced + BLAST</span>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Tier 2: iPHoP integrated prediction (~40-60% of phages)</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>iphop<span class="w"> </span>predict<span class="w"> </span>--fa_file<span class="w"> </span>phages.fa<span class="w"> </span>--db_dir<span class="w"> </span>iphop_db/<span class="w"> </span>--out_dir<span class="w"> </span>iphop/
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a><span class="c1"># Tier 3: Individual methods for remaining phages</span>
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a><span class="c1"># - WIsH (genomic signatures)</span>
</span><span id="__span-4-9"><a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a><span class="c1"># - CHERRY (deep learning)</span>
</span><span id="__span-4-10"><a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a><span class="c1"># - Protein homology (BLAST)</span>
</span><span id="__span-4-11"><a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a>
</span><span id="__span-4-12"><a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a><span class="c1"># Combine with confidence weighting:</span>
</span><span id="__span-4-13"><a id="__codelineno-4-13" name="__codelineno-4-13" href="#__codelineno-4-13"></a><span class="c1"># - CRISPR match: 95% confidence</span>
</span><span id="__span-4-14"><a id="__codelineno-4-14" name="__codelineno-4-14" href="#__codelineno-4-14"></a><span class="c1"># - iPHoP + CRISPR: 90% confidence</span>
</span><span id="__span-4-15"><a id="__codelineno-4-15" name="__codelineno-4-15" href="#__codelineno-4-15"></a><span class="c1"># - iPHoP &gt;90 score: 70% confidence</span>
</span><span id="__span-4-16"><a id="__codelineno-4-16" name="__codelineno-4-16" href="#__codelineno-4-16"></a><span class="c1"># - ≥2 methods agree: 60% confidence</span>
</span><span id="__span-4-17"><a id="__codelineno-4-17" name="__codelineno-4-17" href="#__codelineno-4-17"></a><span class="c1"># - Single method: 30% confidence</span>
</span></code></pre></div>
<p><strong>Implementation:</strong>
<div class="language-python highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="k">def</span><span class="w"> </span><span class="nf">assign_host_confidence</span><span class="p">(</span><span class="n">predictions</span><span class="p">):</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Assign confidence based on prediction method(s)&quot;&quot;&quot;</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>    <span class="n">confidence_map</span> <span class="o">=</span> <span class="p">{</span>
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>        <span class="s1">&#39;CRISPR&#39;</span><span class="p">:</span> <span class="mf">0.95</span><span class="p">,</span>
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>        <span class="s1">&#39;iPHoP_high&#39;</span><span class="p">:</span> <span class="mf">0.70</span><span class="p">,</span>  <span class="c1"># iPHoP score &gt;90</span>
</span><span id="__span-5-6"><a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>        <span class="s1">&#39;iPHoP_medium&#39;</span><span class="p">:</span> <span class="mf">0.50</span><span class="p">,</span>  <span class="c1"># iPHoP score 70-90</span>
</span><span id="__span-5-7"><a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>        <span class="s1">&#39;WIsH&#39;</span><span class="p">:</span> <span class="mf">0.40</span><span class="p">,</span>
</span><span id="__span-5-8"><a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>        <span class="s1">&#39;CHERRY&#39;</span><span class="p">:</span> <span class="mf">0.40</span><span class="p">,</span>
</span><span id="__span-5-9"><a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>        <span class="s1">&#39;Homology&#39;</span><span class="p">:</span> <span class="mf">0.45</span><span class="p">,</span>
</span><span id="__span-5-10"><a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a>        <span class="s1">&#39;Consensus_2&#39;</span><span class="p">:</span> <span class="mf">0.60</span><span class="p">,</span>  <span class="c1"># ≥2 methods agree</span>
</span><span id="__span-5-11"><a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a>        <span class="s1">&#39;Consensus_3&#39;</span><span class="p">:</span> <span class="mf">0.75</span><span class="p">,</span>  <span class="c1"># ≥3 methods agree</span>
</span><span id="__span-5-12"><a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a>    <span class="p">}</span>
</span><span id="__span-5-13"><a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a>
</span><span id="__span-5-14"><a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a>    <span class="c1"># Logic to determine confidence</span>
</span><span id="__span-5-15"><a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a>    <span class="k">if</span> <span class="s1">&#39;CRISPR&#39;</span> <span class="ow">in</span> <span class="n">predictions</span><span class="p">[</span><span class="s1">&#39;methods&#39;</span><span class="p">]:</span>
</span><span id="__span-5-16"><a id="__codelineno-5-16" name="__codelineno-5-16" href="#__codelineno-5-16"></a>        <span class="k">return</span> <span class="n">confidence_map</span><span class="p">[</span><span class="s1">&#39;CRISPR&#39;</span><span class="p">]</span>
</span><span id="__span-5-17"><a id="__codelineno-5-17" name="__codelineno-5-17" href="#__codelineno-5-17"></a>    <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="n">predictions</span><span class="p">[</span><span class="s1">&#39;methods&#39;</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="mi">3</span><span class="p">:</span>
</span><span id="__span-5-18"><a id="__codelineno-5-18" name="__codelineno-5-18" href="#__codelineno-5-18"></a>        <span class="k">return</span> <span class="n">confidence_map</span><span class="p">[</span><span class="s1">&#39;Consensus_3&#39;</span><span class="p">]</span>
</span><span id="__span-5-19"><a id="__codelineno-5-19" name="__codelineno-5-19" href="#__codelineno-5-19"></a>    <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="n">predictions</span><span class="p">[</span><span class="s1">&#39;methods&#39;</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="mi">2</span><span class="p">:</span>
</span><span id="__span-5-20"><a id="__codelineno-5-20" name="__codelineno-5-20" href="#__codelineno-5-20"></a>        <span class="k">return</span> <span class="n">confidence_map</span><span class="p">[</span><span class="s1">&#39;Consensus_2&#39;</span><span class="p">]</span>
</span><span id="__span-5-21"><a id="__codelineno-5-21" name="__codelineno-5-21" href="#__codelineno-5-21"></a>    <span class="c1"># ... etc</span>
</span></code></pre></div></p>
<h3 id="when-to-use-each-host-prediction-tool">When to Use Each Host Prediction Tool</h3>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Use When</th>
<th>Avoid When</th>
<th>Typical Success Rate</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>CRISPR</strong></td>
<td>Bacterial MAGs available</td>
<td>Limited bacterial data</td>
<td>5-10% (very high confidence)</td>
</tr>
<tr>
<td><strong>iPHoP</strong></td>
<td>Any phages</td>
<td>Time/resource limited</td>
<td>40-60% (medium-high confidence)</td>
</tr>
<tr>
<td><strong>WIsH</strong></td>
<td>Bacterial genomes available</td>
<td>Only viral databases</td>
<td>30-50% (medium confidence)</td>
</tr>
<tr>
<td><strong>CHERRY</strong></td>
<td>Any phages, especially novel</td>
<td>Need genus-level precision</td>
<td>35-55% (medium confidence)</td>
</tr>
<tr>
<td><strong>VirHostMatcher</strong></td>
<td>Related hosts in database</td>
<td>Completely novel phages</td>
<td>25-45% (low-medium confidence)</td>
</tr>
</tbody>
</table>
<h2 id="assembly-tool-combinations">Assembly Tool Combinations</h2>
<h3 id="single-assembly-vs-co-assembly-vs-hybrid">Single Assembly vs Co-Assembly vs Hybrid</h3>
<p><strong>Scenario 1: Few Samples (1-5)</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="c1"># Use individual assemblies per sample</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="k">for</span><span class="w"> </span>sample<span class="w"> </span><span class="k">in</span><span class="w"> </span>sample1<span class="w"> </span>sample2<span class="w"> </span>sample3<span class="p">;</span><span class="w"> </span><span class="k">do</span>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="w">    </span>metaspades.py<span class="w"> </span>--metaviral<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a><span class="w">        </span>-1<span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_R1.fq<span class="w"> </span>-2<span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_R2.fq<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a><span class="w">        </span>-o<span class="w"> </span>assembly/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="k">done</span>
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a>
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a><span class="c1"># Dereplicate across samples</span>
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>cat<span class="w"> </span>assembly/*/contigs.fasta<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a><span class="w">    </span>cd-hit-est<span class="w"> </span>-c<span class="w"> </span><span class="m">0</span>.95<span class="w"> </span>-aS<span class="w"> </span><span class="m">0</span>.85<span class="w"> </span>-o<span class="w"> </span>derep_contigs.fa
</span></code></pre></div></p>
<p><strong>Scenario 2: Many Similar Samples (10-50)</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Co-assemble all samples together</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>cat<span class="w"> </span>sample*_R1.fq.gz<span class="w"> </span>&gt;<span class="w"> </span>all_R1.fq.gz
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>cat<span class="w"> </span>sample*_R2.fq.gz<span class="w"> </span>&gt;<span class="w"> </span>all_R2.fq.gz
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>metaspades.py<span class="w"> </span>--metaviral<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="w">    </span>-1<span class="w"> </span>all_R1.fq.gz<span class="w"> </span>-2<span class="w"> </span>all_R2.fq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="w">    </span>-o<span class="w"> </span>co_assembly/<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-m<span class="w"> </span><span class="m">250</span>
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a><span class="c1"># Better for shared viruses, misses rare sample-specific viruses</span>
</span></code></pre></div></p>
<p><strong>Scenario 3: Complex Design (Recommended for Most Studies)</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1"># Hybrid approach:</span>
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="c1"># 1. Co-assemble within groups (timepoints, treatments, etc.)</span>
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a><span class="c1"># 2. Individual assemblies for each sample</span>
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a><span class="c1"># 3. Combine and dereplicate</span>
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="c1"># Co-assembly per timepoint</span>
</span><span id="__span-8-7"><a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a>metaspades.py<span class="w"> </span>--metaviral<span class="w"> </span>-1<span class="w"> </span>T1_all_R1.fq<span class="w"> </span>-2<span class="w"> </span>T1_all_R2.fq<span class="w"> </span>-o<span class="w"> </span>T1_coasm/
</span><span id="__span-8-8"><a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a>metaspades.py<span class="w"> </span>--metaviral<span class="w"> </span>-1<span class="w"> </span>T2_all_R1.fq<span class="w"> </span>-2<span class="w"> </span>T2_all_R2.fq<span class="w"> </span>-o<span class="w"> </span>T2_coasm/
</span><span id="__span-8-9"><a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>
</span><span id="__span-8-10"><a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a><span class="c1"># Individual assemblies</span>
</span><span id="__span-8-11"><a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a><span class="k">for</span><span class="w"> </span>sample<span class="w"> </span><span class="k">in</span><span class="w"> </span>T1_R1<span class="w"> </span>T1_R2<span class="w"> </span>T2_R1<span class="w"> </span>T2_R2<span class="p">;</span><span class="w"> </span><span class="k">do</span>
</span><span id="__span-8-12"><a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a><span class="w">    </span>metaspades.py<span class="w"> </span>--metaviral<span class="w"> </span>-1<span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_R1.fq<span class="w"> </span>-2<span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_R2.fq<span class="w"> </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_asm/
</span><span id="__span-8-13"><a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a><span class="k">done</span>
</span><span id="__span-8-14"><a id="__codelineno-8-14" name="__codelineno-8-14" href="#__codelineno-8-14"></a>
</span><span id="__span-8-15"><a id="__codelineno-8-15" name="__codelineno-8-15" href="#__codelineno-8-15"></a><span class="c1"># Combine and dereplicate</span>
</span><span id="__span-8-16"><a id="__codelineno-8-16" name="__codelineno-8-16" href="#__codelineno-8-16"></a>cat<span class="w"> </span>*_coasm/contigs.fasta<span class="w"> </span>*_asm/contigs.fasta<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-17"><a id="__codelineno-8-17" name="__codelineno-8-17" href="#__codelineno-8-17"></a><span class="w">    </span>cd-hit-est<span class="w"> </span>-c<span class="w"> </span><span class="m">0</span>.95<span class="w"> </span>-aS<span class="w"> </span><span class="m">0</span>.85<span class="w"> </span>-M<span class="w"> </span><span class="m">64000</span><span class="w"> </span>-T<span class="w"> </span><span class="m">16</span><span class="w"> </span>-o<span class="w"> </span>final_derep.fa
</span></code></pre></div></p>
<h2 id="annotation-tool-combinations">Annotation Tool Combinations</h2>
<h3 id="phage-annotation-pipeline">Phage Annotation Pipeline</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1"># Step 1: Pharokka (fast, phage-specific)</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>pharokka.py<span class="w"> </span>-i<span class="w"> </span>phages.fa<span class="w"> </span>-o<span class="w"> </span>pharokka/<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-d<span class="w"> </span>pharokka_db/
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="c1"># Step 2: DRAMv (metabolic annotation)</span>
</span><span id="__span-9-5"><a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>DRAM-v.py<span class="w"> </span>annotate<span class="w"> </span>-i<span class="w"> </span>phages.fa<span class="w"> </span>-o<span class="w"> </span>dramv/<span class="w"> </span>--threads<span class="w"> </span><span class="m">8</span>
</span><span id="__span-9-6"><a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>DRAM-v.py<span class="w"> </span>distill<span class="w"> </span>-i<span class="w"> </span>dramv/annotations.tsv<span class="w"> </span>-o<span class="w"> </span>dramv_distill/
</span><span id="__span-9-7"><a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>
</span><span id="__span-9-8"><a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a><span class="c1"># Step 3: PHROGs (for additional functional annotation)</span>
</span><span id="__span-9-9"><a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a>hmmsearch<span class="w"> </span>--tblout<span class="w"> </span>phrogs_hits.txt<span class="w"> </span>-E<span class="w"> </span>1e-5<span class="w"> </span>--cpu<span class="w"> </span><span class="m">8</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-10"><a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a><span class="w">    </span>PHROGs.hmm<span class="w"> </span>pharokka/proteins.faa
</span><span id="__span-9-11"><a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a>
</span><span id="__span-9-12"><a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a><span class="c1"># Combine annotations (take best from each tool)</span>
</span><span id="__span-9-13"><a id="__codelineno-9-13" name="__codelineno-9-13" href="#__codelineno-9-13"></a>python3<span class="w"> </span>combine_annotations.py<span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-14"><a id="__codelineno-9-14" name="__codelineno-9-14" href="#__codelineno-9-14"></a><span class="w">    </span>--pharokka<span class="w"> </span>pharokka/*.gff<span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-15"><a id="__codelineno-9-15" name="__codelineno-9-15" href="#__codelineno-9-15"></a><span class="w">    </span>--dramv<span class="w"> </span>dramv/annotations.tsv<span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-16"><a id="__codelineno-9-16" name="__codelineno-9-16" href="#__codelineno-9-16"></a><span class="w">    </span>--phrogs<span class="w"> </span>phrogs_hits.txt<span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-17"><a id="__codelineno-9-17" name="__codelineno-9-17" href="#__codelineno-9-17"></a><span class="w">    </span>--output<span class="w"> </span>combined_annotations.tsv
</span></code></pre></div>
<p><strong>When to use each:</strong>
- <strong>Pharokka:</strong> Primary phage annotation (fast, comprehensive)
- <strong>DRAMv:</strong> Auxiliary metabolic genes (AMGs), metabolic pathways
- <strong>PHROGs:</strong> Additional functional categories
- <strong>PROKKA:</strong> If Pharokka fails (less phage-specific)</p>
<h2 id="taxonomic-classification-combinations">Taxonomic Classification Combinations</h2>
<h3 id="multi-method-taxonomy">Multi-Method Taxonomy</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Method 1: BLAST (sequence similarity)</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>blastn<span class="w"> </span>-query<span class="w"> </span>viral.fa<span class="w"> </span>-db<span class="w"> </span>nt<span class="w"> </span>-outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>-max_target_seqs<span class="w"> </span><span class="m">5</span><span class="w"> </span>&gt;<span class="w"> </span>blast.txt
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="c1"># Method 2: vConTACT2 (protein sharing network)</span>
</span><span id="__span-10-5"><a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a>vcontact2<span class="w"> </span>--raw-proteins<span class="w"> </span>proteins.faa<span class="w"> </span>--db<span class="w"> </span>ProkaryoticViralRefSeq<span class="w"> </span>--output<span class="w"> </span>vcontact2/
</span><span id="__span-10-6"><a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a>
</span><span id="__span-10-7"><a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a><span class="c1"># Method 3: PhaGCN (graph convolutional network)</span>
</span><span id="__span-10-8"><a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a>python<span class="w"> </span>PhaTYP.py<span class="w"> </span>--contigs<span class="w"> </span>viral.fa<span class="w"> </span>--threads<span class="w"> </span><span class="m">8</span>
</span><span id="__span-10-9"><a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a>
</span><span id="__span-10-10"><a id="__codelineno-10-10" name="__codelineno-10-10" href="#__codelineno-10-10"></a><span class="c1"># Combine:</span>
</span><span id="__span-10-11"><a id="__codelineno-10-11" name="__codelineno-10-11" href="#__codelineno-10-11"></a><span class="c1"># - BLAST for known viruses (&gt;80% identity)</span>
</span><span id="__span-10-12"><a id="__codelineno-10-12" name="__codelineno-10-12" href="#__codelineno-10-12"></a><span class="c1"># - vConTACT2 for viral clusters (species-level)</span>
</span><span id="__span-10-13"><a id="__codelineno-10-13" name="__codelineno-10-13" href="#__codelineno-10-13"></a><span class="c1"># - PhaGCN for novel viruses (family-level)</span>
</span></code></pre></div>
<p><strong>Decision tree:</strong>
<div class="language-text highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>For each virus:
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>├─ BLAST identity &gt;90%? → Use BLAST taxonomy (high confidence)
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>├─ BLAST identity 70-90%? → Use BLAST family + vConTACT2 genus (medium confidence)
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>├─ vConTACT2 cluster with references? → Use cluster taxonomy (medium confidence)
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>└─ No hits? → Use PhaGCN family prediction (low confidence) + mark as novel
</span></code></pre></div></p>
<h2 id="abundance-estimation-combinations">Abundance Estimation Combinations</h2>
<h3 id="read-mapping-strategy">Read Mapping Strategy</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1"># Method 1: BBMap (sensitive, slower)</span>
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>bbmap.sh<span class="w"> </span><span class="k">in</span><span class="o">=</span>reads.fq<span class="w"> </span><span class="nv">ref</span><span class="o">=</span>viral.fa<span class="w"> </span><span class="nv">out</span><span class="o">=</span>bbmap.sam<span class="w"> </span><span class="nv">covstats</span><span class="o">=</span>bbmap_cov.txt
</span><span id="__span-12-3"><a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>
</span><span id="__span-12-4"><a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a><span class="c1"># Method 2: Bowtie2 (fast, standard)</span>
</span><span id="__span-12-5"><a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a>bowtie2-build<span class="w"> </span>viral.fa<span class="w"> </span>viral_idx
</span><span id="__span-12-6"><a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a>bowtie2<span class="w"> </span>-x<span class="w"> </span>viral_idx<span class="w"> </span>-1<span class="w"> </span>R1.fq<span class="w"> </span>-2<span class="w"> </span>R2.fq<span class="w"> </span>-S<span class="w"> </span>bowtie2.sam
</span><span id="__span-12-7"><a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a>
</span><span id="__span-12-8"><a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a><span class="c1"># Method 3: CoverM (batch processing, multiple metrics)</span>
</span><span id="__span-12-9"><a id="__codelineno-12-9" name="__codelineno-12-9" href="#__codelineno-12-9"></a>coverm<span class="w"> </span>contig<span class="w"> </span>--coupled<span class="w"> </span>*_R1.fq<span class="w"> </span>*_R2.fq<span class="w"> </span>--reference<span class="w"> </span>viral.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-12-10"><a id="__codelineno-12-10" name="__codelineno-12-10" href="#__codelineno-12-10"></a><span class="w">    </span>--methods<span class="w"> </span>mean<span class="w"> </span>trimmed_mean<span class="w"> </span>covered_fraction<span class="w"> </span>variance<span class="w"> </span><span class="se">\</span>
</span><span id="__span-12-11"><a id="__codelineno-12-11" name="__codelineno-12-11" href="#__codelineno-12-11"></a><span class="w">    </span>--output-file<span class="w"> </span>coverm_abundance.tsv
</span><span id="__span-12-12"><a id="__codelineno-12-12" name="__codelineno-12-12" href="#__codelineno-12-12"></a>
</span><span id="__span-12-13"><a id="__codelineno-12-13" name="__codelineno-12-13" href="#__codelineno-12-13"></a><span class="c1"># Recommendation: Use CoverM for multi-sample studies</span>
</span></code></pre></div>
<p><strong>Coverage metrics comparison:</strong></p>
<table>
<thead>
<tr>
<th>Metric</th>
<th>Robust to Outliers</th>
<th>Good for Low Coverage</th>
<th>Best Use</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Mean</strong></td>
<td>No</td>
<td>Yes</td>
<td>Even coverage, high depth</td>
</tr>
<tr>
<td><strong>Trimmed Mean</strong></td>
<td>Yes</td>
<td>Yes</td>
<td><strong>Recommended for most cases</strong></td>
</tr>
<tr>
<td><strong>Median</strong></td>
<td>Yes</td>
<td>No</td>
<td>Very uneven coverage</td>
</tr>
<tr>
<td><strong>RPKM</strong></td>
<td>No</td>
<td>No</td>
<td>Cross-sample comparison</td>
</tr>
<tr>
<td><strong>Covered Fraction</strong></td>
<td>N/A</td>
<td>Yes</td>
<td>Presence/absence</td>
</tr>
</tbody>
</table>
<h2 id="statistical-analysis-combinations">Statistical Analysis Combinations</h2>
<h3 id="diversity-analysis">Diversity Analysis</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="c1"># R script combining multiple diversity metrics</span>
</span></code></pre></div>
<div class="language-R highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">vegan</span><span class="p">)</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">phyloseq</span><span class="p">)</span>
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a>
</span><span id="__span-14-4"><a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="c1"># Alpha diversity (within-sample)</span>
</span><span id="__span-14-5"><a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="n">richness</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">specnumber</span><span class="p">(</span><span class="n">abundance</span><span class="p">)</span><span class="w">  </span><span class="c1"># Species richness</span>
</span><span id="__span-14-6"><a id="__codelineno-14-6" name="__codelineno-14-6" href="#__codelineno-14-6"></a><span class="n">shannon</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">diversity</span><span class="p">(</span><span class="n">abundance</span><span class="p">,</span><span class="w"> </span><span class="n">index</span><span class="o">=</span><span class="s">&quot;shannon&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1"># Shannon</span>
</span><span id="__span-14-7"><a id="__codelineno-14-7" name="__codelineno-14-7" href="#__codelineno-14-7"></a><span class="n">simpson</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">diversity</span><span class="p">(</span><span class="n">abundance</span><span class="p">,</span><span class="w"> </span><span class="n">index</span><span class="o">=</span><span class="s">&quot;simpson&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1"># Simpson</span>
</span><span id="__span-14-8"><a id="__codelineno-14-8" name="__codelineno-14-8" href="#__codelineno-14-8"></a>
</span><span id="__span-14-9"><a id="__codelineno-14-9" name="__codelineno-14-9" href="#__codelineno-14-9"></a><span class="c1"># Beta diversity (between-sample)</span>
</span><span id="__span-14-10"><a id="__codelineno-14-10" name="__codelineno-14-10" href="#__codelineno-14-10"></a><span class="n">bray</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vegdist</span><span class="p">(</span><span class="n">abundance</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s">&quot;bray&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1"># Bray-Curtis</span>
</span><span id="__span-14-11"><a id="__codelineno-14-11" name="__codelineno-14-11" href="#__codelineno-14-11"></a><span class="n">jaccard</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vegdist</span><span class="p">(</span><span class="n">abundance</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s">&quot;jaccard&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">binary</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">  </span><span class="c1"># Jaccard</span>
</span><span id="__span-14-12"><a id="__codelineno-14-12" name="__codelineno-14-12" href="#__codelineno-14-12"></a>
</span><span id="__span-14-13"><a id="__codelineno-14-13" name="__codelineno-14-13" href="#__codelineno-14-13"></a><span class="c1"># Ordination</span>
</span><span id="__span-14-14"><a id="__codelineno-14-14" name="__codelineno-14-14" href="#__codelineno-14-14"></a><span class="n">nmds_bray</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">metaMDS</span><span class="p">(</span><span class="n">abundance</span><span class="p">,</span><span class="w"> </span><span class="n">distance</span><span class="o">=</span><span class="s">&quot;bray&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">k</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>
</span><span id="__span-14-15"><a id="__codelineno-14-15" name="__codelineno-14-15" href="#__codelineno-14-15"></a><span class="n">pca</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rda</span><span class="p">(</span><span class="n">abundance</span><span class="p">)</span>
</span><span id="__span-14-16"><a id="__codelineno-14-16" name="__codelineno-14-16" href="#__codelineno-14-16"></a>
</span><span id="__span-14-17"><a id="__codelineno-14-17" name="__codelineno-14-17" href="#__codelineno-14-17"></a><span class="c1"># Statistical tests</span>
</span><span id="__span-14-18"><a id="__codelineno-14-18" name="__codelineno-14-18" href="#__codelineno-14-18"></a><span class="c1"># PERMANOVA for group differences</span>
</span><span id="__span-14-19"><a id="__codelineno-14-19" name="__codelineno-14-19" href="#__codelineno-14-19"></a><span class="nf">adonis2</span><span class="p">(</span><span class="n">abundance</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Treatment</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">metadata</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s">&quot;bray&quot;</span><span class="p">)</span>
</span><span id="__span-14-20"><a id="__codelineno-14-20" name="__codelineno-14-20" href="#__codelineno-14-20"></a>
</span><span id="__span-14-21"><a id="__codelineno-14-21" name="__codelineno-14-21" href="#__codelineno-14-21"></a><span class="c1"># ANOSIM (alternative)</span>
</span><span id="__span-14-22"><a id="__codelineno-14-22" name="__codelineno-14-22" href="#__codelineno-14-22"></a><span class="nf">anosim</span><span class="p">(</span><span class="n">abundance</span><span class="p">,</span><span class="w"> </span><span class="n">metadata</span><span class="o">$</span><span class="n">Treatment</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s">&quot;bray&quot;</span><span class="p">)</span>
</span><span id="__span-14-23"><a id="__codelineno-14-23" name="__codelineno-14-23" href="#__codelineno-14-23"></a>
</span><span id="__span-14-24"><a id="__codelineno-14-24" name="__codelineno-14-24" href="#__codelineno-14-24"></a><span class="c1"># Recommendation: Use PERMANOVA (more powerful)</span>
</span></code></pre></div>
<h3 id="differential-abundance">Differential Abundance</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="c1"># Use DESeq2 for count data (recommended)</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="c1"># Use ANCOM for compositional data (alternative)</span>
</span></code></pre></div>
<div class="language-R highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span>
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">ALDEx2</span><span class="p">)</span>
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="c1"># Method 1: DESeq2 (recommended for most cases)</span>
</span><span id="__span-16-5"><a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeqDataSetFromMatrix</span><span class="p">(</span>
</span><span id="__span-16-6"><a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a><span class="w">    </span><span class="n">countData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">counts</span><span class="p">,</span>
</span><span id="__span-16-7"><a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a><span class="w">    </span><span class="n">colData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">metadata</span><span class="p">,</span>
</span><span id="__span-16-8"><a id="__codelineno-16-8" name="__codelineno-16-8" href="#__codelineno-16-8"></a><span class="w">    </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Treatment</span>
</span><span id="__span-16-9"><a id="__codelineno-16-9" name="__codelineno-16-9" href="#__codelineno-16-9"></a><span class="p">)</span>
</span><span id="__span-16-10"><a id="__codelineno-16-10" name="__codelineno-16-10" href="#__codelineno-16-10"></a><span class="n">dds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">DESeq</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
</span><span id="__span-16-11"><a id="__codelineno-16-11" name="__codelineno-16-11" href="#__codelineno-16-11"></a><span class="n">results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">results</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Treatment&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;A&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;B&quot;</span><span class="p">))</span>
</span><span id="__span-16-12"><a id="__codelineno-16-12" name="__codelineno-16-12" href="#__codelineno-16-12"></a>
</span><span id="__span-16-13"><a id="__codelineno-16-13" name="__codelineno-16-13" href="#__codelineno-16-13"></a><span class="c1"># Method 2: ALDEx2 (for compositional data)</span>
</span><span id="__span-16-14"><a id="__codelineno-16-14" name="__codelineno-16-14" href="#__codelineno-16-14"></a><span class="n">aldex_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">aldex</span><span class="p">(</span><span class="n">counts</span><span class="p">,</span><span class="w"> </span><span class="n">metadata</span><span class="o">$</span><span class="n">Treatment</span><span class="p">,</span><span class="w"> </span><span class="n">mc.samples</span><span class="o">=</span><span class="m">128</span><span class="p">)</span>
</span><span id="__span-16-15"><a id="__codelineno-16-15" name="__codelineno-16-15" href="#__codelineno-16-15"></a>
</span><span id="__span-16-16"><a id="__codelineno-16-16" name="__codelineno-16-16" href="#__codelineno-16-16"></a><span class="c1"># Recommendation:</span>
</span><span id="__span-16-17"><a id="__codelineno-16-17" name="__codelineno-16-17" href="#__codelineno-16-17"></a><span class="c1"># - DESeq2 for most virome studies</span>
</span><span id="__span-16-18"><a id="__codelineno-16-18" name="__codelineno-16-18" href="#__codelineno-16-18"></a><span class="c1"># - ALDEx2 if concerned about compositionality</span>
</span><span id="__span-16-19"><a id="__codelineno-16-19" name="__codelineno-16-19" href="#__codelineno-16-19"></a><span class="c1"># - Validate with both if results differ substantially</span>
</span></code></pre></div>
<h2 id="workflow-integration-examples">Workflow Integration Examples</h2>
<h3 id="complete-virome-workflow-recommended-stack">Complete Virome Workflow (Recommended Stack)</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="c1"># Complete virome analysis combining best practices</span>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a>
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="c1"># 1. QC</span>
</span><span id="__span-17-5"><a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a>fastp<span class="w"> </span>-i<span class="w"> </span>R1.fq.gz<span class="w"> </span>-I<span class="w"> </span>R2.fq.gz<span class="w"> </span>-o<span class="w"> </span>clean_R1.fq<span class="w"> </span>-O<span class="w"> </span>clean_R2.fq
</span><span id="__span-17-6"><a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a>
</span><span id="__span-17-7"><a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a><span class="c1"># 2. Assembly (hybrid approach)</span>
</span><span id="__span-17-8"><a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a>metaspades.py<span class="w"> </span>--metaviral<span class="w"> </span>-1<span class="w"> </span>clean_R1.fq<span class="w"> </span>-2<span class="w"> </span>clean_R2.fq<span class="w"> </span>-o<span class="w"> </span>assembly/
</span><span id="__span-17-9"><a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a>
</span><span id="__span-17-10"><a id="__codelineno-17-10" name="__codelineno-17-10" href="#__codelineno-17-10"></a><span class="c1"># 3. Viral ID (consensus of 3 tools)</span>
</span><span id="__span-17-11"><a id="__codelineno-17-11" name="__codelineno-17-11" href="#__codelineno-17-11"></a>virsorter<span class="w"> </span>run<span class="w"> </span>-i<span class="w"> </span>assembly/contigs.fa<span class="w"> </span>--min-score<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>-w<span class="w"> </span>vs2/<span class="w"> </span>all
</span><span id="__span-17-12"><a id="__codelineno-17-12" name="__codelineno-17-12" href="#__codelineno-17-12"></a>VIBRANT_run.py<span class="w"> </span>-i<span class="w"> </span>assembly/contigs.fa<span class="w"> </span>-folder<span class="w"> </span>vibrant/
</span><span id="__span-17-13"><a id="__codelineno-17-13" name="__codelineno-17-13" href="#__codelineno-17-13"></a>genomad<span class="w"> </span>end-to-end<span class="w"> </span>assembly/contigs.fa<span class="w"> </span>genomad/<span class="w"> </span>genomad_db/
</span><span id="__span-17-14"><a id="__codelineno-17-14" name="__codelineno-17-14" href="#__codelineno-17-14"></a>
</span><span id="__span-17-15"><a id="__codelineno-17-15" name="__codelineno-17-15" href="#__codelineno-17-15"></a><span class="c1"># Consensus</span>
</span><span id="__span-17-16"><a id="__codelineno-17-16" name="__codelineno-17-16" href="#__codelineno-17-16"></a>python3<span class="w"> </span>consensus_viral_prediction.py<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-17"><a id="__codelineno-17-17" name="__codelineno-17-17" href="#__codelineno-17-17"></a><span class="w">    </span>--vs2<span class="w"> </span>vs2/final-viral-combined.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-18"><a id="__codelineno-17-18" name="__codelineno-17-18" href="#__codelineno-17-18"></a><span class="w">    </span>--vibrant<span class="w"> </span>vibrant/phages.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-19"><a id="__codelineno-17-19" name="__codelineno-17-19" href="#__codelineno-17-19"></a><span class="w">    </span>--genomad<span class="w"> </span>genomad/viral.fna<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-20"><a id="__codelineno-17-20" name="__codelineno-17-20" href="#__codelineno-17-20"></a><span class="w">    </span>--min-tools<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-21"><a id="__codelineno-17-21" name="__codelineno-17-21" href="#__codelineno-17-21"></a><span class="w">    </span>--output<span class="w"> </span>consensus_viral.fa
</span><span id="__span-17-22"><a id="__codelineno-17-22" name="__codelineno-17-22" href="#__codelineno-17-22"></a>
</span><span id="__span-17-23"><a id="__codelineno-17-23" name="__codelineno-17-23" href="#__codelineno-17-23"></a><span class="c1"># 4. Quality check</span>
</span><span id="__span-17-24"><a id="__codelineno-17-24" name="__codelineno-17-24" href="#__codelineno-17-24"></a>checkv<span class="w"> </span>end_to_end<span class="w"> </span>consensus_viral.fa<span class="w"> </span>checkv/<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span>
</span><span id="__span-17-25"><a id="__codelineno-17-25" name="__codelineno-17-25" href="#__codelineno-17-25"></a>
</span><span id="__span-17-26"><a id="__codelineno-17-26" name="__codelineno-17-26" href="#__codelineno-17-26"></a><span class="c1"># 5. Annotation</span>
</span><span id="__span-17-27"><a id="__codelineno-17-27" name="__codelineno-17-27" href="#__codelineno-17-27"></a>pharokka.py<span class="w"> </span>-i<span class="w"> </span>checkv/viruses.fna<span class="w"> </span>-o<span class="w"> </span>pharokka/<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span>
</span><span id="__span-17-28"><a id="__codelineno-17-28" name="__codelineno-17-28" href="#__codelineno-17-28"></a>DRAM-v.py<span class="w"> </span>annotate<span class="w"> </span>-i<span class="w"> </span>checkv/viruses.fna<span class="w"> </span>-o<span class="w"> </span>dramv/<span class="w"> </span>--threads<span class="w"> </span><span class="m">8</span>
</span><span id="__span-17-29"><a id="__codelineno-17-29" name="__codelineno-17-29" href="#__codelineno-17-29"></a>
</span><span id="__span-17-30"><a id="__codelineno-17-30" name="__codelineno-17-30" href="#__codelineno-17-30"></a><span class="c1"># 6. Taxonomy</span>
</span><span id="__span-17-31"><a id="__codelineno-17-31" name="__codelineno-17-31" href="#__codelineno-17-31"></a>blastn<span class="w"> </span>-query<span class="w"> </span>checkv/viruses.fna<span class="w"> </span>-db<span class="w"> </span>nt<span class="w"> </span>-outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>&gt;<span class="w"> </span>blast_tax.txt
</span><span id="__span-17-32"><a id="__codelineno-17-32" name="__codelineno-17-32" href="#__codelineno-17-32"></a>vcontact2<span class="w"> </span>--raw-proteins<span class="w"> </span>pharokka/proteins.faa<span class="w"> </span>--output<span class="w"> </span>vcontact2/
</span><span id="__span-17-33"><a id="__codelineno-17-33" name="__codelineno-17-33" href="#__codelineno-17-33"></a>
</span><span id="__span-17-34"><a id="__codelineno-17-34" name="__codelineno-17-34" href="#__codelineno-17-34"></a><span class="c1"># 7. Abundance</span>
</span><span id="__span-17-35"><a id="__codelineno-17-35" name="__codelineno-17-35" href="#__codelineno-17-35"></a>coverm<span class="w"> </span>contig<span class="w"> </span>--coupled<span class="w"> </span>clean_R*.fq<span class="w"> </span>--reference<span class="w"> </span>checkv/viruses.fna<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-36"><a id="__codelineno-17-36" name="__codelineno-17-36" href="#__codelineno-17-36"></a><span class="w">    </span>--methods<span class="w"> </span>trimmed_mean<span class="w"> </span>--output-file<span class="w"> </span>abundance.tsv
</span><span id="__span-17-37"><a id="__codelineno-17-37" name="__codelineno-17-37" href="#__codelineno-17-37"></a>
</span><span id="__span-17-38"><a id="__codelineno-17-38" name="__codelineno-17-38" href="#__codelineno-17-38"></a><span class="c1"># 8. Host prediction</span>
</span><span id="__span-17-39"><a id="__codelineno-17-39" name="__codelineno-17-39" href="#__codelineno-17-39"></a>iphop<span class="w"> </span>predict<span class="w"> </span>--fa_file<span class="w"> </span>checkv/viruses.fna<span class="w"> </span>--db_dir<span class="w"> </span>iphop_db/<span class="w"> </span>--out_dir<span class="w"> </span>iphop/
</span></code></pre></div>
<h2 id="tool-compatibility-matrix">Tool Compatibility Matrix</h2>
<table>
<thead>
<tr>
<th>Upstream Tool</th>
<th>Compatible Downstream Tools</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>metaviralSPAdes</strong></td>
<td>VirSorter2, VIBRANT, geNomad</td>
<td>All viral ID tools</td>
</tr>
<tr>
<td><strong>VirSorter2</strong></td>
<td>CheckV, Pharokka, DRAMv</td>
<td>Standard workflow</td>
</tr>
<tr>
<td><strong>VIBRANT</strong></td>
<td>CheckV, DRAMv</td>
<td>Built-in annotation</td>
</tr>
<tr>
<td><strong>CheckV</strong></td>
<td>Pharokka, DRAMv, vConTACT2</td>
<td>Use <code>viruses.fna</code> output</td>
</tr>
<tr>
<td><strong>Pharokka</strong></td>
<td>vConTACT2, DRAMv</td>
<td>Use protein FAA</td>
</tr>
<tr>
<td><strong>DRAMv</strong></td>
<td>Custom scripts</td>
<td>Metabolic analysis</td>
</tr>
</tbody>
</table>
<h2 id="common-pitfalls-in-tool-combinations">Common Pitfalls in Tool Combinations</h2>
<h3 id="anti-patterns-avoid-these">❌ Anti-Patterns (Avoid These)</h3>
<ol>
<li><strong>Using only one viral ID tool</strong></li>
<li>False positive rate too high (10-30%)</li>
<li>
<p>Solution: Always use ≥2 tools</p>
</li>
<li>
<p><strong>Not running CheckV after viral ID</strong></p>
</li>
<li>Can't assess quality or remove contamination</li>
<li>
<p>Solution: Always run CheckV</p>
</li>
<li>
<p><strong>Over-relying on machine learning tools</strong></p>
</li>
<li>Need validation with sequence-based methods</li>
<li>
<p>Solution: Combine ML with BLAST/CRISPR</p>
</li>
<li>
<p><strong>Ignoring tool version differences</strong></p>
</li>
<li>Databases and algorithms change</li>
<li>
<p>Solution: Record versions, use same version within study</p>
</li>
<li>
<p><strong>Combining incompatible tools</strong></p>
</li>
<li>E.g., using DNA assembler for RNA viruses</li>
<li>Solution: Check tool documentation</li>
</ol>
<h3 id="best-practices">✅ Best Practices</h3>
<ol>
<li><strong>Consensus predictions</strong> (≥2 tools agree)</li>
<li><strong>CheckV filtering</strong> (remove low quality)</li>
<li><strong>Multiple evidence types</strong> (CRISPR + homology + ML)</li>
<li><strong>Version control</strong> (document all tool versions)</li>
<li><strong>Appropriate thresholds</strong> (adjust for your goals)</li>
</ol>
<h2 id="computational-resource-considerations">Computational Resource Considerations</h2>
<table>
<thead>
<tr>
<th>Tool Combination</th>
<th>RAM</th>
<th>CPUs</th>
<th>Runtime (1000 contigs)</th>
</tr>
</thead>
<tbody>
<tr>
<td>VirSorter2 + geNomad</td>
<td>32GB</td>
<td>16</td>
<td>2-3 hours</td>
</tr>
<tr>
<td>VirSorter2 + VIBRANT + geNomad</td>
<td>64GB</td>
<td>16</td>
<td>4-6 hours</td>
</tr>
<tr>
<td>Full stack (assembly + ID + annotation)</td>
<td>128GB</td>
<td>32</td>
<td>12-24 hours</td>
</tr>
</tbody>
</table>
<p><strong>Optimization tips:</strong>
- Run tools in parallel when possible
- Use high-memory nodes for assembly
- Cache databases (don't re-download)</p>
<h2 id="further-reading">Further Reading</h2>
<ul>
<li>Roux, S., et al. (2019). "Minimum Information about an Uncultivated Virus Genome (MIUViG)." <em>Nature Biotechnology</em>.</li>
<li>Camargo, A. P., et al. (2023). "Identification of mobile genetic elements with geNomad." <em>Nature Biotechnology</em>.</li>
<li>Guo, J., et al. (2021). "VirSorter2: a multi-classifier, expert-guided approach." <em>Microbiome</em>.</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <!-- Custom footer -->

<footer class="md-footer">
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; 2023-2025 <a href="https://github.com/shandley">Scott Handley</a> -  <a href="#__consent">Change cookie settings</a>
          </div>
        
        Awesome-Virome Documentation | 
        <a href="https://shandley.github.io/awesome-virome/dashboard.html" target="_blank">
          Dashboard
        </a> |
        <a href="https://shandley.github.io/awesome-virome/citations.html" target="_blank">
          Citation Analytics
        </a>
        <br>
        Made with
        <a href="https://www.mkdocs.org/" target="_blank" rel="noopener">
          MkDocs
        </a> and
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/shandley/awesome-virome" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
    </div>
  </div>
</footer>

<script>
document.addEventListener('DOMContentLoaded', function() {
  window.mermaidConfig = {
    startOnLoad: true,
    theme: 'default',
    flowchart: {
      useMaxWidth: false,
      htmlLabels: true,
      curve: 'basis',
      defaultRenderer: 'elk',
      width: 250
    },
    securityLevel: 'loose'
  };
  
  function initMermaid() {
    try {
      if (typeof window.mermaid !== 'undefined') {
        window.mermaid.initialize(window.mermaidConfig);
        
        // Process all mermaid diagrams that haven't been processed yet
        document.querySelectorAll('.mermaid:not([data-processed="true"])').forEach(function(el) {
          try {
            window.mermaid.init(undefined, el);
          } catch (err) {
            console.error('Mermaid initialization error for specific element:', err);
            // Add an error class to show there was an issue
            el.classList.add('mermaid-error');
            el.textContent = 'Diagram rendering error. Please check console for details.';
          }
        });
      }
    } catch (err) {
      console.error('Mermaid initialization error:', err);
    }
  }
  
  // Try to use mermaid if it's already loaded
  if (typeof window.mermaid !== 'undefined') {
    initMermaid();
  } else {
    // Load mermaid if it's not available
    console.log('Loading mermaid from mermaid.html partial');
    var script = document.createElement('script');
    script.src = 'https://cdn.jsdelivr.net/npm/mermaid@9.4.3/dist/mermaid.min.js';
    script.onload = function() {
      console.log('Mermaid loaded successfully');
      initMermaid();
    };
    script.onerror = function() {
      console.error('Failed to load mermaid script');
    };
    document.head.appendChild(script);
  }
  
  // Secondary initialization after a delay to catch any missed diagrams
  setTimeout(initMermaid, 1000);
});

// Backup initialization on full page load
window.addEventListener('load', function() {
  if (typeof window.mermaid !== 'undefined') {
    console.log('Window loaded, reinitializing mermaid');
    window.mermaid.initialize({
      startOnLoad: true,
      theme: 'default',
      flowchart: {
        useMaxWidth: false,
        htmlLabels: true,
        curve: 'basis',
        defaultRenderer: 'elk',
        width: 250
      },
      securityLevel: 'loose'
    });
    
    // Process any diagrams that still haven't been processed
    document.querySelectorAll('.mermaid:not([data-processed="true"])').forEach(function(el) {
      try {
        window.mermaid.init(undefined, el);
      } catch (err) {
        console.error('Mermaid load-time initialization error:', err);
      }
    });
  }
});
</script>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.indexes", "navigation.top", "navigation.footer", "content.tabs.link", "content.code.copy", "content.code.annotate", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.7a47a382.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.e71a0d61.min.js"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mermaid@9.4.3/dist/mermaid.min.js"></script>
      
        <script src="../../js/mermaid.js"></script>
      
    
  </body>
</html>