# Academic Citations for Awesome-Virome Tools

This document contains academic citations for tools in the Awesome-Virome collection. These citations can be used to properly acknowledge the tool authors in academic publications.

## Table of Contents

- [AlphaFold-Multimer](#alphafold-multimer)
- [AMRFinder](#amrfinder)
- [BACPHLIP](#bacphlip)
- [BERTax](#bertax)
- [BLAST+DIAMOND](#blast+diamond)
- [BVBRC](#bvbrc)
- [Cenote-Taker 2](#cenote-taker-2)
- [CHARMM-GUI](#charmm-gui)
- [CheckV](#checkv)
- [CHERRY](#cherry)
- [CHVD](#chvd)
- [CliqueSNV](#cliquesnv)
- [COBRA](#cobra)
- [CoCoNet](#coconet)
- [coronaSPAdes](#coronaspades)
- [crassus](#crassus)
- [DBSCAN-SWA](#dbscan-swa)
- [Deep6](#deep6)
- [DeePaC](#deepac)
- [DeepHost](#deephost)
- [DeepVirFinder](#deepvirfinder)
- [DePhT](#depht)
- [DePP](#depp)
- [DRAD](#drad)
- [DRAMv](#dramv)
- [efam](#efam)
- [EpiDOCK](#epidock)
- [EVE](#eve)
- [FastViromeExplorer](#fastviromeexplorer)
- [FAVITES](#favites)
- [FDA ARGOS](#fda-argos)
- [GeneMarkS](#genemarks)
- [GeneMarkS-2](#genemarks-2)
- [geNomad](#genomad)
- [GenomePeek](#genomepeek)
- [GLIMMER](#glimmer)
- [GlyConnect](#glyconnect)
- [GPDB](#gpdb)
- [GraViTy](#gravity)
- [GVD](#gvd)
- [hafeZ](#hafez)
- [hecatomb](#hecatomb)
- [HNADOCK](#hnadock)
- [HoloVir](#holovir)
- [HOPS](#hops)
- [HostG](#hostg)
- [HostPhinder](#hostphinder)
- [I-TASSER](#i-tasser)
- [IDseq](#idseq)
- [IEDB-tools](#iedb-tools)
- [INHERIT](#inherit)
- [INSaFLU-TELEVIR](#insaflu-televir)
- [iPHoP](#iphop)
- [isling](#isling)
- [Jaeger](#jaeger)
- [Jovian](#jovian)
- [LazyPipe](#lazypipe)
- [MARVEL](#marvel)
- [MAVERIC](#maveric)
- [MetaCerberus](#metacerberus)
- [metaPhage](#metaphage)
- [MetaPhinder](#metaphinder)
- [MetaProdigal](#metaprodigal)
- [metaviralSPAdes](#metaviralspades)
- [mMGE](#mmge)
- [MVP](#mvp)
- [NCBI GenBank](#ncbi-genbank)
- [NCBI RefSeq](#ncbi-refseq)
- [NetMHCpan](#netmhcpan)
- [NetNGlyc](#netnglyc)
- [OLGenie](#olgenie)
- [Paleovirology](#paleovirology)
- [PastML](#pastml)
- [PB-LKS](#pb-lks)
- [pGlyco](#pglyco)
- [Phables](#phables)
- [PhaBox](#phabox)
- [PHACTS](#phacts)
- [PhaGCN](#phagcn)
- [PhageAI](#phageai)
- [PhageClouds](#phageclouds)
- [PhageDPO](#phagedpo)
- [PhageHostLearn](#phagehostlearn)
- [PhagePromoter](#phagepromoter)
- [PhagesDB](#phagesdb)
- [PhageTB](#phagetb)
- [PhageTerm](#phageterm)
- [PhageTerm](#phageterm)
- [PhageWeb](#phageweb)
- [phaMers](#phamers)
- [PhANNs](#phanns)
- [PHANOTATE](#phanotate)
- [Phanotate](#phanotate)
- [Phanta](#phanta)
- [Pharokka](#pharokka)
- [PHASTER](#phaster)
- [PHERI](#pheri)
- [PHIAF](#phiaf)
- [Phigaro](#phigaro)
- [PHISDetector](#phisdetector)
- [PhiSpy](#phispy)
- [PHIST](#phist)
- [PHP](#php)
- [PHPGCA](#phpgca)
- [PHRED](#phred)
- [PHROGs](#phrogs)
- [PIGv](#pigv)
- [PPR-Meta](#ppr-meta)
- [PredPHI](#predphi)
- [Prodigal](#prodigal)
- [Prophage Tracer](#prophage-tracer)
- [Prophet](#prophet)
- [Prophinder](#prophinder)
- [pVOGs](#pvogs)
- [QuRe](#qure)
- [RaFaH](#rafah)
- [RdRp-scan](#rdrp-scan)
- [Recentrifuge](#recentrifuge)
- [ResFinder](#resfinder)
- [Seeker](#seeker)
- [Serratus](#serratus)
- [ShoRAH](#shorah)
- [SNPGenie](#snpgenie)
- [SpacePHARER](#spacepharer)
- [SpikeHunter](#spikehunter)
- [TepiTool](#tepitool)
- [TrIdent](#trident)
- [V-pipe](#v-pipe)
- [vAMPirus](#vampirus)
- [VAPiD](#vapid)
- [VCFgenie](#vcfgenie)
- [vConTACT](#vcontact)
- [VEGA](#vega)
- [VFM](#vfm)
- [VGAS](#vgas)
- [vHulk](#vhulk)
- [VIBRANT](#vibrant)
- [VICTOR](#victor)
- [VIDHOP](#vidhop)
- [VIGA](#viga)
- [VIGOR](#vigor)
- [VIP](#vip)
- [VIPERA](#vipera)
- [VIPERdb](#viperdb)
- [VIPtree](#viptree)
- [ViQuaS](#viquas)
- [Viral-NGS](#viral-ngs)
- [Viral-Track](#viral-track)
- [ViralCC](#viralcc)
- [viralComplete](#viralcomplete)
- [ViralConsensus](#viralconsensus)
- [ViralMSA](#viralmsa)
- [VIRALpro](#viralpro)
- [viralVerify](#viralverify)
- [ViralWasm](#viralwasm)
- [viraMiner](#viraminer)
- [VirFinder](#virfinder)
- [VirHostMatcher](#virhostmatcher)
- [VirHostMatcher-Net](#virhostmatcher-net)
- [Virhunter](#virhunter)
- [VIRIDIC](#viridic)
- [VIRify](#virify)
- [Virion](#virion)
- [VirMet](#virmet)
- [VirMine](#virmine)
- [virMiner](#virminer)
- [VIRMOTIF](#virmotif)
- [ViromeQC](#viromeqc)
- [ViroProfiler](#viroprofiler)
- [Virosaurus](#virosaurus)
- [VirSorter](#virsorter)
- [VirSorter2](#virsorter2)
- [Virtifier](#virtifier)
- [Virtus](#virtus)
- [virus_prediction](#virus_prediction)
- [Viruses.String](#viruses.string)
- [VirusSeeker](#virusseeker)
- [VirusTAP](#virustap)
- [VirusTaxo](#virustaxo)
- [VogDB](#vogdb)
- [vRhyme](#vrhyme)
- [WIsH](#wish)


## AlphaFold-Multimer

**Tool URL**: [https://github.com/deepmind/alphafold](https://github.com/deepmind/alphafold)

### Publication Information

**Title**: AlphaFold-Multimer predicts cross-kingdom interactions at the plant-pathogen interface.

**Journal**: Nature communications

**Year**: 2023

**Identifiers**:
- DOI: [10.1038/s41467-023-41721-9](https://doi.org/10.1038/s41467-023-41721-9)
- PMID: [37758696](https://pubmed.ncbi.nlm.nih.gov/37758696/)

### Cite this tool

**APA format**:

```
F, H., J, H., & RAL, v.d.H. (2023). AlphaFold-Multimer predicts cross-kingdom interactions at the plant-pathogen interface. Nature communications, 14(1), 6040. https://doi.org/10.1038/s41467-023-41721-9
```

**BibTeX format**:

```bibtex
@article{f2023,
  title = {AlphaFold-Multimer predicts cross-kingdom interactions at the plant-pathogen interface},
  author = {F, Homma and J, Huang and RAL, van der Hoorn},
  year = {2023},
  journal = {Nature communications},
  volume = {14},
  number = {1},
  pages = {6040},
  doi = {10.1038/s41467-023-41721-9},
  pmid = {37758696},
}
```


## AMRFinder

**Tool URL**: [https://github.com/ncbi/amr](https://github.com/ncbi/amr)

### Publication Information

**Title**: Validating the AMRFinder Tool and Resistance Gene Database by Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of Isolates.

**Journal**: Antimicrobial agents and chemotherapy

**Year**: 2019

**Identifiers**:
- DOI: [10.1128/AAC.00483-19](https://doi.org/10.1128/AAC.00483-19)
- PMID: [31427293](https://pubmed.ncbi.nlm.nih.gov/31427293/)

### Cite this tool

**APA format**:

```
M, F., V, B., DH, H., AB, P., DJ, S., I, T., ... W, K. (2019). Validating the AMRFinder Tool and Resistance Gene Database by Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of Isolates. Antimicrobial agents and chemotherapy, 63(11). https://doi.org/10.1128/AAC.00483-19
```

**BibTeX format**:

```bibtex
@article{m2019,
  title = {Validating the AMRFinder Tool and Resistance Gene Database by Using Antimicrobial Resistance Genotype-Phenotype Correlations in a Collection of Isolates},
  author = {M, Feldgarden and V, Brover and DH, Haft and AB, Prasad and DJ, Slotta and I, Tolstoy and GH, Tyson and S, Zhao and CH, Hsu and PF, McDermott and DA, Tadesse and C, Morales and M, Simmons and G, Tillman and J, Wasilenko and JP, Folster and W, Klimke},
  year = {2019},
  journal = {Antimicrobial agents and chemotherapy},
  volume = {63},
  number = {11},
  doi = {10.1128/AAC.00483-19},
  pmid = {31427293},
}
```


## BACPHLIP

**Tool URL**: [https://github.com/adamhockenberry/bacphlip](https://github.com/adamhockenberry/bacphlip)

### Publication Information

**Title**: BACPHLIP: predicting bacteriophage lifestyle from conserved protein domains.

**Journal**: PeerJ

**Year**: 2021

**Identifiers**:
- DOI: [10.7717/peerj.11396](https://doi.org/10.7717/peerj.11396)
- PMID: [33996289](https://pubmed.ncbi.nlm.nih.gov/33996289/)

### Cite this tool

**APA format**:

```
AJ, H., & CO, W. (2021). BACPHLIP: predicting bacteriophage lifestyle from conserved protein domains. PeerJ, 9, e11396. https://doi.org/10.7717/peerj.11396
```

**BibTeX format**:

```bibtex
@article{aj2021,
  title = {BACPHLIP: predicting bacteriophage lifestyle from conserved protein domains},
  author = {AJ, Hockenberry and CO, Wilke},
  year = {2021},
  journal = {PeerJ},
  volume = {9},
  pages = {e11396},
  doi = {10.7717/peerj.11396},
  pmid = {33996289},
}
```


## BERTax

**Tool URL**: [https://github.com/f-kretschmer/bertax](https://github.com/f-kretschmer/bertax)

### Publication Information

**Title**: Improving taxonomic classification with feature space balancing.

**Journal**: Bioinformatics advances

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/bioadv/vbad092](https://doi.org/10.1093/bioadv/vbad092)
- PMID: [37577265](https://pubmed.ncbi.nlm.nih.gov/37577265/)

### Cite this tool

**APA format**:

```
W, F., S, Z., & K, N. (2023). Improving taxonomic classification with feature space balancing. Bioinformatics advances, 3(1), vbad092. https://doi.org/10.1093/bioadv/vbad092
```

**BibTeX format**:

```bibtex
@article{w2023,
  title = {Improving taxonomic classification with feature space balancing},
  author = {W, Fuhl and S, Zabel and K, Nieselt},
  year = {2023},
  journal = {Bioinformatics advances},
  volume = {3},
  number = {1},
  pages = {vbad092},
  doi = {10.1093/bioadv/vbad092},
  pmid = {37577265},
}
```


## BLAST+DIAMOND

**Tool URL**: [https://github.com/bbuchfink/diamond](https://github.com/bbuchfink/diamond)

### Publication Information

**Title**: MEDUSA: A Pipeline for Sensitive Taxonomic Classification and Flexible Functional Annotation of Metagenomic Shotgun Sequences.

**Journal**: Frontiers in genetics

**Year**: 2022

**Identifiers**:
- DOI: [10.3389/fgene.2022.814437](https://doi.org/10.3389/fgene.2022.814437)
- PMID: [35330728](https://pubmed.ncbi.nlm.nih.gov/35330728/)

### Cite this tool

**APA format**:

```
DAA, M., JVF, C., SS, M., MAB, P., & RJS, D. (2022). MEDUSA: A Pipeline for Sensitive Taxonomic Classification and Flexible Functional Annotation of Metagenomic Shotgun Sequences. Frontiers in genetics, 13, 814437. https://doi.org/10.3389/fgene.2022.814437
```

**BibTeX format**:

```bibtex
@article{daa2022,
  title = {MEDUSA: A Pipeline for Sensitive Taxonomic Classification and Flexible Functional Annotation of Metagenomic Shotgun Sequences},
  author = {DAA, Morais and JVF, Cavalcante and SS, Monteiro and MAB, Pasquali and RJS, Dalmolin},
  year = {2022},
  journal = {Frontiers in genetics},
  volume = {13},
  pages = {814437},
  doi = {10.3389/fgene.2022.814437},
  pmid = {35330728},
}
```


## BVBRC

**Tool URL**: [https://www.bv-brc.org/](https://www.bv-brc.org/)

### Publication Information

**Title**: Introducing the Bacterial and Viral Bioinformatics Resource Center (BV-BRC): a resource combining PATRIC, IRD and ViPR.

**Journal**: Nucleic acids research

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/nar/gkac1003](https://doi.org/10.1093/nar/gkac1003)
- PMID: [36350631](https://pubmed.ncbi.nlm.nih.gov/36350631/)

### Cite this tool

**APA format**:

```
RD, O., R, A., T, B., N, C., C, C., JJ, D., ... RL, S. (2023). Introducing the Bacterial and Viral Bioinformatics Resource Center (BV-BRC): a resource combining PATRIC, IRD and ViPR. Nucleic acids research, 51(D1), D678-D689. https://doi.org/10.1093/nar/gkac1003
```

**BibTeX format**:

```bibtex
@article{rd2023,
  title = {Introducing the Bacterial and Viral Bioinformatics Resource Center (BV-BRC): a resource combining PATRIC, IRD and ViPR},
  author = {RD, Olson and R, Assaf and T, Brettin and N, Conrad and C, Cucinell and JJ, Davis and DM, Dempsey and A, Dickerman and EM, Dietrich and RW, Kenyon and M, Kuscuoglu and EJ, Lefkowitz and J, Lu and D, Machi and C, Macken and C, Mao and A, Niewiadomska and M, Nguyen and GJ, Olsen and JC, Overbeek and B, Parrello and V, Parrello and JS, Porter and GD, Pusch and M, Shukla and I, Singh and L, Stewart and G, Tan and C, Thomas and M, VanOeffelen and V, Vonstein and ZS, Wallace and AS, Warren and AR, Wattam and F, Xia and H, Yoo and Y, Zhang and CM, Zmasek and RH, Scheuermann and RL, Stevens},
  year = {2023},
  journal = {Nucleic acids research},
  volume = {51},
  number = {D1},
  pages = {D678-D689},
  doi = {10.1093/nar/gkac1003},
  pmid = {36350631},
}
```


## Cenote-Taker 2

**Tool URL**: [https://github.com/mtisza1/Cenote-Taker2](https://github.com/mtisza1/Cenote-Taker2)

### Publication Information

**Title**: ['Cenote-Taker 2 democratizes virus discovery and sequence annotation']

### Cite this tool

**APA format**:

```
Tisza, M., Belford, A., Domínguez-Huerta, G., Bolduc, B., & Buck, C. (2020). Cenote-Taker 2 democratizes virus discovery and sequence annotation. Virus Evolution, 7(1). https://doi.org/10.1093/ve/veaa100
```

**BibTeX format**:

```bibtex
@article{tisza2020,
  title = {Cenote-Taker 2 democratizes virus discovery and sequence annotation},
  author = {Tisza, M. and Belford, A. and Domínguez-Huerta, G. and Bolduc, B. and Buck, C.},
  year = {2020},
  journal = {Virus Evolution},
  volume = {7},
  number = {1},
  doi = {10.1093/ve/veaa100},
}
```


## CHARMM-GUI

**Tool URL**: [https://www.charmm-gui.org/](https://www.charmm-gui.org/)

### Publication Information

**Title**: CHARMM-GUI Membrane Builder: Past, Current, and Future Developments and Applications.

**Journal**: Journal of chemical theory and computation

**Year**: 2023

**Identifiers**:
- DOI: [10.1021/acs.jctc.2c01246](https://doi.org/10.1021/acs.jctc.2c01246)
- PMID: [37014931](https://pubmed.ncbi.nlm.nih.gov/37014931/)

### Cite this tool

**APA format**:

```
S, F., S, P., YK, C., & W, I. (2023). CHARMM-GUI Membrane Builder: Past, Current, and Future Developments and Applications. Journal of chemical theory and computation, 19(8), 2161-2185. https://doi.org/10.1021/acs.jctc.2c01246
```

**BibTeX format**:

```bibtex
@article{s2023,
  title = {CHARMM-GUI Membrane Builder: Past, Current, and Future Developments and Applications},
  author = {S, Feng and S, Park and YK, Choi and W, Im},
  year = {2023},
  journal = {Journal of chemical theory and computation},
  volume = {19},
  number = {8},
  pages = {2161-2185},
  doi = {10.1021/acs.jctc.2c01246},
  pmid = {37014931},
}
```


## CheckV

**Tool URL**: [https://bitbucket.org/berkeleylab/checkv/](https://bitbucket.org/berkeleylab/checkv/)

### Publication Information

**Title**: CheckV assesses the quality and completeness of metagenome-assembled viral genomes.

**Journal**: Nature biotechnology

**Year**: 2021

**Identifiers**:
- DOI: [10.1038/s41587-020-00774-7](https://doi.org/10.1038/s41587-020-00774-7)
- PMID: [33349699](https://pubmed.ncbi.nlm.nih.gov/33349699/)

### Cite this tool

**APA format**:

```
S, N., AP, C., F, S., E, E., S, R., & NC, K. (2021). CheckV assesses the quality and completeness of metagenome-assembled viral genomes. Nature biotechnology, 39(5), 578-585. https://doi.org/10.1038/s41587-020-00774-7
```

**BibTeX format**:

```bibtex
@article{s2021,
  title = {CheckV assesses the quality and completeness of metagenome-assembled viral genomes},
  author = {S, Nayfach and AP, Camargo and F, Schulz and E, Eloe-Fadrosh and S, Roux and NC, Kyrpides},
  year = {2021},
  journal = {Nature biotechnology},
  volume = {39},
  number = {5},
  pages = {578-585},
  doi = {10.1038/s41587-020-00774-7},
  pmid = {33349699},
}
```


## CHERRY

**Tool URL**: [https://github.com/KennthShang/CHERRY](https://github.com/KennthShang/CHERRY)

### Publication Information

**Title**: The chest pain guidelines revisited: cherry picking from the frequentist tree.

**Journal**: Journal of nuclear cardiology : official publication of the American Society of Nuclear Cardiology

**Year**: 2023

**Identifiers**:
- DOI: [10.1007/s12350-022-03109-y](https://doi.org/10.1007/s12350-022-03109-y)
- PMID: [36258156](https://pubmed.ncbi.nlm.nih.gov/36258156/)

### Cite this tool

**APA format**:

```
TF, C. (2023). The chest pain guidelines revisited: cherry picking from the frequentist tree. Journal of nuclear cardiology : official publication of the American Society of Nuclear Cardiology, 30(1), 23-34. https://doi.org/10.1007/s12350-022-03109-y
```

**BibTeX format**:

```bibtex
@article{tf2023,
  title = {The chest pain guidelines revisited: cherry picking from the frequentist tree},
  author = {TF, Christian},
  year = {2023},
  journal = {Journal of nuclear cardiology : official publication of the American Society of Nuclear Cardiology},
  volume = {30},
  number = {1},
  pages = {23-34},
  doi = {10.1007/s12350-022-03109-y},
  pmid = {36258156},
}
```


## CHVD

**Tool URL**: [https://zenodo.org/record/4498884](https://zenodo.org/record/4498884)

### Publication Information

**Title**: Anthraquinone Emodin Inhibits Tumor Necrosis Factor Alpha-Induced Calcification of Human Aortic Valve Interstitial Cells via the NF-κB Pathway.

**Journal**: Frontiers in pharmacology

**Year**: 2018

**Identifiers**:
- DOI: [10.3389/fphar.2018.01328](https://doi.org/10.3389/fphar.2018.01328)
- PMID: [30510513](https://pubmed.ncbi.nlm.nih.gov/30510513/)

### Cite this tool

**APA format**:

```
K, X., T, Z., Y, H., Q, C., J, S., P, Z., & N, D. (2018). Anthraquinone Emodin Inhibits Tumor Necrosis Factor Alpha-Induced Calcification of Human Aortic Valve Interstitial Cells via the NF-κB Pathway. Frontiers in pharmacology, 9, 1328. https://doi.org/10.3389/fphar.2018.01328
```

**BibTeX format**:

```bibtex
@article{k2018,
  title = {Anthraquinone Emodin Inhibits Tumor Necrosis Factor Alpha-Induced Calcification of Human Aortic Valve Interstitial Cells via the NF-κB Pathway},
  author = {K, Xu and T, Zhou and Y, Huang and Q, Chi and J, Shi and P, Zhu and N, Dong},
  year = {2018},
  journal = {Frontiers in pharmacology},
  volume = {9},
  pages = {1328},
  doi = {10.3389/fphar.2018.01328},
  pmid = {30510513},
}
```


## CliqueSNV

**Tool URL**: [https://github.com/vtsyvina/CliqueSNV](https://github.com/vtsyvina/CliqueSNV)

### Publication Information

**Title**: Accurate assembly of minority viral haplotypes from next-generation sequencing through efficient noise reduction.

**Journal**: Nucleic acids research

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/nar/gkab576](https://doi.org/10.1093/nar/gkab576)
- PMID: [34214168](https://pubmed.ncbi.nlm.nih.gov/34214168/)

### Cite this tool

**APA format**:

```
S, K., V, T., A, S., A, M., A, A., T, M., ... A, Z. (2021). Accurate assembly of minority viral haplotypes from next-generation sequencing through efficient noise reduction. Nucleic acids research, 49(17), e102. https://doi.org/10.1093/nar/gkab576
```

**BibTeX format**:

```bibtex
@article{s2021,
  title = {Accurate assembly of minority viral haplotypes from next-generation sequencing through efficient noise reduction},
  author = {S, Knyazev and V, Tsyvina and A, Shankar and A, Melnyk and A, Artyomenko and T, Malygina and YB, Porozov and EM, Campbell and WM, Switzer and P, Skums and S, Mangul and A, Zelikovsky},
  year = {2021},
  journal = {Nucleic acids research},
  volume = {49},
  number = {17},
  pages = {e102},
  doi = {10.1093/nar/gkab576},
  pmid = {34214168},
}
```


## COBRA

**Tool URL**: [https://github.com/linxingchen/cobra](https://github.com/linxingchen/cobra)

### Publication Information

**Title**: Cobra Venom Cytotoxins as a Tool for Probing Mechanisms of Mitochondrial Energetics and Understanding Mitochondrial Membrane Structure.

**Journal**: Toxins

**Year**: 2024

**Identifiers**:
- DOI: [10.3390/toxins16070287](https://doi.org/10.3390/toxins16070287)
- PMID: [39057927](https://pubmed.ncbi.nlm.nih.gov/39057927/)

### Cite this tool

**APA format**:

```
ES, G., & RK, D. (2024). Cobra Venom Cytotoxins as a Tool for Probing Mechanisms of Mitochondrial Energetics and Understanding Mitochondrial Membrane Structure. Toxins, 16(7). https://doi.org/10.3390/toxins16070287
```

**BibTeX format**:

```bibtex
@article{es2024,
  title = {Cobra Venom Cytotoxins as a Tool for Probing Mechanisms of Mitochondrial Energetics and Understanding Mitochondrial Membrane Structure},
  author = {ES, Gasanoff and RK, Dagda},
  year = {2024},
  journal = {Toxins},
  volume = {16},
  number = {7},
  doi = {10.3390/toxins16070287},
  pmid = {39057927},
}
```


## CoCoNet

**Tool URL**: [https://github.com/Puumanamana/CoCoNet](https://github.com/Puumanamana/CoCoNet)

### Publication Information

**Title**: CoCoNet: an efficient deep learning tool for viral metagenome binning.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab213](https://doi.org/10.1093/bioinformatics/btab213)
- PMID: [33822891](https://pubmed.ncbi.nlm.nih.gov/33822891/)

### Cite this tool

**APA format**:

```
CG, A., OD, N., GF, S., G, P., & M, B. (2021). CoCoNet: an efficient deep learning tool for viral metagenome binning. Bioinformatics (Oxford, England), 37(18), 2803-2810. https://doi.org/10.1093/bioinformatics/btab213
```

**BibTeX format**:

```bibtex
@article{cg2021,
  title = {CoCoNet: an efficient deep learning tool for viral metagenome binning},
  author = {CG, Arisdakessian and OD, Nigro and GF, Steward and G, Poisson and M, Belcaid},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {18},
  pages = {2803-2810},
  doi = {10.1093/bioinformatics/btab213},
  pmid = {33822891},
}
```


## coronaSPAdes

**Tool URL**: [https://github.com/ablab/spades/tree/metaviral_publication](https://github.com/ablab/spades/tree/metaviral_publication)

### Publication Information

**Title**: coronaSPAdes: from biosynthetic gene clusters to RNA viral assemblies.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab597](https://doi.org/10.1093/bioinformatics/btab597)
- PMID: [34406356](https://pubmed.ncbi.nlm.nih.gov/34406356/)

### Cite this tool

**APA format**:

```
D, M., I, H., & A, K. (2021). coronaSPAdes: from biosynthetic gene clusters to RNA viral assemblies. Bioinformatics (Oxford, England), 38(1), 1-8. https://doi.org/10.1093/bioinformatics/btab597
```

**BibTeX format**:

```bibtex
@article{d2021,
  title = {coronaSPAdes: from biosynthetic gene clusters to RNA viral assemblies},
  author = {D, Meleshko and I, Hajirasouliha and A, Korobeynikov},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {38},
  number = {1},
  pages = {1-8},
  doi = {10.1093/bioinformatics/btab597},
  pmid = {34406356},
}
```


## crassus

**Tool URL**: [https://github.com/dcarrillox/CrassUS](https://github.com/dcarrillox/CrassUS)

### Publication Information

**Title**: Comparative Analysis of Proteins of Functionally Different Body Parts of the Fish Parasites Triaenophorus nodulosus and Triaenophorus crassus.

**Journal**: Acta parasitologica

**Year**: 2021

**Identifiers**:
- DOI: [10.1007/s11686-021-00384-6](https://doi.org/10.1007/s11686-021-00384-6)
- PMID: [33818717](https://pubmed.ncbi.nlm.nih.gov/33818717/)

### Cite this tool

**APA format**:

```
E, B., A, K., D, B., I, S., L, S., & I, B. (2021). Comparative Analysis of Proteins of Functionally Different Body Parts of the Fish Parasites Triaenophorus nodulosus and Triaenophorus crassus. Acta parasitologica, 66(4), 1137-1150. https://doi.org/10.1007/s11686-021-00384-6
```

**BibTeX format**:

```bibtex
@article{e2021,
  title = {Comparative Analysis of Proteins of Functionally Different Body Parts of the Fish Parasites Triaenophorus nodulosus and Triaenophorus crassus},
  author = {E, Borvinskaya and A, Kochneva and D, Bedulina and I, Sukhovskaya and L, Smirnov and I, Babkina},
  year = {2021},
  journal = {Acta parasitologica},
  volume = {66},
  number = {4},
  pages = {1137-1150},
  doi = {10.1007/s11686-021-00384-6},
  pmid = {33818717},
}
```


## DBSCAN-SWA

**Tool URL**: [https://github.com/HIT-ImmunologyLab/DBSCAN-SWA/](https://github.com/HIT-ImmunologyLab/DBSCAN-SWA/)

### Publication Information

**Title**: DBSCAN-SWA: An Integrated Tool for Rapid Prophage Detection and Annotation.

**Journal**: Frontiers in genetics

**Year**: 2022

**Identifiers**:
- DOI: [10.3389/fgene.2022.885048](https://doi.org/10.3389/fgene.2022.885048)
- PMID: [35518360](https://pubmed.ncbi.nlm.nih.gov/35518360/)

### Cite this tool

**APA format**:

```
R, G., F, Z., Y, S., H, Y., C, C., C, R., ... F, Z. (2022). DBSCAN-SWA: An Integrated Tool for Rapid Prophage Detection and Annotation. Frontiers in genetics, 13, 885048. https://doi.org/10.3389/fgene.2022.885048
```

**BibTeX format**:

```bibtex
@article{r2022,
  title = {DBSCAN-SWA: An Integrated Tool for Rapid Prophage Detection and Annotation},
  author = {R, Gan and F, Zhou and Y, Si and H, Yang and C, Chen and C, Ren and J, Wu and F, Zhang},
  year = {2022},
  journal = {Frontiers in genetics},
  volume = {13},
  pages = {885048},
  doi = {10.3389/fgene.2022.885048},
  pmid = {35518360},
}
```


## Deep6

**Tool URL**: [https://github.com/janfelix/Deep6](https://github.com/janfelix/Deep6)

### Publication Information

**Title**: Deep6: Classification of Metatranscriptomic Sequences into Cellular Empires and Viral Realms Using Deep Learning Models.

**Journal**: Microbiology resource announcements

**Year**: 2023

**Identifiers**:
- DOI: [10.1128/mra.01079-22](https://doi.org/10.1128/mra.01079-22)
- PMID: [36651740](https://pubmed.ncbi.nlm.nih.gov/36651740/)

### Cite this tool

**APA format**:

```
JF, F., CTE, K., & CA, S. (2023). Deep6: Classification of Metatranscriptomic Sequences into Cellular Empires and Viral Realms Using Deep Learning Models. Microbiology resource announcements, 12(2), e0107922. https://doi.org/10.1128/mra.01079-22
```

**BibTeX format**:

```bibtex
@article{jf2023,
  title = {Deep6: Classification of Metatranscriptomic Sequences into Cellular Empires and Viral Realms Using Deep Learning Models},
  author = {JF, Finke and CTE, Kellogg and CA, Suttle},
  year = {2023},
  journal = {Microbiology resource announcements},
  volume = {12},
  number = {2},
  pages = {e0107922},
  doi = {10.1128/mra.01079-22},
  pmid = {36651740},
}
```


## DeePaC

**Tool URL**: [https://gitlab.com/dacs-hpi/deepac](https://gitlab.com/dacs-hpi/deepac)

### Publication Information

**Title**: DeePaC: predicting pathogenic potential of novel DNA with reverse-complement neural networks.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/bioinformatics/btz541](https://doi.org/10.1093/bioinformatics/btz541)
- PMID: [31298694](https://pubmed.ncbi.nlm.nih.gov/31298694/)

### Cite this tool

**APA format**:

```
JM, B., A, S., R, R., & BY, R. (2020). DeePaC: predicting pathogenic potential of novel DNA with reverse-complement neural networks. Bioinformatics (Oxford, England), 36(1), 81-89. https://doi.org/10.1093/bioinformatics/btz541
```

**BibTeX format**:

```bibtex
@article{jm2020,
  title = {DeePaC: predicting pathogenic potential of novel DNA with reverse-complement neural networks},
  author = {JM, Bartoszewicz and A, Seidel and R, Rentzsch and BY, Renard},
  year = {2020},
  journal = {Bioinformatics (Oxford, England)},
  volume = {36},
  number = {1},
  pages = {81-89},
  doi = {10.1093/bioinformatics/btz541},
  pmid = {31298694},
}
```


## DeepHost

**Tool URL**: [https://github.com/deepomicslab/DeepHost](https://github.com/deepomicslab/DeepHost)

### Publication Information

**Title**: DeepHost: phage host prediction with convolutional neural network.

**Journal**: Briefings in bioinformatics

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/bib/bbab385](https://doi.org/10.1093/bib/bbab385)
- PMID: [34553750](https://pubmed.ncbi.nlm.nih.gov/34553750/)

### Cite this tool

**APA format**:

```
W, R., Z, X., W, J., & LI, S.C. (2022). DeepHost: phage host prediction with convolutional neural network. Briefings in bioinformatics, 23(1). https://doi.org/10.1093/bib/bbab385
```

**BibTeX format**:

```bibtex
@article{w2022,
  title = {DeepHost: phage host prediction with convolutional neural network},
  author = {W, Ruohan and Z, Xianglilan and W, Jianping and LI, Shuai Cheng},
  year = {2022},
  journal = {Briefings in bioinformatics},
  volume = {23},
  number = {1},
  doi = {10.1093/bib/bbab385},
  pmid = {34553750},
}
```


## DeepVirFinder

**Tool URL**: [https://github.com/jessieren/DeepVirFinder](https://github.com/jessieren/DeepVirFinder)

### Publication Information

**Title**: ProkBERT family: genomic language models for microbiome applications.

**Journal**: Frontiers in microbiology

**Year**: 2023

**Identifiers**:
- DOI: [10.3389/fmicb.2023.1331233](https://doi.org/10.3389/fmicb.2023.1331233)
- PMID: [38282738](https://pubmed.ncbi.nlm.nih.gov/38282738/)

### Cite this tool

**APA format**:

```
B, L., I, S., B, B., N, L., & J, J. (2023). ProkBERT family: genomic language models for microbiome applications. Frontiers in microbiology, 14, 1331233. https://doi.org/10.3389/fmicb.2023.1331233
```

**BibTeX format**:

```bibtex
@article{b2023,
  title = {ProkBERT family: genomic language models for microbiome applications},
  author = {B, Ligeti and I, Szepesi-Nagy and B, Bodnár and N, Ligeti-Nagy and J, Juhász},
  year = {2023},
  journal = {Frontiers in microbiology},
  volume = {14},
  pages = {1331233},
  doi = {10.3389/fmicb.2023.1331233},
  pmid = {38282738},
}
```


## DePhT

**Tool URL**: [https://github.com/chg60/DEPhT](https://github.com/chg60/DEPhT)

### Publication Information

**Title**: DEPhT: a novel approach for efficient prophage discovery and precise extraction.

**Journal**: Nucleic acids research

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/nar/gkac273](https://doi.org/10.1093/nar/gkac273)
- PMID: [35451479](https://pubmed.ncbi.nlm.nih.gov/35451479/)

### Cite this tool

**APA format**:

```
CH, G., L, A., AK, V., J, M., DA, R., & GF, H. (2022). DEPhT: a novel approach for efficient prophage discovery and precise extraction. Nucleic acids research, 50(13), e75. https://doi.org/10.1093/nar/gkac273
```

**BibTeX format**:

```bibtex
@article{ch2022,
  title = {DEPhT: a novel approach for efficient prophage discovery and precise extraction},
  author = {CH, Gauthier and L, Abad and AK, Venbakkam and J, Malnak and DA, Russell and GF, Hatfull},
  year = {2022},
  journal = {Nucleic acids research},
  volume = {50},
  number = {13},
  pages = {e75},
  doi = {10.1093/nar/gkac273},
  pmid = {35451479},
}
```


## DePP

**Tool URL**: [https://web.archive.org/web/20210307090418/https://timskvortsov.github.io/WebDePP/](https://web.archive.org/web/20210307090418/https://timskvortsov.github.io/WebDePP/)

### Publication Information

**Title**: DePolymerase Predictor (DePP): a machine learning tool for the targeted identification of phage depolymerases.

**Journal**: BMC bioinformatics

**Year**: 2023

**Identifiers**:
- DOI: [10.1186/s12859-023-05341-w](https://doi.org/10.1186/s12859-023-05341-w)
- PMID: [37208612](https://pubmed.ncbi.nlm.nih.gov/37208612/)

### Cite this tool

**APA format**:

```
DJ, M., & TA, S. (2023). DePolymerase Predictor (DePP): a machine learning tool for the targeted identification of phage depolymerases. BMC bioinformatics, 24(1), 208. https://doi.org/10.1186/s12859-023-05341-w
```

**BibTeX format**:

```bibtex
@article{dj2023,
  title = {DePolymerase Predictor (DePP): a machine learning tool for the targeted identification of phage depolymerases},
  author = {DJ, Magill and TA, Skvortsov},
  year = {2023},
  journal = {BMC bioinformatics},
  volume = {24},
  number = {1},
  pages = {208},
  doi = {10.1186/s12859-023-05341-w},
  pmid = {37208612},
}
```


## DRAD

**Tool URL**: [https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0001193](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0001193)

### Publication Information

**Title**: Prevalence and risk factors for diabetic retinopathy at diagnosis (DRAD) in patients recently diagnosed with type 2 diabetes (T2D) or latent autoimmune diabetes in the adult (LADA).

**Journal**: Journal of diabetes and its complications

**Year**: 2016

**Identifiers**:
- DOI: [10.1016/j.jdiacomp.2016.08.009](https://doi.org/10.1016/j.jdiacomp.2016.08.009)
- PMID: [27593902](https://pubmed.ncbi.nlm.nih.gov/27593902/)

### Cite this tool

**APA format**:

```
M, M., M, D., J, S., P, S., L, G., & C, G. (2016). Prevalence and risk factors for diabetic retinopathy at diagnosis (DRAD) in patients recently diagnosed with type 2 diabetes (T2D) or latent autoimmune diabetes in the adult (LADA). Journal of diabetes and its complications, 30(8), 1456-1461. https://doi.org/10.1016/j.jdiacomp.2016.08.009
```

**BibTeX format**:

```bibtex
@article{m2016,
  title = {Prevalence and risk factors for diabetic retinopathy at diagnosis (DRAD) in patients recently diagnosed with type 2 diabetes (T2D) or latent autoimmune diabetes in the adult (LADA)},
  author = {M, Martinell and M, Dorkhan and J, Stålhammar and P, Storm and L, Groop and C, Gustavsson},
  year = {2016},
  journal = {Journal of diabetes and its complications},
  volume = {30},
  number = {8},
  pages = {1456-1461},
  doi = {10.1016/j.jdiacomp.2016.08.009},
  pmid = {27593902},
}
```


## DRAMv

**Tool URL**: [https://github.com/WrightonLabCSU/DRAM](https://github.com/WrightonLabCSU/DRAM)

### Publication Information

**Title**: kb_DRAM: annotation and metabolic profiling of genomes with DRAM in KBase.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/bioinformatics/btad110](https://doi.org/10.1093/bioinformatics/btad110)
- PMID: [36857575](https://pubmed.ncbi.nlm.nih.gov/36857575/)

### Cite this tool

**APA format**:

```
M, S., MA, B., B, B., JP, F., RM, F., P, G., ... KC, W. (2023). kb_DRAM: annotation and metabolic profiling of genomes with DRAM in KBase. Bioinformatics (Oxford, England), 39(4). https://doi.org/10.1093/bioinformatics/btad110
```

**BibTeX format**:

```bibtex
@article{m2023,
  title = {kb_DRAM: annotation and metabolic profiling of genomes with DRAM in KBase},
  author = {M, Shaffer and MA, Borton and B, Bolduc and JP, Faria and RM, Flynn and P, Ghadermazi and JN, Edirisinghe and EM, Wood-Charlson and CS, Miller and SHJ, Chan and MB, Sullivan and CS, Henry and KC, Wrighton},
  year = {2023},
  journal = {Bioinformatics (Oxford, England)},
  volume = {39},
  number = {4},
  doi = {10.1093/bioinformatics/btad110},
  pmid = {36857575},
}
```


## efam

**Tool URL**: [https://datacommons.cyverse.org/browse/iplant/home/shared/iVirus/Zayed_efam_2020.1](https://datacommons.cyverse.org/browse/iplant/home/shared/iVirus/Zayed_efam_2020.1)

### Publication Information

**Title**: efam: an expanded, metaproteome-supported HMM profile database of viral protein families.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab451](https://doi.org/10.1093/bioinformatics/btab451)
- PMID: [34132786](https://pubmed.ncbi.nlm.nih.gov/34132786/)

### Cite this tool

**APA format**:

```
AA, Z., D, L., M, M., D, C., B, B., AC, G., ... MB, S. (2021). efam: an expanded, metaproteome-supported HMM profile database of viral protein families. Bioinformatics (Oxford, England), 37(22), 4202-4208. https://doi.org/10.1093/bioinformatics/btab451
```

**BibTeX format**:

```bibtex
@article{aa2021,
  title = {efam: an expanded, metaproteome-supported HMM profile database of viral protein families},
  author = {AA, Zayed and D, Lücking and M, Mohssen and D, Cronin and B, Bolduc and AC, Gregory and KR, Hargreaves and PD, Piehowski and RA, White Iii and EL, Huang and JN, Adkins and S, Roux and C, Moraru and MB, Sullivan},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {22},
  pages = {4202-4208},
  doi = {10.1093/bioinformatics/btab451},
  pmid = {34132786},
}
```


## EpiDOCK

**Tool URL**: [https://epidock.ddg-pharmfac.net/](https://epidock.ddg-pharmfac.net/)

### Publication Information

**Title**: EpiDOCK: a molecular docking-based tool for MHC class II binding prediction.

**Journal**: Protein engineering, design & selection : PEDS

**Year**: 2013

**Identifiers**:
- DOI: [10.1093/protein/gzt018](https://doi.org/10.1093/protein/gzt018)
- PMID: [23661105](https://pubmed.ncbi.nlm.nih.gov/23661105/)

### Cite this tool

**APA format**:

```
M, A., A, P., I, D., DR, F., & I, D. (2013). EpiDOCK: a molecular docking-based tool for MHC class II binding prediction. Protein engineering, design & selection : PEDS, 26(10), 631-4. https://doi.org/10.1093/protein/gzt018
```

**BibTeX format**:

```bibtex
@article{m2013,
  title = {EpiDOCK: a molecular docking-based tool for MHC class II binding prediction},
  author = {M, Atanasova and A, Patronov and I, Dimitrov and DR, Flower and I, Doytchinova},
  year = {2013},
  journal = {Protein engineering, design & selection : PEDS},
  volume = {26},
  number = {10},
  pages = {631-4},
  doi = {10.1093/protein/gzt018},
  pmid = {23661105},
}
```


## EVE

**Tool URL**: [https://github.com/oist/EVE](https://github.com/oist/EVE)

### Publication Information

**Title**: Tattoo: a navigational tool, a lighthouse for biomolecular level precision in surgical endoscopy. Newton's apple or Eve's?

**Journal**: Surgical endoscopy

**Year**: 2009

**Identifiers**:
- DOI: [10.1007/s00464-009-0441-3](https://doi.org/10.1007/s00464-009-0441-3)
- PMID: [19301069](https://pubmed.ncbi.nlm.nih.gov/19301069/)

### Cite this tool

**APA format**:

```
BB, A. (2009). Tattoo: a navigational tool, a lighthouse for biomolecular level precision in surgical endoscopy. Newton's apple or Eve's?. Surgical endoscopy, 23(5), 1166-7. https://doi.org/10.1007/s00464-009-0441-3
```

**BibTeX format**:

```bibtex
@article{bb2009,
  title = {Tattoo: a navigational tool, a lighthouse for biomolecular level precision in surgical endoscopy. Newton's apple or Eve's?},
  author = {BB, Agarwal},
  year = {2009},
  journal = {Surgical endoscopy},
  volume = {23},
  number = {5},
  pages = {1166-7},
  doi = {10.1007/s00464-009-0441-3},
  pmid = {19301069},
}
```


## FastViromeExplorer

**Tool URL**: [https://code.vt.edu/saima5/FastViromeExplorer](https://code.vt.edu/saima5/FastViromeExplorer)

### Publication Information

**Title**: FastViromeExplorer: a pipeline for virus and phage identification and abundance profiling in metagenomics data.

**Journal**: PeerJ

**Year**: 2018

**Identifiers**:
- DOI: [10.7717/peerj.4227](https://doi.org/10.7717/peerj.4227)
- PMID: [29340239](https://pubmed.ncbi.nlm.nih.gov/29340239/)

### Cite this tool

**APA format**:

```
SS, T., FO, A., RV, J., & L, Z. (2018). FastViromeExplorer: a pipeline for virus and phage identification and abundance profiling in metagenomics data. PeerJ, 6, e4227. https://doi.org/10.7717/peerj.4227
```

**BibTeX format**:

```bibtex
@article{ss2018,
  title = {FastViromeExplorer: a pipeline for virus and phage identification and abundance profiling in metagenomics data},
  author = {SS, Tithi and FO, Aylward and RV, Jensen and L, Zhang},
  year = {2018},
  journal = {PeerJ},
  volume = {6},
  pages = {e4227},
  doi = {10.7717/peerj.4227},
  pmid = {29340239},
}
```


## FAVITES

**Tool URL**: [https://github.com/niemasd/FAVITES](https://github.com/niemasd/FAVITES)

### Publication Information

**Title**: FAVITES: simultaneous simulation of transmission networks, phylogenetic trees and sequences.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2019

**Identifiers**:
- DOI: [10.1093/bioinformatics/bty921](https://doi.org/10.1093/bioinformatics/bty921)
- PMID: [30395173](https://pubmed.ncbi.nlm.nih.gov/30395173/)

### Cite this tool

**APA format**:

```
N, M., M, R., JO, W., & S, M. (2019). FAVITES: simultaneous simulation of transmission networks, phylogenetic trees and sequences. Bioinformatics (Oxford, England), 35(11), 1852-1861. https://doi.org/10.1093/bioinformatics/bty921
```

**BibTeX format**:

```bibtex
@article{n2019,
  title = {FAVITES: simultaneous simulation of transmission networks, phylogenetic trees and sequences},
  author = {N, Moshiri and M, Ragonnet-Cronin and JO, Wertheim and S, Mirarab},
  year = {2019},
  journal = {Bioinformatics (Oxford, England)},
  volume = {35},
  number = {11},
  pages = {1852-1861},
  doi = {10.1093/bioinformatics/bty921},
  pmid = {30395173},
}
```


## FDA ARGOS

**Tool URL**: [https://argos.igs.umaryland.edu/](https://argos.igs.umaryland.edu/)

### Publication Information

**Title**: FDA-ARGOS is a database with public quality-controlled reference genomes for diagnostic use and regulatory science.

**Journal**: Nature communications

**Year**: 2019

**Identifiers**:
- DOI: [10.1038/s41467-019-11306-6](https://doi.org/10.1038/s41467-019-11306-6)
- PMID: [31346170](https://pubmed.ncbi.nlm.nih.gov/31346170/)

### Cite this tool

**APA format**:

```
H, S., T, M., Y, Y., C, S., A, H., L, T., ... U, S. (2019). FDA-ARGOS is a database with public quality-controlled reference genomes for diagnostic use and regulatory science. Nature communications, 10(1), 3313. https://doi.org/10.1038/s41467-019-11306-6
```

**BibTeX format**:

```bibtex
@article{h2019,
  title = {FDA-ARGOS is a database with public quality-controlled reference genomes for diagnostic use and regulatory science},
  author = {H, Sichtig and T, Minogue and Y, Yan and C, Stefan and A, Hall and L, Tallon and L, Sadzewicz and S, Nadendla and W, Klimke and E, Hatcher and M, Shumway and DL, Aldea and J, Allen and J, Koehler and T, Slezak and S, Lovell and R, Schoepp and U, Scherf},
  year = {2019},
  journal = {Nature communications},
  volume = {10},
  number = {1},
  pages = {3313},
  doi = {10.1038/s41467-019-11306-6},
  pmid = {31346170},
}
```


## GeneMarkS

**Tool URL**: [http://exon.gatech.edu/GeneMark/](http://exon.gatech.edu/GeneMark/)

### Publication Information

**Title**: Gene identification in prokaryotic genomes, phages, metagenomes, and EST sequences with GeneMarkS suite.

**Journal**: Current protocols in microbiology

**Year**: 2014

**Identifiers**:
- DOI: [10.1002/9780471729259.mc01e07s32](https://doi.org/10.1002/9780471729259.mc01e07s32)
- PMID: [24510847](https://pubmed.ncbi.nlm.nih.gov/24510847/)

### Cite this tool

**APA format**:

```
M, B., & A, L. (2014). Gene identification in prokaryotic genomes, phages, metagenomes, and EST sequences with GeneMarkS suite. Current protocols in microbiology, 32, Unit 1E.7.. https://doi.org/10.1002/9780471729259.mc01e07s32
```

**BibTeX format**:

```bibtex
@article{m2014,
  title = {Gene identification in prokaryotic genomes, phages, metagenomes, and EST sequences with GeneMarkS suite},
  author = {M, Borodovsky and A, Lomsadze},
  year = {2014},
  journal = {Current protocols in microbiology},
  volume = {32},
  pages = {Unit 1E.7.},
  doi = {10.1002/9780471729259.mc01e07s32},
  pmid = {24510847},
}
```


## GeneMarkS-2

**Tool URL**: [http://exon.gatech.edu/GeneMark/genemarks2.cgi](http://exon.gatech.edu/GeneMark/genemarks2.cgi)

### Publication Information

**Title**: GeneMark-HM: improving gene prediction in DNA sequences of human microbiome.

**Journal**: NAR genomics and bioinformatics

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/nargab/lqab047](https://doi.org/10.1093/nargab/lqab047)
- PMID: [34056597](https://pubmed.ncbi.nlm.nih.gov/34056597/)

### Cite this tool

**APA format**:

```
A, L., C, B., F, S., & M, B. (2021). GeneMark-HM: improving gene prediction in DNA sequences of human microbiome. NAR genomics and bioinformatics, 3(2), lqab047. https://doi.org/10.1093/nargab/lqab047
```

**BibTeX format**:

```bibtex
@article{a2021,
  title = {GeneMark-HM: improving gene prediction in DNA sequences of human microbiome},
  author = {A, Lomsadze and C, Bonny and F, Strozzi and M, Borodovsky},
  year = {2021},
  journal = {NAR genomics and bioinformatics},
  volume = {3},
  number = {2},
  pages = {lqab047},
  doi = {10.1093/nargab/lqab047},
  pmid = {34056597},
}
```


## geNomad

**Tool URL**: [https://github.com/apcamargo/genomad](https://github.com/apcamargo/genomad)

### Publication Information

**Title**: Identification of mobile genetic elements with geNomad.

**Journal**: Nature biotechnology

**Year**: 2024

**Identifiers**:
- DOI: [10.1038/s41587-023-01953-y](https://doi.org/10.1038/s41587-023-01953-y)
- PMID: [37735266](https://pubmed.ncbi.nlm.nih.gov/37735266/)

### Cite this tool

**APA format**:

```
AP, C., S, R., F, S., M, B., Y, X., B, H., ... NC, K. (2024). Identification of mobile genetic elements with geNomad. Nature biotechnology, 42(8), 1303-1312. https://doi.org/10.1038/s41587-023-01953-y
```

**BibTeX format**:

```bibtex
@article{ap2024,
  title = {Identification of mobile genetic elements with geNomad},
  author = {AP, Camargo and S, Roux and F, Schulz and M, Babinski and Y, Xu and B, Hu and PSG, Chain and S, Nayfach and NC, Kyrpides},
  year = {2024},
  journal = {Nature biotechnology},
  volume = {42},
  number = {8},
  pages = {1303-1312},
  doi = {10.1038/s41587-023-01953-y},
  pmid = {37735266},
}
```


## GenomePeek

**Tool URL**: [https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0663-4](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0663-4)

### Publication Information

**Title**: GenomePeek-an online tool for prokaryotic genome and metagenome analysis.

**Journal**: PeerJ

**Year**: 2015

**Identifiers**:
- DOI: [10.7717/peerj.1025](https://doi.org/10.7717/peerj.1025)
- PMID: [26157610](https://pubmed.ncbi.nlm.nih.gov/26157610/)

### Cite this tool

**APA format**:

```
K, M., & RA, E. (2015). GenomePeek-an online tool for prokaryotic genome and metagenome analysis. PeerJ, 3, e1025. https://doi.org/10.7717/peerj.1025
```

**BibTeX format**:

```bibtex
@article{k2015,
  title = {GenomePeek-an online tool for prokaryotic genome and metagenome analysis},
  author = {K, McNair and RA, Edwards},
  year = {2015},
  journal = {PeerJ},
  volume = {3},
  pages = {e1025},
  doi = {10.7717/peerj.1025},
  pmid = {26157610},
}
```


## GLIMMER

**Tool URL**: [https://ccb.jhu.edu/software/glimmer/](https://ccb.jhu.edu/software/glimmer/)

### Publication Information

**Title**: A glimmer of light at the end of the tunnel?

**Journal**: Movement disorders : official journal of the Movement Disorder Society

**Year**: 2017

**Identifiers**:
- DOI: [10.1002/mds.27179](https://doi.org/10.1002/mds.27179)
- PMID: [28921664](https://pubmed.ncbi.nlm.nih.gov/28921664/)

### Cite this tool

**APA format**:

```
P, J. (2017). A glimmer of light at the end of the tunnel?. Movement disorders : official journal of the Movement Disorder Society, 32(10), 1343-1344. https://doi.org/10.1002/mds.27179
```

**BibTeX format**:

```bibtex
@article{p2017,
  title = {A glimmer of light at the end of the tunnel?},
  author = {P, Jenner},
  year = {2017},
  journal = {Movement disorders : official journal of the Movement Disorder Society},
  volume = {32},
  number = {10},
  pages = {1343-1344},
  doi = {10.1002/mds.27179},
  pmid = {28921664},
}
```


## GlyConnect

**Tool URL**: [https://glyconnect.expasy.org/](https://glyconnect.expasy.org/)

### Publication Information

**Title**: GlyConnect: Glycoproteomics Goes Visual, Interactive, and Analytical.

**Journal**: Journal of proteome research

**Year**: 2019

**Identifiers**:
- DOI: [10.1021/acs.jproteome.8b00766](https://doi.org/10.1021/acs.jproteome.8b00766)
- PMID: [30574787](https://pubmed.ncbi.nlm.nih.gov/30574787/)

### Cite this tool

**APA format**:

```
D, A., J, M., A, G., E, G., NG, K., D, K., ... F, L. (2019). GlyConnect: Glycoproteomics Goes Visual, Interactive, and Analytical. Journal of proteome research, 18(2), 664-677. https://doi.org/10.1021/acs.jproteome.8b00766
```

**BibTeX format**:

```bibtex
@article{d2019,
  title = {GlyConnect: Glycoproteomics Goes Visual, Interactive, and Analytical},
  author = {D, Alocci and J, Mariethoz and A, Gastaldello and E, Gasteiger and NG, Karlsson and D, Kolarich and NH, Packer and F, Lisacek},
  year = {2019},
  journal = {Journal of proteome research},
  volume = {18},
  number = {2},
  pages = {664-677},
  doi = {10.1021/acs.jproteome.8b00766},
  pmid = {30574787},
}
```


## GPDB

**Tool URL**: [http://ftp.ebi.ac.uk/pub/databases/metagenomics/genome_sets/gut_phage_database/](http://ftp.ebi.ac.uk/pub/databases/metagenomics/genome_sets/gut_phage_database/)

### Publication Information

**Title**: GPCRs, G-proteins, effectors and their interactions: human-gpDB, a database employing visualization tools and data integration techniques.

**Journal**: Database : the journal of biological databases and curation

**Year**: 2010

**Identifiers**:
- DOI: [10.1093/database/baq019](https://doi.org/10.1093/database/baq019)
- PMID: [20689020](https://pubmed.ncbi.nlm.nih.gov/20689020/)

### Cite this tool

**APA format**:

```
VP, S., MC, T., CK, S., GA, P., NC, P., PG, B., ... SJ, H. (2010). GPCRs, G-proteins, effectors and their interactions: human-gpDB, a database employing visualization tools and data integration techniques. Database : the journal of biological databases and curation, 2010, baq019. https://doi.org/10.1093/database/baq019
```

**BibTeX format**:

```bibtex
@article{vp2010,
  title = {GPCRs, G-proteins, effectors and their interactions: human-gpDB, a database employing visualization tools and data integration techniques},
  author = {VP, Satagopam and MC, Theodoropoulou and CK, Stampolakis and GA, Pavlopoulos and NC, Papandreou and PG, Bagos and R, Schneider and SJ, Hamodrakas},
  year = {2010},
  journal = {Database : the journal of biological databases and curation},
  volume = {2010},
  pages = {baq019},
  doi = {10.1093/database/baq019},
  pmid = {20689020},
}
```


## GraViTy

**Tool URL**: [https://github.com/PAiewsakun/GRAViTy](https://github.com/PAiewsakun/GRAViTy)

### Publication Information

**Title**: A rehabilitation tool for functional balance using altered gravity and virtual reality.

**Journal**: Journal of neuroengineering and rehabilitation

**Year**: 2007

**Identifiers**:
- DOI: [10.1186/1743-0003-4-25](https://doi.org/10.1186/1743-0003-4-25)
- PMID: [17623080](https://pubmed.ncbi.nlm.nih.gov/17623080/)

### Cite this tool

**APA format**:

```
LI, O., R, K., J, K., S, I., SR, W., & E, Z. (2007). A rehabilitation tool for functional balance using altered gravity and virtual reality. Journal of neuroengineering and rehabilitation, 4, 25. https://doi.org/10.1186/1743-0003-4-25
```

**BibTeX format**:

```bibtex
@article{li2007,
  title = {A rehabilitation tool for functional balance using altered gravity and virtual reality},
  author = {LI, Oddsson and R, Karlsson and J, Konrad and S, Ince and SR, Williams and E, Zemkova},
  year = {2007},
  journal = {Journal of neuroengineering and rehabilitation},
  volume = {4},
  pages = {25},
  doi = {10.1186/1743-0003-4-25},
  pmid = {17623080},
}
```


## GVD

**Tool URL**: [https://datacommons.cyverse.org/browse/iplant/home/shared/iVirus/Gregory_and_Zablocki_GVD_Jul2020](https://datacommons.cyverse.org/browse/iplant/home/shared/iVirus/Gregory_and_Zablocki_GVD_Jul2020)

### Publication Information

**Title**: 160 Gbit/s-300 km single-channel transmission in the 1.1 μm band with a precise GVD and slope compensation.

**Journal**: Optics express

**Year**: 2013

**Identifiers**:
- DOI: [10.1364/OE.21.004303](https://doi.org/10.1364/OE.21.004303)
- PMID: [23481963](https://pubmed.ncbi.nlm.nih.gov/23481963/)

### Cite this tool

**APA format**:

```
K, K., M, Y., T, H., & M, N. (2013). 160 Gbit/s-300 km single-channel transmission in the 1.1 μm band with a precise GVD and slope compensation. Optics express, 21(4), 4303-10. https://doi.org/10.1364/OE.21.004303
```

**BibTeX format**:

```bibtex
@article{k2013,
  title = {160 Gbit/s-300 km single-channel transmission in the 1.1 μm band with a precise GVD and slope compensation},
  author = {K, Koizumi and M, Yoshida and T, Hirooka and M, Nakazawa},
  year = {2013},
  journal = {Optics express},
  volume = {21},
  number = {4},
  pages = {4303-10},
  doi = {10.1364/OE.21.004303},
  pmid = {23481963},
}
```


## hafeZ

**Tool URL**: [https://github.com/Chrisjrt/hafeZ](https://github.com/Chrisjrt/hafeZ)

### Publication Information

**Title**: ['hafeZ: Active prophage identification through read mapping']

### Cite this tool



## hecatomb

**Tool URL**: [https://github.com/shandley/hecatomb](https://github.com/shandley/hecatomb)

### Publication Information

**Title**: Hecatomb: an integrated software platform for viral metagenomics.

**Journal**: GigaScience

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/gigascience/giae020](https://doi.org/10.1093/gigascience/giae020)
- PMID: [38832467](https://pubmed.ncbi.nlm.nih.gov/38832467/)

### Cite this tool

**APA format**:

```
MJ, R., SJ, B., KA, M., L, W., A, P., LAC, C., ... SA, H. (2024). Hecatomb: an integrated software platform for viral metagenomics. GigaScience, 13. https://doi.org/10.1093/gigascience/giae020
```

**BibTeX format**:

```bibtex
@article{mj2024,
  title = {Hecatomb: an integrated software platform for viral metagenomics},
  author = {MJ, Roach and SJ, Beecroft and KA, Mihindukulasuriya and L, Wang and A, Paredes and LAC, Cárdenas and K, Henry-Cocks and LFO, Lima and EA, Dinsdale and RA, Edwards and SA, Handley},
  year = {2024},
  journal = {GigaScience},
  volume = {13},
  doi = {10.1093/gigascience/giae020},
  pmid = {38832467},
}
```


## HNADOCK

**Tool URL**: [http://huanglab.phys.hust.edu.cn/hnadock/](http://huanglab.phys.hust.edu.cn/hnadock/)

### Publication Information

**Title**: HNADOCK: a nucleic acid docking server for modeling RNA/DNA-RNA/DNA 3D complex structures.

**Journal**: Nucleic acids research

**Year**: 2019

**Identifiers**:
- DOI: [10.1093/nar/gkz412](https://doi.org/10.1093/nar/gkz412)
- PMID: [31114906](https://pubmed.ncbi.nlm.nih.gov/31114906/)

### Cite this tool

**APA format**:

```
J, H., J, W., H, T., Y, X., & SY, H. (2019). HNADOCK: a nucleic acid docking server for modeling RNA/DNA-RNA/DNA 3D complex structures. Nucleic acids research, 47(W1), W35-W42. https://doi.org/10.1093/nar/gkz412
```

**BibTeX format**:

```bibtex
@article{j2019,
  title = {HNADOCK: a nucleic acid docking server for modeling RNA/DNA-RNA/DNA 3D complex structures},
  author = {J, He and J, Wang and H, Tao and Y, Xiao and SY, Huang},
  year = {2019},
  journal = {Nucleic acids research},
  volume = {47},
  number = {W1},
  pages = {W35-W42},
  doi = {10.1093/nar/gkz412},
  pmid = {31114906},
}
```


## HoloVir

**Tool URL**: [https://github.com/plaffy/HoloVir](https://github.com/plaffy/HoloVir)

### Publication Information

**Title**: HoloVir: A Workflow for Investigating the Diversity and Function of Viruses in Invertebrate Holobionts.

**Journal**: Frontiers in microbiology

**Year**: 2016

**Identifiers**:
- DOI: [10.3389/fmicb.2016.00822](https://doi.org/10.3389/fmicb.2016.00822)
- PMID: [27375564](https://pubmed.ncbi.nlm.nih.gov/27375564/)

### Cite this tool

**APA format**:

```
PW, L., EM, W., D, T., KD, W., ES, B., MJ, v.O., ... T, R. (2016). HoloVir: A Workflow for Investigating the Diversity and Function of Viruses in Invertebrate Holobionts. Frontiers in microbiology, 7, 822. https://doi.org/10.3389/fmicb.2016.00822
```

**BibTeX format**:

```bibtex
@article{pw2016,
  title = {HoloVir: A Workflow for Investigating the Diversity and Function of Viruses in Invertebrate Holobionts},
  author = {PW, Laffy and EM, Wood-Charlson and D, Turaev and KD, Weynberg and ES, Botté and MJ, van Oppen and NS, Webster and T, Rattei},
  year = {2016},
  journal = {Frontiers in microbiology},
  volume = {7},
  pages = {822},
  doi = {10.3389/fmicb.2016.00822},
  pmid = {27375564},
}
```


## HOPS

**Tool URL**: [https://github.com/rhuebler/HOPS](https://github.com/rhuebler/HOPS)

### Publication Information

**Title**: A novel approach for identification of biologically active phenolic compounds in complex matrices using hybrid quadrupole-orbitrap mass spectrometer: A promising tool for testing antimicrobial activity of hops.

**Journal**: Talanta

**Year**: 2016

**Identifiers**:
- DOI: [10.1016/j.talanta.2016.05.018](https://doi.org/10.1016/j.talanta.2016.05.018)
- PMID: [27260455](https://pubmed.ncbi.nlm.nih.gov/27260455/)

### Cite this tool

**APA format**:

```
M, D., V, J., P, Č., A, M., & J, O. (2016). A novel approach for identification of biologically active phenolic compounds in complex matrices using hybrid quadrupole-orbitrap mass spectrometer: A promising tool for testing antimicrobial activity of hops. Talanta, 156-157, 209-217. https://doi.org/10.1016/j.talanta.2016.05.018
```

**BibTeX format**:

```bibtex
@article{m2016,
  title = {A novel approach for identification of biologically active phenolic compounds in complex matrices using hybrid quadrupole-orbitrap mass spectrometer: A promising tool for testing antimicrobial activity of hops},
  author = {M, Dušek and V, Jandovská and P, Čermák and A, Mikyška and J, Olšovská},
  year = {2016},
  journal = {Talanta},
  volume = {156-157},
  pages = {209-217},
  doi = {10.1016/j.talanta.2016.05.018},
  pmid = {27260455},
}
```


## HostG

**Tool URL**: [https://github.com/KennthShang/HostG](https://github.com/KennthShang/HostG)

### Publication Information

**Title**: Predicting the hosts of prokaryotic viruses using GCN-based semi-supervised learning.

**Journal**: BMC biology

**Year**: 2021

**Identifiers**:
- DOI: [10.1186/s12915-021-01180-4](https://doi.org/10.1186/s12915-021-01180-4)
- PMID: [34819064](https://pubmed.ncbi.nlm.nih.gov/34819064/)

### Cite this tool

**APA format**:

```
J, S., & Y, S. (2021). Predicting the hosts of prokaryotic viruses using GCN-based semi-supervised learning. BMC biology, 19(1), 250. https://doi.org/10.1186/s12915-021-01180-4
```

**BibTeX format**:

```bibtex
@article{j2021,
  title = {Predicting the hosts of prokaryotic viruses using GCN-based semi-supervised learning},
  author = {J, Shang and Y, Sun},
  year = {2021},
  journal = {BMC biology},
  volume = {19},
  number = {1},
  pages = {250},
  doi = {10.1186/s12915-021-01180-4},
  pmid = {34819064},
}
```


## HostPhinder

**Tool URL**: [https://github.com/julvi/HostPhinder](https://github.com/julvi/HostPhinder)

### Publication Information

**Title**: HostPhinder: A Phage Host Prediction Tool.

**Journal**: Viruses

**Year**: 2016

**Identifiers**:
- DOI: [10.3390/v8050116](https://doi.org/10.3390/v8050116)
- PMID: [27153081](https://pubmed.ncbi.nlm.nih.gov/27153081/)

### Cite this tool

**APA format**:

```
J, V., KA, K., VI, J., H, Z., O, L., M, N., & MV, L. (2016). HostPhinder: A Phage Host Prediction Tool. Viruses, 8(5). https://doi.org/10.3390/v8050116
```

**BibTeX format**:

```bibtex
@article{j2016,
  title = {HostPhinder: A Phage Host Prediction Tool},
  author = {J, Villarroel and KA, Kleinheinz and VI, Jurtz and H, Zschach and O, Lund and M, Nielsen and MV, Larsen},
  year = {2016},
  journal = {Viruses},
  volume = {8},
  number = {5},
  doi = {10.3390/v8050116},
  pmid = {27153081},
}
```


## I-TASSER

**Tool URL**: [https://zhanggroup.org/I-TASSER/](https://zhanggroup.org/I-TASSER/)

### Publication Information

**Title**: GPU-I-TASSER: a GPU accelerated I-TASSER protein structure prediction tool.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab871](https://doi.org/10.1093/bioinformatics/btab871)
- PMID: [34978562](https://pubmed.ncbi.nlm.nih.gov/34978562/)

### Cite this tool

**APA format**:

```
EA, M., C, Z., Y, Z., & DB, K. (2022). GPU-I-TASSER: a GPU accelerated I-TASSER protein structure prediction tool. Bioinformatics (Oxford, England), 38(6), 1754-1755. https://doi.org/10.1093/bioinformatics/btab871
```

**BibTeX format**:

```bibtex
@article{ea2022,
  title = {GPU-I-TASSER: a GPU accelerated I-TASSER protein structure prediction tool},
  author = {EA, MacCarthy and C, Zhang and Y, Zhang and DB, Kc},
  year = {2022},
  journal = {Bioinformatics (Oxford, England)},
  volume = {38},
  number = {6},
  pages = {1754-1755},
  doi = {10.1093/bioinformatics/btab871},
  pmid = {34978562},
}
```


## IDseq

**Tool URL**: [https://github.com/chanzuckerberg/idseq-web](https://github.com/chanzuckerberg/idseq-web)

### Publication Information

**Title**: IDseq-An open source cloud-based pipeline and analysis service for metagenomic pathogen detection and monitoring.

**Journal**: GigaScience

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/gigascience/giaa111](https://doi.org/10.1093/gigascience/giaa111)
- PMID: [33057676](https://pubmed.ncbi.nlm.nih.gov/33057676/)

### Cite this tool

**APA format**:

```
KL, K., T, C., CFA, d.B., B, D., G, D., R, E., ... JL, D. (2020). IDseq-An open source cloud-based pipeline and analysis service for metagenomic pathogen detection and monitoring. GigaScience, 9(10). https://doi.org/10.1093/gigascience/giaa111
```

**BibTeX format**:

```bibtex
@article{kl2020,
  title = {IDseq-An open source cloud-based pipeline and analysis service for metagenomic pathogen detection and monitoring},
  author = {KL, Kalantar and T, Carvalho and CFA, de Bourcy and B, Dimitrov and G, Dingle and R, Egger and J, Han and OB, Holmes and YF, Juan and R, King and A, Kislyuk and MF, Lin and M, Mariano and T, Morse and LV, Reynoso and DR, Cruz and J, Sheu and J, Tang and J, Wang and MA, Zhang and E, Zhong and V, Ahyong and S, Lay and S, Chea and JA, Bohl and JE, Manning and CM, Tato and JL, DeRisi},
  year = {2020},
  journal = {GigaScience},
  volume = {9},
  number = {10},
  doi = {10.1093/gigascience/giaa111},
  pmid = {33057676},
}
```


## IEDB-tools

**Tool URL**: [https://github.com/iedb/iedb-epitope-database](https://github.com/iedb/iedb-epitope-database)

### Publication Information

**Title**: Next-generation IEDB tools: a platform for epitope prediction and analysis.

**Journal**: Nucleic acids research

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/nar/gkae407](https://doi.org/10.1093/nar/gkae407)
- PMID: [38783079](https://pubmed.ncbi.nlm.nih.gov/38783079/)

### Cite this tool

**APA format**:

```
Z, Y., K, K., H, K., B, H., A, G., J, B., ... JA, G. (2024). Next-generation IEDB tools: a platform for epitope prediction and analysis. Nucleic acids research, 52(W1), W526-W532. https://doi.org/10.1093/nar/gkae407
```

**BibTeX format**:

```bibtex
@article{z2024,
  title = {Next-generation IEDB tools: a platform for epitope prediction and analysis},
  author = {Z, Yan and K, Kim and H, Kim and B, Ha and A, Gambiez and J, Bennett and MF, de Almeida Mendes and R, Trevizani and J, Mahita and E, Richardson and D, Marrama and N, Blazeska and Z, Koşaloğlu-Yalçın and M, Nielsen and A, Sette and B, Peters and JA, Greenbaum},
  year = {2024},
  journal = {Nucleic acids research},
  volume = {52},
  number = {W1},
  pages = {W526-W532},
  doi = {10.1093/nar/gkae407},
  pmid = {38783079},
}
```


## INHERIT

**Tool URL**: [https://github.com/Celestial-Bai/INHERIT](https://github.com/Celestial-Bai/INHERIT)

### Publication Information

**Title**: The INHERIT Model: A Tool to Jointly Improve Health, Environmental Sustainability and Health Equity through Behavior and Lifestyle Change.

**Journal**: International journal of environmental research and public health

**Year**: 2018

**Identifiers**:
- DOI: [10.3390/ijerph15071435](https://doi.org/10.3390/ijerph15071435)
- PMID: [29986493](https://pubmed.ncbi.nlm.nih.gov/29986493/)

### Cite this tool

**APA format**:

```
N, v.d.V., B, S., H, K., G, M., C, C., R, B., ... J, S. (2018). The INHERIT Model: A Tool to Jointly Improve Health, Environmental Sustainability and Health Equity through Behavior and Lifestyle Change. International journal of environmental research and public health, 15(7). https://doi.org/10.3390/ijerph15071435
```

**BibTeX format**:

```bibtex
@article{n2018,
  title = {The INHERIT Model: A Tool to Jointly Improve Health, Environmental Sustainability and Health Equity through Behavior and Lifestyle Change},
  author = {N, van der Vliet and B, Staatsen and H, Kruize and G, Morris and C, Costongs and R, Bell and S, Marques and T, Taylor and S, Quiroga and P, Martinez Juarez and V, Máca and M, Ščasný and I, Zvěřinová and F, Tozija and D, Gjorgjev and GA, Espnes and J, Schuit},
  year = {2018},
  journal = {International journal of environmental research and public health},
  volume = {15},
  number = {7},
  doi = {10.3390/ijerph15071435},
  pmid = {29986493},
}
```


## INSaFLU-TELEVIR

**Tool URL**: [https://github.com/INSaFLU/INSaFLU](https://github.com/INSaFLU/INSaFLU)

### Publication Information

**Title**: INSaFLU-TELEVIR: an open web-based bioinformatics suite for viral metagenomic detection and routine genomic surveillance.

**Journal**: Genome medicine

**Year**: 2024

**Identifiers**:
- DOI: [10.1186/s13073-024-01334-3](https://doi.org/10.1186/s13073-024-01334-3)
- PMID: [38659008](https://pubmed.ncbi.nlm.nih.gov/38659008/)

### Cite this tool

**APA format**:

```
JD, S., D, S., M, P., J, I., C, B., M, P., ... V, B. (2024). INSaFLU-TELEVIR: an open web-based bioinformatics suite for viral metagenomic detection and routine genomic surveillance. Genome medicine, 16(1), 61. https://doi.org/10.1186/s13073-024-01334-3
```

**BibTeX format**:

```bibtex
@article{jd2024,
  title = {INSaFLU-TELEVIR: an open web-based bioinformatics suite for viral metagenomic detection and routine genomic surveillance},
  author = {JD, Santos and D, Sobral and M, Pinheiro and J, Isidro and C, Bogaardt and M, Pinto and R, Eusébio and A, Santos and R, Mamede and DL, Horton and JP, Gomes and Consortium, TELEVIR and V, Borges},
  year = {2024},
  journal = {Genome medicine},
  volume = {16},
  number = {1},
  pages = {61},
  doi = {10.1186/s13073-024-01334-3},
  pmid = {38659008},
}
```


## iPHoP

**Tool URL**: [https://github.com/RasmussenLab/PHoP](https://github.com/RasmussenLab/PHoP)

### Publication Information

**Title**: iPHoP: An integrated machine learning framework to maximize host prediction for metagenome-derived viruses of archaea and bacteria.

**Journal**: PLoS biology

**Year**: 2023

**Identifiers**:
- DOI: [10.1371/journal.pbio.3002083](https://doi.org/10.1371/journal.pbio.3002083)
- PMID: [37083735](https://pubmed.ncbi.nlm.nih.gov/37083735/)

### Cite this tool

**APA format**:

```
S, R., AP, C., FH, C., SM, D., BE, D., S, N., & A, T. (2023). iPHoP: An integrated machine learning framework to maximize host prediction for metagenome-derived viruses of archaea and bacteria. PLoS biology, 21(4), e3002083. https://doi.org/10.1371/journal.pbio.3002083
```

**BibTeX format**:

```bibtex
@article{s2023,
  title = {iPHoP: An integrated machine learning framework to maximize host prediction for metagenome-derived viruses of archaea and bacteria},
  author = {S, Roux and AP, Camargo and FH, Coutinho and SM, Dabdoub and BE, Dutilh and S, Nayfach and A, Tritt},
  year = {2023},
  journal = {PLoS biology},
  volume = {21},
  number = {4},
  pages = {e3002083},
  doi = {10.1371/journal.pbio.3002083},
  pmid = {37083735},
}
```


## isling

**Tool URL**: [https://github.com/szsctt/intvi_other-tools](https://github.com/szsctt/intvi_other-tools)

### Publication Information

**Title**: Isling: A Tool for Detecting Integration of Wild-Type Viruses and Clinical Vectors.

**Journal**: Journal of molecular biology

**Year**: 2022

**Identifiers**:
- DOI: [10.1016/j.jmb.2021.167408](https://doi.org/10.1016/j.jmb.2021.167408)
- PMID: [34929203](https://pubmed.ncbi.nlm.nih.gov/34929203/)

### Cite this tool

**APA format**:

```
S, S., CV, H., F, H., S, G., Y, J., IE, A., ... LOW, W. (2022). Isling: A Tool for Detecting Integration of Wild-Type Viruses and Clinical Vectors. Journal of molecular biology, 434(11), 167408. https://doi.org/10.1016/j.jmb.2021.167408
```

**BibTeX format**:

```bibtex
@article{s2022,
  title = {Isling: A Tool for Detecting Integration of Wild-Type Viruses and Clinical Vectors},
  author = {S, Scott and CV, Hallwirth and F, Hartkopf and S, Grigson and Y, Jain and IE, Alexander and DC, Bauer and LOW, Wilson},
  year = {2022},
  journal = {Journal of molecular biology},
  volume = {434},
  number = {11},
  pages = {167408},
  doi = {10.1016/j.jmb.2021.167408},
  pmid = {34929203},
}
```


## Jaeger

**Tool URL**: [https://github.com/Yasas1994/Jaeger](https://github.com/Yasas1994/Jaeger)

### Publication Information

**Title**: Human genome meeting 2016 : Houston, TX, USA. 28 February - 2 March 2016.

**Journal**: Human genomics

**Year**: 2016

**Identifiers**:
- DOI: [10.1186/s40246-016-0063-5](https://doi.org/10.1186/s40246-016-0063-5)
- PMID: [27294413](https://pubmed.ncbi.nlm.nih.gov/27294413/)

### Cite this tool

**APA format**:

```
AK, S., Y, W., R, H., C, S., T, T., L, P., ... A, B. (2016). Human genome meeting 2016 : Houston, TX, USA. 28 February - 2 March 2016. Human genomics, 10 Suppl 1(Suppl 1), 12. https://doi.org/10.1186/s40246-016-0063-5
```

**BibTeX format**:

```bibtex
@article{ak2016,
  title = {Human genome meeting 2016 : Houston, TX, USA. 28 February - 2 March 2016},
  author = {AK, Srivastava and Y, Wang and R, Huang and C, Skinner and T, Thompson and L, Pollard and T, Wood and F, Luo and R, Stevenson and R, Polimanti and J, Gelernter and X, Lin and IY, Lim and Y, Wu and AL, Teh and L, Chen and IM, Aris and SE, Soh and MT, Tint and JL, MacIsaac and F, Yap and K, Kwek and SM, Saw and MS, Kobor and MJ, Meaney and KM, Godfrey and YS, Chong and JD, Holbrook and YS, Lee and PD, Gluckman and N, Karnani and group, GUSTO study and A, Kapoor and D, Lee and A, Chakravarti and C, Maercker and F, Graf and M, Boutros and G, Stamoulis and F, Santoni and P, Makrythanasis and A, Letourneau and M, Guipponi and N, Panousis and M, Garieri and P, Ribaux and E, Falconnet and C, Borel and SE, Antonarakis and S, Kumar and J, Curran and J, Blangero and S, Chatterjee and A, Kapoor and J, Akiyama and D, Auer and C, Berrios and L, Pennacchio and A, Chakravarti and TR, Donti and G, Cappuccio and M, Miller and P, Atwal and A, Kennedy and A, Cardon and C, Bacino and L, Emrick and J, Hertecant and F, Baumer and B, Porter and M, Bainbridge and P, Bonnen and B, Graham and R, Sutton and Q, Sun and S, Elsea and Z, Hu and P, Wang and Y, Zhu and J, Zhao and M, Xiong and DA, Bennett and A, Hidalgo-Miranda and S, Romero-Cordoba and S, Rodriguez-Cuevas and R, Rebollar-Vega and E, Tagliabue and M, Iorio and E, D’Ippolito and S, Baroni and B, Kaczkowski and Y, Tanaka and H, Kawaji and A, Sandelin and R, Andersson and M, Itoh and T, Lassmann and Consortium, The FANTOM5 and Y, Hayashizaki and P, Carninci and ARR, Forrest and CA, Semple and EA, Rosenthal and B, Shirts and L, Amendola and C, Gallego and M, Horike-Pyne and A, Burt and P, Robertson and P, Beyers and C, Nefcy and D, Veenstra and F, Hisama and R, Bennett and M, Dorschner and D, Nickerson and J, Smith and K, Patterson and D, Crosslin and R, Nassir and N, Zubair and T, Harrison and U, Peters and G, Jarvik and Project, NHLBI GO Exome Sequencing and F, Menghi and K, Inaki and X, Woo and P, Kumar and K, Grzeda and A, Malhotra and H, Kim and D, Ucar and P, Shreckengast and K, Karuturi and J, Keck and J, Chuang and ET, Liu and B, Ji and A, Tyler and G, Ananda and G, Carter and H, Nikbakht and M, Montagne and M, Zeinieh and A, Harutyunyan and M, Mcconechy and N, Jabado and P, Lavigne and J, Majewski and JB, Goldstein and M, Overman and G, Varadhachary and R, Shroff and R, Wolff and M, Javle and A, Futreal and D, Fogelman and L, Bravo and W, Fajardo and H, Gomez and C, Castaneda and C, Rolfo and JA, Pinto and KC, Akdemir and L, Chin and A, Futreal and Group, ICGC PCAWG Structural Alterations and S, Patterson and C, Statz and S, Mockus and SN, Nikolaev and XI, Bonilla and L, Parmentier and B, King and F, Bezrukov and G, Kaya and V, Zoete and V, Seplyarskiy and H, Sharpe and T, McKee and A, Letourneau and P, Ribaux and K, Popadin and N, Basset-Seguin and RB, Chaabene and F, Santoni and M, Andrianova and M, Guipponi and M, Garieri and C, Verdan and K, Grosdemange and O, Sumara and M, Eilers and I, Aifantis and O, Michielin and F, de Sauvage and S, Antonarakis and S, Likhitrattanapisal and S, Lincoln and A, Kurian and A, Desmond and S, Yang and Y, Kobayashi and J, Ford and L, Ellisen and TL, Peters and KR, Alvarez and EF, Hollingsworth and DH, Lopez-Terrada and A, Hastie and Z, Dzakula and AW, Pang and ET, Lam and T, Anantharaman and M, Saghbini and H, Cao and Genomics, BioNano and C, Gonzaga-Jauregui and L, Ma and A, King and EB, Rosenzweig and U, Krishnan and JG, Reid and JD, Overton and F, Dewey and WK, Chung and K, Small and A, DeLuca and F, Cremers and RA, Lewis and V, Puech and B, Bakall and R, Silva-Garcia and K, Rohrschneider and M, Leys and FS, Shaya and E, Stone and NL, Sobreira and F, Schiettecatte and H, Ling and E, Pugh and D, Witmer and K, Hetrick and P, Zhang and K, Doheny and D, Valle and A, Hamosh and SN, Jhangiani and ZC, Akdemir and MN, Bainbridge and W, Charng and W, Wiszniewski and T, Gambin and E, Karaca and Y, Bayram and MK, Eldomery and J, Posey and H, Doddapaneni and J, Hu and VR, Sutton and DM, Muzny and EA, Boerwinkle and D, Valle and JR, Lupski and RA, Gibbs and S, Shekar and W, Salerno and A, English and A, Mangubat and J, Bruestle and A, Thorogood and BM, Knoppers and Group, Global Alliance for Genomics and Health - Regulatory and Ethics Working and H, Takahashi and KR, Nitta and A, Kozhuharova and AM, Suzuki and H, Sharma and D, Cotella and C, Santoro and S, Zucchelli and S, Gustincich and P, Carninci and JJ, Mulvihill and G, Baynam and W, Gahl and SC, Groft and K, Kosaki and P, Lasko and B, Melegh and D, Taruscio and R, Ghosh and S, Plon and S, Scherer and X, Qin and R, Sanghvi and K, Walker and T, Chiang and D, Muzny and L, Wang and J, Black and E, Boerwinkle and R, Weinshilboum and R, Gibbs and T, Karpinets and T, Calderone and K, Wani and X, Yu and C, Creasy and C, Haymaker and M, Forget and V, Nanda and J, Roszik and J, Wargo and L, Haydu and X, Song and A, Lazar and J, Gershenwald and M, Davies and C, Bernatchez and J, Zhang and A, Futreal and S, Woodman and EJ, Chesler and T, Reynolds and JA, Bubier and C, Phillips and MA, Langston and EJ, Baker and M, Xiong and L, Ma and N, Lin and C, Amos and N, Lin and P, Wang and Y, Zhu and J, Zhao and V, Calhoun and M, Xiong and O, Dobretsberger and M, Egger and F, Leimgruber and S, Sadedin and A, Oshlack and Alliance, Melbourne Genomics Health and VAA, Antonio and N, Ono and Go, Clark Kendrick C. and Z, Ahmed and M, Bolisetty and S, Zeeshan and E, Anguiano and D, Ucar and A, Sarkar and MR, Nandineni and C, Zeng and J, Shao and H, Cao and A, Hastie and AW, Pang and ET, Lam and T, Liang and K, Pham and M, Saghbini and Z, Dzakula and Y, Chee-Wei and L, Dongsheng and W, Lai-Ping and D, Lian and ROT, Hee and Y, Yunus and F, Aghakhanian and SS, Mokhtar and CV, Lok-Yung and J, Bhak and M, Phipps and X, Shuhua and T, Yik-Ying and V, Kumar and H, Boon-Peng and I, Campbell and MA, Young and P, James and Lifepool and M, Rain and G, Mohammad and R, Kukreti and Q, Pasha and AR, Akilzhanova and C, Guelly and Z, Abilova and S, Rakhimova and A, Akhmetova and U, Kairov and S, Trajanoski and Z, Zhumadilov and M, Bekbossynova and C, Schumacher and S, Sandhu and T, Harkins and V, Makarov and H, Doddapaneni and R, Glenn and Z, Momin and B, Dilrukshi and H, Chao and Q, Meng and B, Gudenkauf and R, Kshitij and J, Jayaseelan and C, Nessner and S, Lee and K, Blankenberg and L, Lewis and J, Hu and Y, Han and H, Dinh and S, Jireh and K, Walker and E, Boerwinkle and D, Muzny and R, Gibbs and J, Hu and K, Walker and C, Buhay and X, Liu and Q, Wang and R, Sanghvi and H, Doddapaneni and Y, Ding and N, Veeraraghavan and Y, Yang and E, Boerwinkle and AL, Beaudet and CM, Eng and DM, Muzny and RA, Gibbs and KCC, Worley and Y, Liu and DST, Hughes and SC, Murali and RA, Harris and AC, English and X, Qin and OA, Hampton and P, Larsen and C, Beck and Y, Han and M, Wang and H, Doddapaneni and CL, Kovar and WJ, Salerno and A, Yoder and S, Richards and J, Rogers and JR, Lupski and DM, Muzny and RA, Gibbs and Q, Meng and M, Bainbridge and M, Wang and H, Doddapaneni and Y, Han and D, Muzny and R, Gibbs and RA, Harris and M, Raveenedran and C, Xue and M, Dahdouli and L, Cox and G, Fan and B, Ferguson and J, Hovarth and Z, Johnson and S, Kanthaswamy and M, Kubisch and M, Platt and D, Smith and E, Vallender and R, Wiseman and X, Liu and J, Below and D, Muzny and R, Gibbs and F, Yu and J, Rogers and J, Lin and Y, Zhang and Z, Ouyang and A, Moore and Z, Wang and J, Hofmann and M, Purdue and R, Stolzenberg-Solomon and S, Weinstein and D, Albanes and CS, Liu and WL, Cheng and TT, Lin and Q, Lan and N, Rothman and S, Berndt and ES, Chen and H, Bahrami and A, Khoshzaban and SH, Keshal and H, Bahrami and A, Khoshzaban and SH, Keshal and KKR, Alharbi and M, Zhalbinova and A, Akilzhanova and S, Rakhimova and M, Bekbosynova and S, Myrzakhmetova and M, Matar and N, Mili and R, Molinari and Y, Ma and S, Guerrier and N, Elhawary and M, Tayeb and N, Bogari and N, Qotb and SA, McClymont and PW, Hook and LA, Goff and A, McCallion and Y, Kong and JR, Charette and WL, Hicks and JK, Naggert and L, Zhao and PM, Nishina and BM, Edrees and M, Athar and FA, Al-Allaf and MM, Taher and W, Khan and A, Bouazzaoui and NA, Harbi and R, Safar and H, Al-Edressi and A, Anazi and N, Altayeb and MA, Ahmed and K, Alansary and Z, Abduljaleel and A, Kratz and P, Beguin and S, Poulain and M, Kaneko and C, Takahiko and A, Matsunaga and S, Kato and AM, Suzuki and N, Bertin and T, Lassmann and R, Vigot and P, Carninci and C, Plessy and T, Launey and D, Graur and D, Lee and A, Kapoor and A, Chakravarti and J, Friis-Nielsen and JM, Izarzugaza and S, Brunak and A, Chakraborty and J, Basak and A, Mukhopadhyay and BS, Soibam and D, Das and N, Biswas and S, Das and S, Sarkar and A, Maitra and C, Panda and P, Majumder and H, Morsy and A, Gaballah and M, Samir and M, Shamseya and H, Mahrous and A, Ghazal and W, Arafat and M, Hashish and JJ, Gruber and N, Jaeger and M, Snyder and K, Patel and S, Bowman and T, Davis and D, Kraushaar and A, Emerman and S, Russello and N, Henig and C, Hendrickson and K, Zhang and M, Rodriguez-Dorantes and CD, Cruz-Hernandez and CDP, Garcia-Tobilla and S, Solorzano-Rosales and N, Jäger and J, Chen and R, Haile and M, Hitchins and JD, Brooks and M, Snyder and S, Jiménez-Morales and M, Ramírez and J, Nuñez and V, Bekker and Y, Leal and E, Jiménez and A, Medina and A, Hidalgo and J, Mejía and V, Halytskiy and J, Naggert and GB, Collin and K, DeMauro and R, Hanusek and PM, Nishina and K, Belhassa and K, Belhassan and L, Bouguenouch and I, Samri and H, Sayel and FZ, moufid and I, El Bouchikhi and S, Trhanint and H, Hamdaoui and I, Elotmani and I, Khtiri and O, Kettani and L, Quibibo and M, Ahagoud and M, Abbassi and K, Ouldim and AV, Marusin and AN, Kornetov and M, Swarovskaya and K, Vagaiceva and V, Stepanov and EMC, De La Paz and R, Sy and J, Nevado and P, Reganit and L, Santos and JD, Magno and FE, Punzalan and D, Ona and E, Llanes and RL, Santos-Cortes and R, Tiongco and J, Aherrera and L, Abrahan and P, Pagauitan-Alan and Group, The Philippine Cardiogenomics Study and KH, Morelli and JS, Domire and N, Pyne and S, Harper and R, Burgess and M, Zhalbinova and A, Akilzhanova and S, Rakhimova and M, Bekbosynova and S, Myrzakhmetova and MA, Gari and A, Dallol and H, Alsehli and A, Gari and M, Gari and A, Abuzenadah and M, Thomas and M, Sukhai and S, Garg and M, Misyura and T, Zhang and A, Schuh and T, Stockley and S, Kamel-Reid and S, Sherry and C, Xiao and D, Slotta and K, Rodarmer and M, Feolo and M, Kimelman and G, Godynskiy and C, O’Sullivan and E, Yaschenko and C, Xiao and E, Yaschenko and S, Sherry and C, Rangel-Escareño and H, Rueda-Zarate and IA, Tayubi and R, Mohammed and 1, on behalf of and I, Ahmed and T, Ahmed and S, Seth and S, Amin and X, Song and X, Mao and H, Sun and RG, Verhaak and A, Futreal and J, Zhang and SJ, Whiite and T, Chiang and A, English and J, Farek and Z, Kahn and W, Salerno and N, Veeraraghavan and E, Boerwinkle and R, Gibbs and T, Kasukawa and M, Lizio and J, Harshbarger and S, Hisashi and J, Severin and A, Imad and S, Sahin and TC, Freeman and K, Baillie and A, Sandelin and P, Carninci and ARR, Forrest and H, Kawaji and Consortium, The FANTOM and W, Salerno and A, English and SN, Shekar and A, Mangubat and J, Bruestle and E, Boerwinkle and RA, Gibbs and AH, Salem and M, Ali and A, Ibrahim and M, Ibrahim and HA, Barrera and L, Garza and JA, Torres and V, Barajas and A, Ulloa-Aguirre and D, Kershenobich and S, Mortaji and P, Guizar and E, Loera and K, Moreno and A, De León and D, Monsiváis and J, Gómez and R, Cardiel and JC, Fernandez-Lopez and V, Bonifaz-Peña and C, Rangel-Escareño and A, Hidalgo-Miranda and AV, Contreras and L, Polfus and Groups, CHARGE and NHLBI Exome Sequence Project Working and X, Wang and V, Philip and G, Carter and AA, Abuzenadah and M, Gari and R, Turki and A, Dallol and A, Uyar and A, Kaygun and S, Zaman and E, Marquez and J, George and D, Ucar and CL, Hendrickson and A, Emerman and D, Kraushaar and S, Bowman and N, Henig and T, Davis and S, Russello and K, Patel and DB, Starr and M, Baird and B, Kirkpatrick and K, Sheets and R, Nitsche and L, Prieto-Lafuente and M, Landrum and J, Lee and W, Rubinstein and D, Maglott and PKR, Thavanati and AE, de Dios and REN, Hernandez and MEA, Aldrate and MRR, Mejia and KRR, Kanala and Z, Abduljaleel and W, Khan and FA, Al-Allaf and M, Athar and MM, Taher and N, Shahzad and A, Bouazzaoui and E, Huber and A, Dan and FA, Al-Allaf and W, Herr and G, Sprotte and J, Köstler and A, Hiergeist and A, Gessner and R, Andreesen and E, Holler and F, Al-Allaf and A, Alashwal and Z, Abduljaleel and M, Taher and A, Bouazzaoui and H, Abalkhail and A, Al-Allaf and R, Bamardadh and M, Athar and O, Filiptsova and M, Kobets and Y, Kobets and I, Burlaka and I, Timoshyna and O, Filiptsova and MN, Kobets and Y, Kobets and I, Burlaka and I, Timoshyna and O, Filiptsova and MN, Kobets and Y, Kobets and I, Burlaka and I, Timoshyna and FA, Al-allaf and MT, Mohiuddin and A, Zainularifeen and A, Mohammed and H, Abalkhail and T, Owaidah and A, Bouazzaoui},
  year = {2016},
  journal = {Human genomics},
  volume = {10 Suppl 1},
  number = {Suppl 1},
  pages = {12},
  doi = {10.1186/s40246-016-0063-5},
  pmid = {27294413},
}
```


## Jovian

**Tool URL**: [https://github.com/DennisSchmitz/Jovian](https://github.com/DennisSchmitz/Jovian)

### Publication Information

**Title**: Accessible viral metagenomics for public health and clinical domains with Jovian.

**Journal**: Scientific reports

**Year**: 2024

**Identifiers**:
- DOI: [10.1038/s41598-024-73785-y](https://doi.org/10.1038/s41598-024-73785-y)
- PMID: [39472593](https://pubmed.ncbi.nlm.nih.gov/39472593/)

### Cite this tool

**APA format**:

```
D, S., F, Z., S, N., TKS, J., J, C., R, V., ... M, d.G. (2024). Accessible viral metagenomics for public health and clinical domains with Jovian. Scientific reports, 14(1), 26018. https://doi.org/10.1038/s41598-024-73785-y
```

**BibTeX format**:

```bibtex
@article{d2024,
  title = {Accessible viral metagenomics for public health and clinical domains with Jovian},
  author = {D, Schmitz and F, Zwagemaker and S, Nooij and TKS, Janssens and J, Cremer and R, Verhagen and H, Vennema and A, Kroneman and MPG, Koopmans and JFJ, Laros and M, de Graaf},
  year = {2024},
  journal = {Scientific reports},
  volume = {14},
  number = {1},
  pages = {26018},
  doi = {10.1038/s41598-024-73785-y},
  pmid = {39472593},
}
```


## LazyPipe

**Tool URL**: [https://www.helsinki.fi/en/projects/lazypipe](https://www.helsinki.fi/en/projects/lazypipe)

### Publication Information

**Title**: Enhanced Viral Metagenomics with Lazypipe 2.

**Journal**: Viruses

**Year**: 2023

**Identifiers**:
- DOI: [10.3390/v15020431](https://doi.org/10.3390/v15020431)
- PMID: [36851645](https://pubmed.ncbi.nlm.nih.gov/36851645/)

### Cite this tool

**APA format**:

```
I, P., O, V., T, S., R, K., & T, S. (2023). Enhanced Viral Metagenomics with Lazypipe 2. Viruses, 15(2). https://doi.org/10.3390/v15020431
```

**BibTeX format**:

```bibtex
@article{i2023,
  title = {Enhanced Viral Metagenomics with Lazypipe 2},
  author = {I, Plyusnin and O, Vapalahti and T, Sironen and R, Kant and T, Smura},
  year = {2023},
  journal = {Viruses},
  volume = {15},
  number = {2},
  doi = {10.3390/v15020431},
  pmid = {36851645},
}
```


## MARVEL

**Tool URL**: [https://github.com/LaboratorioBioinformatica/MARVEL](https://github.com/LaboratorioBioinformatica/MARVEL)

### Publication Information

**Title**: MARVEL, a Tool for Prediction of Bacteriophage Sequences in Metagenomic Bins.

**Journal**: Frontiers in genetics

**Year**: 2018

**Identifiers**:
- DOI: [10.3389/fgene.2018.00304](https://doi.org/10.3389/fgene.2018.00304)
- PMID: [30131825](https://pubmed.ncbi.nlm.nih.gov/30131825/)

### Cite this tool

**APA format**:

```
D, A., LPP, B., AM, d.S., & JC, S. (2018). MARVEL, a Tool for Prediction of Bacteriophage Sequences in Metagenomic Bins. Frontiers in genetics, 9, 304. https://doi.org/10.3389/fgene.2018.00304
```

**BibTeX format**:

```bibtex
@article{d2018,
  title = {MARVEL, a Tool for Prediction of Bacteriophage Sequences in Metagenomic Bins},
  author = {D, Amgarten and LPP, Braga and AM, da Silva and JC, Setubal},
  year = {2018},
  journal = {Frontiers in genetics},
  volume = {9},
  pages = {304},
  doi = {10.3389/fgene.2018.00304},
  pmid = {30131825},
}
```


## MAVERIC

**Tool URL**: [https://web.archive.org/web/20210613041214/https://github.com/GreenleafLab/MAVERIC](https://web.archive.org/web/20210613041214/https://github.com/GreenleafLab/MAVERIC)

### Publication Information

**Title**: Development and testing of an automated computer tablet-based method for self-testing of high and low contrast near visual acuity in ophthalmic patients.

**Journal**: Graefe's archive for clinical and experimental ophthalmology = Albrecht von Graefes Archiv fur klinische und experimentelle Ophthalmologie

**Year**: 2016

**Identifiers**:
- DOI: [10.1007/s00417-016-3293-2](https://doi.org/10.1007/s00417-016-3293-2)
- PMID: [26899899](https://pubmed.ncbi.nlm.nih.gov/26899899/)

### Cite this tool

**APA format**:

```
TM, A., NR, P., IJ, M., M, S., CD, C., N, M., ... HJ, T. (2016). Development and testing of an automated computer tablet-based method for self-testing of high and low contrast near visual acuity in ophthalmic patients. Graefe's archive for clinical and experimental ophthalmology = Albrecht von Graefes Archiv fur klinische und experimentelle Ophthalmologie, 254(5), 891-9. https://doi.org/10.1007/s00417-016-3293-2
```

**BibTeX format**:

```bibtex
@article{tm2016,
  title = {Development and testing of an automated computer tablet-based method for self-testing of high and low contrast near visual acuity in ophthalmic patients},
  author = {TM, Aslam and NR, Parry and IJ, Murray and M, Salleh and CD, Col and N, Mirza and G, Czanner and HJ, Tahir},
  year = {2016},
  journal = {Graefe's archive for clinical and experimental ophthalmology = Albrecht von Graefes Archiv fur klinische und experimentelle Ophthalmologie},
  volume = {254},
  number = {5},
  pages = {891-9},
  doi = {10.1007/s00417-016-3293-2},
  pmid = {26899899},
}
```


## MetaCerberus

**Tool URL**: [https://github.com/raw-lab/MetaCerberus](https://github.com/raw-lab/MetaCerberus)

### Publication Information

**Title**: MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/bioinformatics/btae119](https://doi.org/10.1093/bioinformatics/btae119)
- PMID: [38426351](https://pubmed.ncbi.nlm.nih.gov/38426351/)

### Cite this tool

**APA format**:

```
JL, F.I., E, D., M, B., CR, B., & RA, W.I. (2024). MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life. Bioinformatics (Oxford, England), 40(3). https://doi.org/10.1093/bioinformatics/btae119
```

**BibTeX format**:

```bibtex
@article{jl2024,
  title = {MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life},
  author = {JL, Figueroa Iii and E, Dhungel and M, Bellanger and CR, Brouwer and RA, White Iii},
  year = {2024},
  journal = {Bioinformatics (Oxford, England)},
  volume = {40},
  number = {3},
  doi = {10.1093/bioinformatics/btae119},
  pmid = {38426351},
}
```


## metaPhage

**Tool URL**: [https://mattiapandolfovr.github.io/MetaPhage/](https://mattiapandolfovr.github.io/MetaPhage/)

### Publication Information

**Title**: MetaPhage: an Automated Pipeline for Analyzing, Annotating, and Classifying Bacteriophages in Metagenomics Sequencing Data.

**Journal**: mSystems

**Year**: 2022

**Identifiers**:
- DOI: [10.1128/msystems.00741-22](https://doi.org/10.1128/msystems.00741-22)
- PMID: [36069454](https://pubmed.ncbi.nlm.nih.gov/36069454/)

### Cite this tool

**APA format**:

```
M, P., A, T., G, L., EM, A., & N, V. (2022). MetaPhage: an Automated Pipeline for Analyzing, Annotating, and Classifying Bacteriophages in Metagenomics Sequencing Data. mSystems, 7(5), e0074122. https://doi.org/10.1128/msystems.00741-22
```

**BibTeX format**:

```bibtex
@article{m2022,
  title = {MetaPhage: an Automated Pipeline for Analyzing, Annotating, and Classifying Bacteriophages in Metagenomics Sequencing Data},
  author = {M, Pandolfo and A, Telatin and G, Lazzari and EM, Adriaenssens and N, Vitulo},
  year = {2022},
  journal = {mSystems},
  volume = {7},
  number = {5},
  pages = {e0074122},
  doi = {10.1128/msystems.00741-22},
  pmid = {36069454},
}
```


## MetaPhinder

**Tool URL**: [https://github.com/vanessajurtz/MetaPhinder](https://github.com/vanessajurtz/MetaPhinder)

### Publication Information

**Title**: MetaPhinder-Identifying Bacteriophage Sequences in Metagenomic Data Sets.

**Journal**: PloS one

**Year**: 2016

**Identifiers**:
- DOI: [10.1371/journal.pone.0163111](https://doi.org/10.1371/journal.pone.0163111)
- PMID: [27684958](https://pubmed.ncbi.nlm.nih.gov/27684958/)

### Cite this tool

**APA format**:

```
VI, J., J, V., O, L., M, V.L., & M, N. (2016). MetaPhinder-Identifying Bacteriophage Sequences in Metagenomic Data Sets. PloS one, 11(9), e0163111. https://doi.org/10.1371/journal.pone.0163111
```

**BibTeX format**:

```bibtex
@article{vi2016,
  title = {MetaPhinder-Identifying Bacteriophage Sequences in Metagenomic Data Sets},
  author = {VI, Jurtz and J, Villarroel and O, Lund and M, Voldby Larsen and M, Nielsen},
  year = {2016},
  journal = {PloS one},
  volume = {11},
  number = {9},
  pages = {e0163111},
  doi = {10.1371/journal.pone.0163111},
  pmid = {27684958},
}
```


## MetaProdigal

**Tool URL**: [https://github.com/hyattpd/Prodigal](https://github.com/hyattpd/Prodigal)

### Publication Information

**Title**: Gene and translation initiation site prediction in metagenomic sequences.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2012

**Identifiers**:
- DOI: [10.1093/bioinformatics/bts429](https://doi.org/10.1093/bioinformatics/bts429)
- PMID: [22796954](https://pubmed.ncbi.nlm.nih.gov/22796954/)

### Cite this tool

**APA format**:

```
D, H., PF, L., LJ, H., & EC, U. (2012). Gene and translation initiation site prediction in metagenomic sequences. Bioinformatics (Oxford, England), 28(17), 2223-30. https://doi.org/10.1093/bioinformatics/bts429
```

**BibTeX format**:

```bibtex
@article{d2012,
  title = {Gene and translation initiation site prediction in metagenomic sequences},
  author = {D, Hyatt and PF, LoCascio and LJ, Hauser and EC, Uberbacher},
  year = {2012},
  journal = {Bioinformatics (Oxford, England)},
  volume = {28},
  number = {17},
  pages = {2223-30},
  doi = {10.1093/bioinformatics/bts429},
  pmid = {22796954},
}
```


## metaviralSPAdes

**Tool URL**: [https://github.com/ablab/spades/tree/metaviral_publication](https://github.com/ablab/spades/tree/metaviral_publication)

### Publication Information

**Title**: Metaviral SPAdes: assembly of viruses from metagenomic data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa490](https://doi.org/10.1093/bioinformatics/btaa490)
- PMID: [32413137](https://pubmed.ncbi.nlm.nih.gov/32413137/)

### Cite this tool

**APA format**:

```
D, A., M, R., A, L., & PA, P. (2020). Metaviral SPAdes: assembly of viruses from metagenomic data. Bioinformatics (Oxford, England), 36(14), 4126-4129. https://doi.org/10.1093/bioinformatics/btaa490
```

**BibTeX format**:

```bibtex
@article{d2020,
  title = {Metaviral SPAdes: assembly of viruses from metagenomic data},
  author = {D, Antipov and M, Raiko and A, Lapidus and PA, Pevzner},
  year = {2020},
  journal = {Bioinformatics (Oxford, England)},
  volume = {36},
  number = {14},
  pages = {4126-4129},
  doi = {10.1093/bioinformatics/btaa490},
  pmid = {32413137},
}
```


## mMGE

**Tool URL**: [https://mai.fudan.edu.cn/mgedb/client/index.html#/](https://mai.fudan.edu.cn/mgedb/client/index.html#/)

### Publication Information

**Title**: mMGE: a database for human metagenomic extrachromosomal mobile genetic elements.

**Journal**: Nucleic acids research

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/nar/gkaa869](https://doi.org/10.1093/nar/gkaa869)
- PMID: [33074335](https://pubmed.ncbi.nlm.nih.gov/33074335/)

### Cite this tool

**APA format**:

```
S, L., L, J., B, S., S, P., J, Z., Y, D., ... XM, Z. (2021). mMGE: a database for human metagenomic extrachromosomal mobile genetic elements. Nucleic acids research, 49(D1), D783-D791. https://doi.org/10.1093/nar/gkaa869
```

**BibTeX format**:

```bibtex
@article{s2021,
  title = {mMGE: a database for human metagenomic extrachromosomal mobile genetic elements},
  author = {S, Lai and L, Jia and B, Subramanian and S, Pan and J, Zhang and Y, Dong and WH, Chen and XM, Zhao},
  year = {2021},
  journal = {Nucleic acids research},
  volume = {49},
  number = {D1},
  pages = {D783-D791},
  doi = {10.1093/nar/gkaa869},
  pmid = {33074335},
}
```


## MVP

**Tool URL**: [https://web.archive.org/web/20201204203350/http://mvp.medgenius.info/home](https://web.archive.org/web/20201204203350/http://mvp.medgenius.info/home)

### Publication Information

**Title**: MVP: a modular viromics pipeline to identify, filter, cluster, annotate, and bin viruses from metagenomes.

**Journal**: mSystems

**Year**: 2024

**Identifiers**:
- DOI: [10.1128/msystems.00888-24](https://doi.org/10.1128/msystems.00888-24)
- PMID: [39352141](https://pubmed.ncbi.nlm.nih.gov/39352141/)

### Cite this tool

**APA format**:

```
C, C., AP, C., & S, R. (2024). MVP: a modular viromics pipeline to identify, filter, cluster, annotate, and bin viruses from metagenomes. mSystems, 9(10), e0088824. https://doi.org/10.1128/msystems.00888-24
```

**BibTeX format**:

```bibtex
@article{c2024,
  title = {MVP: a modular viromics pipeline to identify, filter, cluster, annotate, and bin viruses from metagenomes},
  author = {C, Coclet and AP, Camargo and S, Roux},
  year = {2024},
  journal = {mSystems},
  volume = {9},
  number = {10},
  pages = {e0088824},
  doi = {10.1128/msystems.00888-24},
  pmid = {39352141},
}
```


## NCBI GenBank

**Tool URL**: [https://www.ncbi.nlm.nih.gov/genbank/](https://www.ncbi.nlm.nih.gov/genbank/)

### Publication Information

**Title**: VAPiD: a lightweight cross-platform viral annotation pipeline and identification tool to facilitate virus genome submissions to NCBI GenBank.

**Journal**: BMC bioinformatics

**Year**: 2019

**Identifiers**:
- DOI: [10.1186/s12859-019-2606-y](https://doi.org/10.1186/s12859-019-2606-y)
- PMID: [30674273](https://pubmed.ncbi.nlm.nih.gov/30674273/)

### Cite this tool

**APA format**:

```
RC, S., N, M., GD, S., MJ, L., & AL, G. (2019). VAPiD: a lightweight cross-platform viral annotation pipeline and identification tool to facilitate virus genome submissions to NCBI GenBank. BMC bioinformatics, 20(1), 48. https://doi.org/10.1186/s12859-019-2606-y
```

**BibTeX format**:

```bibtex
@article{rc2019,
  title = {VAPiD: a lightweight cross-platform viral annotation pipeline and identification tool to facilitate virus genome submissions to NCBI GenBank},
  author = {RC, Shean and N, Makhsous and GD, Stoddard and MJ, Lin and AL, Greninger},
  year = {2019},
  journal = {BMC bioinformatics},
  volume = {20},
  number = {1},
  pages = {48},
  doi = {10.1186/s12859-019-2606-y},
  pmid = {30674273},
}
```


## NCBI RefSeq

**Tool URL**: [https://www.ncbi.nlm.nih.gov/refseq/](https://www.ncbi.nlm.nih.gov/refseq/)

### Publication Information

**Title**: NCBI RefSeq: reference sequence standards through 25 years of curation and annotation.

**Journal**: Nucleic acids research

**Year**: 2025

**Identifiers**:
- DOI: [10.1093/nar/gkae1038](https://doi.org/10.1093/nar/gkae1038)
- PMID: [39526381](https://pubmed.ncbi.nlm.nih.gov/39526381/)

### Cite this tool

**APA format**:

```
T, G., VK, K., S, P., V, B., B, R., CM, F., ... TD, M. (2025). NCBI RefSeq: reference sequence standards through 25 years of curation and annotation. Nucleic acids research, 53(D1), D243-D257. https://doi.org/10.1093/nar/gkae1038
```

**BibTeX format**:

```bibtex
@article{t2025,
  title = {NCBI RefSeq: reference sequence standards through 25 years of curation and annotation},
  author = {T, Goldfarb and VK, Kodali and S, Pujar and V, Brover and B, Robbertse and CM, Farrell and DH, Oh and A, Astashyn and O, Ermolaeva and D, Haddad and W, Hlavina and J, Hoffman and JD, Jackson and VS, Joardar and D, Kristensen and P, Masterson and KM, McGarvey and R, McVeigh and E, Mozes and MR, Murphy and SS, Schafer and A, Souvorov and B, Spurrier and PK, Strope and H, Sun and AR, Vatsan and C, Wallin and D, Webb and JR, Brister and E, Hatcher and A, Kimchi and W, Klimke and A, Marchler-Bauer and KD, Pruitt and F, Thibaud-Nissen and TD, Murphy},
  year = {2025},
  journal = {Nucleic acids research},
  volume = {53},
  number = {D1},
  pages = {D243-D257},
  doi = {10.1093/nar/gkae1038},
  pmid = {39526381},
}
```


## NetMHCpan

**Tool URL**: [https://services.healthtech.dtu.dk/service.php?NetMHCpan-4.1](https://services.healthtech.dtu.dk/service.php?NetMHCpan-4.1)

### Publication Information

**Title**: NetMHCpan-4.1 and NetMHCIIpan-4.0: improved predictions of MHC antigen presentation by concurrent motif deconvolution and integration of MS MHC eluted ligand data.

**Journal**: Nucleic acids research

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/nar/gkaa379](https://doi.org/10.1093/nar/gkaa379)
- PMID: [32406916](https://pubmed.ncbi.nlm.nih.gov/32406916/)

### Cite this tool

**APA format**:

```
B, R., B, A., S, P., B, P., & M, N. (2020). NetMHCpan-4.1 and NetMHCIIpan-4.0: improved predictions of MHC antigen presentation by concurrent motif deconvolution and integration of MS MHC eluted ligand data. Nucleic acids research, 48(W1), W449-W454. https://doi.org/10.1093/nar/gkaa379
```

**BibTeX format**:

```bibtex
@article{b2020,
  title = {NetMHCpan-4.1 and NetMHCIIpan-4.0: improved predictions of MHC antigen presentation by concurrent motif deconvolution and integration of MS MHC eluted ligand data},
  author = {B, Reynisson and B, Alvarez and S, Paul and B, Peters and M, Nielsen},
  year = {2020},
  journal = {Nucleic acids research},
  volume = {48},
  number = {W1},
  pages = {W449-W454},
  doi = {10.1093/nar/gkaa379},
  pmid = {32406916},
}
```


## NetNGlyc

**Tool URL**: [https://services.healthtech.dtu.dk/service.php?NetNGlyc-1.0](https://services.healthtech.dtu.dk/service.php?NetNGlyc-1.0)

### Publication Information

**Title**: GlycoMine: a machine learning-based approach for predicting N-, C- and O-linked glycosylation in the human proteome.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2015

**Identifiers**:
- DOI: [10.1093/bioinformatics/btu852](https://doi.org/10.1093/bioinformatics/btu852)
- PMID: [25568279](https://pubmed.ncbi.nlm.nih.gov/25568279/)

### Cite this tool

**APA format**:

```
F, L., C, L., M, W., GI, W., Y, Z., JC, W., & J, S. (2015). GlycoMine: a machine learning-based approach for predicting N-, C- and O-linked glycosylation in the human proteome. Bioinformatics (Oxford, England), 31(9), 1411-9. https://doi.org/10.1093/bioinformatics/btu852
```

**BibTeX format**:

```bibtex
@article{f2015,
  title = {GlycoMine: a machine learning-based approach for predicting N-, C- and O-linked glycosylation in the human proteome},
  author = {F, Li and C, Li and M, Wang and GI, Webb and Y, Zhang and JC, Whisstock and J, Song},
  year = {2015},
  journal = {Bioinformatics (Oxford, England)},
  volume = {31},
  number = {9},
  pages = {1411-9},
  doi = {10.1093/bioinformatics/btu852},
  pmid = {25568279},
}
```


## OLGenie

**Tool URL**: [https://github.com/chasewnelson/OLGenie](https://github.com/chasewnelson/OLGenie)

### Publication Information

**Title**: OLGenie: Estimating Natural Selection to Predict Functional Overlapping Genes.

**Journal**: Molecular biology and evolution

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/molbev/msaa087](https://doi.org/10.1093/molbev/msaa087)
- PMID: [32243542](https://pubmed.ncbi.nlm.nih.gov/32243542/)

### Cite this tool

**APA format**:

```
CW, N., Z, A., & X, W. (2020). OLGenie: Estimating Natural Selection to Predict Functional Overlapping Genes. Molecular biology and evolution, 37(8), 2440-2449. https://doi.org/10.1093/molbev/msaa087
```

**BibTeX format**:

```bibtex
@article{cw2020,
  title = {OLGenie: Estimating Natural Selection to Predict Functional Overlapping Genes},
  author = {CW, Nelson and Z, Ardern and X, Wei},
  year = {2020},
  journal = {Molecular biology and evolution},
  volume = {37},
  number = {8},
  pages = {2440-2449},
  doi = {10.1093/molbev/msaa087},
  pmid = {32243542},
}
```


## Paleovirology

**Tool URL**: [https://github.com/giffordlabcvr/DIGS-tool](https://github.com/giffordlabcvr/DIGS-tool)

### Publication Information

**Title**: Paleovirology and virally derived immunity.

**Journal**: Trends in ecology & evolution

**Year**: 2012

**Identifiers**:
- DOI: [10.1016/j.tree.2012.07.007](https://doi.org/10.1016/j.tree.2012.07.007)
- PMID: [22901901](https://pubmed.ncbi.nlm.nih.gov/22901901/)

### Cite this tool

**APA format**:

```
A, A., & A, K. (2012). Paleovirology and virally derived immunity. Trends in ecology & evolution, 27(11), 627-36. https://doi.org/10.1016/j.tree.2012.07.007
```

**BibTeX format**:

```bibtex
@article{a2012,
  title = {Paleovirology and virally derived immunity},
  author = {A, Aswad and A, Katzourakis},
  year = {2012},
  journal = {Trends in ecology & evolution},
  volume = {27},
  number = {11},
  pages = {627-36},
  doi = {10.1016/j.tree.2012.07.007},
  pmid = {22901901},
}
```


## PastML

**Tool URL**: [https://github.com/evolbioinfo/pastml](https://github.com/evolbioinfo/pastml)

### Publication Information

**Title**: Topiary: Pruning the manual labor from ancestral sequence reconstruction.

**Journal**: Protein science : a publication of the Protein Society

**Year**: 2023

**Identifiers**:
- DOI: [10.1002/pro.4551](https://doi.org/10.1002/pro.4551)
- PMID: [36565302](https://pubmed.ncbi.nlm.nih.gov/36565302/)

### Cite this tool

**APA format**:

```
KN, O., SR, P., ZR, S., JL, H., & MJ, H. (2023). Topiary: Pruning the manual labor from ancestral sequence reconstruction. Protein science : a publication of the Protein Society, 32(2), e4551. https://doi.org/10.1002/pro.4551
```

**BibTeX format**:

```bibtex
@article{kn2023,
  title = {Topiary: Pruning the manual labor from ancestral sequence reconstruction},
  author = {KN, Orlandi and SR, Phillips and ZR, Sailer and JL, Harman and MJ, Harms},
  year = {2023},
  journal = {Protein science : a publication of the Protein Society},
  volume = {32},
  number = {2},
  pages = {e4551},
  doi = {10.1002/pro.4551},
  pmid = {36565302},
}
```


## PB-LKS

**Tool URL**: [https://github.com/wanchunnie/PB-LKS](https://github.com/wanchunnie/PB-LKS)

### Publication Information

**Title**: PB-LKS: a python package for predicting phage-bacteria interaction through local K-mer strategy.

**Journal**: Briefings in bioinformatics

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/bib/bbae010](https://doi.org/10.1093/bib/bbae010)
- PMID: [38344864](https://pubmed.ncbi.nlm.nih.gov/38344864/)

### Cite this tool

**APA format**:

```
J, Q., W, N., H, D., J, D., Y, W., D, L., ... T, Q. (2024). PB-LKS: a python package for predicting phage-bacteria interaction through local K-mer strategy. Briefings in bioinformatics, 25(2). https://doi.org/10.1093/bib/bbae010
```

**BibTeX format**:

```bibtex
@article{j2024,
  title = {PB-LKS: a python package for predicting phage-bacteria interaction through local K-mer strategy},
  author = {J, Qiu and W, Nie and H, Ding and J, Dai and Y, Wei and D, Li and Y, Zhang and J, Xie and X, Tian and N, Wu and T, Qiu},
  year = {2024},
  journal = {Briefings in bioinformatics},
  volume = {25},
  number = {2},
  doi = {10.1093/bib/bbae010},
  pmid = {38344864},
}
```


## pGlyco

**Tool URL**: [https://github.com/pFindStudio/pGlyco3](https://github.com/pFindStudio/pGlyco3)

### Publication Information

**Title**: pGlyco: a pipeline for the identification of intact N-glycopeptides by using HCD- and CID-MS/MS and MS3.

**Journal**: Scientific reports

**Year**: 2016

**Identifiers**:
- DOI: [10.1038/srep25102](https://doi.org/10.1038/srep25102)
- PMID: [27139140](https://pubmed.ncbi.nlm.nih.gov/27139140/)

### Cite this tool

**APA format**:

```
WF, Z., MQ, L., Y, Z., JQ, W., P, F., C, P., ... P, Y. (2016). pGlyco: a pipeline for the identification of intact N-glycopeptides by using HCD- and CID-MS/MS and MS3. Scientific reports, 6, 25102. https://doi.org/10.1038/srep25102
```

**BibTeX format**:

```bibtex
@article{wf2016,
  title = {pGlyco: a pipeline for the identification of intact N-glycopeptides by using HCD- and CID-MS/MS and MS3},
  author = {WF, Zeng and MQ, Liu and Y, Zhang and JQ, Wu and P, Fang and C, Peng and A, Nie and G, Yan and W, Cao and C, Liu and H, Chi and RX, Sun and CC, Wong and SM, He and P, Yang},
  year = {2016},
  journal = {Scientific reports},
  volume = {6},
  pages = {25102},
  doi = {10.1038/srep25102},
  pmid = {27139140},
}
```


## Phables

**Tool URL**: [https://github.com/Vini2/phables](https://github.com/Vini2/phables)

### Publication Information

**Title**: Phables: from fragmented assemblies to high-quality bacteriophage genomes.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/bioinformatics/btad586](https://doi.org/10.1093/bioinformatics/btad586)
- PMID: [37738590](https://pubmed.ncbi.nlm.nih.gov/37738590/)

### Cite this tool

**APA format**:

```
V, M., MJ, R., P, D., B, P., SK, G., SR, G., ... RA, E. (2023). Phables: from fragmented assemblies to high-quality bacteriophage genomes. Bioinformatics (Oxford, England), 39(10). https://doi.org/10.1093/bioinformatics/btad586
```

**BibTeX format**:

```bibtex
@article{v2023,
  title = {Phables: from fragmented assemblies to high-quality bacteriophage genomes},
  author = {V, Mallawaarachchi and MJ, Roach and P, Decewicz and B, Papudeshi and SK, Giles and SR, Grigson and G, Bouras and RD, Hesse and LK, Inglis and ALK, Hutton and EA, Dinsdale and RA, Edwards},
  year = {2023},
  journal = {Bioinformatics (Oxford, England)},
  volume = {39},
  number = {10},
  doi = {10.1093/bioinformatics/btad586},
  pmid = {37738590},
}
```


## PhaBox

**Tool URL**: [https://phage.ee.cityu.edu.hk/](https://phage.ee.cityu.edu.hk/)

### Publication Information

**Title**: PhaBOX: a web server for identifying and characterizing phage contigs in metagenomic data.

**Journal**: Bioinformatics advances

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/bioadv/vbad101](https://doi.org/10.1093/bioadv/vbad101)
- PMID: [37641717](https://pubmed.ncbi.nlm.nih.gov/37641717/)

### Cite this tool

**APA format**:

```
J, S., C, P., H, L., X, T., & Y, S. (2023). PhaBOX: a web server for identifying and characterizing phage contigs in metagenomic data. Bioinformatics advances, 3(1), vbad101. https://doi.org/10.1093/bioadv/vbad101
```

**BibTeX format**:

```bibtex
@article{j2023,
  title = {PhaBOX: a web server for identifying and characterizing phage contigs in metagenomic data},
  author = {J, Shang and C, Peng and H, Liao and X, Tang and Y, Sun},
  year = {2023},
  journal = {Bioinformatics advances},
  volume = {3},
  number = {1},
  pages = {vbad101},
  doi = {10.1093/bioadv/vbad101},
  pmid = {37641717},
}
```


## PHACTS

**Tool URL**: [https://github.com/deprekate/PHACTS](https://github.com/deprekate/PHACTS)

### Publication Information

**Title**: PHACTS, a computational approach to classifying the lifestyle of phages.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2012

**Identifiers**:
- DOI: [10.1093/bioinformatics/bts014](https://doi.org/10.1093/bioinformatics/bts014)
- PMID: [22238260](https://pubmed.ncbi.nlm.nih.gov/22238260/)

### Cite this tool

**APA format**:

```
K, M., BA, B., & RA, E. (2012). PHACTS, a computational approach to classifying the lifestyle of phages. Bioinformatics (Oxford, England), 28(5), 614-8. https://doi.org/10.1093/bioinformatics/bts014
```

**BibTeX format**:

```bibtex
@article{k2012,
  title = {PHACTS, a computational approach to classifying the lifestyle of phages},
  author = {K, McNair and BA, Bailey and RA, Edwards},
  year = {2012},
  journal = {Bioinformatics (Oxford, England)},
  volume = {28},
  number = {5},
  pages = {614-8},
  doi = {10.1093/bioinformatics/bts014},
  pmid = {22238260},
}
```


## PhaGCN

**Tool URL**: [https://github.com/KennthShang/PhaGCN](https://github.com/KennthShang/PhaGCN)

### Publication Information

**Title**: Bacteriophage classification for assembled contigs using graph convolutional network.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab293](https://doi.org/10.1093/bioinformatics/btab293)
- PMID: [34252923](https://pubmed.ncbi.nlm.nih.gov/34252923/)

### Cite this tool

**APA format**:

```
J, S., J, J., & Y, S. (2021). Bacteriophage classification for assembled contigs using graph convolutional network. Bioinformatics (Oxford, England), 37(Suppl_1), i25-i33. https://doi.org/10.1093/bioinformatics/btab293
```

**BibTeX format**:

```bibtex
@article{j2021,
  title = {Bacteriophage classification for assembled contigs using graph convolutional network},
  author = {J, Shang and J, Jiang and Y, Sun},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {Suppl_1},
  pages = {i25-i33},
  doi = {10.1093/bioinformatics/btab293},
  pmid = {34252923},
}
```


## PhageAI

**Tool URL**: [https://app.phage.ai/](https://app.phage.ai/)

### Publication Information

**Title**: Comparative genomics and proteomics analysis of phages infecting multi-drug resistant Escherichia coli O177 isolated from cattle faeces.

**Journal**: Scientific reports

**Year**: 2023

**Identifiers**:
- DOI: [10.1038/s41598-023-48788-w](https://doi.org/10.1038/s41598-023-48788-w)
- PMID: [38052835](https://pubmed.ncbi.nlm.nih.gov/38052835/)

### Cite this tool

**APA format**:

```
PK, M., AM, K., F, M., RE, P., V, M., & CN, A. (2023). Comparative genomics and proteomics analysis of phages infecting multi-drug resistant Escherichia coli O177 isolated from cattle faeces. Scientific reports, 13(1), 21426. https://doi.org/10.1038/s41598-023-48788-w
```

**BibTeX format**:

```bibtex
@article{pk2023,
  title = {Comparative genomics and proteomics analysis of phages infecting multi-drug resistant Escherichia coli O177 isolated from cattle faeces},
  author = {PK, Montso and AM, Kropinski and F, Mokoena and RE, Pierneef and V, Mlambo and CN, Ateba},
  year = {2023},
  journal = {Scientific reports},
  volume = {13},
  number = {1},
  pages = {21426},
  doi = {10.1038/s41598-023-48788-w},
  pmid = {38052835},
}
```


## PhageClouds

**Tool URL**: [https://doi.org/10.1089/phage.2021.0008](https://doi.org/10.1089/phage.2021.0008)

### Publication Information

**Title**: From Trees to Clouds: PhageClouds for Fast Comparison of ∼640,000 Phage Genomic Sequences and Host-Centric Visualization Using Genomic Network Graphs.

**Journal**: PHAGE (New Rochelle, N.Y.)

**Year**: 2021

**Identifiers**:
- DOI: [10.1089/phage.2021.0008](https://doi.org/10.1089/phage.2021.0008)
- PMID: [36147515](https://pubmed.ncbi.nlm.nih.gov/36147515/)

### Cite this tool

**APA format**:

```
G, R., A, M., S, M., D, S., K, S., A, R., ... T, S. (2021). From Trees to Clouds: PhageClouds for Fast Comparison of ∼640,000 Phage Genomic Sequences and Host-Centric Visualization Using Genomic Network Graphs. PHAGE (New Rochelle, N.Y.), 2(4), 194-203. https://doi.org/10.1089/phage.2021.0008
```

**BibTeX format**:

```bibtex
@article{g2021,
  title = {From Trees to Clouds: PhageClouds for Fast Comparison of ∼640,000 Phage Genomic Sequences and Host-Centric Visualization Using Genomic Network Graphs},
  author = {G, Rangel-Pineros and A, Millard and S, Michniewski and D, Scanlan and K, Sirén and A, Reyes and B, Petersen and MRJ, Clokie and T, Sicheritz-Pontén},
  year = {2021},
  journal = {PHAGE (New Rochelle, N.Y.)},
  volume = {2},
  number = {4},
  pages = {194-203},
  doi = {10.1089/phage.2021.0008},
  pmid = {36147515},
}
```


## PhageDPO

**Tool URL**: [http://bit.ly/phagedpo](http://bit.ly/phagedpo)

### Publication Information

**Title**: PhageDPO: A machine-learning based computational framework for identifying phage depolymerases.

**Journal**: Computers in biology and medicine

**Year**: 2025

**Identifiers**:
- DOI: [10.1016/j.compbiomed.2025.109836](https://doi.org/10.1016/j.compbiomed.2025.109836)
- PMID: [39951981](https://pubmed.ncbi.nlm.nih.gov/39951981/)

### Cite this tool

**APA format**:

```
MF, V., J, D., R, D., H, O., & O, D. (2025). PhageDPO: A machine-learning based computational framework for identifying phage depolymerases. Computers in biology and medicine, 188, 109836. https://doi.org/10.1016/j.compbiomed.2025.109836
```

**BibTeX format**:

```bibtex
@article{mf2025,
  title = {PhageDPO: A machine-learning based computational framework for identifying phage depolymerases},
  author = {MF, Vieira and J, Duarte and R, Domingues and H, Oliveira and O, Dias},
  year = {2025},
  journal = {Computers in biology and medicine},
  volume = {188},
  pages = {109836},
  doi = {10.1016/j.compbiomed.2025.109836},
  pmid = {39951981},
}
```


## PhageHostLearn

**Tool URL**: [https://github.com/dimiboeckaerts/PhageHostLearn](https://github.com/dimiboeckaerts/PhageHostLearn)

### Publication Information

**Title**: Prediction of Klebsiella phage-host specificity at the strain level.

**Journal**: Nature communications

**Year**: 2024

**Identifiers**:
- DOI: [10.1038/s41467-024-48675-6](https://doi.org/10.1038/s41467-024-48675-6)
- PMID: [38778023](https://pubmed.ncbi.nlm.nih.gov/38778023/)

### Cite this tool

**APA format**:

```
D, B., M, S., C, F., J, O., R, S., P, D., ... Y, B. (2024). Prediction of Klebsiella phage-host specificity at the strain level. Nature communications, 15(1), 4355. https://doi.org/10.1038/s41467-024-48675-6
```

**BibTeX format**:

```bibtex
@article{d2024,
  title = {Prediction of Klebsiella phage-host specificity at the strain level},
  author = {D, Boeckaerts and M, Stock and C, Ferriol-González and J, Oteo-Iglesias and R, Sanjuán and P, Domingo-Calap and B, De Baets and Y, Briers},
  year = {2024},
  journal = {Nature communications},
  volume = {15},
  number = {1},
  pages = {4355},
  doi = {10.1038/s41467-024-48675-6},
  pmid = {38778023},
}
```


## PhagePromoter

**Tool URL**: [https://github.com/martaS95/PhagePromoter](https://github.com/martaS95/PhagePromoter)

### Publication Information

**Title**: Predicting promoters in phage genomes using PhagePromoter.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2019

**Identifiers**:
- DOI: [10.1093/bioinformatics/btz580](https://doi.org/10.1093/bioinformatics/btz580)
- PMID: [31359029](https://pubmed.ncbi.nlm.nih.gov/31359029/)

### Cite this tool

**APA format**:

```
M, S., M, R., H, O., & O, D. (2019). Predicting promoters in phage genomes using PhagePromoter. Bioinformatics (Oxford, England), 35(24), 5301-5302. https://doi.org/10.1093/bioinformatics/btz580
```

**BibTeX format**:

```bibtex
@article{m2019,
  title = {Predicting promoters in phage genomes using PhagePromoter},
  author = {M, Sampaio and M, Rocha and H, Oliveira and O, Dias},
  year = {2019},
  journal = {Bioinformatics (Oxford, England)},
  volume = {35},
  number = {24},
  pages = {5301-5302},
  doi = {10.1093/bioinformatics/btz580},
  pmid = {31359029},
}
```


## PhagesDB

**Tool URL**: [https://phagesdb.org/](https://phagesdb.org/)

### Publication Information

**Title**: PhagesDB: the actinobacteriophage database.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2017

**Identifiers**:
- DOI: [10.1093/bioinformatics/btw711](https://doi.org/10.1093/bioinformatics/btw711)
- PMID: [28365761](https://pubmed.ncbi.nlm.nih.gov/28365761/)

### Cite this tool

**APA format**:

```
DA, R., & GF, H. (2017). PhagesDB: the actinobacteriophage database. Bioinformatics (Oxford, England), 33(5), 784-786. https://doi.org/10.1093/bioinformatics/btw711
```

**BibTeX format**:

```bibtex
@article{da2017,
  title = {PhagesDB: the actinobacteriophage database},
  author = {DA, Russell and GF, Hatfull},
  year = {2017},
  journal = {Bioinformatics (Oxford, England)},
  volume = {33},
  number = {5},
  pages = {784-786},
  doi = {10.1093/bioinformatics/btw711},
  pmid = {28365761},
}
```


## PhageTB

**Tool URL**: [https://github.com/raghavagps/phagetb](https://github.com/raghavagps/phagetb)

### Publication Information

**Title**: An ensemble method for prediction of phage-based therapy against bacterial infections.

**Journal**: Frontiers in microbiology

**Year**: 2023

**Identifiers**:
- DOI: [10.3389/fmicb.2023.1148579](https://doi.org/10.3389/fmicb.2023.1148579)
- PMID: [37032893](https://pubmed.ncbi.nlm.nih.gov/37032893/)

### Cite this tool

**APA format**:

```
S, A., A, D., S, P., S, C., A, A., & GPS, R. (2023). An ensemble method for prediction of phage-based therapy against bacterial infections. Frontiers in microbiology, 14, 1148579. https://doi.org/10.3389/fmicb.2023.1148579
```

**BibTeX format**:

```bibtex
@article{s2023,
  title = {An ensemble method for prediction of phage-based therapy against bacterial infections},
  author = {S, Aggarwal and A, Dhall and S, Patiyal and S, Choudhury and A, Arora and GPS, Raghava},
  year = {2023},
  journal = {Frontiers in microbiology},
  volume = {14},
  pages = {1148579},
  doi = {10.3389/fmicb.2023.1148579},
  pmid = {37032893},
}
```


## PhageTerm

**Tool URL**: [https://sourceforge.net/projects/phageterm/](https://sourceforge.net/projects/phageterm/)

### Publication Information

**Title**: PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data.

**Journal**: Scientific reports

**Year**: 2017

**Identifiers**:
- DOI: [10.1038/s41598-017-07910-5](https://doi.org/10.1038/s41598-017-07910-5)
- PMID: [28811656](https://pubmed.ncbi.nlm.nih.gov/28811656/)

### Cite this tool

**APA format**:

```
JR, G., F, D., LC, F., D, B., & M, M. (2017). PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data. Scientific reports, 7(1), 8292. https://doi.org/10.1038/s41598-017-07910-5
```

**BibTeX format**:

```bibtex
@article{jr2017,
  title = {PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data},
  author = {JR, Garneau and F, Depardieu and LC, Fortier and D, Bikard and M, Monot},
  year = {2017},
  journal = {Scientific reports},
  volume = {7},
  number = {1},
  pages = {8292},
  doi = {10.1038/s41598-017-07910-5},
  pmid = {28811656},
}
```


## PhageTerm

**Tool URL**: [https://gitlab.pasteur.fr/vlegrand/ptv/-/releases](https://gitlab.pasteur.fr/vlegrand/ptv/-/releases)

### Publication Information

**Title**: PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data.

**Journal**: Scientific reports

**Year**: 2017

**Identifiers**:
- DOI: [10.1038/s41598-017-07910-5](https://doi.org/10.1038/s41598-017-07910-5)
- PMID: [28811656](https://pubmed.ncbi.nlm.nih.gov/28811656/)

### Cite this tool

**APA format**:

```
JR, G., F, D., LC, F., D, B., & M, M. (2017). PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data. Scientific reports, 7(1), 8292. https://doi.org/10.1038/s41598-017-07910-5
```

**BibTeX format**:

```bibtex
@article{jr2017,
  title = {PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data},
  author = {JR, Garneau and F, Depardieu and LC, Fortier and D, Bikard and M, Monot},
  year = {2017},
  journal = {Scientific reports},
  volume = {7},
  number = {1},
  pages = {8292},
  doi = {10.1038/s41598-017-07910-5},
  pmid = {28811656},
}
```


## PhageWeb

**Tool URL**: [http://computationalbiology.ufpa.br/phageweb/](http://computationalbiology.ufpa.br/phageweb/)

### Publication Information

**Title**: PhageWeb - Web Interface for Rapid Identification and Characterization of Prophages in Bacterial Genomes.

**Journal**: Frontiers in genetics

**Year**: 2018

**Identifiers**:
- DOI: [10.3389/fgene.2018.00644](https://doi.org/10.3389/fgene.2018.00644)
- PMID: [30619469](https://pubmed.ncbi.nlm.nih.gov/30619469/)

### Cite this tool

**APA format**:

```
AL, d.S., D, M., A, L., EF, F., K, P., F, A., ... RTJ, R. (2018). PhageWeb - Web Interface for Rapid Identification and Characterization of Prophages in Bacterial Genomes. Frontiers in genetics, 9, 644. https://doi.org/10.3389/fgene.2018.00644
```

**BibTeX format**:

```bibtex
@article{al2018,
  title = {PhageWeb - Web Interface for Rapid Identification and Characterization of Prophages in Bacterial Genomes},
  author = {AL, de Sousa and D, Maués and A, Lobato and EF, Franco and K, Pinheiro and F, Araújo and Y, Pantoja and AL, da Costa da Silva and J, Morais and RTJ, Ramos},
  year = {2018},
  journal = {Frontiers in genetics},
  volume = {9},
  pages = {644},
  doi = {10.3389/fgene.2018.00644},
  pmid = {30619469},
}
```


## phaMers

**Tool URL**: [https://github.com/jondeaton/PhaMers](https://github.com/jondeaton/PhaMers)

### Publication Information

**Title**: Mini-Metagenomics and Nucleotide Composition Aid the Identification and Host Association of Novel Bacteriophage Sequences.

**Journal**: Advanced biosystems

**Year**: 2019

**Identifiers**:
- DOI: [10.1002/adbi.201900108](https://doi.org/10.1002/adbi.201900108)
- PMID: [32648690](https://pubmed.ncbi.nlm.nih.gov/32648690/)

### Cite this tool

**APA format**:

```
J, D., FB, Y., & SR, Q. (2019). Mini-Metagenomics and Nucleotide Composition Aid the Identification and Host Association of Novel Bacteriophage Sequences. Advanced biosystems, 3(11), e1900108. https://doi.org/10.1002/adbi.201900108
```

**BibTeX format**:

```bibtex
@article{j2019,
  title = {Mini-Metagenomics and Nucleotide Composition Aid the Identification and Host Association of Novel Bacteriophage Sequences},
  author = {J, Deaton and FB, Yu and SR, Quake},
  year = {2019},
  journal = {Advanced biosystems},
  volume = {3},
  number = {11},
  pages = {e1900108},
  doi = {10.1002/adbi.201900108},
  pmid = {32648690},
}
```


## PhANNs

**Tool URL**: [https://github.com/Adrian-Cantu/PhANNs](https://github.com/Adrian-Cantu/PhANNs)

### Publication Information

**Title**: PhANNs, a fast and accurate tool and web server to classify phage structural proteins.

**Journal**: PLoS computational biology

**Year**: 2020

**Identifiers**:
- DOI: [10.1371/journal.pcbi.1007845](https://doi.org/10.1371/journal.pcbi.1007845)
- PMID: [33137102](https://pubmed.ncbi.nlm.nih.gov/33137102/)

### Cite this tool

**APA format**:

```
VA, C., P, S., V, S., J, R., D, S., RA, E., & AM, S. (2020). PhANNs, a fast and accurate tool and web server to classify phage structural proteins. PLoS computational biology, 16(11), e1007845. https://doi.org/10.1371/journal.pcbi.1007845
```

**BibTeX format**:

```bibtex
@article{va2020,
  title = {PhANNs, a fast and accurate tool and web server to classify phage structural proteins},
  author = {VA, Cantu and P, Salamon and V, Seguritan and J, Redfield and D, Salamon and RA, Edwards and AM, Segall},
  year = {2020},
  journal = {PLoS computational biology},
  volume = {16},
  number = {11},
  pages = {e1007845},
  doi = {10.1371/journal.pcbi.1007845},
  pmid = {33137102},
}
```


## PHANOTATE

**Tool URL**: [https://github.com/deprekate/PHANOTATE](https://github.com/deprekate/PHANOTATE)

### Publication Information

**Title**: PHANOTATE: a novel approach to gene identification in phage genomes.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2019

**Identifiers**:
- DOI: [10.1093/bioinformatics/btz265](https://doi.org/10.1093/bioinformatics/btz265)
- PMID: [31329826](https://pubmed.ncbi.nlm.nih.gov/31329826/)

### Cite this tool

**APA format**:

```
K, M., C, Z., EA, D., B, S., & RA, E. (2019). PHANOTATE: a novel approach to gene identification in phage genomes. Bioinformatics (Oxford, England), 35(22), 4537-4542. https://doi.org/10.1093/bioinformatics/btz265
```

**BibTeX format**:

```bibtex
@article{k2019,
  title = {PHANOTATE: a novel approach to gene identification in phage genomes},
  author = {K, McNair and C, Zhou and EA, Dinsdale and B, Souza and RA, Edwards},
  year = {2019},
  journal = {Bioinformatics (Oxford, England)},
  volume = {35},
  number = {22},
  pages = {4537-4542},
  doi = {10.1093/bioinformatics/btz265},
  pmid = {31329826},
}
```


## Phanotate

**Tool URL**: [https://github.com/deprekate/PHANOTATE](https://github.com/deprekate/PHANOTATE)

### Publication Information

**Title**: Utilizing Amino Acid Composition and Entropy of Potential Open Reading Frames to Identify Protein-Coding Genes.

**Journal**: Microorganisms

**Year**: 2021

**Identifiers**:
- DOI: [10.3390/microorganisms9010129](https://doi.org/10.3390/microorganisms9010129)
- PMID: [33429904](https://pubmed.ncbi.nlm.nih.gov/33429904/)

### Cite this tool

**APA format**:

```
K, M., CL, E.Z., B, S., S, M., & RA, E. (2021). Utilizing Amino Acid Composition and Entropy of Potential Open Reading Frames to Identify Protein-Coding Genes. Microorganisms, 9(1). https://doi.org/10.3390/microorganisms9010129
```

**BibTeX format**:

```bibtex
@article{k2021,
  title = {Utilizing Amino Acid Composition and Entropy of Potential Open Reading Frames to Identify Protein-Coding Genes},
  author = {K, McNair and CL, Ecale Zhou and B, Souza and S, Malfatti and RA, Edwards},
  year = {2021},
  journal = {Microorganisms},
  volume = {9},
  number = {1},
  doi = {10.3390/microorganisms9010129},
  pmid = {33429904},
}
```


## Phanta

**Tool URL**: [https://github.com/bhattlab/phanta](https://github.com/bhattlab/phanta)

### Publication Information

**Title**: ['Phage-inclusive profiling of human gut microbiomes with Phanta']

### Cite this tool

**APA format**:

```
Pinto, Y., Chakraborty, M., Jain, N., & Bhatt, A. (2023). Phage-inclusive profiling of human gut microbiomes with Phanta. Nature Biotechnology, 42(4), 651-662. https://doi.org/10.1038/s41587-023-01799-4
```

**BibTeX format**:

```bibtex
@article{pinto2023,
  title = {Phage-inclusive profiling of human gut microbiomes with Phanta},
  author = {Pinto, Y. and Chakraborty, M. and Jain, N. and Bhatt, A.},
  year = {2023},
  journal = {Nature Biotechnology},
  volume = {42},
  number = {4},
  pages = {651-662},
  doi = {10.1038/s41587-023-01799-4},
}
```


## Pharokka

**Tool URL**: [https://github.com/gbouras13/pharokka](https://github.com/gbouras13/pharokka)

### Publication Information

**Title**: ['Pharokka: a fast scalable bacteriophage annotation tool']

### Cite this tool

**APA format**:

```
Bouras, G., Nepal, R., Houtak, G., Psaltis, A., Wormald, P., & Vreugde, S. (2022). Pharokka: a fast scalable bacteriophage annotation tool. Bioinformatics, 39(1). https://doi.org/10.1093/bioinformatics/btac776
```

**BibTeX format**:

```bibtex
@article{bouras2022,
  title = {Pharokka: a fast scalable bacteriophage annotation tool},
  author = {Bouras, G. and Nepal, R. and Houtak, G. and Psaltis, A. and Wormald, P. and Vreugde, S.},
  year = {2022},
  journal = {Bioinformatics},
  volume = {39},
  number = {1},
  doi = {10.1093/bioinformatics/btac776},
}
```


## PHASTER

**Tool URL**: [https://phaster.ca/](https://phaster.ca/)

### Publication Information

**Title**: PHASTER: a better, faster version of the PHAST phage search tool.

**Journal**: Nucleic acids research

**Year**: 2016

**Identifiers**:
- DOI: [10.1093/nar/gkw387](https://doi.org/10.1093/nar/gkw387)
- PMID: [27141966](https://pubmed.ncbi.nlm.nih.gov/27141966/)

### Cite this tool

**APA format**:

```
D, A., JR, G., A, M., T, S., A, P., Y, L., & DS, W. (2016). PHASTER: a better, faster version of the PHAST phage search tool. Nucleic acids research, 44(W1), W16-21. https://doi.org/10.1093/nar/gkw387
```

**BibTeX format**:

```bibtex
@article{d2016,
  title = {PHASTER: a better, faster version of the PHAST phage search tool},
  author = {D, Arndt and JR, Grant and A, Marcu and T, Sajed and A, Pon and Y, Liang and DS, Wishart},
  year = {2016},
  journal = {Nucleic acids research},
  volume = {44},
  number = {W1},
  pages = {W16-21},
  doi = {10.1093/nar/gkw387},
  pmid = {27141966},
}
```


## PHERI

**Tool URL**: [https://github.com/andynet/pheri](https://github.com/andynet/pheri)

### Publication Information

**Title**: PHERI-Phage Host ExploRation Pipeline.

**Journal**: Microorganisms

**Year**: 2023

**Identifiers**:
- DOI: [10.3390/microorganisms11061398](https://doi.org/10.3390/microorganisms11061398)
- PMID: [37374901](https://pubmed.ncbi.nlm.nih.gov/37374901/)

### Cite this tool

**APA format**:

```
A, B., M, K., J, B., T, S., & J, T. (2023). PHERI-Phage Host ExploRation Pipeline. Microorganisms, 11(6). https://doi.org/10.3390/microorganisms11061398
```

**BibTeX format**:

```bibtex
@article{a2023,
  title = {PHERI-Phage Host ExploRation Pipeline},
  author = {A, Baláž and M, Kajsik and J, Budiš and T, Szemes and J, Turňa},
  year = {2023},
  journal = {Microorganisms},
  volume = {11},
  number = {6},
  doi = {10.3390/microorganisms11061398},
  pmid = {37374901},
}
```


## PHIAF

**Tool URL**: [https://github.com/BioMedicalBigDataMiningLab/PHIAF](https://github.com/BioMedicalBigDataMiningLab/PHIAF)

### Publication Information

**Title**: PHIAF: prediction of phage-host interactions with GAN-based data augmentation and sequence-based feature fusion.

**Journal**: Briefings in bioinformatics

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/bib/bbab348](https://doi.org/10.1093/bib/bbab348)
- PMID: [34472593](https://pubmed.ncbi.nlm.nih.gov/34472593/)

### Cite this tool

**APA format**:

```
M, L., & W, Z. (2022). PHIAF: prediction of phage-host interactions with GAN-based data augmentation and sequence-based feature fusion. Briefings in bioinformatics, 23(1). https://doi.org/10.1093/bib/bbab348
```

**BibTeX format**:

```bibtex
@article{m2022,
  title = {PHIAF: prediction of phage-host interactions with GAN-based data augmentation and sequence-based feature fusion},
  author = {M, Li and W, Zhang},
  year = {2022},
  journal = {Briefings in bioinformatics},
  volume = {23},
  number = {1},
  doi = {10.1093/bib/bbab348},
  pmid = {34472593},
}
```


## Phigaro

**Tool URL**: [https://github.com/bobeobibo/phigaro](https://github.com/bobeobibo/phigaro)

### Publication Information

**Title**: Phigaro: high-throughput prophage sequence annotation.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa250](https://doi.org/10.1093/bioinformatics/btaa250)
- PMID: [32311023](https://pubmed.ncbi.nlm.nih.gov/32311023/)

### Cite this tool

**APA format**:

```
EV, S., PO, T., NA, P., CM, R., EM, Z., EN, I., & VM, G. (2020). Phigaro: high-throughput prophage sequence annotation. Bioinformatics (Oxford, England), 36(12), 3882-3884. https://doi.org/10.1093/bioinformatics/btaa250
```

**BibTeX format**:

```bibtex
@article{ev2020,
  title = {Phigaro: high-throughput prophage sequence annotation},
  author = {EV, Starikova and PO, Tikhonova and NA, Prianichnikov and CM, Rands and EM, Zdobnov and EN, Ilina and VM, Govorun},
  year = {2020},
  journal = {Bioinformatics (Oxford, England)},
  volume = {36},
  number = {12},
  pages = {3882-3884},
  doi = {10.1093/bioinformatics/btaa250},
  pmid = {32311023},
}
```


## PHISDetector

**Tool URL**: [http://www.microbiome-bigdata.com/PHISDetector/index/](http://www.microbiome-bigdata.com/PHISDetector/index/)

### Publication Information

**Title**: PHISDetector: A Tool to Detect Diverse In Silico Phage-host Interaction Signals for Virome Studies.

**Journal**: Genomics, proteomics & bioinformatics

**Year**: 2022

**Identifiers**:
- DOI: [10.1016/j.gpb.2022.02.003](https://doi.org/10.1016/j.gpb.2022.02.003)
- PMID: [35272051](https://pubmed.ncbi.nlm.nih.gov/35272051/)

### Cite this tool

**APA format**:

```
F, Z., R, G., F, Z., C, R., L, Y., Y, S., & Z, H. (2022). PHISDetector: A Tool to Detect Diverse In Silico Phage-host Interaction Signals for Virome Studies. Genomics, proteomics & bioinformatics, 20(3), 508-523. https://doi.org/10.1016/j.gpb.2022.02.003
```

**BibTeX format**:

```bibtex
@article{f2022,
  title = {PHISDetector: A Tool to Detect Diverse In Silico Phage-host Interaction Signals for Virome Studies},
  author = {F, Zhou and R, Gan and F, Zhang and C, Ren and L, Yu and Y, Si and Z, Huang},
  year = {2022},
  journal = {Genomics, proteomics & bioinformatics},
  volume = {20},
  number = {3},
  pages = {508-523},
  doi = {10.1016/j.gpb.2022.02.003},
  pmid = {35272051},
}
```


## PhiSpy

**Tool URL**: [https://github.com/linsalrob/PhiSpy](https://github.com/linsalrob/PhiSpy)

### Publication Information

**Title**: PhiSpy: a novel algorithm for finding prophages in bacterial genomes that combines similarity- and composition-based strategies.

**Journal**: Nucleic acids research

**Year**: 2012

**Identifiers**:
- DOI: [10.1093/nar/gks406](https://doi.org/10.1093/nar/gks406)
- PMID: [22584627](https://pubmed.ncbi.nlm.nih.gov/22584627/)

### Cite this tool

**APA format**:

```
S, A., RK, A., & RA, E. (2012). PhiSpy: a novel algorithm for finding prophages in bacterial genomes that combines similarity- and composition-based strategies. Nucleic acids research, 40(16), e126. https://doi.org/10.1093/nar/gks406
```

**BibTeX format**:

```bibtex
@article{s2012,
  title = {PhiSpy: a novel algorithm for finding prophages in bacterial genomes that combines similarity- and composition-based strategies},
  author = {S, Akhter and RK, Aziz and RA, Edwards},
  year = {2012},
  journal = {Nucleic acids research},
  volume = {40},
  number = {16},
  pages = {e126},
  doi = {10.1093/nar/gks406},
  pmid = {22584627},
}
```


## PHIST

**Tool URL**: [https://github.com/refresh-bio/phist](https://github.com/refresh-bio/phist)

### Publication Information

**Title**: PHIST: fast and accurate prediction of prokaryotic hosts from metagenomic viral sequences.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab837](https://doi.org/10.1093/bioinformatics/btab837)
- PMID: [34904625](https://pubmed.ncbi.nlm.nih.gov/34904625/)

### Cite this tool

**APA format**:

```
A, Z., S, D., & A, G. (2022). PHIST: fast and accurate prediction of prokaryotic hosts from metagenomic viral sequences. Bioinformatics (Oxford, England), 38(5), 1447-1449. https://doi.org/10.1093/bioinformatics/btab837
```

**BibTeX format**:

```bibtex
@article{a2022,
  title = {PHIST: fast and accurate prediction of prokaryotic hosts from metagenomic viral sequences},
  author = {A, Zielezinski and S, Deorowicz and A, Gudyś},
  year = {2022},
  journal = {Bioinformatics (Oxford, England)},
  volume = {38},
  number = {5},
  pages = {1447-1449},
  doi = {10.1093/bioinformatics/btab837},
  pmid = {34904625},
}
```


## PHP

**Tool URL**: [https://github.com/congyulu-bioinfo/PHP](https://github.com/congyulu-bioinfo/PHP)

### Publication Information

**Title**: FCP (http://fibro.biobitfield.com/fcp.php): a bioinformatic tool assisting in PubMed searches for literature on fibrosis-related cytokines.

**Journal**: Arthritis and rheumatism

**Year**: 2003

**Identifiers**:
- DOI: [10.1002/art.11082](https://doi.org/10.1002/art.11082)
- PMID: [12847712](https://pubmed.ncbi.nlm.nih.gov/12847712/)

### Cite this tool

**APA format**:

```
SP, A. (2003). FCP (http://fibro.biobitfield.com/fcp.php): a bioinformatic tool assisting in PubMed searches for literature on fibrosis-related cytokines. Arthritis and rheumatism, 48(7), 2083-4. https://doi.org/10.1002/art.11082
```

**BibTeX format**:

```bibtex
@article{sp2003,
  title = {FCP (http://fibro.biobitfield.com/fcp.php): a bioinformatic tool assisting in PubMed searches for literature on fibrosis-related cytokines},
  author = {SP, Atamas},
  year = {2003},
  journal = {Arthritis and rheumatism},
  volume = {48},
  number = {7},
  pages = {2083-4},
  doi = {10.1002/art.11082},
  pmid = {12847712},
}
```


## PHPGCA

**Tool URL**: [https://github.com/JunPeng-Zhong/PHPGCA](https://github.com/JunPeng-Zhong/PHPGCA)

### Publication Information

**Title**: Prokaryotic virus host prediction with graph contrastive augmentaion.

**Journal**: PLoS computational biology

**Year**: 2023

**Identifiers**:
- DOI: [10.1371/journal.pcbi.1011671](https://doi.org/10.1371/journal.pcbi.1011671)
- PMID: [38039280](https://pubmed.ncbi.nlm.nih.gov/38039280/)

### Cite this tool

**APA format**:

```
ZH, D., JP, Z., Y, L., & JQ, L. (2023). Prokaryotic virus host prediction with graph contrastive augmentaion. PLoS computational biology, 19(12), e1011671. https://doi.org/10.1371/journal.pcbi.1011671
```

**BibTeX format**:

```bibtex
@article{zh2023,
  title = {Prokaryotic virus host prediction with graph contrastive augmentaion},
  author = {ZH, Du and JP, Zhong and Y, Liu and JQ, Li},
  year = {2023},
  journal = {PLoS computational biology},
  volume = {19},
  number = {12},
  pages = {e1011671},
  doi = {10.1371/journal.pcbi.1011671},
  pmid = {38039280},
}
```


## PHRED

**Tool URL**: [https://academic.oup.com/femsle/article/363/4/fnw002/1845417](https://academic.oup.com/femsle/article/363/4/fnw002/1845417)

### Publication Information

**Title**: Phred-Phrap package to analyses tools: a pipeline to facilitate population genetics re-sequencing studies.

**Journal**: Investigative genetics

**Year**: 2011

**Identifiers**:
- DOI: [10.1186/2041-2223-2-3](https://doi.org/10.1186/2041-2223-2-3)
- PMID: [21284835](https://pubmed.ncbi.nlm.nih.gov/21284835/)

### Cite this tool

**APA format**:

```
M, M., WC, M., A, S., B, A., AC, F., SJ, C., ... MR, R. (2011). Phred-Phrap package to analyses tools: a pipeline to facilitate population genetics re-sequencing studies. Investigative genetics, 2(1), 3. https://doi.org/10.1186/2041-2223-2-3
```

**BibTeX format**:

```bibtex
@article{m2011,
  title = {Phred-Phrap package to analyses tools: a pipeline to facilitate population genetics re-sequencing studies},
  author = {M, Machado and WC, Magalhães and A, Sene and B, Araújo and AC, Faria-Campos and SJ, Chanock and L, Scott and G, Oliveira and E, Tarazona-Santos and MR, Rodrigues},
  year = {2011},
  journal = {Investigative genetics},
  volume = {2},
  number = {1},
  pages = {3},
  doi = {10.1186/2041-2223-2-3},
  pmid = {21284835},
}
```


## PHROGs

**Tool URL**: [https://academic.oup.com/nargab/article/3/3/lqab067/6342220](https://academic.oup.com/nargab/article/3/3/lqab067/6342220)

### Publication Information

**Title**: MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/bioinformatics/btae119](https://doi.org/10.1093/bioinformatics/btae119)
- PMID: [38426351](https://pubmed.ncbi.nlm.nih.gov/38426351/)

### Cite this tool

**APA format**:

```
JL, F.I., E, D., M, B., CR, B., & RA, W.I. (2024). MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life. Bioinformatics (Oxford, England), 40(3). https://doi.org/10.1093/bioinformatics/btae119
```

**BibTeX format**:

```bibtex
@article{jl2024,
  title = {MetaCerberus: distributed highly parallelized HMM-based processing for robust functional annotation across the tree of life},
  author = {JL, Figueroa Iii and E, Dhungel and M, Bellanger and CR, Brouwer and RA, White Iii},
  year = {2024},
  journal = {Bioinformatics (Oxford, England)},
  volume = {40},
  number = {3},
  doi = {10.1093/bioinformatics/btae119},
  pmid = {38426351},
}
```


## PIGv

**Tool URL**: [https://github.com/BenMinch/PIGv](https://github.com/BenMinch/PIGv)

### Publication Information

**Title**: Characteristics of Neuroimaging and Behavioural Phenotype in Polish Patients with PIGV-CDG-An Observational Study and Literature Review.

**Journal**: Genes

**Year**: 2023

**Identifiers**:
- DOI: [10.3390/genes14061208](https://doi.org/10.3390/genes14061208)
- PMID: [37372388](https://pubmed.ncbi.nlm.nih.gov/37372388/)

### Cite this tool

**APA format**:

```
M, H., P, L., & A, J. (2023). Characteristics of Neuroimaging and Behavioural Phenotype in Polish Patients with PIGV-CDG-An Observational Study and Literature Review. Genes, 14(6). https://doi.org/10.3390/genes14061208
```

**BibTeX format**:

```bibtex
@article{m2023,
  title = {Characteristics of Neuroimaging and Behavioural Phenotype in Polish Patients with PIGV-CDG-An Observational Study and Literature Review},
  author = {M, Hutny and P, Lipinski and A, Jezela-Stanek},
  year = {2023},
  journal = {Genes},
  volume = {14},
  number = {6},
  doi = {10.3390/genes14061208},
  pmid = {37372388},
}
```


## PPR-Meta

**Tool URL**: [https://github.com/zhenchengfang/PPR-Meta](https://github.com/zhenchengfang/PPR-Meta)

### Publication Information

**Title**: PPR-Meta: a tool for identifying phages and plasmids from metagenomic fragments using deep learning.

**Journal**: GigaScience

**Year**: 2019

**Identifiers**:
- DOI: [10.1093/gigascience/giz066](https://doi.org/10.1093/gigascience/giz066)
- PMID: [31220250](https://pubmed.ncbi.nlm.nih.gov/31220250/)

### Cite this tool

**APA format**:

```
Z, F., J, T., S, W., M, L., C, X., Z, X., & H, Z. (2019). PPR-Meta: a tool for identifying phages and plasmids from metagenomic fragments using deep learning. GigaScience, 8(6). https://doi.org/10.1093/gigascience/giz066
```

**BibTeX format**:

```bibtex
@article{z2019,
  title = {PPR-Meta: a tool for identifying phages and plasmids from metagenomic fragments using deep learning},
  author = {Z, Fang and J, Tan and S, Wu and M, Li and C, Xu and Z, Xie and H, Zhu},
  year = {2019},
  journal = {GigaScience},
  volume = {8},
  number = {6},
  doi = {10.1093/gigascience/giz066},
  pmid = {31220250},
}
```


## PredPHI

**Tool URL**: [https://github.com/xialab-ahu/PredPHI](https://github.com/xialab-ahu/PredPHI)

### Publication Information

**Title**: A Deep Learning-Based Method for Identification of Bacteriophage-Host Interaction.

**Journal**: IEEE/ACM transactions on computational biology and bioinformatics

**Year**: 2021

**Identifiers**:
- DOI: [10.1109/TCBB.2020.3017386](https://doi.org/10.1109/TCBB.2020.3017386)
- PMID: [32813660](https://pubmed.ncbi.nlm.nih.gov/32813660/)

### Cite this tool

**APA format**:

```
M, L., Y, W., F, L., Y, Z., M, L., S, Z., ... J, X. (2021). A Deep Learning-Based Method for Identification of Bacteriophage-Host Interaction. IEEE/ACM transactions on computational biology and bioinformatics, 18(5), 1801-1810. https://doi.org/10.1109/TCBB.2020.3017386
```

**BibTeX format**:

```bibtex
@article{m2021,
  title = {A Deep Learning-Based Method for Identification of Bacteriophage-Host Interaction},
  author = {M, Li and Y, Wang and F, Li and Y, Zhao and M, Liu and S, Zhang and Y, Bin and AI, Smith and GI, Webb and J, Li and J, Song and J, Xia},
  year = {2021},
  journal = {IEEE/ACM transactions on computational biology and bioinformatics},
  volume = {18},
  number = {5},
  pages = {1801-1810},
  doi = {10.1109/TCBB.2020.3017386},
  pmid = {32813660},
}
```


## Prodigal

**Tool URL**: [https://github.com/hyattpd/Prodigal](https://github.com/hyattpd/Prodigal)

### Publication Information

**Title**: Prodigal: prokaryotic gene recognition and translation initiation site identification.

**Journal**: BMC bioinformatics

**Year**: 2010

**Identifiers**:
- DOI: [10.1186/1471-2105-11-119](https://doi.org/10.1186/1471-2105-11-119)
- PMID: [20211023](https://pubmed.ncbi.nlm.nih.gov/20211023/)

### Cite this tool

**APA format**:

```
D, H., GL, C., PF, L., ML, L., FW, L., & LJ, H. (2010). Prodigal: prokaryotic gene recognition and translation initiation site identification. BMC bioinformatics, 11, 119. https://doi.org/10.1186/1471-2105-11-119
```

**BibTeX format**:

```bibtex
@article{d2010,
  title = {Prodigal: prokaryotic gene recognition and translation initiation site identification},
  author = {D, Hyatt and GL, Chen and PF, Locascio and ML, Land and FW, Larimer and LJ, Hauser},
  year = {2010},
  journal = {BMC bioinformatics},
  volume = {11},
  pages = {119},
  doi = {10.1186/1471-2105-11-119},
  pmid = {20211023},
}
```


## Prophage Tracer

**Tool URL**: [https://academic.oup.com/nar/article/49/22/e128/6374144](https://academic.oup.com/nar/article/49/22/e128/6374144)

### Publication Information

**Title**: Prophage Tracer: precisely tracing prophages in prokaryotic genomes using overlapping split-read alignment.

**Journal**: Nucleic acids research

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/nar/gkab824](https://doi.org/10.1093/nar/gkab824)
- PMID: [34551431](https://pubmed.ncbi.nlm.nih.gov/34551431/)

### Cite this tool

**APA format**:

```
K, T., W, W., Y, S., Y, Z., P, W., Y, G., & X, W. (2021). Prophage Tracer: precisely tracing prophages in prokaryotic genomes using overlapping split-read alignment. Nucleic acids research, 49(22), e128. https://doi.org/10.1093/nar/gkab824
```

**BibTeX format**:

```bibtex
@article{k2021,
  title = {Prophage Tracer: precisely tracing prophages in prokaryotic genomes using overlapping split-read alignment},
  author = {K, Tang and W, Wang and Y, Sun and Y, Zhou and P, Wang and Y, Guo and X, Wang},
  year = {2021},
  journal = {Nucleic acids research},
  volume = {49},
  number = {22},
  pages = {e128},
  doi = {10.1093/nar/gkab824},
  pmid = {34551431},
}
```


## Prophet

**Tool URL**: [https://github.com/jaumlrc/ProphET](https://github.com/jaumlrc/ProphET)

### Publication Information

**Title**: ProphET, prophage estimation tool: A stand-alone prophage sequence prediction tool with self-updating reference database.

**Journal**: PloS one

**Year**: 2019

**Identifiers**:
- DOI: [10.1371/journal.pone.0223364](https://doi.org/10.1371/journal.pone.0223364)
- PMID: [31577829](https://pubmed.ncbi.nlm.nih.gov/31577829/)

### Cite this tool

**APA format**:

```
JL, R., DC, B., AL, M., AM, E., & GC, C. (2019). ProphET, prophage estimation tool: A stand-alone prophage sequence prediction tool with self-updating reference database. PloS one, 14(10), e0223364. https://doi.org/10.1371/journal.pone.0223364
```

**BibTeX format**:

```bibtex
@article{jl2019,
  title = {ProphET, prophage estimation tool: A stand-alone prophage sequence prediction tool with self-updating reference database},
  author = {JL, Reis-Cunha and DC, Bartholomeu and AL, Manson and AM, Earl and GC, Cerqueira},
  year = {2019},
  journal = {PloS one},
  volume = {14},
  number = {10},
  pages = {e0223364},
  doi = {10.1371/journal.pone.0223364},
  pmid = {31577829},
}
```


## Prophinder

**Tool URL**: [http://aclame.ulb.ac.be/Tools/Prophinder/](http://aclame.ulb.ac.be/Tools/Prophinder/)

### Publication Information

**Title**: Prophinder: a computational tool for prophage prediction in prokaryotic genomes.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2008

**Identifiers**:
- DOI: [10.1093/bioinformatics/btn043](https://doi.org/10.1093/bioinformatics/btn043)
- PMID: [18238785](https://pubmed.ncbi.nlm.nih.gov/18238785/)

### Cite this tool

**APA format**:

```
G, L., J, V.H., A, T., & R, L. (2008). Prophinder: a computational tool for prophage prediction in prokaryotic genomes. Bioinformatics (Oxford, England), 24(6), 863-5. https://doi.org/10.1093/bioinformatics/btn043
```

**BibTeX format**:

```bibtex
@article{g2008,
  title = {Prophinder: a computational tool for prophage prediction in prokaryotic genomes},
  author = {G, Lima-Mendez and J, Van Helden and A, Toussaint and R, Leplae},
  year = {2008},
  journal = {Bioinformatics (Oxford, England)},
  volume = {24},
  number = {6},
  pages = {863-5},
  doi = {10.1093/bioinformatics/btn043},
  pmid = {18238785},
}
```


## pVOGs

**Tool URL**: [http://pvogs.cvm.iastate.edu/](http://pvogs.cvm.iastate.edu/)

### Publication Information

**Title**: Prokaryotic Virus Orthologous Groups (pVOGs): a resource for comparative genomics and protein family annotation.

**Journal**: Nucleic acids research

**Year**: 2017

**Identifiers**:
- DOI: [10.1093/nar/gkw975](https://doi.org/10.1093/nar/gkw975)
- PMID: [27789703](https://pubmed.ncbi.nlm.nih.gov/27789703/)

### Cite this tool

**APA format**:

```
AL, G., EV, K., & DM, K. (2017). Prokaryotic Virus Orthologous Groups (pVOGs): a resource for comparative genomics and protein family annotation. Nucleic acids research, 45(D1), D491-D498. https://doi.org/10.1093/nar/gkw975
```

**BibTeX format**:

```bibtex
@article{al2017,
  title = {Prokaryotic Virus Orthologous Groups (pVOGs): a resource for comparative genomics and protein family annotation},
  author = {AL, Grazziotin and EV, Koonin and DM, Kristensen},
  year = {2017},
  journal = {Nucleic acids research},
  volume = {45},
  number = {D1},
  pages = {D491-D498},
  doi = {10.1093/nar/gkw975},
  pmid = {27789703},
}
```


## QuRe

**Tool URL**: [https://sourceforge.net/projects/qure/](https://sourceforge.net/projects/qure/)

### Publication Information

**Title**: Radiograph accelerated detection and identification of cancer in the lung (RADICAL): a mixed methods study to assess the clinical effectiveness and acceptability of Qure.ai artificial intelligence software to prioritise chest X-ray (CXR) interpretation.

**Journal**: BMJ open

**Year**: 2024

**Identifiers**:
- DOI: [10.1136/bmjopen-2023-081062](https://doi.org/10.1136/bmjopen-2023-081062)
- PMID: [39306349](https://pubmed.ncbi.nlm.nih.gov/39306349/)

### Cite this tool

**APA format**:

```
SF, D., A, M., J, B., DB, S., JD, M., O, W., ... DJ, L. (2024). Radiograph accelerated detection and identification of cancer in the lung (RADICAL): a mixed methods study to assess the clinical effectiveness and acceptability of Qure.ai artificial intelligence software to prioritise chest X-ray (CXR) interpretation. BMJ open, 14(9), e081062. https://doi.org/10.1136/bmjopen-2023-081062
```

**BibTeX format**:

```bibtex
@article{sf2024,
  title = {Radiograph accelerated detection and identification of cancer in the lung (RADICAL): a mixed methods study to assess the clinical effectiveness and acceptability of Qure.ai artificial intelligence software to prioritise chest X-ray (CXR) interpretation},
  author = {SF, Duncan and A, McConnachie and J, Blackwood and DB, Stobo and JD, Maclay and O, Wu and E, Germeni and D, Robert and B, Bilgili and S, Kumar and M, Hall and DJ, Lowe},
  year = {2024},
  journal = {BMJ open},
  volume = {14},
  number = {9},
  pages = {e081062},
  doi = {10.1136/bmjopen-2023-081062},
  pmid = {39306349},
}
```


## RaFaH

**Tool URL**: [https://sourceforge.net/projects/rafah/](https://sourceforge.net/projects/rafah/)

### Publication Information

**Title**: RaFAH: Host prediction for viruses of Bacteria and Archaea based on protein content.

**Journal**: Patterns (New York, N.Y.)

**Year**: 2021

**Identifiers**:
- DOI: [10.1016/j.patter.2021.100274](https://doi.org/10.1016/j.patter.2021.100274)
- PMID: [34286299](https://pubmed.ncbi.nlm.nih.gov/34286299/)

### Cite this tool

**APA format**:

```
FH, C., A, Z., M, L., J, B., A, Z., BE, D., ... F, R. (2021). RaFAH: Host prediction for viruses of Bacteria and Archaea based on protein content. Patterns (New York, N.Y.), 2(7), 100274. https://doi.org/10.1016/j.patter.2021.100274
```

**BibTeX format**:

```bibtex
@article{fh2021,
  title = {RaFAH: Host prediction for viruses of Bacteria and Archaea based on protein content},
  author = {FH, Coutinho and A, Zaragoza-Solas and M, López-Pérez and J, Barylski and A, Zielezinski and BE, Dutilh and R, Edwards and F, Rodriguez-Valera},
  year = {2021},
  journal = {Patterns (New York, N.Y.)},
  volume = {2},
  number = {7},
  pages = {100274},
  doi = {10.1016/j.patter.2021.100274},
  pmid = {34286299},
}
```


## RdRp-scan

**Tool URL**: [https://github.com/JustineCharon/RdRp-scan/](https://github.com/JustineCharon/RdRp-scan/)

### Publication Information

**Title**: RdRp-scan: A bioinformatic resource to identify and annotate divergent RNA viruses in metagenomic sequence data.

**Journal**: Virus evolution

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/ve/veac082](https://doi.org/10.1093/ve/veac082)
- PMID: [36533143](https://pubmed.ncbi.nlm.nih.gov/36533143/)

### Cite this tool

**APA format**:

```
J, C., JP, B., S, S., & EC, H. (2022). RdRp-scan: A bioinformatic resource to identify and annotate divergent RNA viruses in metagenomic sequence data. Virus evolution, 8(2), veac082. https://doi.org/10.1093/ve/veac082
```

**BibTeX format**:

```bibtex
@article{j2022,
  title = {RdRp-scan: A bioinformatic resource to identify and annotate divergent RNA viruses in metagenomic sequence data},
  author = {J, Charon and JP, Buchmann and S, Sadiq and EC, Holmes},
  year = {2022},
  journal = {Virus evolution},
  volume = {8},
  number = {2},
  pages = {veac082},
  doi = {10.1093/ve/veac082},
  pmid = {36533143},
}
```


## Recentrifuge

**Tool URL**: [https://github.com/khyox/recentrifuge](https://github.com/khyox/recentrifuge)

### Publication Information

**Title**: Recentrifuge: Robust comparative analysis and contamination removal for metagenomics.

**Journal**: PLoS computational biology

**Year**: 2019

**Identifiers**:
- DOI: [10.1371/journal.pcbi.1006967](https://doi.org/10.1371/journal.pcbi.1006967)
- PMID: [30958827](https://pubmed.ncbi.nlm.nih.gov/30958827/)

### Cite this tool

**APA format**:

```
JM, M. (2019). Recentrifuge: Robust comparative analysis and contamination removal for metagenomics. PLoS computational biology, 15(4), e1006967. https://doi.org/10.1371/journal.pcbi.1006967
```

**BibTeX format**:

```bibtex
@article{jm2019,
  title = {Recentrifuge: Robust comparative analysis and contamination removal for metagenomics},
  author = {JM, Martí},
  year = {2019},
  journal = {PLoS computational biology},
  volume = {15},
  number = {4},
  pages = {e1006967},
  doi = {10.1371/journal.pcbi.1006967},
  pmid = {30958827},
}
```


## ResFinder

**Tool URL**: [https://github.com/cadms/resfinder](https://github.com/cadms/resfinder)

### Publication Information

**Title**: ResFinder 4.0 for predictions of phenotypes from genotypes.

**Journal**: The Journal of antimicrobial chemotherapy

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/jac/dkaa345](https://doi.org/10.1093/jac/dkaa345)
- PMID: [32780112](https://pubmed.ncbi.nlm.nih.gov/32780112/)

### Cite this tool

**APA format**:

```
V, B., RS, K., E, R., MC, R., S, S., V, C., ... FM, A. (2020). ResFinder 4.0 for predictions of phenotypes from genotypes. The Journal of antimicrobial chemotherapy, 75(12), 3491-3500. https://doi.org/10.1093/jac/dkaa345
```

**BibTeX format**:

```bibtex
@article{v2020,
  title = {ResFinder 4.0 for predictions of phenotypes from genotypes},
  author = {V, Bortolaia and RS, Kaas and E, Ruppe and MC, Roberts and S, Schwarz and V, Cattoir and A, Philippon and RL, Allesoe and AR, Rebelo and AF, Florensa and L, Fagelhauer and T, Chakraborty and B, Neumann and G, Werner and JK, Bender and K, Stingl and M, Nguyen and J, Coppens and BB, Xavier and S, Malhotra-Kumar and H, Westh and M, Pinholt and MF, Anjum and NA, Duggett and I, Kempf and S, Nykäsenoja and S, Olkkola and K, Wieczorek and A, Amaro and L, Clemente and J, Mossong and S, Losch and C, Ragimbeau and O, Lund and FM, Aarestrup},
  year = {2020},
  journal = {The Journal of antimicrobial chemotherapy},
  volume = {75},
  number = {12},
  pages = {3491-3500},
  doi = {10.1093/jac/dkaa345},
  pmid = {32780112},
}
```


## Seeker

**Tool URL**: [https://github.com/gussow/seeker](https://github.com/gussow/seeker)

### Publication Information

**Title**: Appropriateness of the Asylum Seeker Resource Centre-adapted Refugee Health Assessment Tool.

**Journal**: Australian journal of primary health

**Year**: 2020

**Identifiers**:
- DOI: [10.1071/PY19059](https://doi.org/10.1071/PY19059)
- PMID: [32069438](https://pubmed.ncbi.nlm.nih.gov/32069438/)

### Cite this tool

**APA format**:

```
ESP, N., S, M., & L, N. (2020). Appropriateness of the Asylum Seeker Resource Centre-adapted Refugee Health Assessment Tool. Australian journal of primary health, 26(2), 132-139. https://doi.org/10.1071/PY19059
```

**BibTeX format**:

```bibtex
@article{esp2020,
  title = {Appropriateness of the Asylum Seeker Resource Centre-adapted Refugee Health Assessment Tool},
  author = {ESP, Napwanga and S, McShane and L, Naccarella},
  year = {2020},
  journal = {Australian journal of primary health},
  volume = {26},
  number = {2},
  pages = {132-139},
  doi = {10.1071/PY19059},
  pmid = {32069438},
}
```


## Serratus

**Tool URL**: [https://serratus.io/](https://serratus.io/)

### Publication Information

**Title**: Serratus Anterior and Upper Trapezius Electromyographic Analysis of the Push-Up Plus Exercise: A Systematic Review and Meta-Analysis.

**Journal**: Journal of athletic training

**Year**: 2019

**Identifiers**:
- DOI: [10.4085/1062-6050-237-18](https://doi.org/10.4085/1062-6050-237-18)
- PMID: [31584855](https://pubmed.ncbi.nlm.nih.gov/31584855/)

### Cite this tool

**APA format**:

```
FJ, K., HL, O., KY, L., & JJ, L. (2019). Serratus Anterior and Upper Trapezius Electromyographic Analysis of the Push-Up Plus Exercise: A Systematic Review and Meta-Analysis. Journal of athletic training, 54(11), 1156-1164. https://doi.org/10.4085/1062-6050-237-18
```

**BibTeX format**:

```bibtex
@article{fj2019,
  title = {Serratus Anterior and Upper Trapezius Electromyographic Analysis of the Push-Up Plus Exercise: A Systematic Review and Meta-Analysis},
  author = {FJ, Kang and HL, Ou and KY, Lin and JJ, Lin},
  year = {2019},
  journal = {Journal of athletic training},
  volume = {54},
  number = {11},
  pages = {1156-1164},
  doi = {10.4085/1062-6050-237-18},
  pmid = {31584855},
}
```


## ShoRAH

**Tool URL**: [https://github.com/cbg-ethz/shorah](https://github.com/cbg-ethz/shorah)

### Publication Information

**Title**: ShoRAH: estimating the genetic diversity of a mixed sample from next-generation sequencing data.

**Journal**: BMC bioinformatics

**Year**: 2011

**Identifiers**:
- DOI: [10.1186/1471-2105-12-119](https://doi.org/10.1186/1471-2105-12-119)
- PMID: [21521499](https://pubmed.ncbi.nlm.nih.gov/21521499/)

### Cite this tool

**APA format**:

```
O, Z., A, B., N, E., & N, B. (2011). ShoRAH: estimating the genetic diversity of a mixed sample from next-generation sequencing data. BMC bioinformatics, 12, 119. https://doi.org/10.1186/1471-2105-12-119
```

**BibTeX format**:

```bibtex
@article{o2011,
  title = {ShoRAH: estimating the genetic diversity of a mixed sample from next-generation sequencing data},
  author = {O, Zagordi and A, Bhattacharya and N, Eriksson and N, Beerenwinkel},
  year = {2011},
  journal = {BMC bioinformatics},
  volume = {12},
  pages = {119},
  doi = {10.1186/1471-2105-12-119},
  pmid = {21521499},
}
```


## SNPGenie

**Tool URL**: [https://github.com/chasewnelson/snpgenie](https://github.com/chasewnelson/snpgenie)

### Publication Information

**Title**: SNPGenie: estimating evolutionary parameters to detect natural selection using pooled next-generation sequencing data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2015

**Identifiers**:
- DOI: [10.1093/bioinformatics/btv449](https://doi.org/10.1093/bioinformatics/btv449)
- PMID: [26227143](https://pubmed.ncbi.nlm.nih.gov/26227143/)

### Cite this tool

**APA format**:

```
CW, N., LH, M., & AL, H. (2015). SNPGenie: estimating evolutionary parameters to detect natural selection using pooled next-generation sequencing data. Bioinformatics (Oxford, England), 31(22), 3709-11. https://doi.org/10.1093/bioinformatics/btv449
```

**BibTeX format**:

```bibtex
@article{cw2015,
  title = {SNPGenie: estimating evolutionary parameters to detect natural selection using pooled next-generation sequencing data},
  author = {CW, Nelson and LH, Moncla and AL, Hughes},
  year = {2015},
  journal = {Bioinformatics (Oxford, England)},
  volume = {31},
  number = {22},
  pages = {3709-11},
  doi = {10.1093/bioinformatics/btv449},
  pmid = {26227143},
}
```


## SpacePHARER

**Tool URL**: [https://github.com/soedinglab/spacepharer](https://github.com/soedinglab/spacepharer)

### Publication Information

**Title**: SpacePHARER: sensitive identification of phages from CRISPR spacers in prokaryotic hosts.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab222](https://doi.org/10.1093/bioinformatics/btab222)
- PMID: [33792634](https://pubmed.ncbi.nlm.nih.gov/33792634/)

### Cite this tool

**APA format**:

```
R, Z., M, M., E, L.K., C, N., C, G., & J, S. (2021). SpacePHARER: sensitive identification of phages from CRISPR spacers in prokaryotic hosts. Bioinformatics (Oxford, England), 37(19), 3364-3366. https://doi.org/10.1093/bioinformatics/btab222
```

**BibTeX format**:

```bibtex
@article{r2021,
  title = {SpacePHARER: sensitive identification of phages from CRISPR spacers in prokaryotic hosts},
  author = {R, Zhang and M, Mirdita and E, Levy Karin and C, Norroy and C, Galiez and J, Söding},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {19},
  pages = {3364-3366},
  doi = {10.1093/bioinformatics/btab222},
  pmid = {33792634},
}
```


## SpikeHunter

**Tool URL**: [https://github.com/nlm-irp-jianglab/SpikeHunter](https://github.com/nlm-irp-jianglab/SpikeHunter)

### Publication Information

**Title**: µSpikeHunter: An advanced computational tool for the analysis of neuronal communication and action potential propagation in microfluidic platforms.

**Journal**: Scientific reports

**Year**: 2019

**Identifiers**:
- DOI: [10.1038/s41598-019-42148-3](https://doi.org/10.1038/s41598-019-42148-3)
- PMID: [30962522](https://pubmed.ncbi.nlm.nih.gov/30962522/)

### Cite this tool

**APA format**:

```
K, H., JC, M., CDF, L., E, N., M, L., & P, A. (2019). µSpikeHunter: An advanced computational tool for the analysis of neuronal communication and action potential propagation in microfluidic platforms. Scientific reports, 9(1), 5777. https://doi.org/10.1038/s41598-019-42148-3
```

**BibTeX format**:

```bibtex
@article{k2019,
  title = {µSpikeHunter: An advanced computational tool for the analysis of neuronal communication and action potential propagation in microfluidic platforms},
  author = {K, Heiney and JC, Mateus and CDF, Lopes and E, Neto and M, Lamghari and P, Aguiar},
  year = {2019},
  journal = {Scientific reports},
  volume = {9},
  number = {1},
  pages = {5777},
  doi = {10.1038/s41598-019-42148-3},
  pmid = {30962522},
}
```


## TepiTool

**Tool URL**: [https://tools.iedb.org/tepitool/](https://tools.iedb.org/tepitool/)

### Publication Information

**Title**: TepiTool: A Pipeline for Computational Prediction of T Cell Epitope Candidates.

**Journal**: Current protocols in immunology

**Year**: 2016

**Identifiers**:
- DOI: [10.1002/cpim.12](https://doi.org/10.1002/cpim.12)
- PMID: [27479659](https://pubmed.ncbi.nlm.nih.gov/27479659/)

### Cite this tool

**APA format**:

```
S, P., J, S., A, S., & B, P. (2016). TepiTool: A Pipeline for Computational Prediction of T Cell Epitope Candidates. Current protocols in immunology, 114, 18.19.1-18.19.24. https://doi.org/10.1002/cpim.12
```

**BibTeX format**:

```bibtex
@article{s2016,
  title = {TepiTool: A Pipeline for Computational Prediction of T Cell Epitope Candidates},
  author = {S, Paul and J, Sidney and A, Sette and B, Peters},
  year = {2016},
  journal = {Current protocols in immunology},
  volume = {114},
  pages = {18.19.1-18.19.24},
  doi = {10.1002/cpim.12},
  pmid = {27479659},
}
```


## TrIdent

**Tool URL**: [https://jlmaier12.github.io/TrIdent/](https://jlmaier12.github.io/TrIdent/)

### Publication Information

**Title**: Molecular Monitoring of Enteropathogens in Sewage During NATO Exercise Trident Juncture 2018: Potential Tool in Early Outbreak Warning?

**Journal**: Military medicine

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/milmed/usaa365](https://doi.org/10.1093/milmed/usaa365)
- PMID: [33185665](https://pubmed.ncbi.nlm.nih.gov/33185665/)

### Cite this tool

**APA format**:

```
Ø, Ø., & EH, M. (2021). Molecular Monitoring of Enteropathogens in Sewage During NATO Exercise Trident Juncture 2018: Potential Tool in Early Outbreak Warning?. Military medicine, 186(7-8), e661-e666. https://doi.org/10.1093/milmed/usaa365
```

**BibTeX format**:

```bibtex
@article{ø2021,
  title = {Molecular Monitoring of Enteropathogens in Sewage During NATO Exercise Trident Juncture 2018: Potential Tool in Early Outbreak Warning?},
  author = {Ø, Ørmen and EH, Madslien},
  year = {2021},
  journal = {Military medicine},
  volume = {186},
  number = {7-8},
  pages = {e661-e666},
  doi = {10.1093/milmed/usaa365},
  pmid = {33185665},
}
```


## V-pipe

**Tool URL**: [https://github.com/cbg-ethz/V-pipe](https://github.com/cbg-ethz/V-pipe)

### Publication Information

**Title**: V-pipe: a computational pipeline for assessing viral genetic diversity from high-throughput data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab015](https://doi.org/10.1093/bioinformatics/btab015)
- PMID: [33471068](https://pubmed.ncbi.nlm.nih.gov/33471068/)

### Cite this tool

**APA format**:

```
S, P., D, S., I, T., KP, J., KJ, M., & N, B. (2021). V-pipe: a computational pipeline for assessing viral genetic diversity from high-throughput data. Bioinformatics (Oxford, England), 37(12), 1673-1680. https://doi.org/10.1093/bioinformatics/btab015
```

**BibTeX format**:

```bibtex
@article{s2021,
  title = {V-pipe: a computational pipeline for assessing viral genetic diversity from high-throughput data},
  author = {S, Posada-Céspedes and D, Seifert and I, Topolsky and KP, Jablonski and KJ, Metzner and N, Beerenwinkel},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {12},
  pages = {1673-1680},
  doi = {10.1093/bioinformatics/btab015},
  pmid = {33471068},
}
```


## vAMPirus

**Tool URL**: [https://github.com/Aveglia/vAMPirus](https://github.com/Aveglia/vAMPirus)

### Publication Information

**Title**: vAMPirus: A versatile amplicon processing and analysis program for studying viruses.

**Journal**: Molecular ecology resources

**Year**: 2024

**Identifiers**:
- DOI: [10.1111/1755-0998.13978](https://doi.org/10.1111/1755-0998.13978)
- PMID: [38775206](https://pubmed.ncbi.nlm.nih.gov/38775206/)

### Cite this tool

**APA format**:

```
AJ, V., RE, R., CGB, G., LI, H., & AMS, C. (2024). vAMPirus: A versatile amplicon processing and analysis program for studying viruses. Molecular ecology resources, 24(6), e13978. https://doi.org/10.1111/1755-0998.13978
```

**BibTeX format**:

```bibtex
@article{aj2024,
  title = {vAMPirus: A versatile amplicon processing and analysis program for studying viruses},
  author = {AJ, Veglia and RE, Rivera-Vicéns and CGB, Grupstra and LI, Howe-Kerr and AMS, Correa},
  year = {2024},
  journal = {Molecular ecology resources},
  volume = {24},
  number = {6},
  pages = {e13978},
  doi = {10.1111/1755-0998.13978},
  pmid = {38775206},
}
```


## VAPiD

**Tool URL**: [https://github.com/rcs333/VAPiD](https://github.com/rcs333/VAPiD)

### Publication Information

**Title**: VAPiD: a lightweight cross-platform viral annotation pipeline and identification tool to facilitate virus genome submissions to NCBI GenBank.

**Journal**: BMC bioinformatics

**Year**: 2019

**Identifiers**:
- DOI: [10.1186/s12859-019-2606-y](https://doi.org/10.1186/s12859-019-2606-y)
- PMID: [30674273](https://pubmed.ncbi.nlm.nih.gov/30674273/)

### Cite this tool

**APA format**:

```
RC, S., N, M., GD, S., MJ, L., & AL, G. (2019). VAPiD: a lightweight cross-platform viral annotation pipeline and identification tool to facilitate virus genome submissions to NCBI GenBank. BMC bioinformatics, 20(1), 48. https://doi.org/10.1186/s12859-019-2606-y
```

**BibTeX format**:

```bibtex
@article{rc2019,
  title = {VAPiD: a lightweight cross-platform viral annotation pipeline and identification tool to facilitate virus genome submissions to NCBI GenBank},
  author = {RC, Shean and N, Makhsous and GD, Stoddard and MJ, Lin and AL, Greninger},
  year = {2019},
  journal = {BMC bioinformatics},
  volume = {20},
  number = {1},
  pages = {48},
  doi = {10.1186/s12859-019-2606-y},
  pmid = {30674273},
}
```


## VCFgenie

**Tool URL**: [https://github.com/chasewnelson/VCFgenie](https://github.com/chasewnelson/VCFgenie)

### Publication Information

**Title**: Improved detection of low-frequency within-host variants from deep sequencing: A case study with human papillomavirus.

**Journal**: Virus evolution

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/ve/veae013](https://doi.org/10.1093/ve/veae013)
- PMID: [38455683](https://pubmed.ncbi.nlm.nih.gov/38455683/)

### Cite this tool

**APA format**:

```
SK, M., CW, N., B, Z., M, P., HJ, L., M, D., ... L, M. (2024). Improved detection of low-frequency within-host variants from deep sequencing: A case study with human papillomavirus. Virus evolution, 10(1), veae013. https://doi.org/10.1093/ve/veae013
```

**BibTeX format**:

```bibtex
@article{sk2024,
  title = {Improved detection of low-frequency within-host variants from deep sequencing: A case study with human papillomavirus},
  author = {SK, Mishra and CW, Nelson and B, Zhu and M, Pinheiro and HJ, Lee and M, Dean and L, Burdett and M, Yeager and L, Mirabello},
  year = {2024},
  journal = {Virus evolution},
  volume = {10},
  number = {1},
  pages = {veae013},
  doi = {10.1093/ve/veae013},
  pmid = {38455683},
}
```


## vConTACT

**Tool URL**: [https://bitbucket.org/MAVERICLab/vcontact/src/master/](https://bitbucket.org/MAVERICLab/vcontact/src/master/)

### Publication Information

**Title**: vConTACT: an iVirus tool to classify double-stranded DNA viruses that infect Archaea and Bacteria.

**Journal**: PeerJ

**Year**: 2017

**Identifiers**:
- DOI: [10.7717/peerj.3243](https://doi.org/10.7717/peerj.3243)
- PMID: [28480138](https://pubmed.ncbi.nlm.nih.gov/28480138/)

### Cite this tool

**APA format**:

```
B, B., HB, J., G, D., ZQ, Y., S, R., & MB, S. (2017). vConTACT: an iVirus tool to classify double-stranded DNA viruses that infect Archaea and Bacteria. PeerJ, 5, e3243. https://doi.org/10.7717/peerj.3243
```

**BibTeX format**:

```bibtex
@article{b2017,
  title = {vConTACT: an iVirus tool to classify double-stranded DNA viruses that infect Archaea and Bacteria},
  author = {B, Bolduc and HB, Jang and G, Doulcier and ZQ, You and S, Roux and MB, Sullivan},
  year = {2017},
  journal = {PeerJ},
  volume = {5},
  pages = {e3243},
  doi = {10.7717/peerj.3243},
  pmid = {28480138},
}
```


## VEGA

**Tool URL**: [https://github.com/pauloluniyi/VGEA](https://github.com/pauloluniyi/VGEA)

### Publication Information

**Title**: The VEGA Tool to Check the Applicability Domain Gives Greater Confidence in the Prediction of In Silico Models.

**Journal**: International journal of molecular sciences

**Year**: 2023

**Identifiers**:
- DOI: [10.3390/ijms24129894](https://doi.org/10.3390/ijms24129894)
- PMID: [37373049](https://pubmed.ncbi.nlm.nih.gov/37373049/)

### Cite this tool

**APA format**:

```
A, D., E, C., G, R., A, L., A, R., A, M., ... E, B. (2023). The VEGA Tool to Check the Applicability Domain Gives Greater Confidence in the Prediction of In Silico Models. International journal of molecular sciences, 24(12). https://doi.org/10.3390/ijms24129894
```

**BibTeX format**:

```bibtex
@article{a2023,
  title = {The VEGA Tool to Check the Applicability Domain Gives Greater Confidence in the Prediction of In Silico Models},
  author = {A, Danieli and E, Colombo and G, Raitano and A, Lombardo and A, Roncaglioni and A, Manganaro and A, Sommovigo and E, Carnesecchi and JCM, Dorne and E, Benfenati},
  year = {2023},
  journal = {International journal of molecular sciences},
  volume = {24},
  number = {12},
  doi = {10.3390/ijms24129894},
  pmid = {37373049},
}
```


## VFM

**Tool URL**: [https://github.com/liuql2019/VFM](https://github.com/liuql2019/VFM)

### Publication Information

**Title**: Cardiac VFM visualization and analysis based on YOLO deep learning model and modified 2D continuity equation.

**Journal**: Computerized medical imaging and graphics : the official journal of the Computerized Medical Imaging Society

**Year**: 2020

**Identifiers**:
- DOI: [10.1016/j.compmedimag.2020.101732](https://doi.org/10.1016/j.compmedimag.2020.101732)
- PMID: [32417649](https://pubmed.ncbi.nlm.nih.gov/32417649/)

### Cite this tool

**APA format**:

```
Z, Z., G, L., W, D., ANJ, R., S, Q., J, G., & Y, Y. (2020). Cardiac VFM visualization and analysis based on YOLO deep learning model and modified 2D continuity equation. Computerized medical imaging and graphics : the official journal of the Computerized Medical Imaging Society, 82, 101732. https://doi.org/10.1016/j.compmedimag.2020.101732
```

**BibTeX format**:

```bibtex
@article{z2020,
  title = {Cardiac VFM visualization and analysis based on YOLO deep learning model and modified 2D continuity equation},
  author = {Z, Zhuang and G, Liu and W, Ding and ANJ, Raj and S, Qiu and J, Guo and Y, Yuan},
  year = {2020},
  journal = {Computerized medical imaging and graphics : the official journal of the Computerized Medical Imaging Society},
  volume = {82},
  pages = {101732},
  doi = {10.1016/j.compmedimag.2020.101732},
  pmid = {32417649},
}
```


## VGAS

**Tool URL**: [https://github.com/tianqitang1/VGAS](https://github.com/tianqitang1/VGAS)

### Publication Information

**Title**: Vgas: A Viral Genome Annotation System.

**Journal**: Frontiers in microbiology

**Year**: 2019

**Identifiers**:
- DOI: [10.3389/fmicb.2019.00184](https://doi.org/10.3389/fmicb.2019.00184)
- PMID: [30814982](https://pubmed.ncbi.nlm.nih.gov/30814982/)

### Cite this tool

**APA format**:

```
KY, Z., YZ, G., MZ, D., S, L., C, D., & FB, G. (2019). Vgas: A Viral Genome Annotation System. Frontiers in microbiology, 10, 184. https://doi.org/10.3389/fmicb.2019.00184
```

**BibTeX format**:

```bibtex
@article{ky2019,
  title = {Vgas: A Viral Genome Annotation System},
  author = {KY, Zhang and YZ, Gao and MZ, Du and S, Liu and C, Dong and FB, Guo},
  year = {2019},
  journal = {Frontiers in microbiology},
  volume = {10},
  pages = {184},
  doi = {10.3389/fmicb.2019.00184},
  pmid = {30814982},
}
```


## vHulk

**Tool URL**: [https://www.biorxiv.org/content/10.1101/2020.12.06.413476v1](https://www.biorxiv.org/content/10.1101/2020.12.06.413476v1)

### Publication Information

**Title**: vHULK, a New Tool for Bacteriophage Host Prediction Based on Annotated Genomic Features and Neural Networks.

**Journal**: PHAGE (New Rochelle, N.Y.)

**Year**: 2022

**Identifiers**:
- DOI: [10.1089/phage.2021.0016](https://doi.org/10.1089/phage.2021.0016)
- PMID: [36793881](https://pubmed.ncbi.nlm.nih.gov/36793881/)

### Cite this tool

**APA format**:

```
D, A., BKV, I., CM, P., AM, d.S., & JC, S. (2022). vHULK, a New Tool for Bacteriophage Host Prediction Based on Annotated Genomic Features and Neural Networks. PHAGE (New Rochelle, N.Y.), 3(4), 204-212. https://doi.org/10.1089/phage.2021.0016
```

**BibTeX format**:

```bibtex
@article{d2022,
  title = {vHULK, a New Tool for Bacteriophage Host Prediction Based on Annotated Genomic Features and Neural Networks},
  author = {D, Amgarten and BKV, Iha and CM, Piroupo and AM, da Silva and JC, Setubal},
  year = {2022},
  journal = {PHAGE (New Rochelle, N.Y.)},
  volume = {3},
  number = {4},
  pages = {204-212},
  doi = {10.1089/phage.2021.0016},
  pmid = {36793881},
}
```


## VIBRANT

**Tool URL**: [https://github.com/AnantharamanLab/VIBRANT](https://github.com/AnantharamanLab/VIBRANT)

### Publication Information

**Title**: A computed tomographic data-based vibrant bonebridge visualization tool.

**Journal**: Cochlear implants international

**Year**: 2014

**Identifiers**:
- DOI: [10.1179/1467010014Z.000000000155](https://doi.org/10.1179/1467010014Z.000000000155)
- PMID: [24869450](https://pubmed.ncbi.nlm.nih.gov/24869450/)

### Cite this tool

**APA format**:

```
I, T., H, L., H, R., & A, E. (2014). A computed tomographic data-based vibrant bonebridge visualization tool. Cochlear implants international, 15 Suppl 1, S72-4. https://doi.org/10.1179/1467010014Z.000000000155
```

**BibTeX format**:

```bibtex
@article{i2014,
  title = {A computed tomographic data-based vibrant bonebridge visualization tool},
  author = {I, Todt and H, Lamecker and H, Ramm and A, Ernst},
  year = {2014},
  journal = {Cochlear implants international},
  volume = {15 Suppl 1},
  pages = {S72-4},
  doi = {10.1179/1467010014Z.000000000155},
  pmid = {24869450},
}
```


## VICTOR

**Tool URL**: [https://github.com/vdclab/vdclab-wiki/blob/master/VICTOR.md](https://github.com/vdclab/vdclab-wiki/blob/master/VICTOR.md)

### Publication Information

**Title**: VICTOR: A visual analytics web application for comparing cluster sets.

**Journal**: Computers in biology and medicine

**Year**: 2021

**Identifiers**:
- DOI: [10.1016/j.compbiomed.2021.104557](https://doi.org/10.1016/j.compbiomed.2021.104557)
- PMID: [34139436](https://pubmed.ncbi.nlm.nih.gov/34139436/)

### Cite this tool

**APA format**:

```
E, K., M, G., J, H., FA, B., PI, K., CJ, B., ... GA, P. (2021). VICTOR: A visual analytics web application for comparing cluster sets. Computers in biology and medicine, 135, 104557. https://doi.org/10.1016/j.compbiomed.2021.104557
```

**BibTeX format**:

```bibtex
@article{e2021,
  title = {VICTOR: A visual analytics web application for comparing cluster sets},
  author = {E, Karatzas and M, Gkonta and J, Hotova and FA, Baltoumas and PI, Kontou and CJ, Bobotsis and PG, Bagos and GA, Pavlopoulos},
  year = {2021},
  journal = {Computers in biology and medicine},
  volume = {135},
  pages = {104557},
  doi = {10.1016/j.compbiomed.2021.104557},
  pmid = {34139436},
}
```


## VIDHOP

**Tool URL**: [https://github.com/flomock/vidhop](https://github.com/flomock/vidhop)

### Publication Information

**Title**: VIDHOP, viral host prediction with deep learning.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa705](https://doi.org/10.1093/bioinformatics/btaa705)
- PMID: [32777818](https://pubmed.ncbi.nlm.nih.gov/32777818/)

### Cite this tool

**APA format**:

```
F, M., A, V., E, B., & M, M. (2021). VIDHOP, viral host prediction with deep learning. Bioinformatics (Oxford, England), 37(3), 318-325. https://doi.org/10.1093/bioinformatics/btaa705
```

**BibTeX format**:

```bibtex
@article{f2021,
  title = {VIDHOP, viral host prediction with deep learning},
  author = {F, Mock and A, Viehweger and E, Barth and M, Marz},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {3},
  pages = {318-325},
  doi = {10.1093/bioinformatics/btaa705},
  pmid = {32777818},
}
```


## VIGA

**Tool URL**: [https://github.com/viralInformatics/VIGA](https://github.com/viralInformatics/VIGA)

### Publication Information

**Title**: VIGA: a one-stop tool for eukaryotic virus identification and genome assembly from next-generation-sequencing data.

**Journal**: Briefings in bioinformatics

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/bib/bbad444](https://doi.org/10.1093/bib/bbad444)
- PMID: [38048079](https://pubmed.ncbi.nlm.nih.gov/38048079/)

### Cite this tool

**APA format**:

```
P, F., Y, W., Z, Z., Y, Q., Y, W., & Y, P. (2023). VIGA: a one-stop tool for eukaryotic virus identification and genome assembly from next-generation-sequencing data. Briefings in bioinformatics, 25(1). https://doi.org/10.1093/bib/bbad444
```

**BibTeX format**:

```bibtex
@article{p2023,
  title = {VIGA: a one-stop tool for eukaryotic virus identification and genome assembly from next-generation-sequencing data},
  author = {P, Fu and Y, Wu and Z, Zhang and Y, Qiu and Y, Wang and Y, Peng},
  year = {2023},
  journal = {Briefings in bioinformatics},
  volume = {25},
  number = {1},
  doi = {10.1093/bib/bbad444},
  pmid = {38048079},
}
```


## VIGOR

**Tool URL**: [https://github.com/JCVI-VIRIFX/VIGOR4](https://github.com/JCVI-VIRIFX/VIGOR4)

### Publication Information

**Title**: Dopamine and reward-related vigor in younger and older adults.

**Journal**: Neurobiology of aging

**Year**: 2022

**Identifiers**:
- DOI: [10.1016/j.neurobiolaging.2022.06.003](https://doi.org/10.1016/j.neurobiolaging.2022.06.003)
- PMID: [35858491](https://pubmed.ncbi.nlm.nih.gov/35858491/)

### Cite this tool

**APA format**:

```
EJ, H., U, B., L, D.B., J, A., L, B., & M, G. (2022). Dopamine and reward-related vigor in younger and older adults. Neurobiology of aging, 118, 34-43. https://doi.org/10.1016/j.neurobiolaging.2022.06.003
```

**BibTeX format**:

```bibtex
@article{ej2022,
  title = {Dopamine and reward-related vigor in younger and older adults},
  author = {EJ, Hird and U, Beierholm and L, De Boer and J, Axelsson and L, Backman and M, Guitart-Masip},
  year = {2022},
  journal = {Neurobiology of aging},
  volume = {118},
  pages = {34-43},
  doi = {10.1016/j.neurobiolaging.2022.06.003},
  pmid = {35858491},
}
```


## VIP

**Tool URL**: [https://github.com/keylabivdc/VIP/](https://github.com/keylabivdc/VIP/)

### Publication Information

**Title**: VIP: an integrated pipeline for metagenomics of virus identification and discovery.

**Journal**: Scientific reports

**Year**: 2016

**Identifiers**:
- DOI: [10.1038/srep23774](https://doi.org/10.1038/srep23774)
- PMID: [27026381](https://pubmed.ncbi.nlm.nih.gov/27026381/)

### Cite this tool

**APA format**:

```
Y, L., H, W., K, N., C, Z., Y, Z., J, W., ... X, M. (2016). VIP: an integrated pipeline for metagenomics of virus identification and discovery. Scientific reports, 6, 23774. https://doi.org/10.1038/srep23774
```

**BibTeX format**:

```bibtex
@article{y2016,
  title = {VIP: an integrated pipeline for metagenomics of virus identification and discovery},
  author = {Y, Li and H, Wang and K, Nie and C, Zhang and Y, Zhang and J, Wang and P, Niu and X, Ma},
  year = {2016},
  journal = {Scientific reports},
  volume = {6},
  pages = {23774},
  doi = {10.1038/srep23774},
  pmid = {27026381},
}
```


## VIPERA

**Tool URL**: [https://github.com/PathoGenOmics-Lab/VIPERA](https://github.com/PathoGenOmics-Lab/VIPERA)

### Publication Information

**Title**: VIPERA: Viral Intra-Patient Evolution Reporting and Analysis.

**Journal**: Virus evolution

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/ve/veae018](https://doi.org/10.1093/ve/veae018)
- PMID: [38510921](https://pubmed.ncbi.nlm.nih.gov/38510921/)

### Cite this tool

**APA format**:

```
M, Á., J, S., P, R., A, V., J, V., P, C., ... M, C. (2024). VIPERA: Viral Intra-Patient Evolution Reporting and Analysis. Virus evolution, 10(1), veae018. https://doi.org/10.1093/ve/veae018
```

**BibTeX format**:

```bibtex
@article{m2024,
  title = {VIPERA: Viral Intra-Patient Evolution Reporting and Analysis},
  author = {M, Álvarez-Herrera and J, Sevilla and P, Ruiz-Rodriguez and A, Vergara and J, Vila and P, Cano-Jiménez and F, González-Candelas and I, Comas and M, Coscollá},
  year = {2024},
  journal = {Virus evolution},
  volume = {10},
  number = {1},
  pages = {veae018},
  doi = {10.1093/ve/veae018},
  pmid = {38510921},
}
```


## VIPERdb

**Tool URL**: [http://viperdb.scripps.edu/](http://viperdb.scripps.edu/)

### Publication Information

**Title**: VIPERdb: A Tool for Virus Research.

**Journal**: Annual review of virology

**Year**: 2018

**Identifiers**:
- DOI: [10.1146/annurev-virology-092917-043405](https://doi.org/10.1146/annurev-virology-092917-043405)
- PMID: [30265627](https://pubmed.ncbi.nlm.nih.gov/30265627/)

### Cite this tool

**APA format**:

```
PT, H., DJ, M., JJ, W., M, C., 3rd, B.C., JE, J., & VS, R. (2018). VIPERdb: A Tool for Virus Research. Annual review of virology, 5(1), 477-488. https://doi.org/10.1146/annurev-virology-092917-043405
```

**BibTeX format**:

```bibtex
@article{pt2018,
  title = {VIPERdb: A Tool for Virus Research},
  author = {PT, Ho and DJ, Montiel-Garcia and JJ, Wong and M, Carrillo-Tripp and 3rd, Brooks CL and JE, Johnson and VS, Reddy},
  year = {2018},
  journal = {Annual review of virology},
  volume = {5},
  number = {1},
  pages = {477-488},
  doi = {10.1146/annurev-virology-092917-043405},
  pmid = {30265627},
}
```


## VIPtree

**Tool URL**: [https://github.com/yosuken/ViPTreeGen](https://github.com/yosuken/ViPTreeGen)

### Publication Information

**Title**: ['ViPTree: the viral proteomic tree server']

### Cite this tool

**APA format**:

```
Nishimura, Y., Yoshida, T., Kuronishi, M., Uehara, H., Ogata, H., & Goto, S. (2017). ViPTree: the viral proteomic tree server. Bioinformatics, 33(15), 2379-2380. https://doi.org/10.1093/bioinformatics/btx157
```

**BibTeX format**:

```bibtex
@article{nishimura2017,
  title = {ViPTree: the viral proteomic tree server},
  author = {Nishimura, Y. and Yoshida, T. and Kuronishi, M. and Uehara, H. and Ogata, H. and Goto, S.},
  year = {2017},
  journal = {Bioinformatics},
  volume = {33},
  number = {15},
  pages = {2379-2380},
  doi = {10.1093/bioinformatics/btx157},
}
```


## ViQuaS

**Tool URL**: [https://web.archive.org/web/20190710131744/https://github.com/HadiNW/ViQuaS](https://web.archive.org/web/20190710131744/https://github.com/HadiNW/ViQuaS)

### Publication Information

**Title**: ViQuaS: an improved reconstruction pipeline for viral quasispecies spectra generated by next-generation sequencing.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2015

**Identifiers**:
- DOI: [10.1093/bioinformatics/btu754](https://doi.org/10.1093/bioinformatics/btu754)
- PMID: [25398613](https://pubmed.ncbi.nlm.nih.gov/25398613/)

### Cite this tool

**APA format**:

```
D, J., I, S., S, M., BC, C., SL, T., & SK, H. (2015). ViQuaS: an improved reconstruction pipeline for viral quasispecies spectra generated by next-generation sequencing. Bioinformatics (Oxford, England), 31(6), 886-96. https://doi.org/10.1093/bioinformatics/btu754
```

**BibTeX format**:

```bibtex
@article{d2015,
  title = {ViQuaS: an improved reconstruction pipeline for viral quasispecies spectra generated by next-generation sequencing},
  author = {D, Jayasundara and I, Saeed and S, Maheswararajah and BC, Chang and SL, Tang and SK, Halgamuge},
  year = {2015},
  journal = {Bioinformatics (Oxford, England)},
  volume = {31},
  number = {6},
  pages = {886-96},
  doi = {10.1093/bioinformatics/btu754},
  pmid = {25398613},
}
```


## Viral-NGS

**Tool URL**: [https://github.com/broadinstitute/viral-ngs](https://github.com/broadinstitute/viral-ngs)

### Publication Information

**Title**: VPipe: an Automated Bioinformatics Platform for Assembly and Management of Viral Next-Generation Sequencing Data.

**Journal**: Microbiology spectrum

**Year**: 2022

**Identifiers**:
- DOI: [10.1128/spectrum.02564-21](https://doi.org/10.1128/spectrum.02564-21)
- PMID: [35234489](https://pubmed.ncbi.nlm.nih.gov/35234489/)

### Cite this tool

**APA format**:

```
DD, W., RL, M., E, R., TFF, N., CJ, C., M, O., ... MS, O. (2022). VPipe: an Automated Bioinformatics Platform for Assembly and Management of Viral Next-Generation Sequencing Data. Microbiology spectrum, 10(2), e0256421. https://doi.org/10.1128/spectrum.02564-21
```

**BibTeX format**:

```bibtex
@article{dd2022,
  title = {VPipe: an Automated Bioinformatics Platform for Assembly and Management of Viral Next-Generation Sequencing Data},
  author = {DD, Wagner and RL, Marine and E, Ramos and TFF, Ng and CJ, Castro and M, Okomo-Adhiambo and K, Harvey and G, Doho and R, Kelly and Y, Jain and RL, Tatusov and H, Silva and PA, Rota and AN, Khan and MS, Oberste},
  year = {2022},
  journal = {Microbiology spectrum},
  volume = {10},
  number = {2},
  pages = {e0256421},
  doi = {10.1128/spectrum.02564-21},
  pmid = {35234489},
}
```


## Viral-Track

**Tool URL**: [https://github.com/PierreBSC/Viral-Track](https://github.com/PierreBSC/Viral-Track)

### Publication Information

**Title**: No evidence for active viral infection in unicentric and idiopathic multicentric Castleman disease by Viral-Track analysis.

**Journal**: Scientific reports

**Year**: 2025

**Identifiers**:
- DOI: [10.1038/s41598-025-85193-x](https://doi.org/10.1038/s41598-025-85193-x)
- PMID: [39799155](https://pubmed.ncbi.nlm.nih.gov/39799155/)

### Cite this tool

**APA format**:

```
I, M., MD, M., S, S., M, S.B., P, H., MV, G., & DC, F. (2025). No evidence for active viral infection in unicentric and idiopathic multicentric Castleman disease by Viral-Track analysis. Scientific reports, 15(1), 1676. https://doi.org/10.1038/s41598-025-85193-x
```

**BibTeX format**:

```bibtex
@article{i2025,
  title = {No evidence for active viral infection in unicentric and idiopathic multicentric Castleman disease by Viral-Track analysis},
  author = {I, Miller and MD, Mumau and S, Shyamsundar and M, Sarmiento Bustamante and P, Horna and MV, Gonzalez and DC, Fajgenbaum},
  year = {2025},
  journal = {Scientific reports},
  volume = {15},
  number = {1},
  pages = {1676},
  doi = {10.1038/s41598-025-85193-x},
  pmid = {39799155},
}
```


## ViralCC

**Tool URL**: [https://github.com/dyxstat/ViralCC](https://github.com/dyxstat/ViralCC)

### Publication Information

**Title**: ViralCC retrieves complete viral genomes and virus-host pairs from metagenomic Hi-C data.

**Journal**: Nature communications

**Year**: 2023

**Identifiers**:
- DOI: [10.1038/s41467-023-35945-y](https://doi.org/10.1038/s41467-023-35945-y)
- PMID: [36720887](https://pubmed.ncbi.nlm.nih.gov/36720887/)

### Cite this tool

**APA format**:

```
Y, D., JA, F., & F, S. (2023). ViralCC retrieves complete viral genomes and virus-host pairs from metagenomic Hi-C data. Nature communications, 14(1), 502. https://doi.org/10.1038/s41467-023-35945-y
```

**BibTeX format**:

```bibtex
@article{y2023,
  title = {ViralCC retrieves complete viral genomes and virus-host pairs from metagenomic Hi-C data},
  author = {Y, Du and JA, Fuhrman and F, Sun},
  year = {2023},
  journal = {Nature communications},
  volume = {14},
  number = {1},
  pages = {502},
  doi = {10.1038/s41467-023-35945-y},
  pmid = {36720887},
}
```


## viralComplete

**Tool URL**: [https://github.com/ablab/viralComplete/](https://github.com/ablab/viralComplete/)

### Publication Information

**Title**: Metaviral SPAdes: assembly of viruses from metagenomic data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa490](https://doi.org/10.1093/bioinformatics/btaa490)
- PMID: [32413137](https://pubmed.ncbi.nlm.nih.gov/32413137/)

### Cite this tool

**APA format**:

```
D, A., M, R., A, L., & PA, P. (2020). Metaviral SPAdes: assembly of viruses from metagenomic data. Bioinformatics (Oxford, England), 36(14), 4126-4129. https://doi.org/10.1093/bioinformatics/btaa490
```

**BibTeX format**:

```bibtex
@article{d2020,
  title = {Metaviral SPAdes: assembly of viruses from metagenomic data},
  author = {D, Antipov and M, Raiko and A, Lapidus and PA, Pevzner},
  year = {2020},
  journal = {Bioinformatics (Oxford, England)},
  volume = {36},
  number = {14},
  pages = {4126-4129},
  doi = {10.1093/bioinformatics/btaa490},
  pmid = {32413137},
}
```


## ViralConsensus

**Tool URL**: [https://github.com/niemasd/ViralConsensus](https://github.com/niemasd/ViralConsensus)

### Publication Information

**Title**: ViralConsensus: a fast and memory-efficient tool for calling viral consensus genome sequences directly from read alignment data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2023

**Identifiers**:
- DOI: [10.1093/bioinformatics/btad317](https://doi.org/10.1093/bioinformatics/btad317)
- PMID: [37171896](https://pubmed.ncbi.nlm.nih.gov/37171896/)

### Cite this tool

**APA format**:

```
N, M. (2023). ViralConsensus: a fast and memory-efficient tool for calling viral consensus genome sequences directly from read alignment data. Bioinformatics (Oxford, England), 39(5). https://doi.org/10.1093/bioinformatics/btad317
```

**BibTeX format**:

```bibtex
@article{n2023,
  title = {ViralConsensus: a fast and memory-efficient tool for calling viral consensus genome sequences directly from read alignment data},
  author = {N, Moshiri},
  year = {2023},
  journal = {Bioinformatics (Oxford, England)},
  volume = {39},
  number = {5},
  doi = {10.1093/bioinformatics/btad317},
  pmid = {37171896},
}
```


## ViralMSA

**Tool URL**: [https://github.com/niemasd/ViralMSA](https://github.com/niemasd/ViralMSA)

### Publication Information

**Title**: ViralMSA: massively scalable reference-guided multiple sequence alignment of viral genomes.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa743](https://doi.org/10.1093/bioinformatics/btaa743)
- PMID: [32814953](https://pubmed.ncbi.nlm.nih.gov/32814953/)

### Cite this tool

**APA format**:

```
N, M. (2021). ViralMSA: massively scalable reference-guided multiple sequence alignment of viral genomes. Bioinformatics (Oxford, England), 37(5), 714-716. https://doi.org/10.1093/bioinformatics/btaa743
```

**BibTeX format**:

```bibtex
@article{n2021,
  title = {ViralMSA: massively scalable reference-guided multiple sequence alignment of viral genomes},
  author = {N, Moshiri},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {5},
  pages = {714-716},
  doi = {10.1093/bioinformatics/btaa743},
  pmid = {32814953},
}
```


## VIRALpro

**Tool URL**: [http://scratch.proteomics.ics.uci.edu/](http://scratch.proteomics.ics.uci.edu/)

### Publication Information

**Title**: VIRALpro: a tool to identify viral capsid and tail sequences.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2016

**Identifiers**:
- DOI: [10.1093/bioinformatics/btv727](https://doi.org/10.1093/bioinformatics/btv727)
- PMID: [26733451](https://pubmed.ncbi.nlm.nih.gov/26733451/)

### Cite this tool

**APA format**:

```
C, G., CN, M., F, C., & P, B. (2016). VIRALpro: a tool to identify viral capsid and tail sequences. Bioinformatics (Oxford, England), 32(9), 1405-7. https://doi.org/10.1093/bioinformatics/btv727
```

**BibTeX format**:

```bibtex
@article{c2016,
  title = {VIRALpro: a tool to identify viral capsid and tail sequences},
  author = {C, Galiez and CN, Magnan and F, Coste and P, Baldi},
  year = {2016},
  journal = {Bioinformatics (Oxford, England)},
  volume = {32},
  number = {9},
  pages = {1405-7},
  doi = {10.1093/bioinformatics/btv727},
  pmid = {26733451},
}
```


## viralVerify

**Tool URL**: [https://github.com/ablab/viralVerify/](https://github.com/ablab/viralVerify/)

### Publication Information

**Title**: Metaviral SPAdes: assembly of viruses from metagenomic data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa490](https://doi.org/10.1093/bioinformatics/btaa490)
- PMID: [32413137](https://pubmed.ncbi.nlm.nih.gov/32413137/)

### Cite this tool

**APA format**:

```
D, A., M, R., A, L., & PA, P. (2020). Metaviral SPAdes: assembly of viruses from metagenomic data. Bioinformatics (Oxford, England), 36(14), 4126-4129. https://doi.org/10.1093/bioinformatics/btaa490
```

**BibTeX format**:

```bibtex
@article{d2020,
  title = {Metaviral SPAdes: assembly of viruses from metagenomic data},
  author = {D, Antipov and M, Raiko and A, Lapidus and PA, Pevzner},
  year = {2020},
  journal = {Bioinformatics (Oxford, England)},
  volume = {36},
  number = {14},
  pages = {4126-4129},
  doi = {10.1093/bioinformatics/btaa490},
  pmid = {32413137},
}
```


## ViralWasm

**Tool URL**: [https://niema-lab.github.io/ViralWasm](https://niema-lab.github.io/ViralWasm)

### Publication Information

**Title**: ViralWasm: a client-side user-friendly web application suite for viral genomics.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2024

**Identifiers**:
- DOI: [10.1093/bioinformatics/btae018](https://doi.org/10.1093/bioinformatics/btae018)
- PMID: [38200583](https://pubmed.ncbi.nlm.nih.gov/38200583/)

### Cite this tool

**APA format**:

```
D, J., R, A., & N, M. (2024). ViralWasm: a client-side user-friendly web application suite for viral genomics. Bioinformatics (Oxford, England), 40(1). https://doi.org/10.1093/bioinformatics/btae018
```

**BibTeX format**:

```bibtex
@article{d2024,
  title = {ViralWasm: a client-side user-friendly web application suite for viral genomics},
  author = {D, Ji and R, Aboukhalil and N, Moshiri},
  year = {2024},
  journal = {Bioinformatics (Oxford, England)},
  volume = {40},
  number = {1},
  doi = {10.1093/bioinformatics/btae018},
  pmid = {38200583},
}
```


## viraMiner

**Tool URL**: [https://github.com/NeuroCSUT/ViraMiner](https://github.com/NeuroCSUT/ViraMiner)

### Publication Information

**Title**: ViraMiner: Deep learning on raw DNA sequences for identifying viral genomes in human samples.

**Journal**: PloS one

**Year**: 2019

**Identifiers**:
- DOI: [10.1371/journal.pone.0222271](https://doi.org/10.1371/journal.pone.0222271)
- PMID: [31509583](https://pubmed.ncbi.nlm.nih.gov/31509583/)

### Cite this tool

**APA format**:

```
A, T., Z, B., J, D., & R, V. (2019). ViraMiner: Deep learning on raw DNA sequences for identifying viral genomes in human samples. PloS one, 14(9), e0222271. https://doi.org/10.1371/journal.pone.0222271
```

**BibTeX format**:

```bibtex
@article{a2019,
  title = {ViraMiner: Deep learning on raw DNA sequences for identifying viral genomes in human samples},
  author = {A, Tampuu and Z, Bzhalava and J, Dillner and R, Vicente},
  year = {2019},
  journal = {PloS one},
  volume = {14},
  number = {9},
  pages = {e0222271},
  doi = {10.1371/journal.pone.0222271},
  pmid = {31509583},
}
```


## VirFinder

**Tool URL**: [https://github.com/jessieren/VirFinder](https://github.com/jessieren/VirFinder)

### Publication Information

**Title**: VirFinder: a novel k-mer based tool for identifying viral sequences from assembled metagenomic data.

**Journal**: Microbiome

**Year**: 2017

**Identifiers**:
- DOI: [10.1186/s40168-017-0283-5](https://doi.org/10.1186/s40168-017-0283-5)
- PMID: [28683828](https://pubmed.ncbi.nlm.nih.gov/28683828/)

### Cite this tool

**APA format**:

```
J, R., NA, A., YY, L., JA, F., & F, S. (2017). VirFinder: a novel k-mer based tool for identifying viral sequences from assembled metagenomic data. Microbiome, 5(1), 69. https://doi.org/10.1186/s40168-017-0283-5
```

**BibTeX format**:

```bibtex
@article{j2017,
  title = {VirFinder: a novel k-mer based tool for identifying viral sequences from assembled metagenomic data},
  author = {J, Ren and NA, Ahlgren and YY, Lu and JA, Fuhrman and F, Sun},
  year = {2017},
  journal = {Microbiome},
  volume = {5},
  number = {1},
  pages = {69},
  doi = {10.1186/s40168-017-0283-5},
  pmid = {28683828},
}
```


## VirHostMatcher

**Tool URL**: [https://github.com/jessieren/VirHostMatcher](https://github.com/jessieren/VirHostMatcher)

### Publication Information

**Title**: Prokaryotic virus host predictor: a Gaussian model for host prediction of prokaryotic viruses in metagenomics.

**Journal**: BMC biology

**Year**: 2021

**Identifiers**:
- DOI: [10.1186/s12915-020-00938-6](https://doi.org/10.1186/s12915-020-00938-6)
- PMID: [33441133](https://pubmed.ncbi.nlm.nih.gov/33441133/)

### Cite this tool

**APA format**:

```
C, L., Z, Z., Z, C., Z, Z., Y, Q., A, W., ... Y, P. (2021). Prokaryotic virus host predictor: a Gaussian model for host prediction of prokaryotic viruses in metagenomics. BMC biology, 19(1), 5. https://doi.org/10.1186/s12915-020-00938-6
```

**BibTeX format**:

```bibtex
@article{c2021,
  title = {Prokaryotic virus host predictor: a Gaussian model for host prediction of prokaryotic viruses in metagenomics},
  author = {C, Lu and Z, Zhang and Z, Cai and Z, Zhu and Y, Qiu and A, Wu and T, Jiang and H, Zheng and Y, Peng},
  year = {2021},
  journal = {BMC biology},
  volume = {19},
  number = {1},
  pages = {5},
  doi = {10.1186/s12915-020-00938-6},
  pmid = {33441133},
}
```


## VirHostMatcher-Net

**Tool URL**: [https://github.com/WeiliWw/VirHostMatcher-Net](https://github.com/WeiliWw/VirHostMatcher-Net)

### Publication Information

**Title**: A network-based integrated framework for predicting virus-prokaryote interactions.

**Journal**: NAR genomics and bioinformatics

**Year**: 2020

**Identifiers**:
- DOI: [10.1093/nargab/lqaa044](https://doi.org/10.1093/nargab/lqaa044)
- PMID: [32626849](https://pubmed.ncbi.nlm.nih.gov/32626849/)

### Cite this tool

**APA format**:

```
W, W., J, R., K, T., E, D., JC, I., JA, F., ... NA, A. (2020). A network-based integrated framework for predicting virus-prokaryote interactions. NAR genomics and bioinformatics, 2(2), lqaa044. https://doi.org/10.1093/nargab/lqaa044
```

**BibTeX format**:

```bibtex
@article{w2020,
  title = {A network-based integrated framework for predicting virus-prokaryote interactions},
  author = {W, Wang and J, Ren and K, Tang and E, Dart and JC, Ignacio-Espinoza and JA, Fuhrman and J, Braun and F, Sun and NA, Ahlgren},
  year = {2020},
  journal = {NAR genomics and bioinformatics},
  volume = {2},
  number = {2},
  pages = {lqaa044},
  doi = {10.1093/nargab/lqaa044},
  pmid = {32626849},
}
```


## Virhunter

**Tool URL**: [https://github.com/cbib/virhunter](https://github.com/cbib/virhunter)

### Publication Information

**Title**: VirHunter: A Deep Learning-Based Method for Detection of Novel RNA Viruses in Plant Sequencing Data.

**Journal**: Frontiers in bioinformatics

**Year**: 2022

**Identifiers**:
- DOI: [10.3389/fbinf.2022.867111](https://doi.org/10.3389/fbinf.2022.867111)
- PMID: [36304258](https://pubmed.ncbi.nlm.nih.gov/36304258/)

### Cite this tool

**APA format**:

```
G, S., M, K., O, G., T, C., A, M., & M, N. (2022). VirHunter: A Deep Learning-Based Method for Detection of Novel RNA Viruses in Plant Sequencing Data. Frontiers in bioinformatics, 2, 867111. https://doi.org/10.3389/fbinf.2022.867111
```

**BibTeX format**:

```bibtex
@article{g2022,
  title = {VirHunter: A Deep Learning-Based Method for Detection of Novel RNA Viruses in Plant Sequencing Data},
  author = {G, Sukhorukov and M, Khalili and O, Gascuel and T, Candresse and A, Marais-Colombel and M, Nikolski},
  year = {2022},
  journal = {Frontiers in bioinformatics},
  volume = {2},
  pages = {867111},
  doi = {10.3389/fbinf.2022.867111},
  pmid = {36304258},
}
```


## VIRIDIC

**Tool URL**: [https://www.mdpi.com/1999-4915/12/11/1268](https://www.mdpi.com/1999-4915/12/11/1268)

### Publication Information

**Title**: VIRIDIC-A Novel Tool to Calculate the Intergenomic Similarities of Prokaryote-Infecting Viruses.

**Journal**: Viruses

**Year**: 2020

**Identifiers**:
- DOI: [10.3390/v12111268](https://doi.org/10.3390/v12111268)
- PMID: [33172115](https://pubmed.ncbi.nlm.nih.gov/33172115/)

### Cite this tool

**APA format**:

```
C, M., A, V., & AM, K. (2020). VIRIDIC-A Novel Tool to Calculate the Intergenomic Similarities of Prokaryote-Infecting Viruses. Viruses, 12(11). https://doi.org/10.3390/v12111268
```

**BibTeX format**:

```bibtex
@article{c2020,
  title = {VIRIDIC-A Novel Tool to Calculate the Intergenomic Similarities of Prokaryote-Infecting Viruses},
  author = {C, Moraru and A, Varsani and AM, Kropinski},
  year = {2020},
  journal = {Viruses},
  volume = {12},
  number = {11},
  doi = {10.3390/v12111268},
  pmid = {33172115},
}
```


## VIRify

**Tool URL**: [https://github.com/EBI-Metagenomics/emg-viral-pipeline](https://github.com/EBI-Metagenomics/emg-viral-pipeline)

### Publication Information

**Title**: VIRify: An integrated detection, annotation and taxonomic classification pipeline using virus-specific protein profile hidden Markov models.

**Journal**: PLoS computational biology

**Year**: 2023

**Identifiers**:
- DOI: [10.1371/journal.pcbi.1011422](https://doi.org/10.1371/journal.pcbi.1011422)
- PMID: [37639475](https://pubmed.ncbi.nlm.nih.gov/37639475/)

### Cite this tool

**APA format**:

```
G, R., A, A., M, B., E, S., M, M., A, R.M., ... RD, F. (2023). VIRify: An integrated detection, annotation and taxonomic classification pipeline using virus-specific protein profile hidden Markov models. PLoS computational biology, 19(8), e1011422. https://doi.org/10.1371/journal.pcbi.1011422
```

**BibTeX format**:

```bibtex
@article{g2023,
  title = {VIRify: An integrated detection, annotation and taxonomic classification pipeline using virus-specific protein profile hidden Markov models},
  author = {G, Rangel-Pineros and A, Almeida and M, Beracochea and E, Sakharova and M, Marz and A, Reyes Muñoz and M, Hölzer and RD, Finn},
  year = {2023},
  journal = {PLoS computational biology},
  volume = {19},
  number = {8},
  pages = {e1011422},
  doi = {10.1371/journal.pcbi.1011422},
  pmid = {37639475},
}
```


## Virion

**Tool URL**: [https://github.com/viralemergence/virion](https://github.com/viralemergence/virion)

### Publication Information

**Title**: Isolation and Characterization of Monoclonal Antibodies Against a Virion Core Protein of Orf Virus Strain NA1/11 As Potential Diagnostic Tool for Orf Viruses.

**Journal**: Monoclonal antibodies in immunodiagnosis and immunotherapy

**Year**: 2015

**Identifiers**:
- DOI: [10.1089/mab.2014.0101](https://doi.org/10.1089/mab.2014.0101)
- PMID: [26301926](https://pubmed.ncbi.nlm.nih.gov/26301926/)

### Cite this tool

**APA format**:

```
X, W., J, Z., W, H., Y, P., H, L., W, L., ... S, L. (2015). Isolation and Characterization of Monoclonal Antibodies Against a Virion Core Protein of Orf Virus Strain NA1/11 As Potential Diagnostic Tool for Orf Viruses. Monoclonal antibodies in immunodiagnosis and immunotherapy, 34(4), 233-45. https://doi.org/10.1089/mab.2014.0101
```

**BibTeX format**:

```bibtex
@article{x2015,
  title = {Isolation and Characterization of Monoclonal Antibodies Against a Virion Core Protein of Orf Virus Strain NA1/11 As Potential Diagnostic Tool for Orf Viruses},
  author = {X, Wang and J, Zhang and W, Hao and Y, Peng and H, Li and W, Li and M, Li and S, Luo},
  year = {2015},
  journal = {Monoclonal antibodies in immunodiagnosis and immunotherapy},
  volume = {34},
  number = {4},
  pages = {233-45},
  doi = {10.1089/mab.2014.0101},
  pmid = {26301926},
}
```


## VirMet

**Tool URL**: [https://github.com/medvir/VirMet](https://github.com/medvir/VirMet)

### Publication Information

**Title**: Benchmark of thirteen bioinformatic pipelines for metagenomic virus diagnostics using datasets from clinical samples.

**Journal**: Journal of clinical virology : the official publication of the Pan American Society for Clinical Virology

**Year**: 2021

**Identifiers**:
- DOI: [10.1016/j.jcv.2021.104908](https://doi.org/10.1016/j.jcv.2021.104908)
- PMID: [34273858](https://pubmed.ncbi.nlm.nih.gov/34273858/)

### Cite this tool

**APA format**:

```
JJC, d.V., JR, B., N, F., IA, S., S, M., J, H., ... ECJ, C. (2021). Benchmark of thirteen bioinformatic pipelines for metagenomic virus diagnostics using datasets from clinical samples. Journal of clinical virology : the official publication of the Pan American Society for Clinical Virology, 141, 104908. https://doi.org/10.1016/j.jcv.2021.104908
```

**BibTeX format**:

```bibtex
@article{jjc2021,
  title = {Benchmark of thirteen bioinformatic pipelines for metagenomic virus diagnostics using datasets from clinical samples},
  author = {JJC, de Vries and JR, Brown and N, Fischer and IA, Sidorov and S, Morfopoulou and J, Huang and BBO, Munnink and A, Sayiner and A, Bulgurcu and C, Rodriguez and G, Gricourt and E, Keyaerts and L, Beller and C, Bachofen and J, Kubacki and C, Samuel and L, Florian and S, Dennis and M, Beer and D, Hoeper and M, Huber and V, Kufner and M, Zaheri and A, Lebrand and A, Papa and S, van Boheemen and ACM, Kroes and J, Breuer and FX, Lopez-Labrador and ECJ, Claas},
  year = {2021},
  journal = {Journal of clinical virology : the official publication of the Pan American Society for Clinical Virology},
  volume = {141},
  pages = {104908},
  doi = {10.1016/j.jcv.2021.104908},
  pmid = {34273858},
}
```


## VirMine

**Tool URL**: [https://github.com/thatzopoulos/virMine](https://github.com/thatzopoulos/virMine)

### Publication Information

**Title**: virMine: automated detection of viral sequences from complex metagenomic samples.

**Journal**: PeerJ

**Year**: 2019

**Identifiers**:
- DOI: [10.7717/peerj.6695](https://doi.org/10.7717/peerj.6695)
- PMID: [30993039](https://pubmed.ncbi.nlm.nih.gov/30993039/)

### Cite this tool

**APA format**:

```
A, G., T, H., & C, P. (2019). virMine: automated detection of viral sequences from complex metagenomic samples. PeerJ, 7, e6695. https://doi.org/10.7717/peerj.6695
```

**BibTeX format**:

```bibtex
@article{a2019,
  title = {virMine: automated detection of viral sequences from complex metagenomic samples},
  author = {A, Garretto and T, Hatzopoulos and C, Putonti},
  year = {2019},
  journal = {PeerJ},
  volume = {7},
  pages = {e6695},
  doi = {10.7717/peerj.6695},
  pmid = {30993039},
}
```


## virMiner

**Tool URL**: [https://github.com/TingtZHENG/VirMiner](https://github.com/TingtZHENG/VirMiner)

### Publication Information

**Title**: Mining, analyzing, and integrating viral signals from metagenomic data.

**Journal**: Microbiome

**Year**: 2019

**Identifiers**:
- DOI: [10.1186/s40168-019-0657-y](https://doi.org/10.1186/s40168-019-0657-y)
- PMID: [30890181](https://pubmed.ncbi.nlm.nih.gov/30890181/)

### Cite this tool

**APA format**:

```
T, Z., J, L., Y, N., K, K., MA, M., L, I., ... G, P. (2019). Mining, analyzing, and integrating viral signals from metagenomic data. Microbiome, 7(1), 42. https://doi.org/10.1186/s40168-019-0657-y
```

**BibTeX format**:

```bibtex
@article{t2019,
  title = {Mining, analyzing, and integrating viral signals from metagenomic data},
  author = {T, Zheng and J, Li and Y, Ni and K, Kang and MA, Misiakou and L, Imamovic and BKC, Chow and AA, Rode and P, Bytzer and M, Sommer and G, Panagiotou},
  year = {2019},
  journal = {Microbiome},
  volume = {7},
  number = {1},
  pages = {42},
  doi = {10.1186/s40168-019-0657-y},
  pmid = {30890181},
}
```


## VIRMOTIF

**Tool URL**: [https://gitlab.com/pedram56rajaii/virmotif](https://gitlab.com/pedram56rajaii/virmotif)

### Publication Information

**Title**: VIRMOTIF: A User-Friendly Tool for Viral Sequence Analysis.

**Journal**: Genes

**Year**: 2021

**Identifiers**:
- DOI: [10.3390/genes12020186](https://doi.org/10.3390/genes12020186)
- PMID: [33514039](https://pubmed.ncbi.nlm.nih.gov/33514039/)

### Cite this tool

**APA format**:

```
P, R., KH, J., A, B., SS, B., A, D., & H, A. (2021). VIRMOTIF: A User-Friendly Tool for Viral Sequence Analysis. Genes, 12(2). https://doi.org/10.3390/genes12020186
```

**BibTeX format**:

```bibtex
@article{p2021,
  title = {VIRMOTIF: A User-Friendly Tool for Viral Sequence Analysis},
  author = {P, Rajaei and KH, Jahanian and A, Beheshti and SS, Band and A, Dehzangi and H, Alinejad-Rokny},
  year = {2021},
  journal = {Genes},
  volume = {12},
  number = {2},
  doi = {10.3390/genes12020186},
  pmid = {33514039},
}
```


## ViromeQC

**Tool URL**: [https://github.com/SegataLab/viromeqc](https://github.com/SegataLab/viromeqc)

### Publication Information

**Title**: Detecting contamination in viromes using ViromeQC.

**Journal**: Nature biotechnology

**Year**: 2019

**Identifiers**:
- DOI: [10.1038/s41587-019-0334-5](https://doi.org/10.1038/s41587-019-0334-5)
- PMID: [31748692](https://pubmed.ncbi.nlm.nih.gov/31748692/)

### Cite this tool

**APA format**:

```
M, Z., F, P., F, A., P, M., A, T., FD, B., & N, S. (2019). Detecting contamination in viromes using ViromeQC. Nature biotechnology, 37(12), 1408-1412. https://doi.org/10.1038/s41587-019-0334-5
```

**BibTeX format**:

```bibtex
@article{m2019,
  title = {Detecting contamination in viromes using ViromeQC},
  author = {M, Zolfo and F, Pinto and F, Asnicar and P, Manghi and A, Tett and FD, Bushman and N, Segata},
  year = {2019},
  journal = {Nature biotechnology},
  volume = {37},
  number = {12},
  pages = {1408-1412},
  doi = {10.1038/s41587-019-0334-5},
  pmid = {31748692},
}
```


## ViroProfiler

**Tool URL**: [https://github.com/deng-lab/viroprofiler](https://github.com/deng-lab/viroprofiler)

### Publication Information

**Title**: ViroProfiler: a containerized bioinformatics pipeline for viral metagenomic data analysis.

**Journal**: Gut microbes

**Year**: 2023

**Identifiers**:
- DOI: [10.1080/19490976.2023.2192522](https://doi.org/10.1080/19490976.2023.2192522)
- PMID: [36998174](https://pubmed.ncbi.nlm.nih.gov/36998174/)

### Cite this tool

**APA format**:

```
J, R., M, K.M., J, X., X, P., & L, D. (2023). ViroProfiler: a containerized bioinformatics pipeline for viral metagenomic data analysis. Gut microbes, 15(1), 2192522. https://doi.org/10.1080/19490976.2023.2192522
```

**BibTeX format**:

```bibtex
@article{j2023,
  title = {ViroProfiler: a containerized bioinformatics pipeline for viral metagenomic data analysis},
  author = {J, Ru and M, Khan Mirzaei and J, Xue and X, Peng and L, Deng},
  year = {2023},
  journal = {Gut microbes},
  volume = {15},
  number = {1},
  pages = {2192522},
  doi = {10.1080/19490976.2023.2192522},
  pmid = {36998174},
}
```


## Virosaurus

**Tool URL**: [https://viralzone.expasy.org/8676](https://viralzone.expasy.org/8676)

### Publication Information

**Title**: Virosaurus A Reference to Explore and Capture Virus Genetic Diversity.

**Journal**: Viruses

**Year**: 2020

**Identifiers**:
- DOI: [10.3390/v12111248](https://doi.org/10.3390/v12111248)
- PMID: [33139591](https://pubmed.ncbi.nlm.nih.gov/33139591/)

### Cite this tool

**APA format**:

```
A, G., F, L., N, G., C, I., T, J., S, C., ... PL, M. (2020). Virosaurus A Reference to Explore and Capture Virus Genetic Diversity. Viruses, 12(11). https://doi.org/10.3390/v12111248
```

**BibTeX format**:

```bibtex
@article{a2020,
  title = {Virosaurus A Reference to Explore and Capture Virus Genetic Diversity},
  author = {A, Gleizes and F, Laubscher and N, Guex and C, Iseli and T, Junier and S, Cordey and J, Fellay and I, Xenarios and L, Kaiser and PL, Mercier},
  year = {2020},
  journal = {Viruses},
  volume = {12},
  number = {11},
  doi = {10.3390/v12111248},
  pmid = {33139591},
}
```


## VirSorter

**Tool URL**: [https://github.com/simroux/VirSorter](https://github.com/simroux/VirSorter)

### Publication Information

**Title**: VirSorter: mining viral signal from microbial genomic data.

**Journal**: PeerJ

**Year**: 2015

**Identifiers**:
- DOI: [10.7717/peerj.985](https://doi.org/10.7717/peerj.985)
- PMID: [26038737](https://pubmed.ncbi.nlm.nih.gov/26038737/)

### Cite this tool

**APA format**:

```
S, R., F, E., BL, H., & MB, S. (2015). VirSorter: mining viral signal from microbial genomic data. PeerJ, 3, e985. https://doi.org/10.7717/peerj.985
```

**BibTeX format**:

```bibtex
@article{s2015,
  title = {VirSorter: mining viral signal from microbial genomic data},
  author = {S, Roux and F, Enault and BL, Hurwitz and MB, Sullivan},
  year = {2015},
  journal = {PeerJ},
  volume = {3},
  pages = {e985},
  doi = {10.7717/peerj.985},
  pmid = {26038737},
}
```


## VirSorter2

**Tool URL**: [https://bitbucket.org/MAVERICLab/virsorter2/](https://bitbucket.org/MAVERICLab/virsorter2/)

### Publication Information

**Title**: VirSorter2: a multi-classifier, expert-guided approach to detect diverse DNA and RNA viruses.

**Journal**: Microbiome

**Year**: 2021

**Identifiers**:
- DOI: [10.1186/s40168-020-00990-y](https://doi.org/10.1186/s40168-020-00990-y)
- PMID: [33522966](https://pubmed.ncbi.nlm.nih.gov/33522966/)

### Cite this tool

**APA format**:

```
J, G., B, B., AA, Z., A, V., G, D., TO, D., ... S, R. (2021). VirSorter2: a multi-classifier, expert-guided approach to detect diverse DNA and RNA viruses. Microbiome, 9(1), 37. https://doi.org/10.1186/s40168-020-00990-y
```

**BibTeX format**:

```bibtex
@article{j2021,
  title = {VirSorter2: a multi-classifier, expert-guided approach to detect diverse DNA and RNA viruses},
  author = {J, Guo and B, Bolduc and AA, Zayed and A, Varsani and G, Dominguez-Huerta and TO, Delmont and AA, Pratama and MC, Gazitúa and D, Vik and MB, Sullivan and S, Roux},
  year = {2021},
  journal = {Microbiome},
  volume = {9},
  number = {1},
  pages = {37},
  doi = {10.1186/s40168-020-00990-y},
  pmid = {33522966},
}
```


## Virtifier

**Tool URL**: [https://github.com/crazyinter/Seq2Vec](https://github.com/crazyinter/Seq2Vec)

### Publication Information

**Title**: Virtifier: a deep learning-based identifier for viral sequences from metagenomes.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/bioinformatics/btab845](https://doi.org/10.1093/bioinformatics/btab845)
- PMID: [34908121](https://pubmed.ncbi.nlm.nih.gov/34908121/)

### Cite this tool

**APA format**:

```
Y, M., F, L., T, H., & Y, L. (2022). Virtifier: a deep learning-based identifier for viral sequences from metagenomes. Bioinformatics (Oxford, England), 38(5), 1216-1222. https://doi.org/10.1093/bioinformatics/btab845
```

**BibTeX format**:

```bibtex
@article{y2022,
  title = {Virtifier: a deep learning-based identifier for viral sequences from metagenomes},
  author = {Y, Miao and F, Liu and T, Hou and Y, Liu},
  year = {2022},
  journal = {Bioinformatics (Oxford, England)},
  volume = {38},
  number = {5},
  pages = {1216-1222},
  doi = {10.1093/bioinformatics/btab845},
  pmid = {34908121},
}
```


## Virtus

**Tool URL**: [https://github.com/yyoshiaki/VIRTUS](https://github.com/yyoshiaki/VIRTUS)

### Publication Information

**Title**: VIRTUS: a pipeline for comprehensive virus analysis from conventional RNA-seq data.

**Journal**: Bioinformatics (Oxford, England)

**Year**: 2021

**Identifiers**:
- DOI: [10.1093/bioinformatics/btaa859](https://doi.org/10.1093/bioinformatics/btaa859)
- PMID: [33017003](https://pubmed.ncbi.nlm.nih.gov/33017003/)

### Cite this tool

**APA format**:

```
Y, Y., A, H., S, S., & N, O. (2021). VIRTUS: a pipeline for comprehensive virus analysis from conventional RNA-seq data. Bioinformatics (Oxford, England), 37(10), 1465-1467. https://doi.org/10.1093/bioinformatics/btaa859
```

**BibTeX format**:

```bibtex
@article{y2021,
  title = {VIRTUS: a pipeline for comprehensive virus analysis from conventional RNA-seq data},
  author = {Y, Yasumizu and A, Hara and S, Sakaguchi and N, Ohkura},
  year = {2021},
  journal = {Bioinformatics (Oxford, England)},
  volume = {37},
  number = {10},
  pages = {1465-1467},
  doi = {10.1093/bioinformatics/btaa859},
  pmid = {33017003},
}
```


## virus_prediction

**Tool URL**: [https://github.com/rujinlong/virus_prediction](https://github.com/rujinlong/virus_prediction)

### Publication Information

**Title**: Zika Virus Prediction Using AI-Driven Technology and Hybrid Optimization Algorithm in Healthcare.

**Journal**: Journal of healthcare engineering

**Year**: 2022

**Identifiers**:
- DOI: [10.1155/2022/2793850](https://doi.org/10.1155/2022/2793850)
- PMID: [35070231](https://pubmed.ncbi.nlm.nih.gov/35070231/)

### Cite this tool

**APA format**:

```
P, D., A, M., S, T., S, K., P, S., S, G., & HK, A. (2022). Zika Virus Prediction Using AI-Driven Technology and Hybrid Optimization Algorithm in Healthcare. Journal of healthcare engineering, 2022, 2793850. https://doi.org/10.1155/2022/2793850
```

**BibTeX format**:

```bibtex
@article{p2022,
  title = {Zika Virus Prediction Using AI-Driven Technology and Hybrid Optimization Algorithm in Healthcare},
  author = {P, Dadheech and A, Mehbodniya and S, Tiwari and S, Kumar and P, Singh and S, Gupta and HK, Atiglah},
  year = {2022},
  journal = {Journal of healthcare engineering},
  volume = {2022},
  pages = {2793850},
  doi = {10.1155/2022/2793850},
  pmid = {35070231},
}
```


## Viruses.String

**Tool URL**: [http://viruses.string-db.org/](http://viruses.string-db.org/)

### Publication Information

**Title**: Viruses.STRING: A Virus-Host Protein-Protein Interaction Database.

**Journal**: Viruses

**Year**: 2018

**Identifiers**:
- DOI: [10.3390/v10100519](https://doi.org/10.3390/v10100519)
- PMID: [30249048](https://pubmed.ncbi.nlm.nih.gov/30249048/)

### Cite this tool

**APA format**:

```
HV, C., NT, D., D, S., C, v.M., & LJ, J. (2018). Viruses.STRING: A Virus-Host Protein-Protein Interaction Database. Viruses, 10(10). https://doi.org/10.3390/v10100519
```

**BibTeX format**:

```bibtex
@article{hv2018,
  title = {Viruses.STRING: A Virus-Host Protein-Protein Interaction Database},
  author = {HV, Cook and NT, Doncheva and D, Szklarczyk and C, von Mering and LJ, Jensen},
  year = {2018},
  journal = {Viruses},
  volume = {10},
  number = {10},
  doi = {10.3390/v10100519},
  pmid = {30249048},
}
```


## VirusSeeker

**Tool URL**: [https://wupathlabs.wustl.edu/virusseeker/](https://wupathlabs.wustl.edu/virusseeker/)

### Publication Information

**Title**: VirusSeeker, a computational pipeline for virus discovery and virome composition analysis.

**Journal**: Virology

**Year**: 2017

**Identifiers**:
- DOI: [10.1016/j.virol.2017.01.005](https://doi.org/10.1016/j.virol.2017.01.005)
- PMID: [28110145](https://pubmed.ncbi.nlm.nih.gov/28110145/)

### Cite this tool

**APA format**:

```
G, Z., G, W., ES, L., L, D., S, K., DH, B., ... D, W. (2017). VirusSeeker, a computational pipeline for virus discovery and virome composition analysis. Virology, 503, 21-30. https://doi.org/10.1016/j.virol.2017.01.005
```

**BibTeX format**:

```bibtex
@article{g2017,
  title = {VirusSeeker, a computational pipeline for virus discovery and virome composition analysis},
  author = {G, Zhao and G, Wu and ES, Lim and L, Droit and S, Krishnamurthy and DH, Barouch and HW, Virgin and D, Wang},
  year = {2017},
  journal = {Virology},
  volume = {503},
  pages = {21-30},
  doi = {10.1016/j.virol.2017.01.005},
  pmid = {28110145},
}
```


## VirusTAP

**Tool URL**: [https://web.archive.org/web/20190320032516/https://github.com/bioinformatics-toyama/VirusTAP](https://web.archive.org/web/20190320032516/https://github.com/bioinformatics-toyama/VirusTAP)

### Publication Information

**Title**: VirusTAP: Viral Genome-Targeted Assembly Pipeline.

**Journal**: Frontiers in microbiology

**Year**: 2016

**Identifiers**:
- DOI: [10.3389/fmicb.2016.00032](https://doi.org/10.3389/fmicb.2016.00032)
- PMID: [26870004](https://pubmed.ncbi.nlm.nih.gov/26870004/)

### Cite this tool

**APA format**:

```
A, Y., T, S., & M, K. (2016). VirusTAP: Viral Genome-Targeted Assembly Pipeline. Frontiers in microbiology, 7, 32. https://doi.org/10.3389/fmicb.2016.00032
```

**BibTeX format**:

```bibtex
@article{a2016,
  title = {VirusTAP: Viral Genome-Targeted Assembly Pipeline},
  author = {A, Yamashita and T, Sekizuka and M, Kuroda},
  year = {2016},
  journal = {Frontiers in microbiology},
  volume = {7},
  pages = {32},
  doi = {10.3389/fmicb.2016.00032},
  pmid = {26870004},
}
```


## VirusTaxo

**Tool URL**: [https://github.com/nahid18/virustaxo-wf](https://github.com/nahid18/virustaxo-wf)

### Publication Information

**Title**: ['VirusTaxo: Taxonomic classification of viruses from the genome sequence using k-mer enrichment']

### Cite this tool

**APA format**:

```
Raju, R., Al Nahid, A., Chondrow Dev, P., & Islam, R. (2022). VirusTaxo: Taxonomic classification of viruses from the genome sequence using k-mer enrichment. Genomics, 114(4), 110414. https://doi.org/10.1016/j.ygeno.2022.110414
```

**BibTeX format**:

```bibtex
@article{raju2022,
  title = {VirusTaxo: Taxonomic classification of viruses from the genome sequence using k-mer enrichment},
  author = {Raju, R. and Al Nahid, A. and Chondrow Dev, P. and Islam, R.},
  year = {2022},
  journal = {Genomics},
  volume = {114},
  number = {4},
  pages = {110414},
  doi = {10.1016/j.ygeno.2022.110414},
}
```


## VogDB

**Tool URL**: [http://vogdb.org/](http://vogdb.org/)

### Publication Information

**Title**: VOGDB-Database of Virus Orthologous Groups.

**Journal**: Viruses

**Year**: 2024

**Identifiers**:
- DOI: [10.3390/v16081191](https://doi.org/10.3390/v16081191)
- PMID: [39205165](https://pubmed.ncbi.nlm.nih.gov/39205165/)

### Cite this tool

**APA format**:

```
L, T., HJ, H., J, M., A, P., D, F., M, K., ... T, R. (2024). VOGDB-Database of Virus Orthologous Groups. Viruses, 16(8). https://doi.org/10.3390/v16081191
```

**BibTeX format**:

```bibtex
@article{l2024,
  title = {VOGDB-Database of Virus Orthologous Groups},
  author = {L, Trgovec-Greif and HJ, Hellinger and J, Mainguy and A, Pfundner and D, Frishman and M, Kiening and NS, Webster and PW, Laffy and M, Feichtinger and T, Rattei},
  year = {2024},
  journal = {Viruses},
  volume = {16},
  number = {8},
  doi = {10.3390/v16081191},
  pmid = {39205165},
}
```


## vRhyme

**Tool URL**: [https://github.com/AnantharamanLab/vRhyme](https://github.com/AnantharamanLab/vRhyme)

### Publication Information

**Title**: vRhyme enables binning of viral genomes from metagenomes.

**Journal**: Nucleic acids research

**Year**: 2022

**Identifiers**:
- DOI: [10.1093/nar/gkac341](https://doi.org/10.1093/nar/gkac341)
- PMID: [35544285](https://pubmed.ncbi.nlm.nih.gov/35544285/)

### Cite this tool

**APA format**:

```
K, K., A, A., R, S., L, K., & K, A. (2022). vRhyme enables binning of viral genomes from metagenomes. Nucleic acids research, 50(14), e83. https://doi.org/10.1093/nar/gkac341
```

**BibTeX format**:

```bibtex
@article{k2022,
  title = {vRhyme enables binning of viral genomes from metagenomes},
  author = {K, Kieft and A, Adams and R, Salamzade and L, Kalan and K, Anantharaman},
  year = {2022},
  journal = {Nucleic acids research},
  volume = {50},
  number = {14},
  pages = {e83},
  doi = {10.1093/nar/gkac341},
  pmid = {35544285},
}
```


## WIsH

**Tool URL**: [https://github.com/soedinglab/WIsH](https://github.com/soedinglab/WIsH)

### Publication Information

**Title**: What tool or method do you wish existed?

**Journal**: Cell

**Year**: 2024

**Identifiers**:
- DOI: [10.1016/j.cell.2024.07.043](https://doi.org/10.1016/j.cell.2024.07.043)
- PMID: [39178826](https://pubmed.ncbi.nlm.nih.gov/39178826/)

### Cite this tool

**APA format**:

```
YY, C., CL, E., M, M., ER, D., M, M., M, B., ... X, S. (2024). What tool or method do you wish existed?. Cell, 187(17), 4433-4438. https://doi.org/10.1016/j.cell.2024.07.043
```

**BibTeX format**:

```bibtex
@article{yy2024,
  title = {What tool or method do you wish existed?},
  author = {YY, Chen and CL, Evavold and M, Mann and ER, Davenport and M, McFall-Ngai and M, Bienko and HR, Ueda and L, Tian and N, Tjahjono and P, Anikeeva and JG, Liu and TL, Deans and X, Shen},
  year = {2024},
  journal = {Cell},
  volume = {187},
  number = {17},
  pages = {4433-4438},
  doi = {10.1016/j.cell.2024.07.043},
  pmid = {39178826},
}
```
