
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://shandley.github.io/awesome-virome/docs/latest/tutorials/host-prediction-workflows/">
      
      
        <link rel="prev" href="../comparative-virome/">
      
      
        <link rel="next" href="../../best-practices/quality-control/">
      
      
        
      
      
      <link rel="icon" href="../../assets/awesome-virome-hex.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.0">
    
    
      
        <title>Host Prediction Workflows - Awesome-Virome</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.618322db.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      
  
  
    
      
      
    
  
  
  <style>:root{--md-admonition-icon--tool:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22m21.71%2020.29-1.42%201.42a1%201%200%200%201-1.41%200L7%209.85A3.8%203.8%200%200%201%206%2010a4%204%200%200%201-3.78-5.3l2.54%202.54.53-.53%201.42-1.42.53-.53L4.7%202.22A4%204%200%200%201%2010%206a3.8%203.8%200%200%201-.15%201l11.86%2011.88a1%201%200%200%201%200%201.41M2.29%2018.88a1%201%200%200%200%200%201.41l1.42%201.42a1%201%200%200%200%201.41%200l5.47-5.46-2.83-2.83M20%202l-4%202v2l-2.17%202.17%202%202L18%208h2l2-4Z%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#tutorial-5-host-prediction-workflows" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../home.html" title="Awesome-Virome" class="md-header__button md-logo" aria-label="Awesome-Virome" data-md-component="logo">
      
  <img src="../../assets/awesome-virome-hex.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Awesome-Virome
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Host Prediction Workflows
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/shandley/awesome-virome" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    shandley/awesome-virome
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../intro/" class="md-tabs__link">
          
  
  
    
  
  Getting Started

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../fundamentals/" class="md-tabs__link">
          
  
  
    
  
  Fundamentals

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../" class="md-tabs__link">
          
  
  
    
  
  Tutorials

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../best-practices/quality-control/" class="md-tabs__link">
          
  
  
    
  
  Best Practices

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tools/overview/" class="md-tabs__link">
          
  
  
    
  
  Core Analysis Tools

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../troubleshooting/" class="md-tabs__link">
          
  
  
    
  
  Troubleshooting

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../api/overview/" class="md-tabs__link">
          
  
  
    
  
  API Reference

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../contributing/guidelines/" class="md-tabs__link">
          
  
  
    
  
  Contributing

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../home.html" title="Awesome-Virome" class="md-nav__button md-logo" aria-label="Awesome-Virome" data-md-component="logo">
      
  <img src="../../assets/awesome-virome-hex.png" alt="logo">

    </a>
    Awesome-Virome
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/shandley/awesome-virome" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    shandley/awesome-virome
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../intro/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../intro/workflows/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Typical Workflows
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../intro/versioning/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Versioning
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../fundamentals/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Fundamentals
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_3" id="__nav_3_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Fundamentals
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../fundamentals/sample-preparation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Sample Preparation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../fundamentals/sequencing-strategies/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Sequencing Strategies
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Tutorials
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_4" id="__nav_4_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Tutorials
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../basic-metagenome-virome/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Basic Metagenome Virome
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../rna-virus-discovery/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RNA Virus Discovery
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../prophage-identification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Prophage Identification
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comparative-virome/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Comparative Virome Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction Workflows
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction Workflows
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      
        Overview
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prerequisites
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Prerequisites">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required-software" class="md-nav__link">
    <span class="md-ellipsis">
      
        Required Software
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#system-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      
        System Requirements
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#background-knowledge" class="md-nav__link">
    <span class="md-ellipsis">
      
        Background Knowledge
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-prepare-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1: Prepare Data
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1: Prepare Data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      
        Download Dataset
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inspect-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Inspect Data
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-crispr-spacer-based-host-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2: CRISPR Spacer-Based Host Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: CRISPR Spacer-Based Host Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-extract-crispr-spacers-from-bacterial-mags" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.1 Extract CRISPR Spacers from Bacterial MAGs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-combine-all-crispr-spacers" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.2 Combine All CRISPR Spacers
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-blast-spacers-against-phage-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.3 BLAST Spacers Against Phage Contigs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-sequence-homology-based-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3: Sequence Homology-Based Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Sequence Homology-Based Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-protein-blast-against-bacterial-proteins" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.1 Protein BLAST Against Bacterial Proteins
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-machine-learning-based-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 4: Machine Learning-Based Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 4: Machine Learning-Based Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-wish-genome-composition-based" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.1 WIsH (Genome Composition-Based)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#42-iphop-integrated-approach" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.2 iPHoP (Integrated Approach)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-co-occurrence-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 5: Co-occurrence Analysis
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 5: Co-occurrence Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-calculate-abundance-correlations" class="md-nav__link">
    <span class="md-ellipsis">
      
        5.1 Calculate Abundance Correlations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-consensus-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 6: Consensus Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 6: Consensus Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#61-merge-all-predictions" class="md-nav__link">
    <span class="md-ellipsis">
      
        6.1 Merge All Predictions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-validation-and-confidence-assessment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 7: Validation and Confidence Assessment
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 7: Validation and Confidence Assessment">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#71-assess-prediction-quality" class="md-nav__link">
    <span class="md-ellipsis">
      
        7.1 Assess Prediction Quality
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#72-create-final-recommendation-table" class="md-nav__link">
    <span class="md-ellipsis">
      
        7.2 Create Final Recommendation Table
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-8-summary-and-interpretation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 8: Summary and Interpretation
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 8: Summary and Interpretation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#expected-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Expected Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-confidence-levels" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interpreting Confidence Levels
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#validation-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Validation Strategies
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting" class="md-nav__link">
    <span class="md-ellipsis">
      
        Troubleshooting
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Troubleshooting">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#problem-very-few-crispr-matches" class="md-nav__link">
    <span class="md-ellipsis">
      
        Problem: Very Few CRISPR Matches
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problem-conflicting-predictions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Problem: Conflicting Predictions
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problem-no-predictions-for-most-phages" class="md-nav__link">
    <span class="md-ellipsis">
      
        Problem: No Predictions for Most Phages
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      
        Next Steps
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      
        Further Reading
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Best Practices
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Best Practices
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Control
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/tool-combinations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Tool Combinations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/statistical-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Statistical Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/validation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/reproducibility/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Core Analysis Tools
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Core Analysis Tools
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/virus-identification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Virus Identification
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/host-prediction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/selection-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Selection Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../troubleshooting/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Troubleshooting
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    API Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            
  
    API Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/endpoints/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Endpoints
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            
  
    Contributing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/guidelines/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guidelines
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/validation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/updating-docs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Updating Docs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      
        Overview
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      
        Prerequisites
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Prerequisites">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#required-software" class="md-nav__link">
    <span class="md-ellipsis">
      
        Required Software
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#system-requirements" class="md-nav__link">
    <span class="md-ellipsis">
      
        System Requirements
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#background-knowledge" class="md-nav__link">
    <span class="md-ellipsis">
      
        Background Knowledge
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-prepare-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 1: Prepare Data
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1: Prepare Data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      
        Download Dataset
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inspect-data" class="md-nav__link">
    <span class="md-ellipsis">
      
        Inspect Data
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-crispr-spacer-based-host-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 2: CRISPR Spacer-Based Host Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: CRISPR Spacer-Based Host Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-extract-crispr-spacers-from-bacterial-mags" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.1 Extract CRISPR Spacers from Bacterial MAGs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-combine-all-crispr-spacers" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.2 Combine All CRISPR Spacers
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-blast-spacers-against-phage-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        2.3 BLAST Spacers Against Phage Contigs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-sequence-homology-based-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 3: Sequence Homology-Based Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Sequence Homology-Based Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-protein-blast-against-bacterial-proteins" class="md-nav__link">
    <span class="md-ellipsis">
      
        3.1 Protein BLAST Against Bacterial Proteins
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-machine-learning-based-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 4: Machine Learning-Based Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 4: Machine Learning-Based Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-wish-genome-composition-based" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.1 WIsH (Genome Composition-Based)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#42-iphop-integrated-approach" class="md-nav__link">
    <span class="md-ellipsis">
      
        4.2 iPHoP (Integrated Approach)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-co-occurrence-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 5: Co-occurrence Analysis
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 5: Co-occurrence Analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-calculate-abundance-correlations" class="md-nav__link">
    <span class="md-ellipsis">
      
        5.1 Calculate Abundance Correlations
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-consensus-prediction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 6: Consensus Prediction
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 6: Consensus Prediction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#61-merge-all-predictions" class="md-nav__link">
    <span class="md-ellipsis">
      
        6.1 Merge All Predictions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-validation-and-confidence-assessment" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 7: Validation and Confidence Assessment
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 7: Validation and Confidence Assessment">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#71-assess-prediction-quality" class="md-nav__link">
    <span class="md-ellipsis">
      
        7.1 Assess Prediction Quality
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#72-create-final-recommendation-table" class="md-nav__link">
    <span class="md-ellipsis">
      
        7.2 Create Final Recommendation Table
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-8-summary-and-interpretation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Step 8: Summary and Interpretation
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 8: Summary and Interpretation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#expected-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Expected Results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-confidence-levels" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interpreting Confidence Levels
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#validation-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Validation Strategies
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting" class="md-nav__link">
    <span class="md-ellipsis">
      
        Troubleshooting
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Troubleshooting">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#problem-very-few-crispr-matches" class="md-nav__link">
    <span class="md-ellipsis">
      
        Problem: Very Few CRISPR Matches
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problem-conflicting-predictions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Problem: Conflicting Predictions
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#problem-no-predictions-for-most-phages" class="md-nav__link">
    <span class="md-ellipsis">
      
        Problem: No Predictions for Most Phages
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      
        Next Steps
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      
        Further Reading
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="tutorial-5-host-prediction-workflows">Tutorial 5: Host Prediction Workflows</h1>
<blockquote>
<p><strong>Last Updated:</strong> November 29, 2025
<strong>Level:</strong> Advanced | <strong>Time:</strong> 6-10 hours | <strong>Data Size:</strong> 200MB</p>
</blockquote>
<h2 id="overview">Overview</h2>
<p>Predicting the bacterial hosts of newly discovered phages is one of the most challenging problems in virome analysis. This tutorial covers multiple complementary approaches for host prediction, from high-confidence CRISPR spacer matching to machine learning-based predictions.</p>
<p><strong>What you'll learn:</strong>
- CRISPR spacer-based host assignment (gold standard)
- Sequence homology approaches (BLAST, protein clustering)
- Machine learning predictions (iPHoP, CHERRY, WIsH)
- Co-occurrence and correlation analysis
- Consensus prediction strategies
- Validation and confidence assessment</p>
<p><strong>Sample dataset:</strong>
Environmental phage contigs (500 contigs) + matching bacterial metagenome (100 MAGs)</p>
<div class="admonition warning">
<p class="admonition-title">Realistic Expectations</p>
<p>Host prediction is difficult! Even with multiple methods, expect:
- <strong>30-60% prediction rate</strong> (many phages will have no confident prediction)
- <strong>30-50% accuracy</strong> at genus level for novel phages
- <strong>High confidence only for ~10-20%</strong> of predictions (CRISPR spacer matches)</p>
</div>
<h2 id="prerequisites">Prerequisites</h2>
<h3 id="required-software">Required Software</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Create environment for host prediction</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>conda<span class="w"> </span>create<span class="w"> </span>-n<span class="w"> </span>host_prediction<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>.9
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>conda<span class="w"> </span>activate<span class="w"> </span>host_prediction
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a><span class="c1"># Install host prediction tools</span>
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>bioconda<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a><span class="w">    </span><span class="nv">blast</span><span class="o">=</span><span class="m">2</span>.14.0<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a><span class="w">    </span><span class="nv">hmmer</span><span class="o">=</span><span class="m">3</span>.3.2<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a><span class="w">    </span><span class="nv">prodigal</span><span class="o">=</span><span class="m">2</span>.6.3<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a><span class="w">    </span><span class="nv">minced</span><span class="o">=</span><span class="m">0</span>.4.2<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a><span class="w">    </span><span class="nv">pilercr</span><span class="o">=</span><span class="m">1</span>.06<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a><span class="w">    </span><span class="nv">bedtools</span><span class="o">=</span><span class="m">2</span>.30.0<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-13"><a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a><span class="w">    </span><span class="nv">seqkit</span><span class="o">=</span><span class="m">2</span>.5.1<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-14"><a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a><span class="w">    </span><span class="nv">mash</span><span class="o">=</span><span class="m">2</span>.3<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-15"><a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a><span class="w">    </span><span class="nv">diamond</span><span class="o">=</span><span class="m">2</span>.1.0<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-16"><a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a><span class="w">    </span><span class="nv">mmseqs2</span><span class="o">=</span><span class="m">14</span>.7e284
</span><span id="__span-0-17"><a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>
</span><span id="__span-0-18"><a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a><span class="c1"># Install iPHoP (comprehensive host prediction)</span>
</span><span id="__span-0-19"><a id="__codelineno-0-19" name="__codelineno-0-19" href="#__codelineno-0-19"></a>conda<span class="w"> </span>install<span class="w"> </span>-c<span class="w"> </span>conda-forge<span class="w"> </span>-c<span class="w"> </span>bioconda<span class="w"> </span><span class="nv">iphop</span><span class="o">=</span><span class="m">1</span>.3.0
</span><span id="__span-0-20"><a id="__codelineno-0-20" name="__codelineno-0-20" href="#__codelineno-0-20"></a>
</span><span id="__span-0-21"><a id="__codelineno-0-21" name="__codelineno-0-21" href="#__codelineno-0-21"></a><span class="c1"># Download iPHoP database (large, ~120GB)</span>
</span><span id="__span-0-22"><a id="__codelineno-0-22" name="__codelineno-0-22" href="#__codelineno-0-22"></a>iphop<span class="w"> </span>download<span class="w"> </span>--db_path<span class="w"> </span>iphop_db/
</span><span id="__span-0-23"><a id="__codelineno-0-23" name="__codelineno-0-23" href="#__codelineno-0-23"></a>
</span><span id="__span-0-24"><a id="__codelineno-0-24" name="__codelineno-0-24" href="#__codelineno-0-24"></a><span class="c1"># Install CHERRY (deep learning)</span>
</span><span id="__span-0-25"><a id="__codelineno-0-25" name="__codelineno-0-25" href="#__codelineno-0-25"></a>pip<span class="w"> </span>install<span class="w"> </span>cherry-phage
</span><span id="__span-0-26"><a id="__codelineno-0-26" name="__codelineno-0-26" href="#__codelineno-0-26"></a>
</span><span id="__span-0-27"><a id="__codelineno-0-27" name="__codelineno-0-27" href="#__codelineno-0-27"></a><span class="c1"># Install WIsH (homology-based)</span>
</span><span id="__span-0-28"><a id="__codelineno-0-28" name="__codelineno-0-28" href="#__codelineno-0-28"></a>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/soedinglab/WIsH.git<span class="w"> </span>~/tools/WIsH
</span><span id="__span-0-29"><a id="__codelineno-0-29" name="__codelineno-0-29" href="#__codelineno-0-29"></a><span class="nb">cd</span><span class="w"> </span>~/tools/WIsH
</span><span id="__span-0-30"><a id="__codelineno-0-30" name="__codelineno-0-30" href="#__codelineno-0-30"></a>cmake<span class="w"> </span>.
</span><span id="__span-0-31"><a id="__codelineno-0-31" name="__codelineno-0-31" href="#__codelineno-0-31"></a>make
</span><span id="__span-0-32"><a id="__codelineno-0-32" name="__codelineno-0-32" href="#__codelineno-0-32"></a><span class="nb">export</span><span class="w"> </span><span class="nv">PATH</span><span class="o">=</span><span class="nv">$PATH</span>:<span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span>/bin
</span><span id="__span-0-33"><a id="__codelineno-0-33" name="__codelineno-0-33" href="#__codelineno-0-33"></a>
</span><span id="__span-0-34"><a id="__codelineno-0-34" name="__codelineno-0-34" href="#__codelineno-0-34"></a><span class="nb">cd</span><span class="w"> </span>~/host_prediction
</span></code></pre></div>
<h3 id="system-requirements">System Requirements</h3>
<ul>
<li><strong>RAM:</strong> 64GB minimum (iPHoP database is large)</li>
<li><strong>Disk Space:</strong> 150GB (databases)</li>
<li><strong>CPU:</strong> 16+ cores recommended</li>
<li><strong>OS:</strong> Linux (some tools are Linux-only)</li>
</ul>
<h3 id="background-knowledge">Background Knowledge</h3>
<p>Complete <a href="../basic-metagenome-virome/">Tutorial 1</a> and review:
- CRISPR-Cas systems and spacer acquisition
- Phage-host interactions
- Metagenome-assembled genomes (MAGs)</p>
<h2 id="step-1-prepare-data">Step 1: Prepare Data</h2>
<h3 id="download-dataset">Download Dataset</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Create project directory</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>~/host_prediction
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a><span class="nb">cd</span><span class="w"> </span>~/host_prediction
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a><span class="c1"># Download phage contigs (from virome analysis)</span>
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>00_data
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a><span class="c1"># Simulated dataset (replace with actual Zenodo)</span>
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a><span class="c1"># wget https://zenodo.org/record/EXAMPLE/files/environmental_phages.fasta</span>
</span><span id="__span-1-10"><a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a><span class="c1"># wget https://zenodo.org/record/EXAMPLE/files/bacterial_MAGs.tar.gz</span>
</span><span id="__span-1-11"><a id="__codelineno-1-11" name="__codelineno-1-11" href="#__codelineno-1-11"></a>
</span><span id="__span-1-12"><a id="__codelineno-1-12" name="__codelineno-1-12" href="#__codelineno-1-12"></a><span class="c1"># For tutorial: 500 phage contigs, 100 bacterial MAGs</span>
</span><span id="__span-1-13"><a id="__codelineno-1-13" name="__codelineno-1-13" href="#__codelineno-1-13"></a><span class="c1"># tar -xzf bacterial_MAGs.tar.gz -C 00_data/MAGs/</span>
</span></code></pre></div>
<h3 id="inspect-data">Inspect Data</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1"># Check phage contigs</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>seqkit<span class="w"> </span>stats<span class="w"> </span>00_data/environmental_phages.fasta
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="c1"># Check bacterial MAGs</span>
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="k">for</span><span class="w"> </span>mag<span class="w"> </span><span class="k">in</span><span class="w"> </span>00_data/MAGs/*.fa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a><span class="w">    </span>seqkit<span class="w"> </span>stats<span class="w"> </span><span class="nv">$mag</span>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="k">done</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>head
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="c1"># Expected:</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a><span class="c1"># 500 phage contigs, 5-150 kb each</span>
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a><span class="c1"># 100 bacterial MAGs, 1-6 Mb each, 50-100% complete</span>
</span></code></pre></div>
<h2 id="step-2-crispr-spacer-based-host-prediction">Step 2: CRISPR Spacer-Based Host Prediction</h2>
<p>CRISPR spacer matches are the "gold standard" for host prediction (highest confidence).</p>
<h3 id="21-extract-crispr-spacers-from-bacterial-mags">2.1 Extract CRISPR Spacers from Bacterial MAGs</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Create CRISPR directory</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>01_crispr
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="c1"># Extract CRISPR arrays using PILER-CR</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>01_crispr/pilercr
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="k">for</span><span class="w"> </span>mag<span class="w"> </span><span class="k">in</span><span class="w"> </span>00_data/MAGs/*.fa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a><span class="w">    </span><span class="nv">basename</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$mag</span><span class="w"> </span>.fa<span class="k">)</span>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Extracting CRISPRs from </span><span class="nv">$basename</span><span class="s2">...&quot;</span>
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a>
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a><span class="w">    </span>pilercr<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a><span class="w">        </span>-in<span class="w"> </span><span class="nv">$mag</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-13"><a id="__codelineno-3-13" name="__codelineno-3-13" href="#__codelineno-3-13"></a><span class="w">        </span>-out<span class="w"> </span>01_crispr/pilercr/<span class="si">${</span><span class="nv">basename</span><span class="si">}</span>_pilercr.txt<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-14"><a id="__codelineno-3-14" name="__codelineno-3-14" href="#__codelineno-3-14"></a><span class="w">        </span>-noinfo
</span><span id="__span-3-15"><a id="__codelineno-3-15" name="__codelineno-3-15" href="#__codelineno-3-15"></a>
</span><span id="__span-3-16"><a id="__codelineno-3-16" name="__codelineno-3-16" href="#__codelineno-3-16"></a><span class="k">done</span>
</span><span id="__span-3-17"><a id="__codelineno-3-17" name="__codelineno-3-17" href="#__codelineno-3-17"></a>
</span><span id="__span-3-18"><a id="__codelineno-3-18" name="__codelineno-3-18" href="#__codelineno-3-18"></a><span class="c1"># Also use minced (alternative CRISPR finder)</span>
</span><span id="__span-3-19"><a id="__codelineno-3-19" name="__codelineno-3-19" href="#__codelineno-3-19"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>01_crispr/minced
</span><span id="__span-3-20"><a id="__codelineno-3-20" name="__codelineno-3-20" href="#__codelineno-3-20"></a>
</span><span id="__span-3-21"><a id="__codelineno-3-21" name="__codelineno-3-21" href="#__codelineno-3-21"></a><span class="k">for</span><span class="w"> </span>mag<span class="w"> </span><span class="k">in</span><span class="w"> </span>00_data/MAGs/*.fa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
</span><span id="__span-3-22"><a id="__codelineno-3-22" name="__codelineno-3-22" href="#__codelineno-3-22"></a><span class="w">    </span><span class="nv">basename</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$mag</span><span class="w"> </span>.fa<span class="k">)</span>
</span><span id="__span-3-23"><a id="__codelineno-3-23" name="__codelineno-3-23" href="#__codelineno-3-23"></a>
</span><span id="__span-3-24"><a id="__codelineno-3-24" name="__codelineno-3-24" href="#__codelineno-3-24"></a><span class="w">    </span>minced<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-25"><a id="__codelineno-3-25" name="__codelineno-3-25" href="#__codelineno-3-25"></a><span class="w">        </span>-gff<span class="w"> </span>01_crispr/minced/<span class="si">${</span><span class="nv">basename</span><span class="si">}</span>_minced.gff<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-26"><a id="__codelineno-3-26" name="__codelineno-3-26" href="#__codelineno-3-26"></a><span class="w">        </span>-spacers<span class="w"> </span>01_crispr/minced/<span class="si">${</span><span class="nv">basename</span><span class="si">}</span>_spacers.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-27"><a id="__codelineno-3-27" name="__codelineno-3-27" href="#__codelineno-3-27"></a><span class="w">        </span><span class="nv">$mag</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-28"><a id="__codelineno-3-28" name="__codelineno-3-28" href="#__codelineno-3-28"></a><span class="w">        </span>01_crispr/minced/<span class="si">${</span><span class="nv">basename</span><span class="si">}</span>_minced.txt
</span><span id="__span-3-29"><a id="__codelineno-3-29" name="__codelineno-3-29" href="#__codelineno-3-29"></a>
</span><span id="__span-3-30"><a id="__codelineno-3-30" name="__codelineno-3-30" href="#__codelineno-3-30"></a><span class="k">done</span>
</span></code></pre></div>
<h3 id="22-combine-all-crispr-spacers">2.2 Combine All CRISPR Spacers</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Combine all spacers into single file</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>cat<span class="w"> </span>01_crispr/minced/*_spacers.fa<span class="w"> </span>&gt;<span class="w"> </span>01_crispr/all_crispr_spacers.fasta
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Add MAG name to spacer headers</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a><span class="s">from Bio import SeqIO</span>
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a>
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a><span class="s">spacers_with_source = []</span>
</span><span id="__span-4-9"><a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>
</span><span id="__span-4-10"><a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a><span class="s">for record in SeqIO.parse(&#39;01_crispr/all_crispr_spacers.fasta&#39;, &#39;fasta&#39;):</span>
</span><span id="__span-4-11"><a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a><span class="s">    # Extract MAG name from spacer ID</span>
</span><span id="__span-4-12"><a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a><span class="s">    # Assuming format: MAG_001 CRISPR1 Spacer1</span>
</span><span id="__span-4-13"><a id="__codelineno-4-13" name="__codelineno-4-13" href="#__codelineno-4-13"></a><span class="s">    parts = record.description.split()</span>
</span><span id="__span-4-14"><a id="__codelineno-4-14" name="__codelineno-4-14" href="#__codelineno-4-14"></a><span class="s">    mag_id = parts[0]</span>
</span><span id="__span-4-15"><a id="__codelineno-4-15" name="__codelineno-4-15" href="#__codelineno-4-15"></a>
</span><span id="__span-4-16"><a id="__codelineno-4-16" name="__codelineno-4-16" href="#__codelineno-4-16"></a><span class="s">    # Modify header to include MAG source</span>
</span><span id="__span-4-17"><a id="__codelineno-4-17" name="__codelineno-4-17" href="#__codelineno-4-17"></a><span class="s">    record.id = f&quot;{mag_id}_{record.id}&quot;</span>
</span><span id="__span-4-18"><a id="__codelineno-4-18" name="__codelineno-4-18" href="#__codelineno-4-18"></a><span class="s">    record.description = f&quot;MAG={mag_id} {record.description}&quot;</span>
</span><span id="__span-4-19"><a id="__codelineno-4-19" name="__codelineno-4-19" href="#__codelineno-4-19"></a>
</span><span id="__span-4-20"><a id="__codelineno-4-20" name="__codelineno-4-20" href="#__codelineno-4-20"></a><span class="s">    spacers_with_source.append(record)</span>
</span><span id="__span-4-21"><a id="__codelineno-4-21" name="__codelineno-4-21" href="#__codelineno-4-21"></a>
</span><span id="__span-4-22"><a id="__codelineno-4-22" name="__codelineno-4-22" href="#__codelineno-4-22"></a><span class="s">SeqIO.write(spacers_with_source, &#39;01_crispr/spacers_with_source.fasta&#39;, &#39;fasta&#39;)</span>
</span><span id="__span-4-23"><a id="__codelineno-4-23" name="__codelineno-4-23" href="#__codelineno-4-23"></a><span class="s">print(f&quot;Total CRISPR spacers: {len(spacers_with_source)}&quot;)</span>
</span><span id="__span-4-24"><a id="__codelineno-4-24" name="__codelineno-4-24" href="#__codelineno-4-24"></a><span class="s">EOF</span>
</span><span id="__span-4-25"><a id="__codelineno-4-25" name="__codelineno-4-25" href="#__codelineno-4-25"></a>
</span><span id="__span-4-26"><a id="__codelineno-4-26" name="__codelineno-4-26" href="#__codelineno-4-26"></a><span class="c1"># Count spacers per MAG</span>
</span><span id="__span-4-27"><a id="__codelineno-4-27" name="__codelineno-4-27" href="#__codelineno-4-27"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Spacers per MAG:&quot;</span>
</span><span id="__span-4-28"><a id="__codelineno-4-28" name="__codelineno-4-28" href="#__codelineno-4-28"></a>grep<span class="w"> </span><span class="s2">&quot;&gt;&quot;</span><span class="w"> </span>01_crispr/spacers_with_source.fasta<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-d<span class="s1">&#39;_&#39;</span><span class="w"> </span>-f1-2<span class="w"> </span><span class="p">|</span><span class="w"> </span>sort<span class="w"> </span><span class="p">|</span><span class="w"> </span>uniq<span class="w"> </span>-c<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-20
</span></code></pre></div>
<p><strong>Expected CRISPR results:</strong>
- <strong>MAGs with CRISPRs:</strong> 30-60% (not all bacteria have CRISPR systems)
- <strong>Spacers per MAG:</strong> 0-200 (highly variable)
- <strong>Total spacers:</strong> 1,000-10,000</p>
<h3 id="23-blast-spacers-against-phage-contigs">2.3 BLAST Spacers Against Phage Contigs</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="c1"># Create BLAST database from phage contigs</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>makeblastdb<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="w">    </span>-in<span class="w"> </span>00_data/environmental_phages.fasta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="w">    </span>-dbtype<span class="w"> </span>nucl<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a><span class="w">    </span>-out<span class="w"> </span>01_crispr/phage_db
</span><span id="__span-5-6"><a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>
</span><span id="__span-5-7"><a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a><span class="c1"># BLAST spacers against phages (very stringent parameters)</span>
</span><span id="__span-5-8"><a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>blastn<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-9"><a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a><span class="w">    </span>-query<span class="w"> </span>01_crispr/spacers_with_source.fasta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-10"><a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a><span class="w">    </span>-db<span class="w"> </span>01_crispr/phage_db<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-11"><a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a><span class="w">    </span>-out<span class="w"> </span>01_crispr/spacer_phage_matches.txt<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-12"><a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a><span class="w">    </span>-outfmt<span class="w"> </span><span class="s1">&#39;6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen&#39;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-13"><a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a><span class="w">    </span>-evalue<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-14"><a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a><span class="w">    </span>-word_size<span class="w"> </span><span class="m">7</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-15"><a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a><span class="w">    </span>-reward<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-16"><a id="__codelineno-5-16" name="__codelineno-5-16" href="#__codelineno-5-16"></a><span class="w">    </span>-penalty<span class="w"> </span>-1<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-17"><a id="__codelineno-5-17" name="__codelineno-5-17" href="#__codelineno-5-17"></a><span class="w">    </span>-gapopen<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-18"><a id="__codelineno-5-18" name="__codelineno-5-18" href="#__codelineno-5-18"></a><span class="w">    </span>-gapextend<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-19"><a id="__codelineno-5-19" name="__codelineno-5-19" href="#__codelineno-5-19"></a><span class="w">    </span>-num_threads<span class="w"> </span><span class="m">16</span>
</span><span id="__span-5-20"><a id="__codelineno-5-20" name="__codelineno-5-20" href="#__codelineno-5-20"></a>
</span><span id="__span-5-21"><a id="__codelineno-5-21" name="__codelineno-5-21" href="#__codelineno-5-21"></a><span class="c1"># Filter for high-confidence matches</span>
</span><span id="__span-5-22"><a id="__codelineno-5-22" name="__codelineno-5-22" href="#__codelineno-5-22"></a><span class="c1"># Criteria: 95% identity, 90% coverage, 1 mismatch</span>
</span><span id="__span-5-23"><a id="__codelineno-5-23" name="__codelineno-5-23" href="#__codelineno-5-23"></a>awk<span class="w"> </span><span class="s1">&#39;$3 &gt;= 95 &amp;&amp; $4 &gt;= 0.9*$13 &amp;&amp; $5 &lt;= 1 {print $0}&#39;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-24"><a id="__codelineno-5-24" name="__codelineno-5-24" href="#__codelineno-5-24"></a><span class="w">    </span>01_crispr/spacer_phage_matches.txt<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-25"><a id="__codelineno-5-25" name="__codelineno-5-25" href="#__codelineno-5-25"></a><span class="w">    </span>&gt;<span class="w"> </span>01_crispr/high_confidence_spacer_matches.txt
</span><span id="__span-5-26"><a id="__codelineno-5-26" name="__codelineno-5-26" href="#__codelineno-5-26"></a>
</span><span id="__span-5-27"><a id="__codelineno-5-27" name="__codelineno-5-27" href="#__codelineno-5-27"></a><span class="c1"># Extract host predictions from CRISPR matches</span>
</span><span id="__span-5-28"><a id="__codelineno-5-28" name="__codelineno-5-28" href="#__codelineno-5-28"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-5-29"><a id="__codelineno-5-29" name="__codelineno-5-29" href="#__codelineno-5-29"></a><span class="s">import pandas as pd</span>
</span><span id="__span-5-30"><a id="__codelineno-5-30" name="__codelineno-5-30" href="#__codelineno-5-30"></a>
</span><span id="__span-5-31"><a id="__codelineno-5-31" name="__codelineno-5-31" href="#__codelineno-5-31"></a><span class="s"># Load CRISPR matches</span>
</span><span id="__span-5-32"><a id="__codelineno-5-32" name="__codelineno-5-32" href="#__codelineno-5-32"></a><span class="s">matches = pd.read_csv(&#39;01_crispr/high_confidence_spacer_matches.txt&#39;, sep=&#39;\t&#39;,</span>
</span><span id="__span-5-33"><a id="__codelineno-5-33" name="__codelineno-5-33" href="#__codelineno-5-33"></a><span class="s">                      header=None, names=[&#39;spacer&#39;, &#39;phage&#39;, &#39;pident&#39;, &#39;length&#39;,</span>
</span><span id="__span-5-34"><a id="__codelineno-5-34" name="__codelineno-5-34" href="#__codelineno-5-34"></a><span class="s">                                          &#39;mismatch&#39;, &#39;gapopen&#39;, &#39;qstart&#39;, &#39;qend&#39;,</span>
</span><span id="__span-5-35"><a id="__codelineno-5-35" name="__codelineno-5-35" href="#__codelineno-5-35"></a><span class="s">                                          &#39;sstart&#39;, &#39;send&#39;, &#39;evalue&#39;, &#39;bitscore&#39;,</span>
</span><span id="__span-5-36"><a id="__codelineno-5-36" name="__codelineno-5-36" href="#__codelineno-5-36"></a><span class="s">                                          &#39;qlen&#39;, &#39;slen&#39;])</span>
</span><span id="__span-5-37"><a id="__codelineno-5-37" name="__codelineno-5-37" href="#__codelineno-5-37"></a>
</span><span id="__span-5-38"><a id="__codelineno-5-38" name="__codelineno-5-38" href="#__codelineno-5-38"></a><span class="s"># Extract MAG ID from spacer name</span>
</span><span id="__span-5-39"><a id="__codelineno-5-39" name="__codelineno-5-39" href="#__codelineno-5-39"></a><span class="s">matches[&#39;MAG_ID&#39;] = matches[&#39;spacer&#39;].str.split(&#39;_&#39;).str[0:2].str.join(&#39;_&#39;)</span>
</span><span id="__span-5-40"><a id="__codelineno-5-40" name="__codelineno-5-40" href="#__codelineno-5-40"></a>
</span><span id="__span-5-41"><a id="__codelineno-5-41" name="__codelineno-5-41" href="#__codelineno-5-41"></a><span class="s"># Extract phage ID (contig name)</span>
</span><span id="__span-5-42"><a id="__codelineno-5-42" name="__codelineno-5-42" href="#__codelineno-5-42"></a><span class="s">matches[&#39;Phage_ID&#39;] = matches[&#39;phage&#39;]</span>
</span><span id="__span-5-43"><a id="__codelineno-5-43" name="__codelineno-5-43" href="#__codelineno-5-43"></a>
</span><span id="__span-5-44"><a id="__codelineno-5-44" name="__codelineno-5-44" href="#__codelineno-5-44"></a><span class="s"># Create host predictions</span>
</span><span id="__span-5-45"><a id="__codelineno-5-45" name="__codelineno-5-45" href="#__codelineno-5-45"></a><span class="s">host_predictions_crispr = matches[[&#39;Phage_ID&#39;, &#39;MAG_ID&#39;, &#39;pident&#39;, &#39;mismatch&#39;]].copy()</span>
</span><span id="__span-5-46"><a id="__codelineno-5-46" name="__codelineno-5-46" href="#__codelineno-5-46"></a><span class="s">host_predictions_crispr[&#39;Method&#39;] = &#39;CRISPR&#39;</span>
</span><span id="__span-5-47"><a id="__codelineno-5-47" name="__codelineno-5-47" href="#__codelineno-5-47"></a><span class="s">host_predictions_crispr[&#39;Confidence&#39;] = &#39;High&#39;</span>
</span><span id="__span-5-48"><a id="__codelineno-5-48" name="__codelineno-5-48" href="#__codelineno-5-48"></a>
</span><span id="__span-5-49"><a id="__codelineno-5-49" name="__codelineno-5-49" href="#__codelineno-5-49"></a><span class="s"># Save</span>
</span><span id="__span-5-50"><a id="__codelineno-5-50" name="__codelineno-5-50" href="#__codelineno-5-50"></a><span class="s">host_predictions_crispr.to_csv(&#39;01_crispr/crispr_host_predictions.tsv&#39;,</span>
</span><span id="__span-5-51"><a id="__codelineno-5-51" name="__codelineno-5-51" href="#__codelineno-5-51"></a><span class="s">                               sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-5-52"><a id="__codelineno-5-52" name="__codelineno-5-52" href="#__codelineno-5-52"></a>
</span><span id="__span-5-53"><a id="__codelineno-5-53" name="__codelineno-5-53" href="#__codelineno-5-53"></a><span class="s">print(f&quot;CRISPR-based host predictions: {len(host_predictions_crispr)}&quot;)</span>
</span><span id="__span-5-54"><a id="__codelineno-5-54" name="__codelineno-5-54" href="#__codelineno-5-54"></a><span class="s">print(f&quot;Phages with CRISPR match: {host_predictions_crispr[&#39;Phage_ID&#39;].nunique()}&quot;)</span>
</span><span id="__span-5-55"><a id="__codelineno-5-55" name="__codelineno-5-55" href="#__codelineno-5-55"></a><span class="s">print(f&quot;Prediction rate: {host_predictions_crispr[&#39;Phage_ID&#39;].nunique() / 500 * 100:.1f}%&quot;)</span>
</span><span id="__span-5-56"><a id="__codelineno-5-56" name="__codelineno-5-56" href="#__codelineno-5-56"></a><span class="s">EOF</span>
</span></code></pre></div>
<p><strong>Expected CRISPR predictions:</strong>
- <strong>Phages with CRISPR match:</strong> 10-50 (2-10% of phages)
- <strong>Confidence:</strong> Very high (these are the best predictions)</p>
<div class="admonition success">
<p class="admonition-title">CRISPR Spacer Matches</p>
<p>CRISPR spacer matches are the highest-confidence host predictions (~90-95% accuracy). However, they're only available for a small fraction of phages.</p>
</div>
<h2 id="step-3-sequence-homology-based-prediction">Step 3: Sequence Homology-Based Prediction</h2>
<h3 id="31-protein-blast-against-bacterial-proteins">3.1 Protein BLAST Against Bacterial Proteins</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="c1"># Create homology directory</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>02_homology
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a><span class="c1"># Predict phage proteins</span>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>prodigal<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="w">    </span>-i<span class="w"> </span>00_data/environmental_phages.fasta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a><span class="w">    </span>-a<span class="w"> </span>02_homology/phage_proteins.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a><span class="w">    </span>-p<span class="w"> </span>meta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a><span class="w">    </span>-q
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a>
</span><span id="__span-6-11"><a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a><span class="c1"># Predict MAG proteins</span>
</span><span id="__span-6-12"><a id="__codelineno-6-12" name="__codelineno-6-12" href="#__codelineno-6-12"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>02_homology/MAG_proteins
</span><span id="__span-6-13"><a id="__codelineno-6-13" name="__codelineno-6-13" href="#__codelineno-6-13"></a>
</span><span id="__span-6-14"><a id="__codelineno-6-14" name="__codelineno-6-14" href="#__codelineno-6-14"></a><span class="k">for</span><span class="w"> </span>mag<span class="w"> </span><span class="k">in</span><span class="w"> </span>00_data/MAGs/*.fa<span class="p">;</span><span class="w"> </span><span class="k">do</span>
</span><span id="__span-6-15"><a id="__codelineno-6-15" name="__codelineno-6-15" href="#__codelineno-6-15"></a><span class="w">    </span><span class="nv">basename</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$mag</span><span class="w"> </span>.fa<span class="k">)</span>
</span><span id="__span-6-16"><a id="__codelineno-6-16" name="__codelineno-6-16" href="#__codelineno-6-16"></a>
</span><span id="__span-6-17"><a id="__codelineno-6-17" name="__codelineno-6-17" href="#__codelineno-6-17"></a><span class="w">    </span>prodigal<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-18"><a id="__codelineno-6-18" name="__codelineno-6-18" href="#__codelineno-6-18"></a><span class="w">        </span>-i<span class="w"> </span><span class="nv">$mag</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-19"><a id="__codelineno-6-19" name="__codelineno-6-19" href="#__codelineno-6-19"></a><span class="w">        </span>-a<span class="w"> </span>02_homology/MAG_proteins/<span class="si">${</span><span class="nv">basename</span><span class="si">}</span>.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-20"><a id="__codelineno-6-20" name="__codelineno-6-20" href="#__codelineno-6-20"></a><span class="w">        </span>-p<span class="w"> </span>single<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-21"><a id="__codelineno-6-21" name="__codelineno-6-21" href="#__codelineno-6-21"></a><span class="w">        </span>-q
</span><span id="__span-6-22"><a id="__codelineno-6-22" name="__codelineno-6-22" href="#__codelineno-6-22"></a><span class="k">done</span>
</span><span id="__span-6-23"><a id="__codelineno-6-23" name="__codelineno-6-23" href="#__codelineno-6-23"></a>
</span><span id="__span-6-24"><a id="__codelineno-6-24" name="__codelineno-6-24" href="#__codelineno-6-24"></a><span class="c1"># Combine all MAG proteins</span>
</span><span id="__span-6-25"><a id="__codelineno-6-25" name="__codelineno-6-25" href="#__codelineno-6-25"></a>cat<span class="w"> </span>02_homology/MAG_proteins/*.faa<span class="w"> </span>&gt;<span class="w"> </span>02_homology/all_MAG_proteins.faa
</span><span id="__span-6-26"><a id="__codelineno-6-26" name="__codelineno-6-26" href="#__codelineno-6-26"></a>
</span><span id="__span-6-27"><a id="__codelineno-6-27" name="__codelineno-6-27" href="#__codelineno-6-27"></a><span class="c1"># Create DIAMOND database (faster than BLAST)</span>
</span><span id="__span-6-28"><a id="__codelineno-6-28" name="__codelineno-6-28" href="#__codelineno-6-28"></a>diamond<span class="w"> </span>makedb<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-29"><a id="__codelineno-6-29" name="__codelineno-6-29" href="#__codelineno-6-29"></a><span class="w">    </span>--in<span class="w"> </span>02_homology/all_MAG_proteins.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-30"><a id="__codelineno-6-30" name="__codelineno-6-30" href="#__codelineno-6-30"></a><span class="w">    </span>--db<span class="w"> </span>02_homology/MAG_proteins_db
</span><span id="__span-6-31"><a id="__codelineno-6-31" name="__codelineno-6-31" href="#__codelineno-6-31"></a>
</span><span id="__span-6-32"><a id="__codelineno-6-32" name="__codelineno-6-32" href="#__codelineno-6-32"></a><span class="c1"># BLAST phage proteins against MAG proteins</span>
</span><span id="__span-6-33"><a id="__codelineno-6-33" name="__codelineno-6-33" href="#__codelineno-6-33"></a>diamond<span class="w"> </span>blastp<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-34"><a id="__codelineno-6-34" name="__codelineno-6-34" href="#__codelineno-6-34"></a><span class="w">    </span>--query<span class="w"> </span>02_homology/phage_proteins.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-35"><a id="__codelineno-6-35" name="__codelineno-6-35" href="#__codelineno-6-35"></a><span class="w">    </span>--db<span class="w"> </span>02_homology/MAG_proteins_db<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-36"><a id="__codelineno-6-36" name="__codelineno-6-36" href="#__codelineno-6-36"></a><span class="w">    </span>--out<span class="w"> </span>02_homology/phage_MAG_blastp.txt<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-37"><a id="__codelineno-6-37" name="__codelineno-6-37" href="#__codelineno-6-37"></a><span class="w">    </span>--outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>qseqid<span class="w"> </span>sseqid<span class="w"> </span>pident<span class="w"> </span>length<span class="w"> </span>evalue<span class="w"> </span>bitscore<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-38"><a id="__codelineno-6-38" name="__codelineno-6-38" href="#__codelineno-6-38"></a><span class="w">    </span>--evalue<span class="w"> </span>1e-5<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-39"><a id="__codelineno-6-39" name="__codelineno-6-39" href="#__codelineno-6-39"></a><span class="w">    </span>--max-target-seqs<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-40"><a id="__codelineno-6-40" name="__codelineno-6-40" href="#__codelineno-6-40"></a><span class="w">    </span>--threads<span class="w"> </span><span class="m">16</span>
</span><span id="__span-6-41"><a id="__codelineno-6-41" name="__codelineno-6-41" href="#__codelineno-6-41"></a>
</span><span id="__span-6-42"><a id="__codelineno-6-42" name="__codelineno-6-42" href="#__codelineno-6-42"></a><span class="c1"># Count hits per phage contig</span>
</span><span id="__span-6-43"><a id="__codelineno-6-43" name="__codelineno-6-43" href="#__codelineno-6-43"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-6-44"><a id="__codelineno-6-44" name="__codelineno-6-44" href="#__codelineno-6-44"></a><span class="s">import pandas as pd</span>
</span><span id="__span-6-45"><a id="__codelineno-6-45" name="__codelineno-6-45" href="#__codelineno-6-45"></a>
</span><span id="__span-6-46"><a id="__codelineno-6-46" name="__codelineno-6-46" href="#__codelineno-6-46"></a><span class="s"># Load BLAST results</span>
</span><span id="__span-6-47"><a id="__codelineno-6-47" name="__codelineno-6-47" href="#__codelineno-6-47"></a><span class="s">blast = pd.read_csv(&#39;02_homology/phage_MAG_blastp.txt&#39;, sep=&#39;\t&#39;,</span>
</span><span id="__span-6-48"><a id="__codelineno-6-48" name="__codelineno-6-48" href="#__codelineno-6-48"></a><span class="s">                    header=None, names=[&#39;phage_protein&#39;, &#39;mag_protein&#39;,</span>
</span><span id="__span-6-49"><a id="__codelineno-6-49" name="__codelineno-6-49" href="#__codelineno-6-49"></a><span class="s">                                        &#39;pident&#39;, &#39;length&#39;, &#39;evalue&#39;, &#39;bitscore&#39;])</span>
</span><span id="__span-6-50"><a id="__codelineno-6-50" name="__codelineno-6-50" href="#__codelineno-6-50"></a>
</span><span id="__span-6-51"><a id="__codelineno-6-51" name="__codelineno-6-51" href="#__codelineno-6-51"></a><span class="s"># Extract contig and MAG IDs</span>
</span><span id="__span-6-52"><a id="__codelineno-6-52" name="__codelineno-6-52" href="#__codelineno-6-52"></a><span class="s">blast[&#39;Phage_ID&#39;] = blast[&#39;phage_protein&#39;].str.rsplit(&#39;_&#39;, n=1).str[0]</span>
</span><span id="__span-6-53"><a id="__codelineno-6-53" name="__codelineno-6-53" href="#__codelineno-6-53"></a><span class="s">blast[&#39;MAG_ID&#39;] = blast[&#39;mag_protein&#39;].str.rsplit(&#39;_&#39;, n=1).str[0]</span>
</span><span id="__span-6-54"><a id="__codelineno-6-54" name="__codelineno-6-54" href="#__codelineno-6-54"></a>
</span><span id="__span-6-55"><a id="__codelineno-6-55" name="__codelineno-6-55" href="#__codelineno-6-55"></a><span class="s"># Count hits per phage-MAG pair</span>
</span><span id="__span-6-56"><a id="__codelineno-6-56" name="__codelineno-6-56" href="#__codelineno-6-56"></a><span class="s">hit_counts = blast.groupby([&#39;Phage_ID&#39;, &#39;MAG_ID&#39;]).size().reset_index(name=&#39;Hit_Count&#39;)</span>
</span><span id="__span-6-57"><a id="__codelineno-6-57" name="__codelineno-6-57" href="#__codelineno-6-57"></a>
</span><span id="__span-6-58"><a id="__codelineno-6-58" name="__codelineno-6-58" href="#__codelineno-6-58"></a><span class="s"># For each phage, select MAG with most hits</span>
</span><span id="__span-6-59"><a id="__codelineno-6-59" name="__codelineno-6-59" href="#__codelineno-6-59"></a><span class="s">best_matches = hit_counts.loc[hit_counts.groupby(&#39;Phage_ID&#39;)[&#39;Hit_Count&#39;].idxmax()]</span>
</span><span id="__span-6-60"><a id="__codelineno-6-60" name="__codelineno-6-60" href="#__codelineno-6-60"></a>
</span><span id="__span-6-61"><a id="__codelineno-6-61" name="__codelineno-6-61" href="#__codelineno-6-61"></a><span class="s"># Filter: require 3 protein hits</span>
</span><span id="__span-6-62"><a id="__codelineno-6-62" name="__codelineno-6-62" href="#__codelineno-6-62"></a><span class="s">best_matches = best_matches[best_matches[&#39;Hit_Count&#39;] &gt;= 3]</span>
</span><span id="__span-6-63"><a id="__codelineno-6-63" name="__codelineno-6-63" href="#__codelineno-6-63"></a>
</span><span id="__span-6-64"><a id="__codelineno-6-64" name="__codelineno-6-64" href="#__codelineno-6-64"></a><span class="s"># Save</span>
</span><span id="__span-6-65"><a id="__codelineno-6-65" name="__codelineno-6-65" href="#__codelineno-6-65"></a><span class="s">best_matches[&#39;Method&#39;] = &#39;Protein_Homology&#39;</span>
</span><span id="__span-6-66"><a id="__codelineno-6-66" name="__codelineno-6-66" href="#__codelineno-6-66"></a><span class="s">best_matches[&#39;Confidence&#39;] = &#39;Medium&#39;</span>
</span><span id="__span-6-67"><a id="__codelineno-6-67" name="__codelineno-6-67" href="#__codelineno-6-67"></a><span class="s">best_matches.to_csv(&#39;02_homology/homology_host_predictions.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-6-68"><a id="__codelineno-6-68" name="__codelineno-6-68" href="#__codelineno-6-68"></a>
</span><span id="__span-6-69"><a id="__codelineno-6-69" name="__codelineno-6-69" href="#__codelineno-6-69"></a><span class="s">print(f&quot;Homology-based host predictions: {len(best_matches)}&quot;)</span>
</span><span id="__span-6-70"><a id="__codelineno-6-70" name="__codelineno-6-70" href="#__codelineno-6-70"></a><span class="s">print(f&quot;Prediction rate: {len(best_matches) / 500 * 100:.1f}%&quot;)</span>
</span><span id="__span-6-71"><a id="__codelineno-6-71" name="__codelineno-6-71" href="#__codelineno-6-71"></a><span class="s">EOF</span>
</span></code></pre></div>
<p><strong>Expected homology predictions:</strong>
- <strong>Phages with homology match:</strong> 100-250 (20-50%)
- <strong>Confidence:</strong> Medium (some may be horizontal gene transfer, not true host)</p>
<h2 id="step-4-machine-learning-based-prediction">Step 4: Machine Learning-Based Prediction</h2>
<h3 id="41-wish-genome-composition-based">4.1 WIsH (Genome Composition-Based)</h3>
<p>WIsH predicts hosts based on k-mer composition similarity.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Create WIsH directory</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>03_wish
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="c1"># Run WIsH</span>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>WIsH<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="w">    </span>-c<span class="w"> </span>00_data/environmental_phages.fasta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="w">    </span>-b<span class="w"> </span>00_data/MAGs<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a><span class="w">    </span>-o<span class="w"> </span>03_wish/wish_predictions.txt<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a><span class="w">    </span>-n<span class="w"> </span><span class="m">16</span>
</span><span id="__span-7-10"><a id="__codelineno-7-10" name="__codelineno-7-10" href="#__codelineno-7-10"></a>
</span><span id="__span-7-11"><a id="__codelineno-7-11" name="__codelineno-7-11" href="#__codelineno-7-11"></a><span class="c1"># Parse WIsH results</span>
</span><span id="__span-7-12"><a id="__codelineno-7-12" name="__codelineno-7-12" href="#__codelineno-7-12"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-7-13"><a id="__codelineno-7-13" name="__codelineno-7-13" href="#__codelineno-7-13"></a><span class="s">import pandas as pd</span>
</span><span id="__span-7-14"><a id="__codelineno-7-14" name="__codelineno-7-14" href="#__codelineno-7-14"></a>
</span><span id="__span-7-15"><a id="__codelineno-7-15" name="__codelineno-7-15" href="#__codelineno-7-15"></a><span class="s"># Load WIsH predictions</span>
</span><span id="__span-7-16"><a id="__codelineno-7-16" name="__codelineno-7-16" href="#__codelineno-7-16"></a><span class="s"># WIsH output format: phage, host, log-likelihood, p-value</span>
</span><span id="__span-7-17"><a id="__codelineno-7-17" name="__codelineno-7-17" href="#__codelineno-7-17"></a><span class="s">wish = pd.read_csv(&#39;03_wish/wish_predictions.txt&#39;, sep=&#39;\t&#39;,</span>
</span><span id="__span-7-18"><a id="__codelineno-7-18" name="__codelineno-7-18" href="#__codelineno-7-18"></a><span class="s">                   header=None, names=[&#39;Phage_ID&#39;, &#39;MAG_ID&#39;, &#39;LogLikelihood&#39;, &#39;Pvalue&#39;])</span>
</span><span id="__span-7-19"><a id="__codelineno-7-19" name="__codelineno-7-19" href="#__codelineno-7-19"></a>
</span><span id="__span-7-20"><a id="__codelineno-7-20" name="__codelineno-7-20" href="#__codelineno-7-20"></a><span class="s"># Filter by p-value (&lt; 0.05)</span>
</span><span id="__span-7-21"><a id="__codelineno-7-21" name="__codelineno-7-21" href="#__codelineno-7-21"></a><span class="s">wish_filtered = wish[wish[&#39;Pvalue&#39;] &lt; 0.05].copy()</span>
</span><span id="__span-7-22"><a id="__codelineno-7-22" name="__codelineno-7-22" href="#__codelineno-7-22"></a>
</span><span id="__span-7-23"><a id="__codelineno-7-23" name="__codelineno-7-23" href="#__codelineno-7-23"></a><span class="s"># Take best prediction per phage (highest log-likelihood)</span>
</span><span id="__span-7-24"><a id="__codelineno-7-24" name="__codelineno-7-24" href="#__codelineno-7-24"></a><span class="s">best_wish = wish_filtered.loc[wish_filtered.groupby(&#39;Phage_ID&#39;)[&#39;LogLikelihood&#39;].idxmax()]</span>
</span><span id="__span-7-25"><a id="__codelineno-7-25" name="__codelineno-7-25" href="#__codelineno-7-25"></a>
</span><span id="__span-7-26"><a id="__codelineno-7-26" name="__codelineno-7-26" href="#__codelineno-7-26"></a><span class="s">best_wish[&#39;Method&#39;] = &#39;WIsH&#39;</span>
</span><span id="__span-7-27"><a id="__codelineno-7-27" name="__codelineno-7-27" href="#__codelineno-7-27"></a><span class="s">best_wish[&#39;Confidence&#39;] = &#39;Low-Medium&#39;</span>
</span><span id="__span-7-28"><a id="__codelineno-7-28" name="__codelineno-7-28" href="#__codelineno-7-28"></a><span class="s">best_wish.to_csv(&#39;03_wish/wish_host_predictions.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-7-29"><a id="__codelineno-7-29" name="__codelineno-7-29" href="#__codelineno-7-29"></a>
</span><span id="__span-7-30"><a id="__codelineno-7-30" name="__codelineno-7-30" href="#__codelineno-7-30"></a><span class="s">print(f&quot;WIsH-based host predictions: {len(best_wish)}&quot;)</span>
</span><span id="__span-7-31"><a id="__codelineno-7-31" name="__codelineno-7-31" href="#__codelineno-7-31"></a><span class="s">print(f&quot;Prediction rate: {len(best_wish) / 500 * 100:.1f}%&quot;)</span>
</span><span id="__span-7-32"><a id="__codelineno-7-32" name="__codelineno-7-32" href="#__codelineno-7-32"></a><span class="s">EOF</span>
</span></code></pre></div>
<h3 id="42-iphop-integrated-approach">4.2 iPHoP (Integrated Approach)</h3>
<p>iPHoP combines multiple methods including CRISPR, homology, and genomic signatures.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1"># Create iPHoP directory</span>
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>04_iphop
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a><span class="c1"># Run iPHoP (all-in-one tool)</span>
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>iphop<span class="w"> </span>predict<span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="w">    </span>--fa_file<span class="w"> </span>00_data/environmental_phages.fasta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-7"><a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a><span class="w">    </span>--db_dir<span class="w"> </span>~/databases/iphop_db/<span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-8"><a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a><span class="w">    </span>--out_dir<span class="w"> </span>04_iphop<span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-9"><a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a><span class="w">    </span>--num_threads<span class="w"> </span><span class="m">16</span>
</span><span id="__span-8-10"><a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a>
</span><span id="__span-8-11"><a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a><span class="c1"># iPHoP outputs comprehensive results</span>
</span><span id="__span-8-12"><a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a>cat<span class="w"> </span>04_iphop/Host_prediction_to_genus_m90.csv
</span></code></pre></div>
<p><strong>iPHoP output files:</strong>
- <code>Host_prediction_to_genus_m90.csv</code>: Genus-level predictions (90% confidence)
- <code>Host_prediction_to_genome_m90.csv</code>: Genome-level predictions
- <code>Detailed_output_by_tool.csv</code>: Breakdown by method</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1"># Parse iPHoP predictions</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="s">import pandas as pd</span>
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>
</span><span id="__span-9-5"><a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a><span class="s"># Load iPHoP genus-level predictions</span>
</span><span id="__span-9-6"><a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a><span class="s">iphop = pd.read_csv(&#39;04_iphop/Host_prediction_to_genus_m90.csv&#39;)</span>
</span><span id="__span-9-7"><a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>
</span><span id="__span-9-8"><a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a><span class="s"># Filter for confident predictions (confidence score &gt; 90)</span>
</span><span id="__span-9-9"><a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a><span class="s">iphop_confident = iphop[iphop[&#39;Host genus confidence score&#39;] &gt; 90].copy()</span>
</span><span id="__span-9-10"><a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a>
</span><span id="__span-9-11"><a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a><span class="s">iphop_confident[&#39;Method&#39;] = &#39;iPHoP&#39;</span>
</span><span id="__span-9-12"><a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a><span class="s">iphop_confident[&#39;Confidence&#39;] = &#39;Medium-High&#39;</span>
</span><span id="__span-9-13"><a id="__codelineno-9-13" name="__codelineno-9-13" href="#__codelineno-9-13"></a>
</span><span id="__span-9-14"><a id="__codelineno-9-14" name="__codelineno-9-14" href="#__codelineno-9-14"></a><span class="s"># Save</span>
</span><span id="__span-9-15"><a id="__codelineno-9-15" name="__codelineno-9-15" href="#__codelineno-9-15"></a><span class="s">iphop_confident.to_csv(&#39;04_iphop/iphop_host_predictions.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-9-16"><a id="__codelineno-9-16" name="__codelineno-9-16" href="#__codelineno-9-16"></a>
</span><span id="__span-9-17"><a id="__codelineno-9-17" name="__codelineno-9-17" href="#__codelineno-9-17"></a><span class="s">print(f&quot;iPHoP host predictions: {len(iphop_confident)}&quot;)</span>
</span><span id="__span-9-18"><a id="__codelineno-9-18" name="__codelineno-9-18" href="#__codelineno-9-18"></a><span class="s">print(f&quot;Prediction rate: {len(iphop_confident) / 500 * 100:.1f}%&quot;)</span>
</span><span id="__span-9-19"><a id="__codelineno-9-19" name="__codelineno-9-19" href="#__codelineno-9-19"></a><span class="s">EOF</span>
</span></code></pre></div>
<p><strong>Expected iPHoP predictions:</strong>
- <strong>Phages with prediction:</strong> 150-350 (30-70%)
- <strong>Confidence:</strong> Variable (check confidence scores)</p>
<h2 id="step-5-co-occurrence-analysis">Step 5: Co-occurrence Analysis</h2>
<p>Phages and their hosts may co-occur in metagenomic samples.</p>
<h3 id="51-calculate-abundance-correlations">5.1 Calculate Abundance Correlations</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># This requires abundance data from multiple samples</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="c1"># Assuming you have abundance tables for phages and MAGs across samples</span>
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>05_cooccurrence
</span><span id="__span-10-5"><a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a>
</span><span id="__span-10-6"><a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a><span class="c1"># Create mock abundance data for tutorial</span>
</span><span id="__span-10-7"><a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a><span class="c1"># In practice, you&#39;d use CoverM or similar to get real abundances</span>
</span><span id="__span-10-8"><a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a>
</span><span id="__span-10-9"><a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a><span class="c1"># Calculate Spearman correlations</span>
</span><span id="__span-10-10"><a id="__codelineno-10-10" name="__codelineno-10-10" href="#__codelineno-10-10"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-10-11"><a id="__codelineno-10-11" name="__codelineno-10-11" href="#__codelineno-10-11"></a><span class="s">import pandas as pd</span>
</span><span id="__span-10-12"><a id="__codelineno-10-12" name="__codelineno-10-12" href="#__codelineno-10-12"></a><span class="s">import numpy as np</span>
</span><span id="__span-10-13"><a id="__codelineno-10-13" name="__codelineno-10-13" href="#__codelineno-10-13"></a><span class="s">from scipy.stats import spearmanr</span>
</span><span id="__span-10-14"><a id="__codelineno-10-14" name="__codelineno-10-14" href="#__codelineno-10-14"></a>
</span><span id="__span-10-15"><a id="__codelineno-10-15" name="__codelineno-10-15" href="#__codelineno-10-15"></a><span class="s"># Load abundance data (simulated for tutorial)</span>
</span><span id="__span-10-16"><a id="__codelineno-10-16" name="__codelineno-10-16" href="#__codelineno-10-16"></a><span class="s"># phage_abundance = pd.read_csv(&#39;phage_abundance_matrix.tsv&#39;, sep=&#39;\t&#39;, index_col=0)</span>
</span><span id="__span-10-17"><a id="__codelineno-10-17" name="__codelineno-10-17" href="#__codelineno-10-17"></a><span class="s"># mag_abundance = pd.read_csv(&#39;mag_abundance_matrix.tsv&#39;, sep=&#39;\t&#39;, index_col=0)</span>
</span><span id="__span-10-18"><a id="__codelineno-10-18" name="__codelineno-10-18" href="#__codelineno-10-18"></a>
</span><span id="__span-10-19"><a id="__codelineno-10-19" name="__codelineno-10-19" href="#__codelineno-10-19"></a><span class="s"># For tutorial, create mock data</span>
</span><span id="__span-10-20"><a id="__codelineno-10-20" name="__codelineno-10-20" href="#__codelineno-10-20"></a><span class="s">np.random.seed(42)</span>
</span><span id="__span-10-21"><a id="__codelineno-10-21" name="__codelineno-10-21" href="#__codelineno-10-21"></a><span class="s">n_samples = 20</span>
</span><span id="__span-10-22"><a id="__codelineno-10-22" name="__codelineno-10-22" href="#__codelineno-10-22"></a><span class="s">n_phages = 500</span>
</span><span id="__span-10-23"><a id="__codelineno-10-23" name="__codelineno-10-23" href="#__codelineno-10-23"></a><span class="s">n_mags = 100</span>
</span><span id="__span-10-24"><a id="__codelineno-10-24" name="__codelineno-10-24" href="#__codelineno-10-24"></a>
</span><span id="__span-10-25"><a id="__codelineno-10-25" name="__codelineno-10-25" href="#__codelineno-10-25"></a><span class="s">phage_abundance = pd.DataFrame(</span>
</span><span id="__span-10-26"><a id="__codelineno-10-26" name="__codelineno-10-26" href="#__codelineno-10-26"></a><span class="s">    np.random.lognormal(0, 2, (n_phages, n_samples)),</span>
</span><span id="__span-10-27"><a id="__codelineno-10-27" name="__codelineno-10-27" href="#__codelineno-10-27"></a><span class="s">    index=[f&quot;phage_{i}&quot; for i in range(n_phages)],</span>
</span><span id="__span-10-28"><a id="__codelineno-10-28" name="__codelineno-10-28" href="#__codelineno-10-28"></a><span class="s">    columns=[f&quot;sample_{i}&quot; for i in range(n_samples)]</span>
</span><span id="__span-10-29"><a id="__codelineno-10-29" name="__codelineno-10-29" href="#__codelineno-10-29"></a><span class="s">)</span>
</span><span id="__span-10-30"><a id="__codelineno-10-30" name="__codelineno-10-30" href="#__codelineno-10-30"></a>
</span><span id="__span-10-31"><a id="__codelineno-10-31" name="__codelineno-10-31" href="#__codelineno-10-31"></a><span class="s">mag_abundance = pd.DataFrame(</span>
</span><span id="__span-10-32"><a id="__codelineno-10-32" name="__codelineno-10-32" href="#__codelineno-10-32"></a><span class="s">    np.random.lognormal(0, 2, (n_mags, n_samples)),</span>
</span><span id="__span-10-33"><a id="__codelineno-10-33" name="__codelineno-10-33" href="#__codelineno-10-33"></a><span class="s">    index=[f&quot;MAG_{i:03d}&quot; for i in range(n_mags)],</span>
</span><span id="__span-10-34"><a id="__codelineno-10-34" name="__codelineno-10-34" href="#__codelineno-10-34"></a><span class="s">    columns=[f&quot;sample_{i}&quot; for i in range(n_samples)]</span>
</span><span id="__span-10-35"><a id="__codelineno-10-35" name="__codelineno-10-35" href="#__codelineno-10-35"></a><span class="s">)</span>
</span><span id="__span-10-36"><a id="__codelineno-10-36" name="__codelineno-10-36" href="#__codelineno-10-36"></a>
</span><span id="__span-10-37"><a id="__codelineno-10-37" name="__codelineno-10-37" href="#__codelineno-10-37"></a><span class="s"># Calculate correlations</span>
</span><span id="__span-10-38"><a id="__codelineno-10-38" name="__codelineno-10-38" href="#__codelineno-10-38"></a><span class="s">correlations = []</span>
</span><span id="__span-10-39"><a id="__codelineno-10-39" name="__codelineno-10-39" href="#__codelineno-10-39"></a>
</span><span id="__span-10-40"><a id="__codelineno-10-40" name="__codelineno-10-40" href="#__codelineno-10-40"></a><span class="s">for phage_id in phage_abundance.index:</span>
</span><span id="__span-10-41"><a id="__codelineno-10-41" name="__codelineno-10-41" href="#__codelineno-10-41"></a><span class="s">    phage_abund = phage_abundance.loc[phage_id].values</span>
</span><span id="__span-10-42"><a id="__codelineno-10-42" name="__codelineno-10-42" href="#__codelineno-10-42"></a>
</span><span id="__span-10-43"><a id="__codelineno-10-43" name="__codelineno-10-43" href="#__codelineno-10-43"></a><span class="s">    for mag_id in mag_abundance.index:</span>
</span><span id="__span-10-44"><a id="__codelineno-10-44" name="__codelineno-10-44" href="#__codelineno-10-44"></a><span class="s">        mag_abund = mag_abundance.loc[mag_id].values</span>
</span><span id="__span-10-45"><a id="__codelineno-10-45" name="__codelineno-10-45" href="#__codelineno-10-45"></a>
</span><span id="__span-10-46"><a id="__codelineno-10-46" name="__codelineno-10-46" href="#__codelineno-10-46"></a><span class="s">        # Spearman correlation</span>
</span><span id="__span-10-47"><a id="__codelineno-10-47" name="__codelineno-10-47" href="#__codelineno-10-47"></a><span class="s">        rho, pval = spearmanr(phage_abund, mag_abund)</span>
</span><span id="__span-10-48"><a id="__codelineno-10-48" name="__codelineno-10-48" href="#__codelineno-10-48"></a>
</span><span id="__span-10-49"><a id="__codelineno-10-49" name="__codelineno-10-49" href="#__codelineno-10-49"></a><span class="s">        correlations.append({</span>
</span><span id="__span-10-50"><a id="__codelineno-10-50" name="__codelineno-10-50" href="#__codelineno-10-50"></a><span class="s">            &#39;Phage_ID&#39;: phage_id,</span>
</span><span id="__span-10-51"><a id="__codelineno-10-51" name="__codelineno-10-51" href="#__codelineno-10-51"></a><span class="s">            &#39;MAG_ID&#39;: mag_id,</span>
</span><span id="__span-10-52"><a id="__codelineno-10-52" name="__codelineno-10-52" href="#__codelineno-10-52"></a><span class="s">            &#39;Spearman_Rho&#39;: rho,</span>
</span><span id="__span-10-53"><a id="__codelineno-10-53" name="__codelineno-10-53" href="#__codelineno-10-53"></a><span class="s">            &#39;P_value&#39;: pval</span>
</span><span id="__span-10-54"><a id="__codelineno-10-54" name="__codelineno-10-54" href="#__codelineno-10-54"></a><span class="s">        })</span>
</span><span id="__span-10-55"><a id="__codelineno-10-55" name="__codelineno-10-55" href="#__codelineno-10-55"></a>
</span><span id="__span-10-56"><a id="__codelineno-10-56" name="__codelineno-10-56" href="#__codelineno-10-56"></a><span class="s"># Convert to DataFrame</span>
</span><span id="__span-10-57"><a id="__codelineno-10-57" name="__codelineno-10-57" href="#__codelineno-10-57"></a><span class="s">corr_df = pd.DataFrame(correlations)</span>
</span><span id="__span-10-58"><a id="__codelineno-10-58" name="__codelineno-10-58" href="#__codelineno-10-58"></a>
</span><span id="__span-10-59"><a id="__codelineno-10-59" name="__codelineno-10-59" href="#__codelineno-10-59"></a><span class="s"># Filter for significant positive correlations</span>
</span><span id="__span-10-60"><a id="__codelineno-10-60" name="__codelineno-10-60" href="#__codelineno-10-60"></a><span class="s">corr_sig = corr_df[(corr_df[&#39;P_value&#39;] &lt; 0.01) &amp; (corr_df[&#39;Spearman_Rho&#39;] &gt; 0.6)].copy()</span>
</span><span id="__span-10-61"><a id="__codelineno-10-61" name="__codelineno-10-61" href="#__codelineno-10-61"></a>
</span><span id="__span-10-62"><a id="__codelineno-10-62" name="__codelineno-10-62" href="#__codelineno-10-62"></a><span class="s"># Take best match per phage</span>
</span><span id="__span-10-63"><a id="__codelineno-10-63" name="__codelineno-10-63" href="#__codelineno-10-63"></a><span class="s">best_corr = corr_sig.loc[corr_sig.groupby(&#39;Phage_ID&#39;)[&#39;Spearman_Rho&#39;].idxmax()]</span>
</span><span id="__span-10-64"><a id="__codelineno-10-64" name="__codelineno-10-64" href="#__codelineno-10-64"></a>
</span><span id="__span-10-65"><a id="__codelineno-10-65" name="__codelineno-10-65" href="#__codelineno-10-65"></a><span class="s">best_corr[&#39;Method&#39;] = &#39;Co-occurrence&#39;</span>
</span><span id="__span-10-66"><a id="__codelineno-10-66" name="__codelineno-10-66" href="#__codelineno-10-66"></a><span class="s">best_corr[&#39;Confidence&#39;] = &#39;Low&#39;</span>
</span><span id="__span-10-67"><a id="__codelineno-10-67" name="__codelineno-10-67" href="#__codelineno-10-67"></a><span class="s">best_corr.to_csv(&#39;05_cooccurrence/cooccurrence_host_predictions.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-10-68"><a id="__codelineno-10-68" name="__codelineno-10-68" href="#__codelineno-10-68"></a>
</span><span id="__span-10-69"><a id="__codelineno-10-69" name="__codelineno-10-69" href="#__codelineno-10-69"></a><span class="s">print(f&quot;Co-occurrence-based predictions: {len(best_corr)}&quot;)</span>
</span><span id="__span-10-70"><a id="__codelineno-10-70" name="__codelineno-10-70" href="#__codelineno-10-70"></a><span class="s">print(f&quot;Prediction rate: {len(best_corr) / n_phages * 100:.1f}%&quot;)</span>
</span><span id="__span-10-71"><a id="__codelineno-10-71" name="__codelineno-10-71" href="#__codelineno-10-71"></a><span class="s">EOF</span>
</span></code></pre></div>
<p><strong>Expected co-occurrence predictions:</strong>
- <strong>Phages with correlation:</strong> 50-150 (10-30%)
- <strong>Confidence:</strong> Low (correlation  causation, many false positives)</p>
<h2 id="step-6-consensus-prediction">Step 6: Consensus Prediction</h2>
<p>Combine predictions from multiple methods for higher confidence.</p>
<h3 id="61-merge-all-predictions">6.1 Merge All Predictions</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Create consensus directory</span>
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>06_consensus
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a><span class="c1"># Combine all prediction tables</span>
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-11-6"><a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a><span class="s">import pandas as pd</span>
</span><span id="__span-11-7"><a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a>
</span><span id="__span-11-8"><a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a><span class="s"># Load all prediction tables</span>
</span><span id="__span-11-9"><a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a><span class="s">crispr = pd.read_csv(&#39;01_crispr/crispr_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-11-10"><a id="__codelineno-11-10" name="__codelineno-11-10" href="#__codelineno-11-10"></a><span class="s">homology = pd.read_csv(&#39;02_homology/homology_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-11-11"><a id="__codelineno-11-11" name="__codelineno-11-11" href="#__codelineno-11-11"></a><span class="s">wish = pd.read_csv(&#39;03_wish/wish_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-11-12"><a id="__codelineno-11-12" name="__codelineno-11-12" href="#__codelineno-11-12"></a><span class="s">iphop = pd.read_csv(&#39;04_iphop/iphop_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-11-13"><a id="__codelineno-11-13" name="__codelineno-11-13" href="#__codelineno-11-13"></a><span class="s"># cooccur = pd.read_csv(&#39;05_cooccurrence/cooccurrence_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-11-14"><a id="__codelineno-11-14" name="__codelineno-11-14" href="#__codelineno-11-14"></a>
</span><span id="__span-11-15"><a id="__codelineno-11-15" name="__codelineno-11-15" href="#__codelineno-11-15"></a><span class="s"># Standardize columns</span>
</span><span id="__span-11-16"><a id="__codelineno-11-16" name="__codelineno-11-16" href="#__codelineno-11-16"></a><span class="s">def standardize(df, method_name):</span>
</span><span id="__span-11-17"><a id="__codelineno-11-17" name="__codelineno-11-17" href="#__codelineno-11-17"></a><span class="s">    return df[[&#39;Phage_ID&#39;, &#39;MAG_ID&#39;, &#39;Method&#39;, &#39;Confidence&#39;]].copy()</span>
</span><span id="__span-11-18"><a id="__codelineno-11-18" name="__codelineno-11-18" href="#__codelineno-11-18"></a>
</span><span id="__span-11-19"><a id="__codelineno-11-19" name="__codelineno-11-19" href="#__codelineno-11-19"></a><span class="s">crispr_std = standardize(crispr, &#39;CRISPR&#39;)</span>
</span><span id="__span-11-20"><a id="__codelineno-11-20" name="__codelineno-11-20" href="#__codelineno-11-20"></a><span class="s">homology_std = standardize(homology, &#39;Homology&#39;)</span>
</span><span id="__span-11-21"><a id="__codelineno-11-21" name="__codelineno-11-21" href="#__codelineno-11-21"></a><span class="s">wish_std = standardize(wish, &#39;WIsH&#39;)</span>
</span><span id="__span-11-22"><a id="__codelineno-11-22" name="__codelineno-11-22" href="#__codelineno-11-22"></a>
</span><span id="__span-11-23"><a id="__codelineno-11-23" name="__codelineno-11-23" href="#__codelineno-11-23"></a><span class="s"># Note: iPHoP output format is different, needs custom parsing</span>
</span><span id="__span-11-24"><a id="__codelineno-11-24" name="__codelineno-11-24" href="#__codelineno-11-24"></a><span class="s"># For tutorial, we&#39;ll just use the first three</span>
</span><span id="__span-11-25"><a id="__codelineno-11-25" name="__codelineno-11-25" href="#__codelineno-11-25"></a>
</span><span id="__span-11-26"><a id="__codelineno-11-26" name="__codelineno-11-26" href="#__codelineno-11-26"></a><span class="s"># Combine</span>
</span><span id="__span-11-27"><a id="__codelineno-11-27" name="__codelineno-11-27" href="#__codelineno-11-27"></a><span class="s">all_predictions = pd.concat([crispr_std, homology_std, wish_std], ignore_index=True)</span>
</span><span id="__span-11-28"><a id="__codelineno-11-28" name="__codelineno-11-28" href="#__codelineno-11-28"></a>
</span><span id="__span-11-29"><a id="__codelineno-11-29" name="__codelineno-11-29" href="#__codelineno-11-29"></a><span class="s"># Save all predictions</span>
</span><span id="__span-11-30"><a id="__codelineno-11-30" name="__codelineno-11-30" href="#__codelineno-11-30"></a><span class="s">all_predictions.to_csv(&#39;06_consensus/all_host_predictions.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-11-31"><a id="__codelineno-11-31" name="__codelineno-11-31" href="#__codelineno-11-31"></a>
</span><span id="__span-11-32"><a id="__codelineno-11-32" name="__codelineno-11-32" href="#__codelineno-11-32"></a><span class="s"># Count predictions per phage</span>
</span><span id="__span-11-33"><a id="__codelineno-11-33" name="__codelineno-11-33" href="#__codelineno-11-33"></a><span class="s">pred_counts = all_predictions.groupby(&#39;Phage_ID&#39;).size().reset_index(name=&#39;Num_Methods&#39;)</span>
</span><span id="__span-11-34"><a id="__codelineno-11-34" name="__codelineno-11-34" href="#__codelineno-11-34"></a>
</span><span id="__span-11-35"><a id="__codelineno-11-35" name="__codelineno-11-35" href="#__codelineno-11-35"></a><span class="s">print(f&quot;Total predictions: {len(all_predictions)}&quot;)</span>
</span><span id="__span-11-36"><a id="__codelineno-11-36" name="__codelineno-11-36" href="#__codelineno-11-36"></a><span class="s">print(f&quot;Phages with 1 prediction: {all_predictions[&#39;Phage_ID&#39;].nunique()}&quot;)</span>
</span><span id="__span-11-37"><a id="__codelineno-11-37" name="__codelineno-11-37" href="#__codelineno-11-37"></a><span class="s">print(f&quot;Phages with 2 methods agreeing: {len(pred_counts[pred_counts[&#39;Num_Methods&#39;] &gt;= 2])}&quot;)</span>
</span><span id="__span-11-38"><a id="__codelineno-11-38" name="__codelineno-11-38" href="#__codelineno-11-38"></a>
</span><span id="__span-11-39"><a id="__codelineno-11-39" name="__codelineno-11-39" href="#__codelineno-11-39"></a><span class="s"># Find consensus predictions (2 methods predicting same host)</span>
</span><span id="__span-11-40"><a id="__codelineno-11-40" name="__codelineno-11-40" href="#__codelineno-11-40"></a><span class="s">consensus_predictions = []</span>
</span><span id="__span-11-41"><a id="__codelineno-11-41" name="__codelineno-11-41" href="#__codelineno-11-41"></a>
</span><span id="__span-11-42"><a id="__codelineno-11-42" name="__codelineno-11-42" href="#__codelineno-11-42"></a><span class="s">for phage in all_predictions[&#39;Phage_ID&#39;].unique():</span>
</span><span id="__span-11-43"><a id="__codelineno-11-43" name="__codelineno-11-43" href="#__codelineno-11-43"></a><span class="s">    phage_preds = all_predictions[all_predictions[&#39;Phage_ID&#39;] == phage]</span>
</span><span id="__span-11-44"><a id="__codelineno-11-44" name="__codelineno-11-44" href="#__codelineno-11-44"></a>
</span><span id="__span-11-45"><a id="__codelineno-11-45" name="__codelineno-11-45" href="#__codelineno-11-45"></a><span class="s">    # Count predictions for each MAG</span>
</span><span id="__span-11-46"><a id="__codelineno-11-46" name="__codelineno-11-46" href="#__codelineno-11-46"></a><span class="s">    mag_counts = phage_preds[&#39;MAG_ID&#39;].value_counts()</span>
</span><span id="__span-11-47"><a id="__codelineno-11-47" name="__codelineno-11-47" href="#__codelineno-11-47"></a>
</span><span id="__span-11-48"><a id="__codelineno-11-48" name="__codelineno-11-48" href="#__codelineno-11-48"></a><span class="s">    if mag_counts.max() &gt;= 2:  # At least 2 methods agree</span>
</span><span id="__span-11-49"><a id="__codelineno-11-49" name="__codelineno-11-49" href="#__codelineno-11-49"></a><span class="s">        predicted_mag = mag_counts.idxmax()</span>
</span><span id="__span-11-50"><a id="__codelineno-11-50" name="__codelineno-11-50" href="#__codelineno-11-50"></a><span class="s">        num_methods = mag_counts.max()</span>
</span><span id="__span-11-51"><a id="__codelineno-11-51" name="__codelineno-11-51" href="#__codelineno-11-51"></a>
</span><span id="__span-11-52"><a id="__codelineno-11-52" name="__codelineno-11-52" href="#__codelineno-11-52"></a><span class="s">        # Get list of methods</span>
</span><span id="__span-11-53"><a id="__codelineno-11-53" name="__codelineno-11-53" href="#__codelineno-11-53"></a><span class="s">        methods = phage_preds[phage_preds[&#39;MAG_ID&#39;] == predicted_mag][&#39;Method&#39;].tolist()</span>
</span><span id="__span-11-54"><a id="__codelineno-11-54" name="__codelineno-11-54" href="#__codelineno-11-54"></a>
</span><span id="__span-11-55"><a id="__codelineno-11-55" name="__codelineno-11-55" href="#__codelineno-11-55"></a><span class="s">        consensus_predictions.append({</span>
</span><span id="__span-11-56"><a id="__codelineno-11-56" name="__codelineno-11-56" href="#__codelineno-11-56"></a><span class="s">            &#39;Phage_ID&#39;: phage,</span>
</span><span id="__span-11-57"><a id="__codelineno-11-57" name="__codelineno-11-57" href="#__codelineno-11-57"></a><span class="s">            &#39;Predicted_Host&#39;: predicted_mag,</span>
</span><span id="__span-11-58"><a id="__codelineno-11-58" name="__codelineno-11-58" href="#__codelineno-11-58"></a><span class="s">            &#39;Num_Methods_Agreeing&#39;: num_methods,</span>
</span><span id="__span-11-59"><a id="__codelineno-11-59" name="__codelineno-11-59" href="#__codelineno-11-59"></a><span class="s">            &#39;Methods&#39;: &#39;,&#39;.join(methods),</span>
</span><span id="__span-11-60"><a id="__codelineno-11-60" name="__codelineno-11-60" href="#__codelineno-11-60"></a><span class="s">            &#39;Confidence&#39;: &#39;High&#39; if &#39;CRISPR&#39; in methods else &#39;Medium&#39;</span>
</span><span id="__span-11-61"><a id="__codelineno-11-61" name="__codelineno-11-61" href="#__codelineno-11-61"></a><span class="s">        })</span>
</span><span id="__span-11-62"><a id="__codelineno-11-62" name="__codelineno-11-62" href="#__codelineno-11-62"></a>
</span><span id="__span-11-63"><a id="__codelineno-11-63" name="__codelineno-11-63" href="#__codelineno-11-63"></a><span class="s">consensus_df = pd.DataFrame(consensus_predictions)</span>
</span><span id="__span-11-64"><a id="__codelineno-11-64" name="__codelineno-11-64" href="#__codelineno-11-64"></a><span class="s">consensus_df.to_csv(&#39;06_consensus/consensus_host_predictions.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-11-65"><a id="__codelineno-11-65" name="__codelineno-11-65" href="#__codelineno-11-65"></a>
</span><span id="__span-11-66"><a id="__codelineno-11-66" name="__codelineno-11-66" href="#__codelineno-11-66"></a><span class="s">print(f&quot;\n=== Consensus Predictions ===&quot;)</span>
</span><span id="__span-11-67"><a id="__codelineno-11-67" name="__codelineno-11-67" href="#__codelineno-11-67"></a><span class="s">print(f&quot;Phages with consensus prediction (2 methods): {len(consensus_df)}&quot;)</span>
</span><span id="__span-11-68"><a id="__codelineno-11-68" name="__codelineno-11-68" href="#__codelineno-11-68"></a><span class="s">print(f&quot;Prediction rate: {len(consensus_df) / 500 * 100:.1f}%&quot;)</span>
</span><span id="__span-11-69"><a id="__codelineno-11-69" name="__codelineno-11-69" href="#__codelineno-11-69"></a><span class="s">print(f&quot;High confidence (includes CRISPR): {len(consensus_df[consensus_df[&#39;Confidence&#39;] == &#39;High&#39;])}&quot;)</span>
</span><span id="__span-11-70"><a id="__codelineno-11-70" name="__codelineno-11-70" href="#__codelineno-11-70"></a><span class="s">EOF</span>
</span></code></pre></div>
<h2 id="step-7-validation-and-confidence-assessment">Step 7: Validation and Confidence Assessment</h2>
<h3 id="71-assess-prediction-quality">7.1 Assess Prediction Quality</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1"># Create validation directory</span>
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>07_validation
</span><span id="__span-12-3"><a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>
</span><span id="__span-12-4"><a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a><span class="c1"># Analyze prediction confidence</span>
</span><span id="__span-12-5"><a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-12-6"><a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a><span class="s">import pandas as pd</span>
</span><span id="__span-12-7"><a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a><span class="s">import matplotlib.pyplot as plt</span>
</span><span id="__span-12-8"><a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a>
</span><span id="__span-12-9"><a id="__codelineno-12-9" name="__codelineno-12-9" href="#__codelineno-12-9"></a><span class="s"># Load consensus predictions</span>
</span><span id="__span-12-10"><a id="__codelineno-12-10" name="__codelineno-12-10" href="#__codelineno-12-10"></a><span class="s">consensus = pd.read_csv(&#39;06_consensus/consensus_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-12-11"><a id="__codelineno-12-11" name="__codelineno-12-11" href="#__codelineno-12-11"></a><span class="s">all_preds = pd.read_csv(&#39;06_consensus/all_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-12-12"><a id="__codelineno-12-12" name="__codelineno-12-12" href="#__codelineno-12-12"></a>
</span><span id="__span-12-13"><a id="__codelineno-12-13" name="__codelineno-12-13" href="#__codelineno-12-13"></a><span class="s"># Summary statistics</span>
</span><span id="__span-12-14"><a id="__codelineno-12-14" name="__codelineno-12-14" href="#__codelineno-12-14"></a><span class="s">print(&quot;=== Host Prediction Summary ===\n&quot;)</span>
</span><span id="__span-12-15"><a id="__codelineno-12-15" name="__codelineno-12-15" href="#__codelineno-12-15"></a>
</span><span id="__span-12-16"><a id="__codelineno-12-16" name="__codelineno-12-16" href="#__codelineno-12-16"></a><span class="s">print(f&quot;Total phages: 500&quot;)</span>
</span><span id="__span-12-17"><a id="__codelineno-12-17" name="__codelineno-12-17" href="#__codelineno-12-17"></a><span class="s">print(f&quot;Phages with any prediction: {all_preds[&#39;Phage_ID&#39;].nunique()} ({all_preds[&#39;Phage_ID&#39;].nunique()/500*100:.1f}%)&quot;)</span>
</span><span id="__span-12-18"><a id="__codelineno-12-18" name="__codelineno-12-18" href="#__codelineno-12-18"></a><span class="s">print(f&quot;Phages with consensus prediction (2 methods): {len(consensus)} ({len(consensus)/500*100:.1f}%)&quot;)</span>
</span><span id="__span-12-19"><a id="__codelineno-12-19" name="__codelineno-12-19" href="#__codelineno-12-19"></a><span class="s">print(f&quot;Phages with high confidence prediction: {len(consensus[consensus[&#39;Confidence&#39;]==&#39;High&#39;])} ({len(consensus[consensus[&#39;Confidence&#39;]==&#39;High&#39;])/500*100:.1f}%)&quot;)</span>
</span><span id="__span-12-20"><a id="__codelineno-12-20" name="__codelineno-12-20" href="#__codelineno-12-20"></a>
</span><span id="__span-12-21"><a id="__codelineno-12-21" name="__codelineno-12-21" href="#__codelineno-12-21"></a><span class="s">print(&quot;\nPredictions by method:&quot;)</span>
</span><span id="__span-12-22"><a id="__codelineno-12-22" name="__codelineno-12-22" href="#__codelineno-12-22"></a><span class="s">print(all_preds[&#39;Method&#39;].value_counts())</span>
</span><span id="__span-12-23"><a id="__codelineno-12-23" name="__codelineno-12-23" href="#__codelineno-12-23"></a>
</span><span id="__span-12-24"><a id="__codelineno-12-24" name="__codelineno-12-24" href="#__codelineno-12-24"></a><span class="s">print(&quot;\nConsensus predictions by number of agreeing methods:&quot;)</span>
</span><span id="__span-12-25"><a id="__codelineno-12-25" name="__codelineno-12-25" href="#__codelineno-12-25"></a><span class="s">print(consensus[&#39;Num_Methods_Agreeing&#39;].value_counts())</span>
</span><span id="__span-12-26"><a id="__codelineno-12-26" name="__codelineno-12-26" href="#__codelineno-12-26"></a>
</span><span id="__span-12-27"><a id="__codelineno-12-27" name="__codelineno-12-27" href="#__codelineno-12-27"></a><span class="s"># Plot prediction rates</span>
</span><span id="__span-12-28"><a id="__codelineno-12-28" name="__codelineno-12-28" href="#__codelineno-12-28"></a><span class="s">methods = all_preds.groupby(&#39;Method&#39;)[&#39;Phage_ID&#39;].nunique()</span>
</span><span id="__span-12-29"><a id="__codelineno-12-29" name="__codelineno-12-29" href="#__codelineno-12-29"></a>
</span><span id="__span-12-30"><a id="__codelineno-12-30" name="__codelineno-12-30" href="#__codelineno-12-30"></a><span class="s">plt.figure(figsize=(10, 6))</span>
</span><span id="__span-12-31"><a id="__codelineno-12-31" name="__codelineno-12-31" href="#__codelineno-12-31"></a><span class="s">methods.plot(kind=&#39;bar&#39;, color=&#39;steelblue&#39;)</span>
</span><span id="__span-12-32"><a id="__codelineno-12-32" name="__codelineno-12-32" href="#__codelineno-12-32"></a><span class="s">plt.ylabel(&#39;Number of Phages with Prediction&#39;)</span>
</span><span id="__span-12-33"><a id="__codelineno-12-33" name="__codelineno-12-33" href="#__codelineno-12-33"></a><span class="s">plt.xlabel(&#39;Method&#39;)</span>
</span><span id="__span-12-34"><a id="__codelineno-12-34" name="__codelineno-12-34" href="#__codelineno-12-34"></a><span class="s">plt.title(&#39;Host Prediction Rate by Method&#39;)</span>
</span><span id="__span-12-35"><a id="__codelineno-12-35" name="__codelineno-12-35" href="#__codelineno-12-35"></a><span class="s">plt.xticks(rotation=45, ha=&#39;right&#39;)</span>
</span><span id="__span-12-36"><a id="__codelineno-12-36" name="__codelineno-12-36" href="#__codelineno-12-36"></a><span class="s">plt.tight_layout()</span>
</span><span id="__span-12-37"><a id="__codelineno-12-37" name="__codelineno-12-37" href="#__codelineno-12-37"></a><span class="s">plt.savefig(&#39;07_validation/prediction_rates_by_method.pdf&#39;)</span>
</span><span id="__span-12-38"><a id="__codelineno-12-38" name="__codelineno-12-38" href="#__codelineno-12-38"></a>
</span><span id="__span-12-39"><a id="__codelineno-12-39" name="__codelineno-12-39" href="#__codelineno-12-39"></a><span class="s"># Confidence distribution</span>
</span><span id="__span-12-40"><a id="__codelineno-12-40" name="__codelineno-12-40" href="#__codelineno-12-40"></a><span class="s">plt.figure(figsize=(8, 6))</span>
</span><span id="__span-12-41"><a id="__codelineno-12-41" name="__codelineno-12-41" href="#__codelineno-12-41"></a><span class="s">all_preds[&#39;Confidence&#39;].value_counts().plot(kind=&#39;bar&#39;, color=&#39;coral&#39;)</span>
</span><span id="__span-12-42"><a id="__codelineno-12-42" name="__codelineno-12-42" href="#__codelineno-12-42"></a><span class="s">plt.ylabel(&#39;Number of Predictions&#39;)</span>
</span><span id="__span-12-43"><a id="__codelineno-12-43" name="__codelineno-12-43" href="#__codelineno-12-43"></a><span class="s">plt.xlabel(&#39;Confidence Level&#39;)</span>
</span><span id="__span-12-44"><a id="__codelineno-12-44" name="__codelineno-12-44" href="#__codelineno-12-44"></a><span class="s">plt.title(&#39;Prediction Confidence Distribution&#39;)</span>
</span><span id="__span-12-45"><a id="__codelineno-12-45" name="__codelineno-12-45" href="#__codelineno-12-45"></a><span class="s">plt.xticks(rotation=45, ha=&#39;right&#39;)</span>
</span><span id="__span-12-46"><a id="__codelineno-12-46" name="__codelineno-12-46" href="#__codelineno-12-46"></a><span class="s">plt.tight_layout()</span>
</span><span id="__span-12-47"><a id="__codelineno-12-47" name="__codelineno-12-47" href="#__codelineno-12-47"></a><span class="s">plt.savefig(&#39;07_validation/confidence_distribution.pdf&#39;)</span>
</span><span id="__span-12-48"><a id="__codelineno-12-48" name="__codelineno-12-48" href="#__codelineno-12-48"></a>
</span><span id="__span-12-49"><a id="__codelineno-12-49" name="__codelineno-12-49" href="#__codelineno-12-49"></a><span class="s">print(&quot;\nPlots saved to 07_validation/&quot;)</span>
</span><span id="__span-12-50"><a id="__codelineno-12-50" name="__codelineno-12-50" href="#__codelineno-12-50"></a><span class="s">EOF</span>
</span></code></pre></div>
<h3 id="72-create-final-recommendation-table">7.2 Create Final Recommendation Table</h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="c1"># Generate final host predictions with recommendations</span>
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>python3<span class="w"> </span><span class="s">&lt;&lt; &#39;EOF&#39;</span>
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="s">import pandas as pd</span>
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a><span class="s"># Load all predictions</span>
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a><span class="s">all_preds = pd.read_csv(&#39;06_consensus/all_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a><span class="s">consensus = pd.read_csv(&#39;06_consensus/consensus_host_predictions.tsv&#39;, sep=&#39;\t&#39;)</span>
</span><span id="__span-13-8"><a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a>
</span><span id="__span-13-9"><a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a><span class="s"># Create final recommendation table</span>
</span><span id="__span-13-10"><a id="__codelineno-13-10" name="__codelineno-13-10" href="#__codelineno-13-10"></a><span class="s">recommendations = []</span>
</span><span id="__span-13-11"><a id="__codelineno-13-11" name="__codelineno-13-11" href="#__codelineno-13-11"></a>
</span><span id="__span-13-12"><a id="__codelineno-13-12" name="__codelineno-13-12" href="#__codelineno-13-12"></a><span class="s">for phage in set(all_preds[&#39;Phage_ID&#39;].unique()):</span>
</span><span id="__span-13-13"><a id="__codelineno-13-13" name="__codelineno-13-13" href="#__codelineno-13-13"></a><span class="s">    phage_preds = all_preds[all_preds[&#39;Phage_ID&#39;] == phage]</span>
</span><span id="__span-13-14"><a id="__codelineno-13-14" name="__codelineno-13-14" href="#__codelineno-13-14"></a>
</span><span id="__span-13-15"><a id="__codelineno-13-15" name="__codelineno-13-15" href="#__codelineno-13-15"></a><span class="s">    # Check if in consensus</span>
</span><span id="__span-13-16"><a id="__codelineno-13-16" name="__codelineno-13-16" href="#__codelineno-13-16"></a><span class="s">    if phage in consensus[&#39;Phage_ID&#39;].values:</span>
</span><span id="__span-13-17"><a id="__codelineno-13-17" name="__codelineno-13-17" href="#__codelineno-13-17"></a><span class="s">        cons_row = consensus[consensus[&#39;Phage_ID&#39;] == phage].iloc[0]</span>
</span><span id="__span-13-18"><a id="__codelineno-13-18" name="__codelineno-13-18" href="#__codelineno-13-18"></a>
</span><span id="__span-13-19"><a id="__codelineno-13-19" name="__codelineno-13-19" href="#__codelineno-13-19"></a><span class="s">        recommendations.append({</span>
</span><span id="__span-13-20"><a id="__codelineno-13-20" name="__codelineno-13-20" href="#__codelineno-13-20"></a><span class="s">            &#39;Phage_ID&#39;: phage,</span>
</span><span id="__span-13-21"><a id="__codelineno-13-21" name="__codelineno-13-21" href="#__codelineno-13-21"></a><span class="s">            &#39;Recommended_Host&#39;: cons_row[&#39;Predicted_Host&#39;],</span>
</span><span id="__span-13-22"><a id="__codelineno-13-22" name="__codelineno-13-22" href="#__codelineno-13-22"></a><span class="s">            &#39;Confidence&#39;: cons_row[&#39;Confidence&#39;],</span>
</span><span id="__span-13-23"><a id="__codelineno-13-23" name="__codelineno-13-23" href="#__codelineno-13-23"></a><span class="s">            &#39;Num_Methods&#39;: cons_row[&#39;Num_Methods_Agreeing&#39;],</span>
</span><span id="__span-13-24"><a id="__codelineno-13-24" name="__codelineno-13-24" href="#__codelineno-13-24"></a><span class="s">            &#39;Methods&#39;: cons_row[&#39;Methods&#39;],</span>
</span><span id="__span-13-25"><a id="__codelineno-13-25" name="__codelineno-13-25" href="#__codelineno-13-25"></a><span class="s">            &#39;Recommendation&#39;: &#39;Use this prediction&#39; if cons_row[&#39;Confidence&#39;] == &#39;High&#39; else &#39;Validate experimentally&#39;</span>
</span><span id="__span-13-26"><a id="__codelineno-13-26" name="__codelineno-13-26" href="#__codelineno-13-26"></a><span class="s">        })</span>
</span><span id="__span-13-27"><a id="__codelineno-13-27" name="__codelineno-13-27" href="#__codelineno-13-27"></a><span class="s">    else:</span>
</span><span id="__span-13-28"><a id="__codelineno-13-28" name="__codelineno-13-28" href="#__codelineno-13-28"></a><span class="s">        # Single method prediction - lower confidence</span>
</span><span id="__span-13-29"><a id="__codelineno-13-29" name="__codelineno-13-29" href="#__codelineno-13-29"></a><span class="s">        # Take CRISPR if available, otherwise best available</span>
</span><span id="__span-13-30"><a id="__codelineno-13-30" name="__codelineno-13-30" href="#__codelineno-13-30"></a><span class="s">        if &#39;CRISPR&#39; in phage_preds[&#39;Method&#39;].values:</span>
</span><span id="__span-13-31"><a id="__codelineno-13-31" name="__codelineno-13-31" href="#__codelineno-13-31"></a><span class="s">            pred_row = phage_preds[phage_preds[&#39;Method&#39;] == &#39;CRISPR&#39;].iloc[0]</span>
</span><span id="__span-13-32"><a id="__codelineno-13-32" name="__codelineno-13-32" href="#__codelineno-13-32"></a><span class="s">            conf = &#39;High&#39;</span>
</span><span id="__span-13-33"><a id="__codelineno-13-33" name="__codelineno-13-33" href="#__codelineno-13-33"></a><span class="s">            rec = &#39;Use this prediction (CRISPR match)&#39;</span>
</span><span id="__span-13-34"><a id="__codelineno-13-34" name="__codelineno-13-34" href="#__codelineno-13-34"></a><span class="s">        else:</span>
</span><span id="__span-13-35"><a id="__codelineno-13-35" name="__codelineno-13-35" href="#__codelineno-13-35"></a><span class="s">            # Take highest confidence method</span>
</span><span id="__span-13-36"><a id="__codelineno-13-36" name="__codelineno-13-36" href="#__codelineno-13-36"></a><span class="s">            pred_row = phage_preds.iloc[0]</span>
</span><span id="__span-13-37"><a id="__codelineno-13-37" name="__codelineno-13-37" href="#__codelineno-13-37"></a><span class="s">            conf = &#39;Low&#39;</span>
</span><span id="__span-13-38"><a id="__codelineno-13-38" name="__codelineno-13-38" href="#__codelineno-13-38"></a><span class="s">            rec = &#39;Low confidence - validate experimentally or use with caution&#39;</span>
</span><span id="__span-13-39"><a id="__codelineno-13-39" name="__codelineno-13-39" href="#__codelineno-13-39"></a>
</span><span id="__span-13-40"><a id="__codelineno-13-40" name="__codelineno-13-40" href="#__codelineno-13-40"></a><span class="s">        recommendations.append({</span>
</span><span id="__span-13-41"><a id="__codelineno-13-41" name="__codelineno-13-41" href="#__codelineno-13-41"></a><span class="s">            &#39;Phage_ID&#39;: phage,</span>
</span><span id="__span-13-42"><a id="__codelineno-13-42" name="__codelineno-13-42" href="#__codelineno-13-42"></a><span class="s">            &#39;Recommended_Host&#39;: pred_row[&#39;MAG_ID&#39;],</span>
</span><span id="__span-13-43"><a id="__codelineno-13-43" name="__codelineno-13-43" href="#__codelineno-13-43"></a><span class="s">            &#39;Confidence&#39;: conf,</span>
</span><span id="__span-13-44"><a id="__codelineno-13-44" name="__codelineno-13-44" href="#__codelineno-13-44"></a><span class="s">            &#39;Num_Methods&#39;: 1,</span>
</span><span id="__span-13-45"><a id="__codelineno-13-45" name="__codelineno-13-45" href="#__codelineno-13-45"></a><span class="s">            &#39;Methods&#39;: pred_row[&#39;Method&#39;],</span>
</span><span id="__span-13-46"><a id="__codelineno-13-46" name="__codelineno-13-46" href="#__codelineno-13-46"></a><span class="s">            &#39;Recommendation&#39;: rec</span>
</span><span id="__span-13-47"><a id="__codelineno-13-47" name="__codelineno-13-47" href="#__codelineno-13-47"></a><span class="s">        })</span>
</span><span id="__span-13-48"><a id="__codelineno-13-48" name="__codelineno-13-48" href="#__codelineno-13-48"></a>
</span><span id="__span-13-49"><a id="__codelineno-13-49" name="__codelineno-13-49" href="#__codelineno-13-49"></a><span class="s">rec_df = pd.DataFrame(recommendations)</span>
</span><span id="__span-13-50"><a id="__codelineno-13-50" name="__codelineno-13-50" href="#__codelineno-13-50"></a><span class="s">rec_df.to_csv(&#39;07_validation/final_host_recommendations.tsv&#39;, sep=&#39;\t&#39;, index=False)</span>
</span><span id="__span-13-51"><a id="__codelineno-13-51" name="__codelineno-13-51" href="#__codelineno-13-51"></a>
</span><span id="__span-13-52"><a id="__codelineno-13-52" name="__codelineno-13-52" href="#__codelineno-13-52"></a><span class="s">print(f&quot;Final recommendations created for {len(rec_df)} phages&quot;)</span>
</span><span id="__span-13-53"><a id="__codelineno-13-53" name="__codelineno-13-53" href="#__codelineno-13-53"></a><span class="s">print(&quot;\nRecommendation breakdown:&quot;)</span>
</span><span id="__span-13-54"><a id="__codelineno-13-54" name="__codelineno-13-54" href="#__codelineno-13-54"></a><span class="s">print(rec_df[&#39;Recommendation&#39;].value_counts())</span>
</span><span id="__span-13-55"><a id="__codelineno-13-55" name="__codelineno-13-55" href="#__codelineno-13-55"></a><span class="s">EOF</span>
</span></code></pre></div>
<h2 id="step-8-summary-and-interpretation">Step 8: Summary and Interpretation</h2>
<h3 id="expected-results">Expected Results</h3>
<p><strong>Typical prediction rates:</strong>
- <strong>CRISPR matches:</strong> 2-10% of phages (highest confidence)
- <strong>Homology matches:</strong> 20-50% of phages (medium confidence)
- <strong>Machine learning:</strong> 30-70% of phages (variable confidence)
- <strong>Consensus (2 methods):</strong> 10-30% of phages (recommended for use)</p>
<h3 id="interpreting-confidence-levels">Interpreting Confidence Levels</h3>
<table>
<thead>
<tr>
<th>Confidence</th>
<th>Criteria</th>
<th>Accuracy (Expected)</th>
<th>Recommendation</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>High</strong></td>
<td>CRISPR match OR 3+ methods agree</td>
<td>~80-95%</td>
<td>Use with confidence</td>
</tr>
<tr>
<td><strong>Medium-High</strong></td>
<td>2+ methods agree (incl. homology)</td>
<td>~50-70%</td>
<td>Reasonable for most analyses, validate key findings</td>
</tr>
<tr>
<td><strong>Medium</strong></td>
<td>iPHoP &gt;90 score OR homology + ML</td>
<td>~30-50%</td>
<td>Use with caution, validate if important</td>
</tr>
<tr>
<td><strong>Low</strong></td>
<td>Single method (non-CRISPR)</td>
<td>~20-40%</td>
<td>Hypothesis only, requires validation</td>
</tr>
</tbody>
</table>
<h3 id="validation-strategies">Validation Strategies</h3>
<p><strong>Experimental validation:</strong>
1. <strong>Culture-based:</strong> Infect predicted host with phage isolate
2. <strong>qPCR:</strong> Check phage and host co-occurrence in samples
3. <strong>Hi-C:</strong> Proximity ligation shows phage-host interactions
4. <strong>BONCAT:</strong> Label newly synthesized proteins during infection</p>
<p><strong>Computational validation:</strong>
1. <strong>Prophage analysis:</strong> Check if phage integrates in predicted host lineage
2. <strong>Coverage correlation:</strong> Phage and host should co-vary across samples
3. <strong>Literature:</strong> Known phages from same family infecting predicted host genus</p>
<h2 id="troubleshooting">Troubleshooting</h2>
<h3 id="problem-very-few-crispr-matches">Problem: Very Few CRISPR Matches</h3>
<p><strong>Causes:</strong>
- Few MAGs have CRISPR systems
- Phages are novel and not yet encountered by hosts
- CRISPR spacers are too old and diverged</p>
<p><strong>Solutions:</strong>
- Include more MAGs from same environment
- Use public CRISPR spacer databases (IMG/VR, PADLOC)
- Relax CRISPR matching criteria slightly (allow 2 mismatches)</p>
<h3 id="problem-conflicting-predictions">Problem: Conflicting Predictions</h3>
<p><strong>When methods disagree:</strong>
- Prioritize CRISPR &gt; Homology &gt; Machine Learning
- Check taxonomic consistency (predicted hosts should be related)
- Look at prediction confidence scores
- Use consensus only (2 methods agree)</p>
<h3 id="problem-no-predictions-for-most-phages">Problem: No Predictions for Most Phages</h3>
<p><strong>This is normal!</strong> Expected for:
- Highly novel phages
- Undersampled environments
- Limited MAG database</p>
<p><strong>Solutions:</strong>
- Expand MAG collection
- Use broader host databases (IMG/VR, GTDB)
- Accept that many phages will have unknown hosts</p>
<h2 id="next-steps">Next Steps</h2>
<p><strong>Improve predictions:</strong>
- Add more MAGs from your environment
- Include metatranscriptome data (active infections)
- Perform targeted validation experiments</p>
<p><strong>Downstream analyses:</strong>
- Host range analysis (broad vs narrow)
- Network analysis (phage-host interaction networks)
- Link phage auxiliary metabolic genes to host metabolism</p>
<h2 id="further-reading">Further Reading</h2>
<ul>
<li>Galiez, C., et al. (2017). "WIsH: who is the host?" <em>Bioinformatics</em>, 33(19), 3113-3114.</li>
<li>Roux, S., et al. (2023). "iPHoP: An integrated machine learning framework to maximize host prediction for metagenome-derived viruses." <em>PLoS Biology</em>, 21(4), e3002083.</li>
<li>Dion, M. B., et al. (2021). "Streamlining CRISPR spacer-based bacterial host predictions." <em>PeerJ</em>, 9, e11059.</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <!-- Custom footer -->

<footer class="md-footer">
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; 2023-2025 <a href="https://github.com/shandley">Scott Handley</a> -  <a href="#__consent">Change cookie settings</a>
          </div>
        
        Awesome-Virome Documentation | 
        <a href="https://shandley.github.io/awesome-virome/dashboard.html" target="_blank">
          Dashboard
        </a> |
        <a href="https://shandley.github.io/awesome-virome/citations.html" target="_blank">
          Citation Analytics
        </a>
        <br>
        Made with
        <a href="https://www.mkdocs.org/" target="_blank" rel="noopener">
          MkDocs
        </a> and
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/shandley/awesome-virome" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
    </div>
  </div>
</footer>

<script>
document.addEventListener('DOMContentLoaded', function() {
  window.mermaidConfig = {
    startOnLoad: true,
    theme: 'default',
    flowchart: {
      useMaxWidth: false,
      htmlLabels: true,
      curve: 'basis',
      defaultRenderer: 'elk',
      width: 250
    },
    securityLevel: 'loose'
  };
  
  function initMermaid() {
    try {
      if (typeof window.mermaid !== 'undefined') {
        window.mermaid.initialize(window.mermaidConfig);
        
        // Process all mermaid diagrams that haven't been processed yet
        document.querySelectorAll('.mermaid:not([data-processed="true"])').forEach(function(el) {
          try {
            window.mermaid.init(undefined, el);
          } catch (err) {
            console.error('Mermaid initialization error for specific element:', err);
            // Add an error class to show there was an issue
            el.classList.add('mermaid-error');
            el.textContent = 'Diagram rendering error. Please check console for details.';
          }
        });
      }
    } catch (err) {
      console.error('Mermaid initialization error:', err);
    }
  }
  
  // Try to use mermaid if it's already loaded
  if (typeof window.mermaid !== 'undefined') {
    initMermaid();
  } else {
    // Load mermaid if it's not available
    console.log('Loading mermaid from mermaid.html partial');
    var script = document.createElement('script');
    script.src = 'https://cdn.jsdelivr.net/npm/mermaid@9.4.3/dist/mermaid.min.js';
    script.onload = function() {
      console.log('Mermaid loaded successfully');
      initMermaid();
    };
    script.onerror = function() {
      console.error('Failed to load mermaid script');
    };
    document.head.appendChild(script);
  }
  
  // Secondary initialization after a delay to catch any missed diagrams
  setTimeout(initMermaid, 1000);
});

// Backup initialization on full page load
window.addEventListener('load', function() {
  if (typeof window.mermaid !== 'undefined') {
    console.log('Window loaded, reinitializing mermaid');
    window.mermaid.initialize({
      startOnLoad: true,
      theme: 'default',
      flowchart: {
        useMaxWidth: false,
        htmlLabels: true,
        curve: 'basis',
        defaultRenderer: 'elk',
        width: 250
      },
      securityLevel: 'loose'
    });
    
    // Process any diagrams that still haven't been processed
    document.querySelectorAll('.mermaid:not([data-processed="true"])').forEach(function(el) {
      try {
        window.mermaid.init(undefined, el);
      } catch (err) {
        console.error('Mermaid load-time initialization error:', err);
      }
    });
  }
});
</script>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.indexes", "navigation.top", "navigation.footer", "content.tabs.link", "content.code.copy", "content.code.annotate", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.7a47a382.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.e71a0d61.min.js"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mermaid@9.4.3/dist/mermaid.min.js"></script>
      
        <script src="../../js/mermaid.js"></script>
      
    
  </body>
</html>