
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://shandley.github.io/awesome-virome/latest/fundamentals/sequencing-strategies/">
      
      
        <link rel="prev" href="../sample-preparation/">
      
      
        <link rel="next" href="../../tutorials/">
      
      
        
      
      
      <link rel="icon" href="../../assets/awesome-virome-hex.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.0">
    
    
      
        <title>Sequencing Strategies - Awesome-Virome</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.618322db.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      
  
  
    
      
      
    
  
  
  <style>:root{--md-admonition-icon--tool:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22m21.71%2020.29-1.42%201.42a1%201%200%200%201-1.41%200L7%209.85A3.8%203.8%200%200%201%206%2010a4%204%200%200%201-3.78-5.3l2.54%202.54.53-.53%201.42-1.42.53-.53L4.7%202.22A4%204%200%200%201%2010%206a3.8%203.8%200%200%201-.15%201l11.86%2011.88a1%201%200%200%201%200%201.41M2.29%2018.88a1%201%200%200%200%200%201.41l1.42%201.42a1%201%200%200%200%201.41%200l5.47-5.46-2.83-2.83M20%202l-4%202v2l-2.17%202.17%202%202L18%208h2l2-4Z%22/%3E%3C/svg%3E');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#sequencing-strategies-for-virome-analysis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../home.html" title="Awesome-Virome" class="md-header__button md-logo" aria-label="Awesome-Virome" data-md-component="logo">
      
  <img src="../../assets/awesome-virome-hex.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Awesome-Virome
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Sequencing Strategies
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="teal"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/shandley/awesome-virome" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    shandley/awesome-virome
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../intro/" class="md-tabs__link">
          
  
  
    
  
  Getting Started

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../" class="md-tabs__link">
          
  
  
    
  
  Fundamentals

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tutorials/" class="md-tabs__link">
          
  
  
    
  
  Tutorials

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../best-practices/quality-control/" class="md-tabs__link">
          
  
  
    
  
  Best Practices

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tools/overview/" class="md-tabs__link">
          
  
  
    
  
  Core Analysis Tools

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../troubleshooting/" class="md-tabs__link">
          
  
  
    
  
  Troubleshooting

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../api/overview/" class="md-tabs__link">
          
  
  
    
  
  API Reference

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../contributing/guidelines/" class="md-tabs__link">
          
  
  
    
  
  Contributing

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../home.html" title="Awesome-Virome" class="md-nav__button md-logo" aria-label="Awesome-Virome" data-md-component="logo">
      
  <img src="../../assets/awesome-virome-hex.png" alt="logo">

    </a>
    Awesome-Virome
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/shandley/awesome-virome" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    shandley/awesome-virome
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../intro/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../intro/workflows/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Typical Workflows
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../intro/versioning/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Versioning
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Fundamentals
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_3" id="__nav_3_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Fundamentals
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sample-preparation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Sample Preparation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Sequencing Strategies
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Sequencing Strategies
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#sequencing-platform-overview" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequencing Platform Overview
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sequencing Platform Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#platform-comparison" class="md-nav__link">
    <span class="md-ellipsis">
      
        Platform Comparison
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#illumina-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Illumina Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pacbio-hifi-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        PacBio HiFi Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#oxford-nanopore-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Oxford Nanopore Sequencing
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#library-preparation-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Library Preparation Strategies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Library Preparation Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#standard-shotgun-metagenomics" class="md-nav__link">
    <span class="md-ellipsis">
      
        Standard Shotgun Metagenomics
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amplicon-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Amplicon Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rna-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        RNA Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#low-input-library-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Low-Input Library Preparation
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#coverage-and-sequencing-depth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Coverage and Sequencing Depth
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Coverage and Sequencing Depth">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#how-much-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        How Much Sequencing?
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#recommended-depths" class="md-nav__link">
    <span class="md-ellipsis">
      
        Recommended Depths
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Recommended Depths">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#diversitycommunity-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diversity/Community Analysis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deep-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      
        Deep Characterization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#complete-genome-recovery" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete Genome Recovery
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rarefaction-and-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Rarefaction and Coverage
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#paired-end-vs-single-end" class="md-nav__link">
    <span class="md-ellipsis">
      
        Paired-End vs. Single-End
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Paired-End vs. Single-End">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#paired-end-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Paired-End Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-end-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Single-End Sequencing
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-length-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Length Considerations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Read Length Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#short-reads-50-150-bp" class="md-nav__link">
    <span class="md-ellipsis">
      
        Short Reads (50-150 bp)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#medium-reads-150-300-bp" class="md-nav__link">
    <span class="md-ellipsis">
      
        Medium Reads (150-300 bp)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#long-reads-10-kb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Long Reads (&gt;10 kb)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hybrid-sequencing-approaches" class="md-nav__link">
    <span class="md-ellipsis">
      
        Hybrid Sequencing Approaches
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Hybrid Sequencing Approaches">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#illumina-pacbionanopore" class="md-nav__link">
    <span class="md-ellipsis">
      
        Illumina + PacBio/Nanopore
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adaptivetargeted-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Adaptive/Targeted Sequencing
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-multiplexing-and-barcoding" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sample Multiplexing and Barcoding
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sample Multiplexing and Barcoding">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#when-to-multiplex" class="md-nav__link">
    <span class="md-ellipsis">
      
        When to Multiplex?
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiplexing-guidelines" class="md-nav__link">
    <span class="md-ellipsis">
      
        Multiplexing Guidelines
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#barcoding-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Barcoding Strategies
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cost-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Cost Considerations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Cost Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-sample-cost-estimates-2025" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Sample Cost Estimates (2025)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#special-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Special Considerations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Special Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#metagenome-vs-virome-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Metagenome vs. Virome Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantitative-vs-qualitative" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quantitative vs. Qualitative
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#time-series-and-longitudinal-studies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Time-Series and Longitudinal Studies
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#decision-tree-choosing-your-strategy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Decision Tree: Choosing Your Strategy
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#recommended-strategies-by-study-type" class="md-nav__link">
    <span class="md-ellipsis">
      
        Recommended Strategies by Study Type
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Recommended Strategies by Study Type">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#study-type-1-exploratory-virome-diversity" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 1: Exploratory Virome Diversity
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#study-type-2-novel-virus-discovery" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 2: Novel Virus Discovery
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#study-type-3-outbreak-investigation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 3: Outbreak Investigation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#study-type-4-large-scale-epidemiology" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 4: Large-Scale Epidemiology
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      
        Next Steps
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      
        Further Reading
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../tutorials/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Tutorials
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_4" id="__nav_4_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Tutorials
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/basic-metagenome-virome/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Basic Metagenome Virome
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/rna-virus-discovery/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    RNA Virus Discovery
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/prophage-identification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Prophage Identification
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/comparative-virome/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Comparative Virome Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tutorials/host-prediction-workflows/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction Workflows
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Best Practices
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Best Practices
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quality Control
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/tool-combinations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Tool Combinations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/statistical-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Statistical Analysis
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/validation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../best-practices/reproducibility/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Reproducibility
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Core Analysis Tools
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Core Analysis Tools
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/virus-identification/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Virus Identification
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/host-prediction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Host Prediction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tools/selection-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Selection Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../troubleshooting/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Troubleshooting
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    API Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            
  
    API Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/endpoints/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Endpoints
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            
  
    Contributing
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/guidelines/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Guidelines
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/validation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Validation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../contributing/updating-docs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Updating Docs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#sequencing-platform-overview" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sequencing Platform Overview
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sequencing Platform Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#platform-comparison" class="md-nav__link">
    <span class="md-ellipsis">
      
        Platform Comparison
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#illumina-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Illumina Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pacbio-hifi-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        PacBio HiFi Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#oxford-nanopore-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Oxford Nanopore Sequencing
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#library-preparation-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Library Preparation Strategies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Library Preparation Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#standard-shotgun-metagenomics" class="md-nav__link">
    <span class="md-ellipsis">
      
        Standard Shotgun Metagenomics
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#amplicon-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Amplicon Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rna-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        RNA Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#low-input-library-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Low-Input Library Preparation
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#coverage-and-sequencing-depth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Coverage and Sequencing Depth
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Coverage and Sequencing Depth">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#how-much-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        How Much Sequencing?
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#recommended-depths" class="md-nav__link">
    <span class="md-ellipsis">
      
        Recommended Depths
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Recommended Depths">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#diversitycommunity-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diversity/Community Analysis
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deep-characterization" class="md-nav__link">
    <span class="md-ellipsis">
      
        Deep Characterization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#complete-genome-recovery" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete Genome Recovery
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rarefaction-and-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Rarefaction and Coverage
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#paired-end-vs-single-end" class="md-nav__link">
    <span class="md-ellipsis">
      
        Paired-End vs. Single-End
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Paired-End vs. Single-End">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#paired-end-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Paired-End Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#single-end-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Single-End Sequencing
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-length-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Length Considerations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Read Length Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#short-reads-50-150-bp" class="md-nav__link">
    <span class="md-ellipsis">
      
        Short Reads (50-150 bp)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#medium-reads-150-300-bp" class="md-nav__link">
    <span class="md-ellipsis">
      
        Medium Reads (150-300 bp)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#long-reads-10-kb" class="md-nav__link">
    <span class="md-ellipsis">
      
        Long Reads (&gt;10 kb)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hybrid-sequencing-approaches" class="md-nav__link">
    <span class="md-ellipsis">
      
        Hybrid Sequencing Approaches
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Hybrid Sequencing Approaches">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#illumina-pacbionanopore" class="md-nav__link">
    <span class="md-ellipsis">
      
        Illumina + PacBio/Nanopore
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adaptivetargeted-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Adaptive/Targeted Sequencing
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sample-multiplexing-and-barcoding" class="md-nav__link">
    <span class="md-ellipsis">
      
        Sample Multiplexing and Barcoding
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sample Multiplexing and Barcoding">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#when-to-multiplex" class="md-nav__link">
    <span class="md-ellipsis">
      
        When to Multiplex?
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiplexing-guidelines" class="md-nav__link">
    <span class="md-ellipsis">
      
        Multiplexing Guidelines
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#barcoding-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Barcoding Strategies
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cost-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Cost Considerations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Cost Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-sample-cost-estimates-2025" class="md-nav__link">
    <span class="md-ellipsis">
      
        Per-Sample Cost Estimates (2025)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#special-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Special Considerations
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Special Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#metagenome-vs-virome-sequencing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Metagenome vs. Virome Sequencing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantitative-vs-qualitative" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quantitative vs. Qualitative
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#time-series-and-longitudinal-studies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Time-Series and Longitudinal Studies
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#decision-tree-choosing-your-strategy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Decision Tree: Choosing Your Strategy
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#recommended-strategies-by-study-type" class="md-nav__link">
    <span class="md-ellipsis">
      
        Recommended Strategies by Study Type
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Recommended Strategies by Study Type">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#study-type-1-exploratory-virome-diversity" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 1: Exploratory Virome Diversity
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#study-type-2-novel-virus-discovery" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 2: Novel Virus Discovery
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#study-type-3-outbreak-investigation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 3: Outbreak Investigation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#study-type-4-large-scale-epidemiology" class="md-nav__link">
    <span class="md-ellipsis">
      
        Study Type 4: Large-Scale Epidemiology
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      
        Next Steps
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      
        Further Reading
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="sequencing-strategies-for-virome-analysis">Sequencing Strategies for Virome Analysis</h1>
<blockquote>
<p><strong>Last Updated:</strong> November 29, 2025</p>
</blockquote>
<p>Choosing the right sequencing strategy is crucial for successful virome analysis. This guide covers sequencing platforms, library preparation methods, coverage requirements, and how to match your strategy to your research question.</p>
<h2 id="sequencing-platform-overview">Sequencing Platform Overview</h2>
<h3 id="platform-comparison">Platform Comparison</h3>
<table>
<thead>
<tr>
<th>Platform</th>
<th>Read Length</th>
<th>Accuracy</th>
<th>Throughput</th>
<th>Cost/Gb</th>
<th>Best For</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Illumina</strong></td>
<td>150-300 bp</td>
<td>&gt;99.9%</td>
<td>Very High</td>
<td>$</td>
<td>Diversity studies, quantification</td>
</tr>
<tr>
<td><strong>PacBio HiFi</strong></td>
<td>10-25 kb</td>
<td>&gt;99.9%</td>
<td>High</td>
<td>$$$</td>
<td>Complete genomes, structural variation</td>
</tr>
<tr>
<td><strong>Oxford Nanopore</strong></td>
<td>1-100+ kb</td>
<td>~95-99%</td>
<td>Medium-High</td>
<td>$$</td>
<td>Long-range assembly, real-time analysis</td>
</tr>
<tr>
<td><strong>MGI/BGI</strong></td>
<td>150-300 bp</td>
<td>&gt;99%</td>
<td>Very High</td>
<td>$</td>
<td>Similar to Illumina, lower cost</td>
</tr>
</tbody>
</table>
<h3 id="illumina-sequencing">Illumina Sequencing</h3>
<p><strong>Platforms:</strong> NovaSeq, NextSeq, MiSeq, HiSeq</p>
<p><strong>Advantages:</strong>
- Highest accuracy (&gt;Q30)
- Well-established protocols
- Extensive computational tools
- Cost-effective for high throughput
- Excellent for paired-end reads</p>
<p><strong>Limitations:</strong>
- Short read length challenges assembly
- Can miss structural variants
- Repetitive regions difficult to resolve
- GC bias in some library prep methods</p>
<p><strong>When to use:</strong>
- Viral diversity and abundance studies
- Comparative viromics (multiple samples)
- Targeted amplicon sequencing
- When high accuracy is paramount
- Budget-constrained projects</p>
<p><strong>Recommended configurations:</strong>
- <strong>Diversity studies</strong>: 2150 bp, 10-20M reads per sample
- <strong>Deep virome</strong>: 2250 bp, 50-100M reads per sample
- <strong>Metagenome with virome</strong>: 2150 bp, 100-300M reads per sample</p>
<h3 id="pacbio-hifi-sequencing">PacBio HiFi Sequencing</h3>
<p><strong>Platform:</strong> Sequel II/IIe, Revio</p>
<p><strong>Advantages:</strong>
- Long, accurate reads (HiFi)
- Complete viral genomes in single reads
- Resolves repeat regions
- No GC bias
- Detects structural variants</p>
<p><strong>Limitations:</strong>
- More expensive per gigabase
- Lower throughput than Illumina
- Requires more input DNA (typically)
- Computational intensity for assembly</p>
<p><strong>When to use:</strong>
- Isolate sequencing (complete genomes)
- Characterizing structural variation
- Resolving complex repeats
- Host-integrated prophages with flanking regions
- Novel virus discovery requiring complete genomes</p>
<p><strong>Recommended configurations:</strong>
- <strong>Single viral isolate</strong>: 1-5 SMRT cells (Gb scale)
- <strong>Complex virome</strong>: 5-10 SMRT cells for deep coverage
- <strong>Hybrid approach</strong>: Combine with Illumina for accuracy + completeness</p>
<h3 id="oxford-nanopore-sequencing">Oxford Nanopore Sequencing</h3>
<p><strong>Platforms:</strong> MinION, GridION, PromethION</p>
<p><strong>Advantages:</strong>
- Very long reads (10-100+ kb)
- Real-time sequencing and analysis
- Portable (MinION)
- Direct RNA sequencing possible
- Rapid turnaround
- Lower capital cost</p>
<p><strong>Limitations:</strong>
- Lower accuracy than Illumina/HiFi (~95-99%)
- Homopolymer errors
- Requires more input DNA
- Computational tools still maturing</p>
<p><strong>When to use:</strong>
- Rapid outbreak investigation
- Field deployment
- Long-range structural analysis
- Direct RNA virus sequencing
- When real-time results needed</p>
<p><strong>Recommended configurations:</strong>
- <strong>MinION</strong>: Quick virome screen, 1-2 flow cells
- <strong>PromethION</strong>: Deep virome, production-scale
- <strong>Adaptive sampling</strong>: Target specific viral sequences in real-time</p>
<h2 id="library-preparation-strategies">Library Preparation Strategies</h2>
<h3 id="standard-shotgun-metagenomics">Standard Shotgun Metagenomics</h3>
<p><strong>Principle:</strong> Fragment all DNA, add adapters, sequence everything.</p>
<p><strong>Protocol:</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="m">1</span>.<span class="w"> </span>Fragment<span class="w"> </span>DNA<span class="w"> </span><span class="o">(</span>enzymatic<span class="w"> </span>or<span class="w"> </span>mechanical<span class="o">)</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="m">2</span>.<span class="w"> </span>End-repair<span class="w"> </span>and<span class="w"> </span>A-tailing
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="m">3</span>.<span class="w"> </span>Adapter<span class="w"> </span>ligation
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="m">4</span>.<span class="w"> </span>Size<span class="w"> </span>selection<span class="w"> </span><span class="o">(</span>optional<span class="o">)</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a><span class="m">5</span>.<span class="w"> </span>PCR<span class="w"> </span>amplification<span class="w"> </span><span class="o">(</span>minimal<span class="w"> </span>cycles<span class="o">)</span>
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a><span class="m">6</span>.<span class="w"> </span>Quantification<span class="w"> </span>and<span class="w"> </span>QC
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a><span class="m">7</span>.<span class="w"> </span>Sequencing
</span></code></pre></div></p>
<p><strong>Advantages:</strong>
- Unbiased representation of viral genomes
- Quantitative (read abundance reflects genome abundance)
- Works for diverse viral types
- Standard, well-supported protocols</p>
<p><strong>Limitations:</strong>
- Requires sufficient input DNA (typically 1-100 ng)
- Host contamination in unenriched samples
- Rare viruses may need high sequencing depth</p>
<p><strong>When to use:</strong>
- Standard virome analysis
- Diversity and abundance studies
- Sufficient input material available</p>
<h3 id="amplicon-sequencing">Amplicon Sequencing</h3>
<p><strong>Principle:</strong> PCR amplify specific viral genes, then sequence amplicons.</p>
<p><strong>Common targets:</strong>
- <strong>DNA viruses</strong>: DNA polymerase genes, capsid genes
- <strong>RNA viruses</strong>: RdRp (RNA-dependent RNA polymerase)
- <strong>Specific groups</strong>: Papillomavirus L1, adenovirus hexon, etc.</p>
<p><strong>Advantages:</strong>
- Extremely sensitive (can detect rare viruses)
- Lower sequencing cost per sample
- Works with low-input samples
- Targeted for specific viral groups</p>
<p><strong>Limitations:</strong>
- PCR bias
- Only detects amplified regions
- Primer design challenges (viral diversity)
- Not quantitative (PCR amplification bias)
- Misses unamplified viruses</p>
<p><strong>When to use:</strong>
- Surveillance for specific viral families
- Low-biomass samples
- Cost-effective screening of many samples
- Phylogenetic studies of specific genes</p>
<h3 id="rna-sequencing">RNA Sequencing</h3>
<p><strong>Specific considerations for RNA viruses:</strong></p>
<p><strong>Library prep options:</strong>
1. <strong>Random priming</strong>: Captures all RNA sequences
2. <strong>rRNA depletion</strong>: Remove ribosomal contamination
3. <strong>PolyA selection</strong>: Only for polyA+ viral RNAs (rare in viruses)
4. <strong>dsRNA enrichment</strong>: Targets viral replication intermediates</p>
<p><strong>Challenges:</strong>
- RNA degradation (use RNase-free everything)
- rRNA contamination (often &gt;90% of reads)
- Variable strand specificity needs</p>
<p><strong>Stranded vs. Non-stranded:</strong>
- <strong>Non-stranded</strong>: Cheaper, more standard
- <strong>Stranded</strong>: Preserves orientation, critical for some RNA viruses</p>
<p><strong>Recommended approach for RNA viromes:</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="m">1</span>.<span class="w"> </span>Extract<span class="w"> </span>RNA<span class="w"> </span><span class="o">(</span>DNase-treated<span class="w"> </span>to<span class="w"> </span>remove<span class="w"> </span>DNA<span class="o">)</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="m">2</span>.<span class="w"> </span>Optional:<span class="w"> </span>dsRNA<span class="w"> </span>enrichment
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a><span class="m">3</span>.<span class="w"> </span>rRNA<span class="w"> </span>depletion<span class="w"> </span><span class="o">(</span>CRITICAL<span class="w"> </span><span class="k">for</span><span class="w"> </span>host-associated<span class="w"> </span>samples<span class="o">)</span>
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a><span class="m">4</span>.<span class="w"> </span>Stranded<span class="w"> </span>library<span class="w"> </span>prep
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a><span class="m">5</span>.<span class="w"> </span>Sequencing<span class="w"> </span><span class="o">(</span><span class="m">2</span>150<span class="w"> </span>bp<span class="w"> </span>typical<span class="o">)</span>
</span></code></pre></div></p>
<h3 id="low-input-library-preparation">Low-Input Library Preparation</h3>
<p><strong>For samples &lt;1 ng DNA:</strong></p>
<p><strong>Options:</strong>
1. <strong>Nextera XT</strong>: Tagmentation-based, works with 1 ng
2. <strong>SMARTer</strong>: Template switching, works with &lt;1 ng
3. <strong>TruSeq Nano</strong>: Standard Illumina, requires 25-200 ng (not low-input)
4. <strong>MDA then standard library</strong>: Amplify first, then make library</p>
<p><strong>Considerations:</strong>
- More PCR cycles = more bias
- Adapter dimers more problematic
- Size selection often critical
- Negative controls absolutely essential</p>
<h2 id="coverage-and-sequencing-depth">Coverage and Sequencing Depth</h2>
<h3 id="how-much-sequencing">How Much Sequencing?</h3>
<p><strong>Factors affecting depth requirements:</strong>
- Research question (diversity vs. complete genomes)
- Sample complexity (simple vs. highly diverse)
- Viral abundance (high vs. low biomass)
- Assembly strategy (reference vs. de novo)</p>
<h3 id="recommended-depths">Recommended Depths</h3>
<h4 id="diversitycommunity-analysis">Diversity/Community Analysis</h4>
<p><strong>Goal:</strong> Detect most abundant viruses, characterize community structure</p>
<p><strong>Illumina:</strong>
- <strong>Simple communities</strong> (isolates, monocultures): 5-10M reads
- <strong>Moderate complexity</strong> (gut, marine): 20-50M reads
- <strong>High complexity</strong> (soil, sediment): 50-200M reads</p>
<p><strong>Expected:</strong>
- Capture dominant viruses (&gt;0.1% abundance)
- Miss rare viruses (&lt;0.01%)
- Sufficient for diversity metrics
- Good for comparative studies</p>
<h4 id="deep-characterization">Deep Characterization</h4>
<p><strong>Goal:</strong> Complete/near-complete genomes of abundant viruses</p>
<p><strong>Illumina:</strong>
- 50-100M reads for gut viromes
- 100-300M reads for environmental viromes
- More for highly diverse samples</p>
<p><strong>Expected:</strong>
- Near-complete genomes for viruses &gt;0.01% abundance
- Better assembly contiguity
- Detect rarer viruses
- Identify strain variation</p>
<h4 id="complete-genome-recovery">Complete Genome Recovery</h4>
<p><strong>Goal:</strong> Closed, circularized viral genomes</p>
<p><strong>Long-read (PacBio/Nanopore):</strong>
- 5-20 Gb for isolates
- 20-100 Gb for complex communities
- Depends on target genome sizes and abundance</p>
<p><strong>Expected:</strong>
- Complete genomes for abundant viruses
- Resolve repeats and structural variations
- Phage-host linkages (if integrated)</p>
<h3 id="rarefaction-and-coverage">Rarefaction and Coverage</h3>
<p><strong>Rarefaction curves</strong> show whether you've sequenced deeply enough:</p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1"># Conceptual rarefaction curve</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="kn">from</span><span class="w"> </span><span class="nn">matplotlib</span><span class="w"> </span><span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="n">reads</span> <span class="o">=</span> <span class="p">[</span><span class="mi">1</span><span class="n">M</span><span class="p">,</span> <span class="mi">5</span><span class="n">M</span><span class="p">,</span> <span class="mi">10</span><span class="n">M</span><span class="p">,</span> <span class="mi">20</span><span class="n">M</span><span class="p">,</span> <span class="mi">50</span><span class="n">M</span><span class="p">,</span> <span class="mi">100</span><span class="n">M</span><span class="p">]</span>
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="n">new_viruses</span> <span class="o">=</span> <span class="p">[</span><span class="mi">100</span><span class="p">,</span> <span class="mi">250</span><span class="p">,</span> <span class="mi">350</span><span class="p">,</span> <span class="mi">425</span><span class="p">,</span> <span class="mi">475</span><span class="p">,</span> <span class="mi">490</span><span class="p">]</span>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">reads</span><span class="p">,</span> <span class="n">new_viruses</span><span class="p">)</span>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a><span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Sequencing Depth (reads)&quot;</span><span class="p">)</span>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Unique Viruses Detected&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong>Interpretation:</strong>
- <strong>Steep curve</strong>: Not saturated, more sequencing will find more viruses
- <strong>Plateau</strong>: Saturated, additional sequencing has diminishing returns
- <strong>No plateau</strong>: Extremely diverse, may never saturate (e.g., ocean viromes)</p>
<h2 id="paired-end-vs-single-end">Paired-End vs. Single-End</h2>
<h3 id="paired-end-sequencing">Paired-End Sequencing</h3>
<p><strong>Configuration:</strong> 2150 bp, 2250 bp (read pairs from both ends of DNA fragment)</p>
<p><strong>Advantages:</strong>
- Improved assembly (scaffolding information)
- Better repeat resolution
- Detects structural variants
- Higher confidence alignments
- Useful for strain differentiation</p>
<p><strong>Limitations:</strong>
- More expensive than single-end
- Slightly lower total bases per run
- Requires paired-end aware tools</p>
<p><strong>When to use:</strong>
- De novo assembly
- Comprehensive virome studies
- When budget allows
- <strong>Recommended for most virome projects</strong></p>
<h3 id="single-end-sequencing">Single-End Sequencing</h3>
<p><strong>Configuration:</strong> 1150 bp, 175 bp</p>
<p><strong>Advantages:</strong>
- Lower cost per sample
- More total bases per run
- Sufficient for some applications
- Faster library prep</p>
<p><strong>When to use:</strong>
- Reference-based mapping only
- Simple viral communities
- Budget-constrained projects
- Amplicon sequencing</p>
<h2 id="read-length-considerations">Read Length Considerations</h2>
<h3 id="short-reads-50-150-bp">Short Reads (50-150 bp)</h3>
<p><strong>Pros:</strong>
- Lower cost
- Higher throughput
- Established tools</p>
<p><strong>Cons:</strong>
- Assembly challenges
- Poor repeat resolution
- Limited for complex genomes</p>
<p><strong>Use for:</strong>
- Reference mapping
- Read-based classification (Kraken, Kaiju)
- Cost-effective screening</p>
<h3 id="medium-reads-150-300-bp">Medium Reads (150-300 bp)</h3>
<p><strong>Pros:</strong>
- Balance of cost and assembly quality
- Good for most viruses
- Paired-end overlap possible</p>
<p><strong>Cons:</strong>
- Still challenging for repeats
- Assembly not as contiguous as long reads</p>
<p><strong>Use for:</strong>
- <strong>Standard virome analysis (most common)</strong>
- Good assembly with paired-end
- Versatile for many applications</p>
<h3 id="long-reads-10-kb">Long Reads (&gt;10 kb)</h3>
<p><strong>Pros:</strong>
- Complete viral genomes
- Resolve structural complexity
- Less assembly ambiguity</p>
<p><strong>Cons:</strong>
- Higher cost
- Lower throughput
- More input DNA needed</p>
<p><strong>Use for:</strong>
- Novel virus characterization
- Structural variation studies
- Prophage architecture
- Host-phage linkages</p>
<h2 id="hybrid-sequencing-approaches">Hybrid Sequencing Approaches</h2>
<h3 id="illumina-pacbionanopore">Illumina + PacBio/Nanopore</h3>
<p><strong>Strategy:</strong> Use long reads for assembly scaffolding, short reads for error correction.</p>
<p><strong>Workflow:</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="m">1</span>.<span class="w"> </span>Sequence<span class="w"> </span>with<span class="w"> </span>long<span class="w"> </span>reads<span class="w"> </span><span class="o">(</span>PacBio/Nanopore<span class="o">)</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="m">2</span>.<span class="w"> </span>Assemble<span class="w"> </span>long<span class="w"> </span>reads<span class="w"> </span>into<span class="w"> </span>contigs
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="m">3</span>.<span class="w"> </span>Sequence<span class="w"> </span>same<span class="w"> </span>sample<span class="w"> </span>with<span class="w"> </span>Illumina<span class="w"> </span><span class="o">(</span>high<span class="w"> </span>coverage<span class="o">)</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="m">4</span>.<span class="w"> </span>Polish<span class="w"> </span>long-read<span class="w"> </span>assembly<span class="w"> </span>with<span class="w"> </span>Illumina<span class="w"> </span>reads
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="m">5</span>.<span class="w"> </span>Result:<span class="w"> </span>Long<span class="w"> </span>contigs<span class="w"> </span>with<span class="w"> </span>high<span class="w"> </span>accuracy
</span></code></pre></div></p>
<p><strong>Advantages:</strong>
- Best of both worlds: length + accuracy
- Cost-effective compromise
- Resolves complex structures accurately</p>
<p><strong>Limitations:</strong>
- Two library preps required
- More computational analysis
- Higher overall cost</p>
<p><strong>When to use:</strong>
- Important novel viruses needing complete genomes
- Prophage characterization
- Research projects with sufficient budget
- Structural variation studies</p>
<h3 id="adaptivetargeted-sequencing">Adaptive/Targeted Sequencing</h3>
<p><strong>Nanopore-specific feature:</strong> Real-time sequence analysis and selective sequencing.</p>
<p><strong>Applications:</strong>
- Enrich for specific viral sequences during run
- Deplete abundant contaminants (host, bacteria)
- Focus sequencing on target viruses</p>
<p><strong>Example workflow:</strong>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="m">1</span>.<span class="w"> </span>Start<span class="w"> </span>sequencing<span class="w"> </span>run
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="m">2</span>.<span class="w"> </span>Real-time<span class="w"> </span>basecalling<span class="w"> </span>identifies<span class="w"> </span>sequences
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="m">3</span>.<span class="w"> </span>Reject<span class="w"> </span>reads<span class="w"> </span>matching<span class="w"> </span>host/bacterial<span class="w"> </span>references
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="m">4</span>.<span class="w"> </span>Continue<span class="w"> </span>sequencing<span class="w"> </span>viral<span class="w"> </span>reads<span class="w"> </span>only
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a><span class="m">5</span>.<span class="w"> </span>Result:<span class="w"> </span>Enriched<span class="w"> </span>viral<span class="w"> </span>dataset
</span></code></pre></div></p>
<h2 id="sample-multiplexing-and-barcoding">Sample Multiplexing and Barcoding</h2>
<h3 id="when-to-multiplex">When to Multiplex?</h3>
<p><strong>Advantages:</strong>
- Reduce cost per sample
- Increase throughput
- Efficient use of sequencing capacity</p>
<p><strong>Considerations:</strong>
- Need enough reads per sample after demultiplexing
- Barcode loss can occur (typically 1-5%)
- Multiplexing level depends on viral complexity</p>
<h3 id="multiplexing-guidelines">Multiplexing Guidelines</h3>
<table>
<thead>
<tr>
<th>Sample Complexity</th>
<th>Platform</th>
<th>Reads Needed/Sample</th>
<th>Max Multiplexing</th>
</tr>
</thead>
<tbody>
<tr>
<td>Low (isolates)</td>
<td>Illumina MiSeq</td>
<td>5-10M</td>
<td>10-20 samples</td>
</tr>
<tr>
<td>Medium (gut)</td>
<td>Illumina NextSeq</td>
<td>20-50M</td>
<td>8-16 samples</td>
</tr>
<tr>
<td>High (soil, ocean)</td>
<td>Illumina NovaSeq</td>
<td>50-200M</td>
<td>20-40 samples</td>
</tr>
</tbody>
</table>
<p><strong>Example:</strong>
- NovaSeq S4 flow cell: ~3-4 billion reads
- Target 50M reads per sample
- Can multiplex 60-80 samples
- Account for 10-20% loss to QC and host filtering</p>
<h3 id="barcoding-strategies">Barcoding Strategies</h3>
<p><strong>Dual indexing</strong> (recommended):
- Reduces barcode hopping/cross-contamination
- Two barcodes per sample (i5 and i7 adapters)
- Critical for high-plex runs</p>
<p><strong>Unique molecular identifiers (UMIs):</strong>
- Tag individual molecules before amplification
- Remove PCR duplicates computationally
- Important for low-input samples with many PCR cycles
- Enables accurate quantification</p>
<h2 id="cost-considerations">Cost Considerations</h2>
<h3 id="per-sample-cost-estimates-2025">Per-Sample Cost Estimates (2025)</h3>
<table>
<thead>
<tr>
<th>Strategy</th>
<th>Platform</th>
<th>Cost/Sample</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td>Basic virome</td>
<td>Illumina (10M reads)</td>
<td>$50-150</td>
<td>Multiplexed, 2150 bp</td>
</tr>
<tr>
<td>Standard virome</td>
<td>Illumina (50M reads)</td>
<td>$150-300</td>
<td>Good for most studies</td>
</tr>
<tr>
<td>Deep virome</td>
<td>Illumina (200M reads)</td>
<td>$300-600</td>
<td>High diversity samples</td>
</tr>
<tr>
<td>Complete genomes</td>
<td>PacBio HiFi</td>
<td>$500-2000</td>
<td>Isolates or low complexity</td>
</tr>
<tr>
<td>Rapid outbreak</td>
<td>Nanopore MinION</td>
<td>$100-500</td>
<td>Lower depth, fast results</td>
</tr>
</tbody>
</table>
<p><strong>Cost-saving strategies:</strong>
- Multiplex aggressively for simple samples
- Use amplicon sequencing for screening
- Reserve deep sequencing for high-priority samples
- Pilot with small-scale sequencing first
- Use appropriate depth (don't over-sequence)</p>
<h2 id="special-considerations">Special Considerations</h2>
<h3 id="metagenome-vs-virome-sequencing">Metagenome vs. Virome Sequencing</h3>
<p><strong>Metagenome (total community DNA):</strong>
- <strong>Pros</strong>: Captures prophages, maintains host context
- <strong>Cons</strong>: Mostly non-viral reads (&gt;90% bacteria/eukarya in unenriched)
- <strong>When</strong>: Studying prophages, have high sequencing budget</p>
<p><strong>Enriched virome (VLP-prepared):</strong>
- <strong>Pros</strong>: Enriched viral signal, more cost-effective
- <strong>Cons</strong>: Misses prophages, sample prep complexity
- <strong>When</strong>: Standard virome studies, focus on free viruses</p>
<h3 id="quantitative-vs-qualitative">Quantitative vs. Qualitative</h3>
<p><strong>For quantitative analysis</strong> (accurate viral abundances):
- Avoid or minimize amplification (MDA/PCR biases)
- Use spike-ins for normalization
- Deep, even coverage
- Careful library prep (minimal PCR cycles)</p>
<p><strong>For qualitative analysis</strong> (presence/absence, diversity):
- Amplification acceptable
- Lower depth may suffice
- Focus on maximizing virus recovery</p>
<h3 id="time-series-and-longitudinal-studies">Time-Series and Longitudinal Studies</h3>
<p><strong>Considerations:</strong>
- Consistent sample prep across timepoints
- Batch effects (sequence all together if possible)
- Sufficient depth to detect changes (50M+ reads)
- Technical replicates for key timepoints
- Spike-in controls for cross-timepoint comparison</p>
<h2 id="decision-tree-choosing-your-strategy">Decision Tree: Choosing Your Strategy</h2>
<div class="mermaid">graph TD
    A[Research Question] --&gt; B{Complete genomes needed?}
    B --&gt;|Yes| C{Budget allows?}
    C --&gt;|Yes| D[PacBio HiFi or Hybrid]
    C --&gt;|No| E[Illumina deep sequencing]
    B --&gt;|No| F{Diversity study?}
    F --&gt;|Yes| G[Illumina 2x150, 20-50M reads]
    F --&gt;|No| H{Targeted group?}
    H --&gt;|Yes| I[Amplicon sequencing]
    H --&gt;|No| J[Standard Illumina virome]</div>
<h2 id="recommended-strategies-by-study-type">Recommended Strategies by Study Type</h2>
<h3 id="study-type-1-exploratory-virome-diversity">Study Type 1: Exploratory Virome Diversity</h3>
<p><strong>Goal:</strong> Characterize viral community, compare between samples</p>
<p><strong>Recommended:</strong>
- Platform: Illumina
- Configuration: 2150 bp paired-end
- Depth: 20-50M reads per sample
- Multiplexing: 10-20 samples per run (NextSeq)
- Library: Standard shotgun metagenomics
- Sample prep: VLP enrichment + DNase</p>
<h3 id="study-type-2-novel-virus-discovery">Study Type 2: Novel Virus Discovery</h3>
<p><strong>Goal:</strong> Complete genomes of unknown viruses</p>
<p><strong>Recommended:</strong>
- Platform: PacBio HiFi or Hybrid (PacBio + Illumina)
- Configuration: PacBio 15-20kb HiFi, Illumina 2250bp
- Depth: 20-50 Gb PacBio, 100M reads Illumina
- Library: Shotgun for both platforms
- Sample prep: Enriched virome or isolate culture</p>
<h3 id="study-type-3-outbreak-investigation">Study Type 3: Outbreak Investigation</h3>
<p><strong>Goal:</strong> Rapid identification of pathogenic viruses</p>
<p><strong>Recommended:</strong>
- Platform: Nanopore MinION or Illumina MiSeq
- Configuration: Nanopore long reads or 2150 bp
- Depth: 5-20M reads
- Turnaround: &lt;24 hours possible with Nanopore
- Library: Rapid library prep kits
- Analysis: Real-time for Nanopore</p>
<h3 id="study-type-4-large-scale-epidemiology">Study Type 4: Large-Scale Epidemiology</h3>
<p><strong>Goal:</strong> Screen hundreds of samples for specific viruses</p>
<p><strong>Recommended:</strong>
- Platform: Illumina NovaSeq
- Configuration: 2150 bp or amplicon approach
- Depth: 5-10M reads per sample (or less for amplicons)
- Multiplexing: 100+ samples per run
- Library: Amplicon (if targeting specific viruses) or shotgun
- Sample prep: Minimal processing for cost efficiency</p>
<h2 id="next-steps">Next Steps</h2>
<ul>
<li><strong><a href="../../tutorials/basic-metagenome-virome/">Complete Tutorials</a></strong> - Follow step-by-step sequencing workflows</li>
<li><strong><a href="../../tools/selection-guide/">Tool Selection Guide</a></strong> - Choose analysis tools</li>
<li><strong><a href="../best-practices/index.md">Best Practices</a></strong> - Ensure high-quality results</li>
</ul>
<h2 id="further-reading">Further Reading</h2>
<ul>
<li>Warwick-Dugdale, J., et al. (2019). "Long-read viral metagenomics captures abundant and microdiverse viral populations." <em>PeerJ</em>, 7, e6800.</li>
<li>Roux, S., et al. (2016). "Towards quantitative viromics for both double-stranded and single-stranded DNA viruses." <em>PeerJ</em>, 4, e2777.</li>
<li>Garalde, D. R., et al. (2018). "Highly parallel direct RNA sequencing on an array of nanopores." <em>Nature Methods</em>, 15(3), 201-206.</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <!-- Custom footer -->

<footer class="md-footer">
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; 2023-2025 <a href="https://github.com/shandley">Scott Handley</a> -  <a href="#__consent">Change cookie settings</a>
          </div>
        
        Awesome-Virome Documentation | 
        <a href="https://shandley.github.io/awesome-virome/dashboard.html" target="_blank">
          Dashboard
        </a> |
        <a href="https://shandley.github.io/awesome-virome/citations.html" target="_blank">
          Citation Analytics
        </a>
        <br>
        Made with
        <a href="https://www.mkdocs.org/" target="_blank" rel="noopener">
          MkDocs
        </a> and
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/shandley/awesome-virome" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
    </div>
  </div>
</footer>

<script>
document.addEventListener('DOMContentLoaded', function() {
  window.mermaidConfig = {
    startOnLoad: true,
    theme: 'default',
    flowchart: {
      useMaxWidth: false,
      htmlLabels: true,
      curve: 'basis',
      defaultRenderer: 'elk',
      width: 250
    },
    securityLevel: 'loose'
  };
  
  function initMermaid() {
    try {
      if (typeof window.mermaid !== 'undefined') {
        window.mermaid.initialize(window.mermaidConfig);
        
        // Process all mermaid diagrams that haven't been processed yet
        document.querySelectorAll('.mermaid:not([data-processed="true"])').forEach(function(el) {
          try {
            window.mermaid.init(undefined, el);
          } catch (err) {
            console.error('Mermaid initialization error for specific element:', err);
            // Add an error class to show there was an issue
            el.classList.add('mermaid-error');
            el.textContent = 'Diagram rendering error. Please check console for details.';
          }
        });
      }
    } catch (err) {
      console.error('Mermaid initialization error:', err);
    }
  }
  
  // Try to use mermaid if it's already loaded
  if (typeof window.mermaid !== 'undefined') {
    initMermaid();
  } else {
    // Load mermaid if it's not available
    console.log('Loading mermaid from mermaid.html partial');
    var script = document.createElement('script');
    script.src = 'https://cdn.jsdelivr.net/npm/mermaid@9.4.3/dist/mermaid.min.js';
    script.onload = function() {
      console.log('Mermaid loaded successfully');
      initMermaid();
    };
    script.onerror = function() {
      console.error('Failed to load mermaid script');
    };
    document.head.appendChild(script);
  }
  
  // Secondary initialization after a delay to catch any missed diagrams
  setTimeout(initMermaid, 1000);
});

// Backup initialization on full page load
window.addEventListener('load', function() {
  if (typeof window.mermaid !== 'undefined') {
    console.log('Window loaded, reinitializing mermaid');
    window.mermaid.initialize({
      startOnLoad: true,
      theme: 'default',
      flowchart: {
        useMaxWidth: false,
        htmlLabels: true,
        curve: 'basis',
        defaultRenderer: 'elk',
        width: 250
      },
      securityLevel: 'loose'
    });
    
    // Process any diagrams that still haven't been processed
    document.querySelectorAll('.mermaid:not([data-processed="true"])').forEach(function(el) {
      try {
        window.mermaid.init(undefined, el);
      } catch (err) {
        console.error('Mermaid load-time initialization error:', err);
      }
    });
  }
});
</script>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.sections", "navigation.indexes", "navigation.top", "navigation.footer", "content.tabs.link", "content.code.copy", "content.code.annotate", "search.highlight", "search.share", "search.suggest"], "search": "../../assets/javascripts/workers/search.7a47a382.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"default": "latest", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.e71a0d61.min.js"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mermaid@9.4.3/dist/mermaid.min.js"></script>
      
        <script src="../../js/mermaid.js"></script>
      
    
  </body>
</html>